{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "006c2bb4-2782-429d-be24-8077f786abee",
   "metadata": {
    "panel-layout": {
     "height": 64.4375,
     "visible": true,
     "width": 100
    }
   },
   "source": [
    "<h1 style=\"font-family: 'Times New Roman'; text-align: center; font-weight: bold;\">\n",
    "    Data Preprocessing\n",
    "</h1></p>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ab542741-b903-475e-88a0-1196e06765a0",
   "metadata": {},
   "source": [
    "<div style=\"font-family: 'Times New Roman'; font-size: 15px; text-align: justify; width: 100%;\">\n",
    "  <div>\n",
    "    <span style=\"display: inline-block; width: 100px;\"><b>Date</b></span>: 4<sup>th</sup> October 2024\n",
    "  </div>\n",
    "  <div>\n",
    "    <span style=\"display: inline-block; width: 100px;\"><b>Author</b></span>: Deepan Kanagarajan Babu\n",
    "  </div>\n",
    "  <div>\n",
    "    <span style=\"display: inline-block; width: 100px;\"><b>Description</b></span>: In this document, the Metagenomic Assembled Genomes (MAGs) from the rice leaf phyllosphere-associated data are preprocessed to run BLAST, RAST, and CoverM analysis. The provided data are <a href=\"https://static-content.springer.com/esm/art%3A10.1038%2Fs41467-023-44335-3/MediaObjects/41467_2023_44335_MOESM4_ESM.xlsx\">Supplementary Data 1</a>, <a href=\"https://static-content.springer.com/esm/art%3A10.1038%2Fs41597-022-01320-7/MediaObjects/41597_2022_1320_MOESM2_ESM.xlsx\">Supplementary Data 2</a> containing MAGs details, MAGs, and CoxL protein sequences. Supplementary Data 1 contains details about the rice varieties used for metagenome sequencing, and Supplementary Data 2 contains details about the MAGs and their assembly, including taxonomic classification, BioSample ID, SRA accession, Sample ID, and scientific name details. Supplimentary Data 1 is from the article <a href=\"https://doi.org/10.1038/s41467-023-44335-3\">\"Microbiome homeostasis on rice leaves is regulated by a precursor molecule of lignin biosynthesis\"</a> and Supplimentary Data 2 is from the article <a href=\"https://doi.org/10.1038/s41597-022-01320-7\">\"Recovery of metagenome-assembled genomes from the phyllosphere of 110 rice genotypes\"</a>\n",
    "  </div>\n",
    "</div>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "40583b9f-6854-4b59-918b-0ab5d35eaea2",
   "metadata": {
    "panel-layout": {
     "height": 50.8125,
     "visible": true,
     "width": 100
    }
   },
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Required Libraries\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "82fc52cb-edcf-4af2-b99e-8a7332bc9b87",
   "metadata": {},
   "outputs": [],
   "source": [
    "# importing necessary libraries\n",
    "import os\n",
    "import shutil\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from Bio import SeqIO\n",
    "import subprocess"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d6cbe16-ec86-48fc-96bc-78a0db88d699",
   "metadata": {
    "panel-layout": {
     "height": 50.8125,
     "visible": true,
     "width": 100
    }
   },
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Required Functions\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "fc2d758a-ad80-4e7d-84e5-9388f3483b4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Defining single function to open multiple file formats (.csv, .txt, .fasta or .fastq (to display only 1st 100 lines), and .xlsx)\n",
    "def open_file(file_path):\n",
    "    if file_path.endswith('.csv'):\n",
    "        # Read CSV file\n",
    "        data = pd.read_csv(file_path)\n",
    "        return data\n",
    "    elif file_path.endswith('.txt'):\n",
    "        # Read TXT file\n",
    "        try:\n",
    "            with open(file_path, 'r') as file:\n",
    "                data = file.read()\n",
    "            return data\n",
    "        except FileNotFoundError:\n",
    "            raise FileNotFoundError(f\"The file {file_path} does not exist.\")\n",
    "    elif file_path.endswith(('.fasta', '.fastq')):\n",
    "        # Read FASTA or FASTQ file\n",
    "        try:\n",
    "            with open(file_path, 'r') as file:\n",
    "                data = []\n",
    "                for _ in range(100):\n",
    "                    line = file.readline()\n",
    "                    if not line:\n",
    "                        break\n",
    "                    data.append(line.strip())\n",
    "            return data\n",
    "        except FileNotFoundError:\n",
    "            raise FileNotFoundError(f\"The file {file_path} does not exist.\")\n",
    "    elif file_path.endswith('.xlsx'):\n",
    "        # Read XLSX file\n",
    "        try:\n",
    "            data = pd.read_excel(file_path)\n",
    "            return data\n",
    "        except FileNotFoundError:\n",
    "            raise FileNotFoundError(f\"The file {file_path} does not exist.\")\n",
    "    else:\n",
    "        raise ValueError(\"Unsupported file format\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "869f5b5e-d21b-4dc7-ac1b-0d3ca18456ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Defining function to list files from the folder\n",
    "def list_file(folder_path):\n",
    "    # List all files in the folder\n",
    "    files = os.listdir(folder_path)\n",
    "    \n",
    "    # Check if there are any files in the folder\n",
    "    if not files:\n",
    "        return None\n",
    "    \n",
    "    # Return the first file in the folder\n",
    "    return files\n",
    "\n",
    "# Defining function to cound number of files in a folder by extension\n",
    "def count_files(folder_path, extensions):\n",
    "    # List all files in the given folder\n",
    "    files = os.listdir(folder_path)\n",
    "    # Initialize a dictionary to hold counts for each extension\n",
    "    counts = {ext: 0 for ext in extensions}\n",
    "    # Count files for each extension\n",
    "    for file in files:\n",
    "        for ext in extensions:\n",
    "            if file.endswith(ext):\n",
    "                counts[ext] += 1\n",
    "    return counts\n",
    "\n",
    "# List of possible extensions to count\n",
    "extensions = ['.docx', '.csv', '.fasta', '.txt', '.fastq']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "cdce376a-10ea-4566-86dc-9ba1ae18650b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Defining function to create database\n",
    "def create_db(input_file, output_folder, db_name):\n",
    "    # Create the new folder if it doesn't exist\n",
    "    os.makedirs(output_folder, exist_ok=True)\n",
    "    \n",
    "    # Define the output database path\n",
    "    output_db = os.path.join(output_folder, db_name)\n",
    "    \n",
    "    # Run the makeblastdb command\n",
    "    subprocess.run([\n",
    "        \"makeblastdb\",\n",
    "        \"-in\", input_file,\n",
    "        \"-dbtype\", \"prot\",\n",
    "        \"-out\", output_db\n",
    "    ])\n",
    "    \n",
    "    print(f\"Database created at {output_db}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2eab2a10-5062-4d72-bc06-55d4688d6b34",
   "metadata": {},
   "outputs": [],
   "source": [
    "def remove_gaps_from_fasta(input_file, output_file):\n",
    "    with open(input_file, 'r') as infile, open(output_file, 'w') as outfile:\n",
    "        for line in infile:\n",
    "            if line.startswith('>'):\n",
    "                outfile.write(line)\n",
    "            else:\n",
    "                outfile.write(line.replace('-', ''))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "796e3e5a-34e1-43a1-8f83-7c8086a8cf82",
   "metadata": {
    "panel-layout": {
     "height": 50.8125,
     "visible": true,
     "width": 100
    }
   },
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Required Directories Path\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "c55ba52a-f246-456d-b0d8-4a1f6ef55eb2",
   "metadata": {
    "panel-layout": {
     "height": 0,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [],
   "source": [
    "# Specify the folder path containing unprocessed MAG files\n",
    "## Raw_Data folder where all the files are located\n",
    "raw_data = \"C:/Users/dkb1g24/Documents/mres/Raw_Data\"\n",
    "\n",
    "## Path to the unprocessed MAGs files\n",
    "unprocessed_mag = \"C:/Users/dkb1g24/Documents/mres/Raw_Data/MAG_unprocessed\"\n",
    "\n",
    "## Path to Rice cultivars details excel file\n",
    "rice_path = \"C:/Users/dkb1g24/Documents/mres/Raw_Data/rice_cultivar_data.xlsx\"\n",
    "\n",
    "## Path to Metagenomic Assembled Genomes (MAGs) details CSV file\n",
    "mag_file_path = \"C:/Users/dkb1g24/Documents/mres/Raw_Data/mag_data.csv\"\n",
    "\n",
    "## Path to CoxL protein sequences FASTA file\n",
    "coxl_fasta_path = \"C:/Users/dkb1g24/Documents/mres/Raw_Data/CoxL_sequences.fa\"\n",
    "\n",
    "## Path to New SRA accession file path\n",
    "sra_path = \"C:/Users/dkb1g24/Documents/mres/Raw_Data/SRA.xlsx\"\n",
    "\n",
    "# Create output directories\n",
    "## Directory to save the process MAG files\n",
    "mags = \"C:/Users/dkb1g24/Documents/mres/BLAST/MAG\"\n",
    "## Ensure the output directory exists\n",
    "os.makedirs(mags, exist_ok=True)\n",
    "\n",
    "## Define coxl db output folders\n",
    "coxl1_db_folder = \"C:/Users/dkb1g24/Documents/mres/BLAST/CoxL1\" # for Form 1 proteins\n",
    "coxl2_db_folder = \"C:/Users/dkb1g24/Documents/mres/BLAST/CoxL2\" # for Form 2 proteins\n",
    "\n",
    "## Ensure the output directories exist\n",
    "os.makedirs(coxl1_db_folder, exist_ok=True)\n",
    "os.makedirs(coxl2_db_folder, exist_ok=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f22fc255-bf9b-407e-9ceb-f17fb96a464d",
   "metadata": {
    "panel-layout": {
     "height": 50.8125,
     "visible": true,
     "width": 100
    }
   },
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Analysis and Preprocess\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "9c801eb6-9f9f-4759-b17b-bd5d7d56a78f",
   "metadata": {
    "panel-layout": {
     "height": 0,
     "visible": true,
     "width": 100
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['maxbin_SP106_1.016_cleaned.fasta', 'maxbin_SP106_1.023_cleaned.fasta', 'maxbin_SP106_2.001.fasta', 'maxbin_SP106_3.024.fasta', 'maxbin_SP108_3.007_cleaned.fasta', 'maxbin_SP112_1.001_cleaned.fasta', 'maxbin_SP118_2.002.fasta', 'maxbin_SP118_2.009.fasta', 'maxbin_SP118_3.007_cleaned.fasta', 'maxbin_SP119_2.004.fasta', 'maxbin_SP122_1.006_cleaned.fasta', 'maxbin_SP122_2.006_cleaned.fasta', 'maxbin_SP123_3.025_sub_cleaned.fasta', 'maxbin_SP128_2.014.fasta', 'maxbin_SP128_2.029_sub_cleaned.fasta', 'maxbin_SP128_3.008.fasta', 'maxbin_SP133_1.009.fasta', 'maxbin_SP133_2.005.fasta', 'maxbin_SP133_3.007_cleaned.fasta', 'maxbin_SP135_3.009_sub_cleaned.fasta', 'maxbin_SP138_3.003.fasta', 'maxbin_SP139_2.003_cleaned.fasta', 'maxbin_SP139_3.013.fasta', 'maxbin_SP143_1.006.fasta', 'maxbin_SP149_1.012_cleaned.fasta', 'maxbin_SP149_2.004_cleaned.fasta', 'maxbin_SP149_3.011_sub_cleaned.fasta', 'maxbin_SP151_2.002_cleaned.fasta', 'maxbin_SP154_3.006_sub_cleaned.fasta', 'maxbin_SP166_2.026_cleaned.fasta', 'maxbin_SP166_3.028_cleaned.fasta', 'maxbin_SP167_1.005.fasta', 'maxbin_SP167_1.018_sub_cleaned.fasta', 'maxbin_SP167_3.015.fasta', 'maxbin_SP168_1.017_sub_cleaned.fasta', 'maxbin_SP184_1.002_sub.fasta', 'maxbin_SP184_3.002_sub_cleaned.fasta', 'maxbin_SP188_3.003.fasta', 'maxbin_SP188_3.010.fasta', 'maxbin_SP188_3.012.fasta', 'maxbin_SP200_2.008.fasta', 'maxbin_SP210_1.037.fasta', 'maxbin_SP210_2.032.fasta', 'maxbin_SP210_3.022.fasta', 'maxbin_SP217_1.002.fasta', 'maxbin_SP21_1.014_cleaned.fasta', 'maxbin_SP21_1.015_sub_cleaned.fasta', 'maxbin_SP223_1.001.fasta', 'maxbin_SP238_1.005.fasta', 'maxbin_SP264_3.001.fasta', 'maxbin_SP265_3.009_cleaned.fasta', 'maxbin_SP266_1.010.fasta', 'maxbin_SP266_1.016_sub_cleaned.fasta', 'maxbin_SP266_2.040.fasta', 'maxbin_SP285_1.001_sub_cleaned.fasta', 'maxbin_SP285_1.013.fasta', 'maxbin_SP285_1.040_cleaned.fasta', 'maxbin_SP287_1.003_sub_cleaned.fasta', 'maxbin_SP287_3.002_sub_cleaned.fasta', 'maxbin_SP301_2.015.fasta', 'maxbin_SP306_1.005_cleaned.fasta', 'maxbin_SP306_2.008_cleaned.fasta', 'maxbin_SP306_2.020_sub_cleaned.fasta', 'maxbin_SP309_3.003.fasta', 'maxbin_SP310_2.008.fasta', 'maxbin_SP339_2.004_cleaned.fasta', 'maxbin_SP349_3.014_cleaned.fasta', 'maxbin_SP350_1.003_cleaned.fasta', 'maxbin_SP350_1.005.fasta', 'maxbin_SP352_2.003.fasta', 'maxbin_SP369_3.019_cleaned.fasta', 'maxbin_SP370_2.004.fasta', 'maxbin_SP373_2.011_cleaned.fasta', 'maxbin_SP397_3.003.fasta', 'maxbin_SP398_1.001.fasta', 'maxbin_SP398_2.010_cleaned.fasta', 'maxbin_SP398_3.010_cleaned.fasta', 'maxbin_SP415_3.004.fasta', 'maxbin_SP424_1.012_sub_cleaned.fasta', 'maxbin_SP426_1.013.fasta', 'maxbin_SP48_1.033_cleaned.fasta', 'maxbin_SP48_2.041.fasta', 'maxbin_SP48_3.042_cleaned.fasta', 'maxbin_SP51_1.004_sub_cleaned.fasta', 'maxbin_SP51_1.021.fasta', 'maxbin_SP51_2.007_sub.fasta', 'maxbin_SP60_2.018_cleaned.fasta', 'maxbin_SP60_2.034_cleaned.fasta', 'maxbin_SP67_1.019_cleaned.fasta', 'maxbin_SP67_2.016.fasta', 'maxbin_SP71_1.009_cleaned.fasta', 'maxbin_SP71_2.012.fasta', 'maxbin_SP71_3.011_cleaned.fasta', 'maxbin_SP77_1.034_sub_cleaned.fasta', 'maxbin_SP80_3.007_cleaned.fasta', 'maxbin_SP80_3.012.fasta', 'maxbin_SP81_2.001_cleaned.fasta', 'maxbin_SP81_2.010.fasta', 'maxbin_SP81_3.008.fasta', 'maxbin_SP81_3.009.fasta', 'maxbin_SP89_1.015_cleaned.fasta', 'maxbin_SP89_2.009.fasta', 'maxbin_SP89_2.019_sub.fasta', 'maxbin_SP89_3.014_sub.fasta', 'maxbin_SP8_2.014.fasta', 'maxbin_SP92_3.016_sub.fasta', 'S1C10298_SP242.fasta', 'S1C108396_SP36.fasta', 'S1C108_SP426.fasta', 'S1C1096_SP62.fasta', 'S1C1163_SP60.fasta', 'S1C118219_SP301.fasta', 'S1C11885_SP265.fasta', 'S1C11945_SP264.fasta', 'S1C12337_SP426.fasta', 'S1C12339_SP29.fasta', 'S1C125195_SP226.fasta', 'S1C1391_SP200.fasta', 'S1C13_SP238.fasta', 'S1C143502_SP285.fasta', 'S1C14366_SP242.fasta', 'S1C14620_SP426.fasta', 'S1C1487_SP403.fasta', 'S1C1497_SP370.fasta', 'S1C16883_SP350.fasta', 'S1C17122_SP350.fasta', 'S1C18108_SP112.fasta', 'S1C19042_SP118.fasta', 'S1C1_SP3.fasta', 'S1C20416_SP29_cleaned.fasta', 'S1C20649_SP200.fasta', 'S1C20_SP93.fasta', 'S1C210_SP29.fasta', 'S1C21702_SP77.fasta', 'S1C234_SP168.fasta', 'S1C2374_SP242.fasta', 'S1C23829_SP242.fasta', 'S1C23932_SP242.fasta', 'S1C26421_SP349.fasta', 'S1C26695_SP210.fasta', 'S1C273_SP108.fasta', 'S1C28558_SP21.fasta', 'S1C28809_SP349.fasta', 'S1C315_SP188.fasta', 'S1C32281_SP236.fasta', 'S1C3268_SP123.fasta', 'S1C35295_SP282_cleaned.fasta', 'S1C36_SP60.fasta', 'S1C36_SP75.fasta', 'S1C41259_SP48_cleaned.fasta', 'S1C47698_SP152.fasta', 'S1C48_SP260.fasta', 'S1C5480_SP242.fasta', 'S1C5636_SP81.fasta', 'S1C58117_SP162.fasta', 'S1C627_SP166.fasta', 'S1C62909_SP162_cleaned.fasta', 'S1C6315_SP302.fasta', 'S1C67_SP62.fasta', 'S1C710_SP310.fasta', 'S1C7250_SP286.fasta', 'S1C72769_SP131.fasta', 'S1C77_SP397.fasta', 'S1C794_SP166.fasta', 'S1C83591_SP131.fasta', 'S1C8761_SP8.fasta', 'S1C92728_SP213.fasta', 'S1C92_SP369.fasta', 'S2C102328_SP39.fasta', 'S2C106_SP217.fasta', 'S2C108735_SP301.fasta', 'S2C11432_SP223.fasta', 'S2C118127_SP301.fasta', 'S2C124_SP48.fasta', 'S2C12621_SP285.fasta', 'S2C12736_SP71.fasta', 'S2C12874_SP309.fasta', 'S2C129_SP75.fasta', 'S2C13626_SP71.fasta', 'S2C16027_SP60.fasta', 'S2C16742_SP242.fasta', 'S2C17225_SP29.fasta', 'S2C17652_SP166.fasta', 'S2C17738_SP200.fasta', 'S2C18454_SP60_cleaned.fasta', 'S2C18479_SP60.fasta', 'S2C19509_SP61.fasta', 'S2C19572_SP51.fasta', 'S2C19659_SP238.fasta', 'S2C199_SP339.fasta', 'S2C21750_SP77.fasta', 'S2C21_SP68.fasta', 'S2C23416_SP67.fasta', 'S2C23696_SP266.fasta', 'S2C23896_SP200.fasta', 'S2C255_SP51.fasta', 'S2C26004_SP21.fasta', 'S2C26139_SP139.fasta', 'S2C27220_SP282.fasta', 'S2C29_SP398.fasta', 'S2C31181_SP210.fasta', 'S2C311_SP75.fasta', 'S2C313_SP266.fasta', 'S2C31_SP92.fasta', 'S2C3242_SP226.fasta', 'S2C32656_SP139.fasta', 'S2C32737_SP223.fasta', 'S2C32993_SP51.fasta', 'S2C332_SP118.fasta', 'S2C33542_SP48.fasta', 'S2C33_SP339.fasta', 'S2C346_SP369.fasta', 'S2C369_SP123.fasta', 'S2C37388_SP108.fasta', 'S2C38726_SP106.fasta', 'S2C41209_SP48.fasta', 'S2C46_SP95.fasta', 'S2C476_SP71.fasta', 'S2C4820_SP77.fasta', 'S2C60_SP8.fasta', 'S2C67_SP310.fasta', 'S2C78296_SP133.fasta', 'S2C7865_SP61.fasta', 'S2C7956_SP352.fasta', 'S2C86_SP60.fasta', 'S2C87093_SP39.fasta', 'S2C87442_SP39.fasta', 'S2C893_SP162.fasta', 'S2C8941_SP8.fasta', 'S2C954_SP236.fasta', 'S3C1012_SP226.fasta', 'S3C10154_SP92.fasta', 'S3C102340_SP39.fasta', 'S3C103065_SP39.fasta', 'S3C10405_SP242.fasta', 'S3C107984_SP36.fasta', 'S3C1086_SP112.fasta', 'S3C11303_SP39.fasta', 'S3C12250_SP29.fasta', 'S3C122_SP339.fasta', 'S3C12306_SP29.fasta', 'S3C1253_SP210.fasta', 'S3C126_SP123.fasta', 'S3C1335_SP223.fasta', 'S3C13875_SP60.fasta', 'S3C142789_SP226.fasta', 'S3C1435_SP48.fasta', 'S3C1439_SP349.fasta', 'S3C152_SP370.fasta', 'S3C1665_SP217.fasta', 'S3C17520_SP143.fasta', 'S3C18099_SP112.fasta', 'S3C18100_SP226.fasta', 'S3C181_SP217.fasta', 'S3C18757_SP77.fasta', 'S3C1876_SP226.fasta', 'S3C190_SP108.fasta', 'S3C19501_SP86.fasta', 'S3C19671_SP238.fasta', 'S3C19732_SP60.fasta', 'S3C208_SP223.fasta', 'S3C215_SP51.fasta', 'S3C221_SP309.fasta', 'S3C22292_SP61.fasta', 'S3C23132_SP350.fasta', 'S3C23698_SP282.fasta', 'S3C25031_SP51.fasta', 'S3C25280_SP266.fasta', 'S3C2533_SP236.fasta', 'S3C26360_SP51.fasta', 'S3C28800_SP106.fasta', 'S3C2912_SP226.fasta', 'S3C299_SP217.fasta', 'S3C31588_SP166.fasta', 'S3C32792_SP266.fasta', 'S3C36278_SP166.fasta', 'S3C36471_SP108.fasta', 'S3C367_SP236.fasta', 'S3C39948_SP80.fasta', 'S3C3_SP3.fasta', 'S3C400_SP29.fasta', 'S3C40919_SP210.fasta', 'S3C4298_SP309.fasta', 'S3C436_SP266.fasta', 'S3C441_SP108.fasta', 'S3C442_SP352.fasta', 'S3C4605_SP217.fasta', 'S3C482_SP67.fasta', 'S3C4844_SP301.fasta', 'S3C519_SP188.fasta', 'S3C5220_SP51.fasta', 'S3C53_SP339.fasta', 'S3C5621_SP81.fasta', 'S3C58_SP421.fasta', 'S3C594_SP118.fasta', 'S3C6073_SP122.fasta', 'S3C63226_SP133.fasta', 'S3C6817_SP11.fasta', 'S3C78723_SP285.fasta', 'S3C7931_SP352.fasta', 'S3C82631_SP236.fasta', 'S3C907_SP139.fasta', 'S3C94_SP310.fasta', 'S3C97460_SP39.fasta', 'S3C979_SP92.fasta', 'S3C99_SP71.fasta', 'SP106_1_metabat2_genome_mining.1.fasta', 'SP106_1_metabat2_genome_mining.9.fasta', 'SP106_2_metabat2_genome_mining.28.fasta', 'SP106_2_metabat2_genome_mining.31.fasta', 'SP106_2_metabat2_genome_mining.33.fasta', 'SP106_2_metabat2_genome_mining.35_cleaned.fasta', 'SP106_3_metabat2_genome_mining.30.fasta', 'SP108_1_metabat2_genome_mining.30.fasta', 'SP108_2_metabat2_genome_mining.22.fasta', 'SP108_2_metabat2_genome_mining.23_cleaned.fasta', 'SP108_2_metabat2_genome_mining.28.fasta', 'SP112_2_metabat2_genome_mining.15.fasta', 'SP112_2_metabat2_genome_mining.16.fasta', 'SP11_1_metabat2_genome_mining.22.fasta', 'SP123_2_metabat2_genome_mining.9_sub.fasta', 'SP123_3_metabat2_genome_mining.2.fasta', 'SP123_3_metabat2_genome_mining.23.fasta', 'SP128_1_metabat2_genome_mining.23_sub.fasta', 'SP128_1_metabat2_genome_mining.6.fasta', 'SP128_2_metabat2_genome_mining.22_sub.fasta', 'SP128_3_metabat2_genome_mining.14_sub.fasta', 'SP131_1_metabat2_genome_mining.18.fasta', 'SP131_1_metabat2_genome_mining.2.fasta', 'SP131_1_metabat2_genome_mining.22.fasta', 'SP131_2_metabat2_genome_mining.10.fasta', 'SP131_2_metabat2_genome_mining.13.fasta', 'SP133_3_metabat2_genome_mining.7_sub.fasta', 'SP135_1_metabat2_genome_mining.5.fasta', 'SP138_1_metabat2_genome_mining.16.fasta', 'SP138_1_metabat2_genome_mining.20.fasta', 'SP138_1_metabat2_genome_mining.26.fasta', 'SP138_2_metabat2_genome_mining.21.fasta', 'SP138_2_metabat2_genome_mining.9.fasta', 'SP138_3_metabat2_genome_mining.19.fasta', 'SP138_3_metabat2_genome_mining.25.fasta', 'SP138_3_metabat2_genome_mining.9.fasta', 'SP139_1_metabat2_genome_mining.12.fasta', 'SP139_1_metabat2_genome_mining.21.fasta', 'SP139_1_metabat2_genome_mining.36_sub.fasta', 'SP139_1_metabat2_genome_mining.38_cleaned.fasta', 'SP139_2_metabat2_genome_mining.27.fasta', 'SP139_2_metabat2_genome_mining.40.fasta', 'SP139_3_metabat2_genome_mining.19_cleaned.fasta', 'SP143_1_metabat2_genome_mining.23.fasta', 'SP143_2_metabat2_genome_mining.23.fasta', 'SP143_3_metabat2_genome_mining.25_cleaned.fasta', 'SP147_2_metabat2_genome_mining.8.fasta', 'SP147_3_metabat2_genome_mining.17.fasta', 'SP149_1_metabat2_genome_mining.5.fasta', 'SP149_2_metabat2_genome_mining.17_sub.fasta', 'SP152_1_metabat2_genome_mining.5.fasta', 'SP152_3_metabat2_genome_mining.14.fasta', 'SP162_1_metabat2_genome_mining.13.fasta', 'SP162_1_metabat2_genome_mining.25.fasta', 'SP164_3_metabat2_genome_mining.9.fasta', 'SP166_1_metabat2_genome_mining.33_sub.fasta', 'SP166_1_metabat2_genome_mining.55.fasta', 'SP166_2_metabat2_genome_mining.10.fasta', 'SP166_2_metabat2_genome_mining.11.fasta', 'SP167_1_metabat2_genome_mining.26.fasta', 'SP167_1_metabat2_genome_mining.27.fasta', 'SP167_3_metabat2_genome_mining.6.fasta', 'SP168_2_metabat2_genome_mining.16_sub_cleaned.fasta', 'SP168_3_metabat2_genome_mining.15.fasta', 'SP188_1_metabat2_genome_mining.17_sub.fasta', 'SP188_1_metabat2_genome_mining.18.fasta', 'SP188_1_metabat2_genome_mining.21.fasta', 'SP188_2_metabat2_genome_mining.14.fasta', 'SP188_3_metabat2_genome_mining.13.fasta', 'SP188_3_metabat2_genome_mining.14.fasta', 'SP200_1_metabat2_genome_mining.38_sub.fasta', 'SP200_1_metabat2_genome_mining.43.fasta', 'SP200_1_metabat2_genome_mining.44.fasta', 'SP200_2_metabat2_genome_mining.26.fasta', 'SP200_2_metabat2_genome_mining.31.fasta', 'SP200_3_metabat2_genome_mining.14.fasta', 'SP200_3_metabat2_genome_mining.62.fasta', 'SP209_1_metabat2_genome_mining.8.fasta', 'SP210_1_metabat2_genome_mining.33_cleaned.fasta', 'SP210_1_metabat2_genome_mining.39.fasta', 'SP210_1_metabat2_genome_mining.48.fasta', 'SP210_2_metabat2_genome_mining.13.fasta', 'SP210_2_metabat2_genome_mining.14.fasta', 'SP210_2_metabat2_genome_mining.17_sub_cleaned.fasta', 'SP210_2_metabat2_genome_mining.20.fasta', 'SP210_2_metabat2_genome_mining.40.fasta', 'SP210_2_metabat2_genome_mining.41.fasta', 'SP210_3_metabat2_genome_mining.10.fasta', 'SP210_3_metabat2_genome_mining.3.fasta', 'SP210_3_metabat2_genome_mining.50.fasta', 'SP210_3_metabat2_genome_mining.57.fasta', 'SP213_1_metabat2_genome_mining.25.fasta', 'SP213_2_metabat2_genome_mining.2_sub.fasta', 'SP213_2_metabat2_genome_mining.4.fasta', 'SP213_3_metabat2_genome_mining.20.fasta', 'SP217_1_metabat2_genome_mining.12.fasta', 'SP21_1_metabat2_genome_mining.10.fasta', 'SP21_1_metabat2_genome_mining.8.fasta', 'SP21_3_metabat2_genome_mining.44_cleaned.fasta', 'SP223_1_metabat2_genome_mining.22_cleaned.fasta', 'SP223_2_metabat2_genome_mining.47_cleaned.fasta', 'SP226_1_metabat2_genome_mining.16_cleaned.fasta', 'SP226_1_metabat2_genome_mining.5.fasta', 'SP226_1_metabat2_genome_mining.6_cleaned.fasta', 'SP226_2_metabat2_genome_mining.20.fasta', 'SP226_2_metabat2_genome_mining.9.fasta', 'SP226_3_metabat2_genome_mining.15.fasta', 'SP236_1_metabat2_genome_mining.17_cleaned.fasta', 'SP236_1_metabat2_genome_mining.8.fasta', 'SP236_2_metabat2_genome_mining.10.fasta', 'SP236_2_metabat2_genome_mining.18.fasta', 'SP236_2_metabat2_genome_mining.32.fasta', 'SP236_3_metabat2_genome_mining.5.fasta', 'SP237_1_metabat2_genome_mining.5.fasta', 'SP238_2_metabat2_genome_mining.24.fasta', 'SP238_3_metabat2_genome_mining.31.fasta', 'SP238_3_metabat2_genome_mining.52.fasta', 'SP242_1_metabat2_genome_mining.23.fasta', 'SP242_1_metabat2_genome_mining.4.fasta', 'SP242_2_metabat2_genome_mining.26_cleaned.fasta', 'SP242_2_metabat2_genome_mining.9_sub_cleaned.fasta', 'SP242_3_metabat2_genome_mining.19.fasta', 'SP242_3_metabat2_genome_mining.52.fasta', 'SP260_1_metabat2_genome_mining.39.fasta', 'SP260_2_metabat2_genome_mining.16.fasta', 'SP260_2_metabat2_genome_mining.31.fasta', 'SP260_2_metabat2_genome_mining.46.fasta', 'SP266_1_metabat2_genome_mining.14.fasta', 'SP266_1_metabat2_genome_mining.36.fasta', 'SP266_1_metabat2_genome_mining.40_cleaned.fasta', 'SP266_1_metabat2_genome_mining.7_sub.fasta', 'SP266_2_metabat2_genome_mining.11.fasta', 'SP266_2_metabat2_genome_mining.38.fasta', 'SP266_2_metabat2_genome_mining.41.fasta', 'SP266_3_metabat2_genome_mining.23.fasta', 'SP266_3_metabat2_genome_mining.25.fasta', 'SP266_3_metabat2_genome_mining.45_sub_cleaned.fasta', 'SP266_3_metabat2_genome_mining.7_cleaned.fasta', 'SP282_3_metabat2_genome_mining.17.fasta', 'SP282_3_metabat2_genome_mining.31_sub.fasta', 'SP282_3_metabat2_genome_mining.36.fasta', 'SP285_1_metabat2_genome_mining.11.fasta', 'SP285_1_metabat2_genome_mining.14.fasta', 'SP285_1_metabat2_genome_mining.9.fasta', 'SP285_2_metabat2_genome_mining.35_sub.fasta', 'SP285_3_metabat2_genome_mining.21.fasta', 'SP286_1_metabat2_genome_mining.17_sub_cleaned.fasta', 'SP286_1_metabat2_genome_mining.28.fasta', 'SP287_2_metabat2_genome_mining.19.fasta', 'SP287_2_metabat2_genome_mining.21_cleaned.fasta', 'SP287_2_metabat2_genome_mining.22_sub.fasta', 'SP287_2_metabat2_genome_mining.26.fasta', 'SP287_2_metabat2_genome_mining.35.fasta', 'SP287_3_metabat2_genome_mining.12.fasta', 'SP287_3_metabat2_genome_mining.19_cleaned.fasta', 'SP287_3_metabat2_genome_mining.23.fasta', 'SP29_1_metabat2_genome_mining.10_sub.fasta', 'SP29_3_metabat2_genome_mining.11.fasta', 'SP29_3_metabat2_genome_mining.23.fasta', 'SP29_3_metabat2_genome_mining.5_cleaned.fasta', 'SP301_1_metabat2_genome_mining.10_sub.fasta', 'SP301_1_metabat2_genome_mining.15_cleaned.fasta', 'SP301_2_metabat2_genome_mining.23_sub_cleaned.fasta', 'SP301_2_metabat2_genome_mining.7.fasta', 'SP302_2_metabat2_genome_mining.12_sub.fasta', 'SP302_2_metabat2_genome_mining.9.fasta', 'SP306_3_metabat2_genome_mining.14_cleaned.fasta', 'SP306_3_metabat2_genome_mining.20.fasta', 'SP309_1_metabat2_genome_mining.36.fasta', 'SP309_3_metabat2_genome_mining.12.fasta', 'SP309_3_metabat2_genome_mining.19.fasta', 'SP310_2_metabat2_genome_mining.9.fasta', 'SP310_3_metabat2_genome_mining.11_sub.fasta', 'SP339_1_metabat2_genome_mining.8.fasta', 'SP339_2_metabat2_genome_mining.19_cleaned.fasta', 'SP349_1_metabat2_genome_mining.8_sub.fasta', 'SP350_1_metabat2_genome_mining.31_cleaned.fasta', 'SP350_1_metabat2_genome_mining.36.fasta', 'SP350_1_metabat2_genome_mining.41.fasta', 'SP350_1_metabat2_genome_mining.7_sub.fasta', 'SP350_1_metabat2_genome_mining.9.fasta', 'SP350_2_metabat2_genome_mining.23.fasta', 'SP350_3_metabat2_genome_mining.1.fasta', 'SP350_3_metabat2_genome_mining.18_sub.fasta', 'SP350_3_metabat2_genome_mining.19_sub.fasta', 'SP350_3_metabat2_genome_mining.40.fasta', 'SP352_1_metabat2_genome_mining.5.fasta', 'SP36_2_metabat2_genome_mining.13.fasta', 'SP36_2_metabat2_genome_mining.16.fasta', 'SP36_3_metabat2_genome_mining.19.fasta', 'SP373_1_metabat2_genome_mining.19_cleaned.fasta', 'SP373_2_metabat2_genome_mining.12.fasta', 'SP39_1_metabat2_genome_mining.3.fasta', 'SP39_2_metabat2_genome_mining.4.fasta', 'SP39_3_metabat2_genome_mining.15.fasta', 'SP39_3_metabat2_genome_mining.9.fasta', 'SP3_3_metabat2_genome_mining.12.fasta', 'SP400_1_metabat2_genome_mining.4.fasta', 'SP403_2_metabat2_genome_mining.12.fasta', 'SP415_1_metabat2_genome_mining.10.fasta', 'SP415_1_metabat2_genome_mining.22.fasta', 'SP415_2_metabat2_genome_mining.16.fasta', 'SP420_2_metabat2_genome_mining.14.fasta', 'SP421_1_metabat2_genome_mining.27.fasta', 'SP421_3_metabat2_genome_mining.10.fasta', 'SP424_2_metabat2_genome_mining.17_cleaned.fasta', 'SP426_1_metabat2_genome_mining.15.fasta', 'SP426_1_metabat2_genome_mining.21_sub.fasta', 'SP48_2_metabat2_genome_mining.27.fasta', 'SP48_2_metabat2_genome_mining.34.fasta', 'SP48_3_metabat2_genome_mining.2.fasta', 'SP51_1_metabat2_genome_mining.8.fasta', 'SP51_2_metabat2_genome_mining.28_sub.fasta', 'SP51_2_metabat2_genome_mining.34_sub.fasta', 'SP51_2_metabat2_genome_mining.35_cleaned.fasta', 'SP51_2_metabat2_genome_mining.37_sub.fasta', 'SP51_2_metabat2_genome_mining.38_sub.fasta', 'SP51_3_metabat2_genome_mining.45.fasta', 'SP51_3_metabat2_genome_mining.49_cleaned.fasta', 'SP51_3_metabat2_genome_mining.8.fasta', 'SP60_1_metabat2_genome_mining.16.fasta', 'SP60_2_metabat2_genome_mining.1.fasta', 'SP60_2_metabat2_genome_mining.21.fasta', 'SP60_2_metabat2_genome_mining.23_sub_cleaned.fasta', 'SP60_2_metabat2_genome_mining.39.fasta', 'SP60_2_metabat2_genome_mining.56.fasta', 'SP61_1_metabat2_genome_mining.34_sub.fasta', 'SP61_3_metabat2_genome_mining.37_sub.fasta', 'SP62_1_metabat2_genome_mining.14_cleaned.fasta', 'SP67_1_metabat2_genome_mining.20.fasta', 'SP67_1_metabat2_genome_mining.5.fasta', 'SP67_2_metabat2_genome_mining.16.fasta', 'SP67_3_metabat2_genome_mining.11.fasta', 'SP67_3_metabat2_genome_mining.36.fasta', 'SP67_3_metabat2_genome_mining.7.fasta', 'SP68_3_metabat2_genome_mining.9.fasta', 'SP71_1_metabat2_genome_mining.12.fasta', 'SP71_3_metabat2_genome_mining.12.fasta', 'SP75_1_metabat2_genome_mining.10_sub.fasta', 'SP77_2_metabat2_genome_mining.21.fasta', 'SP77_3_metabat2_genome_mining.18_sub.fasta', 'SP77_3_metabat2_genome_mining.28.fasta', 'SP80_1_metabat2_genome_mining.32_sub.fasta', 'SP80_2_metabat2_genome_mining.11_sub_cleaned.fasta', 'SP80_2_metabat2_genome_mining.23_sub.fasta', 'SP80_2_metabat2_genome_mining.35.fasta', 'SP80_2_metabat2_genome_mining.39.fasta', 'SP80_2_metabat2_genome_mining.9.fasta', 'SP80_3_metabat2_genome_mining.16_sub.fasta', 'SP81_3_metabat2_genome_mining.4.fasta', 'SP86_3_metabat2_genome_mining.5.fasta', 'SP89_1_metabat2_genome_mining.23_sub.fasta', 'SP89_2_metabat2_genome_mining.28.fasta', 'SP89_3_metabat2_genome_mining.21_sub_cleaned.fasta', 'SP92_2_metabat2_genome_mining.23.fasta', 'SP92_3_metabat2_genome_mining.29.fasta', 'SP92_3_metabat2_genome_mining.37.fasta', 'SP92_3_metabat2_genome_mining.40.fasta', 'SP92_3_metabat2_genome_mining.5_sub.fasta', 'SP92_3_metabat2_genome_mining.9.fasta', 'SP93_2_metabat2_genome_mining.11.fasta', 'SP93_2_metabat2_genome_mining.8.fasta', 'SP93_3_metabat2_genome_mining.5.fasta']\n"
     ]
    }
   ],
   "source": [
    "# List the files in the raw unprocessed MAG files\n",
    "mag_files_list = list_file(unprocessed_mag)\n",
    "print(mag_files_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "39fb9526-07ff-43f0-8611-d387674dbb54",
   "metadata": {
    "panel-layout": {
     "height": 1762.7396240234375,
     "visible": true,
     "width": 100
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['>Genome437_k127_93520 flag=1 multi=4.0000 len=1848',\n",
       " 'TTTATGGATCAGGGGCGTATAGTTGAGCAGGGGCTGGCAAAAGCGTTATTCGCCGACCCGCAACAGCCGC',\n",
       " 'GTACCCGACAGTTCCTGGAAAAATTCCTTCTTCAGTAGAATCTTTGTTCTTTTTTGCTCCAGAAACATTA',\n",
       " 'TCTGGAGCAAATAATTCCCCCTGAAAGTCTTTACATATATTGTTTTATTCTGTCTGTATTGCTGTAACCA',\n",
       " 'CATAAATTCTGTACAAATTCCCCCTTGTGGTCGATATTTTATACATATAAATAAGATTTATGTGTTAGTT',\n",
       " 'TAATCAATCCATAATAATGATTATCTTTTACAGGGGCTTCATTCCGTAAGTATGAAGGATTCAGACTTTT',\n",
       " 'TCACATGGCGACGGGATTGCTATCTTCGGTTTCAGGAAATGACTTCGGCCGATGAGGTATATTCCGAACT',\n",
       " 'TCTGCGACAAACACAGGCACTAGAGTTCGATTATTTTTCGCTTTGTGTCCGTCATCCCGTGCCGTTTACC',\n",
       " 'CGGCCTAAACTCTCATTGCAGAGTAATTATCCGGCACAATGGCTATCGCATTATCAGGCGGAGAATTATT',\n",
       " 'TCGCCATCGATCCGGTACTCAAACCAGAGAATTTTGTGCACGGCCACCTACCCTGGAATGACGCGTTGTT',\n",
       " 'TGCCGATGCGCAACCCTTATGGGATGGCGCGCGGGAGCATGGGCTGCGCAAGGGAATTACCCAGTGCCTG',\n",
       " 'ATGTTACCGAATCATGCGCTCGGTTTTCTGTCCGTCTCCAGCACGGTGCAAAGCATCAATAAGTTCAGCA',\n",
       " 'AAGAAGAGCTGGAGTTGCGTCTGCAAATGCTGGTGCAGATGGCCCTGACCACGCTGTTGCGTCTGGAACA',\n",
       " 'TGAAATGGTGATGCCGCCTGAGATGAAATTCAGCAAGCGCGAGAAAGAGATTCTGAAATGGACGGCGGAA',\n",
       " 'GGTAAAACCTCAGCGGAAATAGCGATTATTCTTTCTATTTCTGAAAATACGGTGAACTTCCATCAGAAGA',\n",
       " 'ACATGCAGAAGAAATTCAATGCGCCAAACAAAACGCAGATAGCATGTTATGCGGCGGCGACGGGCATTAT',\n",
       " 'CTGACGCGTTCTGACGTTCTGCGTGCCAGACGAAAAAACGGCTGAATGCAATGCGTTCAGCCGTTTTTGT',\n",
       " 'TTTTACTGACGGTAAGCTTGTTTAATCTGCTTAACCGTGCTGGAAAATACGTCGGCCTGAGCCTGGTCTT',\n",
       " 'CCATTTGCGCGATCTGCTTTTCCATCTTAAGGATCACACGACCCGCATCAGCCTGGCCCATTGCCTGAAG',\n",
       " 'CATCAGGGTCAGAAGGGATTTCAGACAGGAGACTTCCTGCGCCAGTTCCTGGTTATTTTCCGCAGTGGCG',\n",
       " 'AAATCAGGTGTGCTCATTCATTTTCCTCATTATAAAAACGGCGCATACGCGCTTAGACCTTAGTTAAGGC',\n",
       " 'GGCGGAGTATACCACAAGCTTGACAAAAAAATGCAGCGGTTGTTTTTTGTACTTTTGGCATGCGGTTTAA',\n",
       " 'TTGAATAAACAATCAATAAGCATAAACTTCATCTTGCTTTTACTGACAATTAACATATACCGGCAGTTTT',\n",
       " 'AATGCAGTTAATTATTGTTACAAAAATTTGAAAATTATATTGGTGAACGCAACCCCGCCTGAACAATTGT',\n",
       " 'AACCTTCTGAAAAAAGGGAGCTAAAATTTAGCGACAAGCGTTGAAGTGCCTTCTTATCGTTTCCAGTTAC',\n",
       " 'TGTATAGGTTAATTTCCAAAAATTTGTTCGGTATCGCTTATCTGCATTTTTAACGTTTTAAAGTTGAGGT',\n",
       " 'AAAAAACGGTAATATATACGACATTAGCATTCCATAGCGTTATCCCTATTCTGGAGACTTTCCATTGATC',\n",
       " 'AACGTCCTTCTTGTTGATGACCACGAAC',\n",
       " '>Genome437_k127_71936 flag=1 multi=6.0000 len=4247',\n",
       " 'CGCAGAACGGTTTCCAGACATTCTCCAACGTTGTCTACATAATAATAATCCTGCGGAGATTAGGGTCCCG',\n",
       " 'GCGTGACCGATACCTACCCACCAAACGAAGTTAGTGATGTCCCATCCCCAGTTAATCGTCCTGTTAAGTC',\n",
       " 'CCCATGCTCCAATACCTGTTCCGATAGTGTAAGCGATACAGCCGAAACCGTAGATGAATAGAACCAAGGC',\n",
       " 'AGCATATAATGAAATCCACCATAATTTACCTGCTCTTTCTTCGATAGGTCGTGCAATATCTTCCGTGATA',\n",
       " 'TCGTGATAAGTTTTGTGACCAATAATCAGAGGTTCCCTTATCGGAGCTTCGTAATGTCCTGACATTTTTT',\n",
       " 'ACCTATTTATTATTTAAACTTTATTTTTCTACTCTGTTTCTTACTTTAGTATGATAGAACACGTTTGGTT',\n",
       " 'TGGTTCCGATCTCTTCAAGTAAATAATATCTTCTGTTGTCAGCATATAGCTTTCTCACTTCAGATGCTTT',\n",
       " 'GTCGTTCATGTCTCCGAACTGAATAGATCCGGTAGTACAAGCTTTTGAACAAGCAGTCTGGAACTCGCCG',\n",
       " 'TCTTTTACCACTCTTCCTTCTTTTTTCGCTTCAAGGATGGTAGTCTGTGTCATTTGGATACACAGAGAGC',\n",
       " 'ATTTCTCCATAACCCCTCTTGTTCTTACTACTACGTCAGGATTAAGTACCATTCTTCCAAGGTCGTTGTT',\n",
       " 'CATATTGAAGTCGAACTTGTCATTCAGGTTATAGGTAAACCAGTTGAAACGTCTTACTTTATAAGGACAG',\n",
       " 'TTGTTCGCACAATACCTTGTACCGATACATCTGTTGTATGCCATATGGTTCTGGCCCTGCTGACCGTGAG',\n",
       " 'AAGTTGCCGCTACCGGACAAACCGTTTCACATGGTGCATGGTTACAGTGCTGACACATCACCGGCTGGAA',\n",
       " 'GATAACATCAGGATTTTCGGCAGGGTGGTTCAGAGCACCTCCTTCACCGTTGAATGCCGTTCCGTACAGT',\n",
       " 'TCAGGTACCGCCATTCCTTCCTTAAGACCTTCATAAACCTCTACTTTCTGCTTAGCAGAATAATAGCGGT',\n",
       " 'CAATCCTTAACCAGTACATATCTCTGGACATCCTTACCTCATCCTTACCTACCACCGGAACGTTGTTCTC',\n",
       " 'TGCCTGACATGCAATCACACATGATCCGCATCCCGTACAGGAGTTCAGGTCGATAGACATATTGAAGTGA',\n",
       " 'GGTCCGTCTGTATCATCAAACGCATCCCAAAGGTCAATTTTTCTTGCCGGAAGGGCTCCGCTGATGGTAT',\n",
       " 'GGTACTCCAAAGGCTTGTTCCATCCGTGGTGTTCATCATCAAAAGCTACGTTTAAGAAATCTGCCAAAGG',\n",
       " 'AACTTCTTTCGCAATTTCATAACGTCCCATCAGGGTGTTCTGAAGCTGGATACCTGCAAAAGGATGCATC',\n",
       " 'TCGCCAGTTTTCTCTATTTTAATGTTGGATACCGCCAGGTTGGAACCGTCAAACAAAGGATAAGCATTCA',\n",
       " 'CACCGGTTTCTGCAGTAGCTCCCGAGTTTTTCTTACCGTAACCCAAAGCAAGACCTACGGATCCTTCTGC',\n",
       " 'CTGTCCCGGCTGAATGAATACAGGAACGTCTTTTACCGTCACTCCGTTTACCGTAACGTTCACGATGGAA',\n",
       " 'CCGTCCAGCTGCATTCTTGCATTAAGGTTATTTTCAATACCTAATCTTTCTGCATCTTTAGGAGAAATAG',\n",
       " 'TCAGGTAGTTATCCCAGGACATTCTGGTCAGCGGATCCGGTAACTCCTGAAGCCAAGGGTTGTTGGCCTG',\n",
       " 'TGTACCGTCTCCCATTGCTGTTTTGGTGTACAATACCAATTCCAGGTCAGACGCCTTGAAAGCTCCCAGT',\n",
       " 'TCAGCCAATGCCTGAGCCGCATTTCCGCCGGAATAGGATAAAGCCGTTGTATTGGCTGTAGTATTGAACC',\n",
       " 'CGTTATACAGCGCTTTATTGAAAGAAGTACCTCCTAATGCTGAAGTGGCATTTGCTTTCAGGTAATCGTA',\n",
       " 'ATAATTGTTGGCAGCGTTGTTCTGTCCGTTTTTCCAAACCAGAAGAGATTCTTCAATCTGTCTTGATTTG',\n",
       " 'TAGATTTTCTGGATGGTAGGCTGCATCAAAGTATACATTCCTGTCTGAGGCTCAATATCTCCCCAGGATT',\n",
       " 'CCAGCCAGTTGGCTACAGGAATTACAGCTTTCGCTGCTTTGTACATTTCGTTTTTCTTATCAGCAATAGC',\n",
       " 'TACTACGTAAGGAACTTTGGATAAAGATTTTTTGAAATCTTCACCTTTATGGTAAGAGTAGATCGGGTCT',\n",
       " 'ACGTTGTTGGCGATCAACACGCCTACCTGACCAGCATTTACCCATCCTAAAAATTCCTGGAACCTTGCTT',\n",
       " 'TGTCAAATTCCTTAAGGAAGTTTGCTTTTCCTGTGAAGGCTACTGATCCTAATTTCTGGTTGATCAGATG',\n",
       " 'CGCTAACACCTGAGCTCCTTTGGATCCGTCTGCAAATACAACTGCTTTGCTGCCTTTTGCCTGAAGTTCT',\n",
       " 'TTAGCAATTTCTGAAGCTGTTTTGTCTGAAGTTCCGCCGCCTACGATAGCATTATAAACTTCAACAAGGG',\n",
       " 'TTTTGTTAACTGCACTTGGCTTCAGTCTGTATCTTGAATCTGCATTGGCTCCTGTTAAAGACATATTGGA',\n",
       " 'TTCAACCTGGATATGTCTCAGCATGGTTGCTCCCGGCTTTCTTGCTGCTGCATAAGAAGACTCAAGGCTT',\n",
       " 'GCTGCATTGTAATCTCCTAAGAAGTCAGCCTGGAAACCTACTACCAATTCGCTTCCTCTAAGGTCATATA',\n",
       " 'CCGGTAATGCTCTGGTTCCGAAAACTTCCTGAGCGGCATCAAGGGCAGCAGCATATGGGAAGGCATCGTA',\n",
       " 'TGTTACCAGTTCGGCAGTAGGGTATTTTGCCTTGAATTCAGCAAATAATTTCTTAAAAGTAGGGGAAGGC',\n",
       " 'AAAGAATGTGATAAAAGCACGATCTTTTTCCCTGCCGTATTAGCTTCTGCCAAACCTTTAAGGATGAAAG',\n",
       " 'CGTCCACTTTGTCAAAAGTTTCATCTTTCCCGTCCAGTTTAGGCTGCTTTACTTTATCGTTGTCATATAA',\n",
       " 'AGAAAGAACGCTTGCCTGAGCTCTTGCATTGGTTTTACCCAAATCTCCTGCAGCAGTGTTAGGTTCAATC',\n",
       " 'TTAATTGGTCTTCCTTCTCTGGTTTTTACCAATACGCTGGCAAAATCAAAACCGTCGAAATATGTTGAAG',\n",
       " 'CGTAATAGTTGGGAATCCCCGGGATAATGTCATGCGGCTTTACCACATAAGGAATCGTTTTGATTACCGG',\n",
       " 'AGCCTCACATGCAGCAAGCGTTACCGCTGCAGTGGAGAACCCTAATAGTTTTAAAAAGTCTCTTCTTGAA',\n",
       " 'GTACTAGATCCGTTCTTTTCAGCATCTCCGAGAAAATCTTCTACCGGAATTTCTTCCTGGAACTCTTTTT',\n",
       " 'GAGCCAGCTTATTATTTAAAGCCGGGTCTTTAAGTTCATGAATACTTCTAAATTGTATTTTGTTTGAAGC',\n",
       " 'CATTTATACTTCTAATTTTTAGTTATTAATAATGACATTTACCACATTCAATACCTCCAATTGCATCTAC',\n",
       " 'AGTAATCTTACCACCGTCTTTAGGGTATTGTTTTTTCAGTTTTTCGTGTAGATTCTTGAAGTACTCTTTA',\n",
       " 'TTATAACCGTTGTTCATATCAACTTCAGTCGTTCTGTGGCACTCAATACACCATCCCATGGTAAAGTCAT',\n",
       " 'TGGCCATCTGAACAACATTCATAGTGTCTATTTTACCGTGACAAGCTTTACATACTACATCGATTTTGTT',\n",
       " 'ATTAGGATTCTTTTTGTTGAAAGAATTGATGATGGCCTGTTCACCGGCTACTACGTGCTGGGCGTGGTTA',\n",
       " 'AAGTAAACGAAATCCGGCATGTTGTGGATTCTGGTCCATTCCACCGGTTGTGTTTTTCCTGTATACTGCT',\n",
       " 'GTTTCTGGGAATCCCATCCTGTAGCAGCGTAGATCTTCTGGATTTCTCCGTCATAGAATGCCTTGTCTTT',\n",
       " 'TCCTGGCTCCATATAATGCTCTGCATTGTATTCGGAAATTGTTCTGTGACAGTTCATACAAACGTTCATA',\n",
       " 'GAAGGGATTTCAGATACTTTACCGTACTTGGCACTGGAGTGACATAGTTGACAGTCGATTTTTTGTTCTC',\n",
       " 'CGGCGTGAATCTTATGCGAGAAGTAAATCGGCTGTTCAGGTTTATAACCTTTGTAAACCCCGATCCACAT',\n",
       " 'CAGCCAGTTCCATACGCCATAAGCAGCAAGGATGGCAAGGATTGCCAGTAAAGCTTTACCTACATAATGA',\n",
       " 'TACTTTTCATAGATTTCACGGAAAGAACGAACTCTGGTTTCGTTAAG',\n",
       " '>Genome437_k127_28785 flag=1 multi=4.0000 len=1961',\n",
       " 'AGATCCGGTACTGAATGCTTCCGGTACCACAATTCAAATTTAAAATTTTATATTTGTACTGACCGTTTCA',\n",
       " 'GGAATAAATACGAGATCACCTATGGAATTGAGATGGAGTAACAGAAAATTGGCCATACAAAACATAAGCA',\n",
       " 'TTCTGATGATTGCATCTGTACTGGCTGTTTCATGCGGGAGTTCTAAAAATATACCCGTAAAAAGAAAAAA',\n",
       " 'AGTAACGGTAAACGTACCGAGGTCGGAAGAGCTGAGGGTTTTAAATTCAGATTTCACGGGTAAAATTCCC',\n",
       " 'GAGTCCGTTACCGAACTGCTGAAAGATGCAGAAAAGTACCTTGGTGCCCCCTATAAATTCGGAGGCGATA',\n",
       " 'CGTCCTCTGGATTCGACTGCTCCGGACTGGCCCTTAAAGTCTTTGAAGAAAATGATTACAGGCTTCCCAG',\n",
       " 'AAGGTCGGCAGACCAGGCAGAAACCGGAAAGCAGATCGATATCCGTGATGTAAAGCCCGGAGACCTGCTG',\n",
       " 'TTTTTTGCTACCGCCGGAGGAAGCAGGGTTTCCCATGTGGGGATCGTCCATGATATCGGCAGAGACGGGG',\n",
       " 'AAGTAAAATTCATCCATTCTTCTACCAGCAAAGGAGTCATCATTTCTTCCCTGAACGAAAAATACTGGAA']"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check opening one of the fasta files\n",
    "## Specify the fasta file path\n",
    "fasta_files_path = os.path.join(unprocessed_mag, \"maxbin_SP8_2.014.fasta\")\n",
    "\n",
    "## Open first 100 line of fasta file\n",
    "open_file(fasta_files_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "16f911cd-d03f-43dd-b43c-3a5e0974e818",
   "metadata": {
    "panel-layout": {
     "height": 332.66668701171875,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>SRA_accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Tax_ID</th>\n",
       "      <th>Scientific_name</th>\n",
       "      <th>Sequencing_method</th>\n",
       "      <th>Assembly_Software</th>\n",
       "      <th>Completeness_score</th>\n",
       "      <th>Completeness_software</th>\n",
       "      <th>Contamination_score</th>\n",
       "      <th>Binning_software</th>\n",
       "      <th>Investigation_type</th>\n",
       "      <th>Binning_parameters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S1C35295_SP282_cleaned</td>\n",
       "      <td>ERS6626560</td>\n",
       "      <td>SAMEA8943724</td>\n",
       "      <td>114707</td>\n",
       "      <td>uncultured Pseudomonas sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>99</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>0</td>\n",
       "      <td>vamb</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S1C41259_SP48_cleaned</td>\n",
       "      <td>ERS6626561</td>\n",
       "      <td>SAMEA8943725</td>\n",
       "      <td>158751</td>\n",
       "      <td>uncultured Acidovorax sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>99</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>0</td>\n",
       "      <td>vamb</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S1C20416_SP29_cleaned</td>\n",
       "      <td>ERS6626562</td>\n",
       "      <td>SAMEA8943726</td>\n",
       "      <td>331964</td>\n",
       "      <td>uncultured Curtobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>98</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>0</td>\n",
       "      <td>vamb</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S1C62909_SP162_cleaned</td>\n",
       "      <td>ERS6626563</td>\n",
       "      <td>SAMEA8943727</td>\n",
       "      <td>114707</td>\n",
       "      <td>uncultured Pseudomonas sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>91</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>vamb</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S2C18454_SP60_cleaned</td>\n",
       "      <td>ERS6626564</td>\n",
       "      <td>SAMEA8943728</td>\n",
       "      <td>165433</td>\n",
       "      <td>uncultured Acinetobacter sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>91</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>vamb</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Sample_ID SRA_accession     BioSample  Tax_ID  \\\n",
       "0  S1C35295_SP282_cleaned    ERS6626560  SAMEA8943724  114707   \n",
       "1   S1C41259_SP48_cleaned    ERS6626561  SAMEA8943725  158751   \n",
       "2   S1C20416_SP29_cleaned    ERS6626562  SAMEA8943726  331964   \n",
       "3  S1C62909_SP162_cleaned    ERS6626563  SAMEA8943727  114707   \n",
       "4   S2C18454_SP60_cleaned    ERS6626564  SAMEA8943728  165433   \n",
       "\n",
       "                 Scientific_name      Sequencing_method Assembly_Software  \\\n",
       "0     uncultured Pseudomonas sp.  Illumina NovaSeq 6000           Megahit   \n",
       "1      uncultured Acidovorax sp.  Illumina NovaSeq 6000           Megahit   \n",
       "2  uncultured Curtobacterium sp.  Illumina NovaSeq 6000           Megahit   \n",
       "3     uncultured Pseudomonas sp.  Illumina NovaSeq 6000           Megahit   \n",
       "4   uncultured Acinetobacter sp.  Illumina NovaSeq 6000           Megahit   \n",
       "\n",
       "   Completeness_score Completeness_software  Contamination_score  \\\n",
       "0                  99                CheckM                    0   \n",
       "1                  99                CheckM                    0   \n",
       "2                  98                CheckM                    0   \n",
       "3                  91                CheckM                    1   \n",
       "4                  91                CheckM                    1   \n",
       "\n",
       "  Binning_software           Investigation_type Binning_parameters  \n",
       "0             vamb  metagenome-assembled genome            default  \n",
       "1             vamb  metagenome-assembled genome            default  \n",
       "2             vamb  metagenome-assembled genome            default  \n",
       "3             vamb  metagenome-assembled genome            default  \n",
       "4             vamb  metagenome-assembled genome            default  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load MAGs CSV file\n",
    "mag_df = open_file(mag_file_path)\n",
    "mag_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "b025c965-7e22-4b1d-a419-8f6d802feacb",
   "metadata": {
    "panel-layout": {
     "height": 0,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All done\n"
     ]
    }
   ],
   "source": [
    "# Create a .csv file with the list of MAG fasta files name\n",
    "## List to store file names\n",
    "fasta_names = []\n",
    "\n",
    "## Append file names\n",
    "for file_name in os.listdir(unprocessed_mag):\n",
    "    if file_name.endswith('.fasta'):\n",
    "        # Remove extension\n",
    "        fasta_names.append(file_name)\n",
    "\n",
    "## Create csv file for file names\n",
    "fasta_list = pd.DataFrame({'file_name': fasta_names})\n",
    "\n",
    "## Save the DataFrame to a csv file\n",
    "fasta_out_list = 'fasta_list.csv'\n",
    "fasta_list.to_csv(fasta_out_list, index=False)\n",
    "\n",
    "print(\"All done\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "2289d409-875d-4e7a-839e-e037acd5e4e4",
   "metadata": {
    "panel-layout": {
     "height": 170.6666717529297,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>maxbin_SP106_1.016_cleaned.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>maxbin_SP106_1.023_cleaned.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>maxbin_SP106_2.001.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>maxbin_SP106_3.024.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>maxbin_SP108_3.007_cleaned.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          file_name\n",
       "0  maxbin_SP106_1.016_cleaned.fasta\n",
       "1  maxbin_SP106_1.023_cleaned.fasta\n",
       "2          maxbin_SP106_2.001.fasta\n",
       "3          maxbin_SP106_3.024.fasta\n",
       "4  maxbin_SP108_3.007_cleaned.fasta"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Open the fasta list .csv file\n",
    "fasta_files = open_file('fasta_list.csv')\n",
    "fasta_files.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "ce696663-deb9-4fbb-8f9d-8de68405494f",
   "metadata": {
    "panel-layout": {
     "height": 170.6666717529297,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_name</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>maxbin_SP106_1.016_cleaned.fasta</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>maxbin_SP106_1.023_cleaned.fasta</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>maxbin_SP106_2.001.fasta</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>maxbin_SP106_3.024.fasta</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>maxbin_SP108_3.007_cleaned.fasta</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          file_name                          ID\n",
       "0  maxbin_SP106_1.016_cleaned.fasta  maxbin_SP106_1.016_cleaned\n",
       "1  maxbin_SP106_1.023_cleaned.fasta  maxbin_SP106_1.023_cleaned\n",
       "2          maxbin_SP106_2.001.fasta          maxbin_SP106_2.001\n",
       "3          maxbin_SP106_3.024.fasta          maxbin_SP106_3.024\n",
       "4  maxbin_SP108_3.007_cleaned.fasta  maxbin_SP108_3.007_cleaned"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remove the extension and save the ids to ID column\n",
    "fasta_files['ID'] = fasta_files.iloc[:,0].str[:-6] # last 6 character of the file name is file extensions\n",
    "fasta_files.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "af157f04-afe5-4077-9600-f299e0b28665",
   "metadata": {
    "panel-layout": {
     "height": 0,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [],
   "source": [
    "# Saving the modification to fasta_list.csv\n",
    "fasta_files.to_csv('fasta_list.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "cebf779f-4d7b-4980-8e27-ac33650a2f52",
   "metadata": {
    "panel-layout": {
     "height": 170.6666717529297,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_name</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>maxbin_SP106_1.016_cleaned.fasta</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>maxbin_SP106_1.023_cleaned.fasta</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>maxbin_SP106_2.001.fasta</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>maxbin_SP106_3.024.fasta</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>maxbin_SP108_3.007_cleaned.fasta</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          file_name                          ID\n",
       "0  maxbin_SP106_1.016_cleaned.fasta  maxbin_SP106_1.016_cleaned\n",
       "1  maxbin_SP106_1.023_cleaned.fasta  maxbin_SP106_1.023_cleaned\n",
       "2          maxbin_SP106_2.001.fasta          maxbin_SP106_2.001\n",
       "3          maxbin_SP106_3.024.fasta          maxbin_SP106_3.024\n",
       "4  maxbin_SP108_3.007_cleaned.fasta  maxbin_SP108_3.007_cleaned"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Opening fasta file name list\n",
    "fasta_files = open_file('fasta_list.csv')\n",
    "fasta_files.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "38441d53-2d13-4730-a55f-2122ffaf193f",
   "metadata": {
    "panel-layout": {
     "height": 332.66668701171875,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>SRA_accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Tax_ID</th>\n",
       "      <th>Scientific_name</th>\n",
       "      <th>Sequencing_method</th>\n",
       "      <th>Assembly_Software</th>\n",
       "      <th>Completeness_score</th>\n",
       "      <th>Completeness_software</th>\n",
       "      <th>Contamination_score</th>\n",
       "      <th>Binning_software</th>\n",
       "      <th>Investigation_type</th>\n",
       "      <th>Binning_parameters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S1C35295_SP282_cleaned</td>\n",
       "      <td>ERS6626560</td>\n",
       "      <td>SAMEA8943724</td>\n",
       "      <td>114707</td>\n",
       "      <td>uncultured Pseudomonas sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>99</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>0</td>\n",
       "      <td>vamb</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S1C41259_SP48_cleaned</td>\n",
       "      <td>ERS6626561</td>\n",
       "      <td>SAMEA8943725</td>\n",
       "      <td>158751</td>\n",
       "      <td>uncultured Acidovorax sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>99</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>0</td>\n",
       "      <td>vamb</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S1C20416_SP29_cleaned</td>\n",
       "      <td>ERS6626562</td>\n",
       "      <td>SAMEA8943726</td>\n",
       "      <td>331964</td>\n",
       "      <td>uncultured Curtobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>98</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>0</td>\n",
       "      <td>vamb</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S1C62909_SP162_cleaned</td>\n",
       "      <td>ERS6626563</td>\n",
       "      <td>SAMEA8943727</td>\n",
       "      <td>114707</td>\n",
       "      <td>uncultured Pseudomonas sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>91</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>vamb</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S2C18454_SP60_cleaned</td>\n",
       "      <td>ERS6626564</td>\n",
       "      <td>SAMEA8943728</td>\n",
       "      <td>165433</td>\n",
       "      <td>uncultured Acinetobacter sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>91</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>vamb</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Sample_ID SRA_accession     BioSample  Tax_ID  \\\n",
       "0  S1C35295_SP282_cleaned    ERS6626560  SAMEA8943724  114707   \n",
       "1   S1C41259_SP48_cleaned    ERS6626561  SAMEA8943725  158751   \n",
       "2   S1C20416_SP29_cleaned    ERS6626562  SAMEA8943726  331964   \n",
       "3  S1C62909_SP162_cleaned    ERS6626563  SAMEA8943727  114707   \n",
       "4   S2C18454_SP60_cleaned    ERS6626564  SAMEA8943728  165433   \n",
       "\n",
       "                 Scientific_name      Sequencing_method Assembly_Software  \\\n",
       "0     uncultured Pseudomonas sp.  Illumina NovaSeq 6000           Megahit   \n",
       "1      uncultured Acidovorax sp.  Illumina NovaSeq 6000           Megahit   \n",
       "2  uncultured Curtobacterium sp.  Illumina NovaSeq 6000           Megahit   \n",
       "3     uncultured Pseudomonas sp.  Illumina NovaSeq 6000           Megahit   \n",
       "4   uncultured Acinetobacter sp.  Illumina NovaSeq 6000           Megahit   \n",
       "\n",
       "   Completeness_score Completeness_software  Contamination_score  \\\n",
       "0                  99                CheckM                    0   \n",
       "1                  99                CheckM                    0   \n",
       "2                  98                CheckM                    0   \n",
       "3                  91                CheckM                    1   \n",
       "4                  91                CheckM                    1   \n",
       "\n",
       "  Binning_software           Investigation_type Binning_parameters  \n",
       "0             vamb  metagenome-assembled genome            default  \n",
       "1             vamb  metagenome-assembled genome            default  \n",
       "2             vamb  metagenome-assembled genome            default  \n",
       "3             vamb  metagenome-assembled genome            default  \n",
       "4             vamb  metagenome-assembled genome            default  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking the MAGs CSV file\n",
    "mag_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "12a4c985-738b-4fde-a859-23fc8f5c6390",
   "metadata": {
    "panel-layout": {
     "height": 0,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Merged and Removed\n"
     ]
    }
   ],
   "source": [
    "# Merging the data frames with matching values of ID from fasta file name and Sample_ID from MAGs CSV file\n",
    "merged_df = pd.merge(fasta_files, mag_df, left_on='ID', right_on='Sample_ID')\n",
    "\n",
    "# Saving the merged file as 'mag_data_v2.csv' (version 2) file\n",
    "output_file = os.path.join(raw_data, 'mag_data_v2.csv')\n",
    "merged_df.to_csv(output_file, index=False)\n",
    "\n",
    "# Deleting the unwanted variable fasta_files and .csv file fasta_list.csv\n",
    "del fasta_files\n",
    "os.remove('fasta_list.csv')\n",
    "\n",
    "# To verigy the process are done\n",
    "print(\"Merged and Removed\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "9275dccd-62cc-4763-9669-ba61cc653517",
   "metadata": {
    "panel-layout": {
     "height": 332.66668701171875,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(569, 15)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>SRA_accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Tax_ID</th>\n",
       "      <th>Scientific_name</th>\n",
       "      <th>Sequencing_method</th>\n",
       "      <th>Assembly_Software</th>\n",
       "      <th>Completeness_score</th>\n",
       "      <th>Completeness_software</th>\n",
       "      <th>Contamination_score</th>\n",
       "      <th>Binning_software</th>\n",
       "      <th>Investigation_type</th>\n",
       "      <th>Binning_parameters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>maxbin_SP106_1.016_cleaned.fasta</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned</td>\n",
       "      <td>ERS6626984</td>\n",
       "      <td>SAMEA8944148</td>\n",
       "      <td>157277</td>\n",
       "      <td>uncultured Agrobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>72</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>6</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>maxbin_SP106_1.023_cleaned.fasta</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "      <td>ERS6626983</td>\n",
       "      <td>SAMEA8944147</td>\n",
       "      <td>278208</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>72</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>maxbin_SP106_2.001.fasta</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "      <td>ERS6626844</td>\n",
       "      <td>SAMEA8944008</td>\n",
       "      <td>227322</td>\n",
       "      <td>uncultured Paenibacillus sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>100</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>maxbin_SP106_3.024.fasta</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "      <td>ERS6626982</td>\n",
       "      <td>SAMEA8944146</td>\n",
       "      <td>278208</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>71</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>2</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>maxbin_SP108_3.007_cleaned.fasta</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned</td>\n",
       "      <td>ERS6626962</td>\n",
       "      <td>SAMEA8944126</td>\n",
       "      <td>797541</td>\n",
       "      <td>uncultured Mucilaginibacter sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>63</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>7</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          file_name                          ID  \\\n",
       "0  maxbin_SP106_1.016_cleaned.fasta  maxbin_SP106_1.016_cleaned   \n",
       "1  maxbin_SP106_1.023_cleaned.fasta  maxbin_SP106_1.023_cleaned   \n",
       "2          maxbin_SP106_2.001.fasta          maxbin_SP106_2.001   \n",
       "3          maxbin_SP106_3.024.fasta          maxbin_SP106_3.024   \n",
       "4  maxbin_SP108_3.007_cleaned.fasta  maxbin_SP108_3.007_cleaned   \n",
       "\n",
       "                    Sample_ID SRA_accession     BioSample  Tax_ID  \\\n",
       "0  maxbin_SP106_1.016_cleaned    ERS6626984  SAMEA8944148  157277   \n",
       "1  maxbin_SP106_1.023_cleaned    ERS6626983  SAMEA8944147  278208   \n",
       "2          maxbin_SP106_2.001    ERS6626844  SAMEA8944008  227322   \n",
       "3          maxbin_SP106_3.024    ERS6626982  SAMEA8944146  278208   \n",
       "4  maxbin_SP108_3.007_cleaned    ERS6626962  SAMEA8944126  797541   \n",
       "\n",
       "                   Scientific_name      Sequencing_method Assembly_Software  \\\n",
       "0     uncultured Agrobacterium sp.  Illumina NovaSeq 6000           Megahit   \n",
       "1         uncultured Spirosoma sp.  Illumina NovaSeq 6000           Megahit   \n",
       "2     uncultured Paenibacillus sp.  Illumina NovaSeq 6000           Megahit   \n",
       "3         uncultured Spirosoma sp.  Illumina NovaSeq 6000           Megahit   \n",
       "4  uncultured Mucilaginibacter sp.  Illumina NovaSeq 6000           Megahit   \n",
       "\n",
       "   Completeness_score Completeness_software  Contamination_score  \\\n",
       "0                  72                CheckM                    6   \n",
       "1                  72                CheckM                    1   \n",
       "2                 100                CheckM                    1   \n",
       "3                  71                CheckM                    2   \n",
       "4                  63                CheckM                    7   \n",
       "\n",
       "  Binning_software           Investigation_type Binning_parameters  \n",
       "0           maxbin  metagenome-assembled genome            default  \n",
       "1           maxbin  metagenome-assembled genome            default  \n",
       "2           maxbin  metagenome-assembled genome            default  \n",
       "3           maxbin  metagenome-assembled genome            default  \n",
       "4           maxbin  metagenome-assembled genome            default  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Opening the MAGs CSV file version 2\n",
    "mag_df2 = open_file(output_file)\n",
    "print(mag_df2.shape) # Checking the data frame shape to verify the number of MAGs\n",
    "mag_df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "cc2399ae-20ca-42dc-b4cb-e797a35aa416",
   "metadata": {
    "panel-layout": {
     "height": 27.135417938232422,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ True])"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Comparing the ID column with Sample_ID to verify the file names are properly assigned to the right organsim\n",
    "compare = mag_df2['ID'] == mag_df2['Sample_ID']\n",
    "\n",
    "# Verifying all the matches are true and no mismatch is found\n",
    "compare.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "c78b635f-549c-4fc2-bc65-e022f0292db7",
   "metadata": {
    "panel-layout": {
     "height": 27.135417938232422,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "569"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Cross checking the length of comparison is as same as number of MAG files\n",
    "len(compare)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "080b0c95-babb-4cc1-a996-8642d8d5ab31",
   "metadata": {
    "panel-layout": {
     "height": 0,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updated\n"
     ]
    }
   ],
   "source": [
    "# Droping the extra sample ID\n",
    "mag_df2 = mag_df2.drop(columns=[\"ID\"])\n",
    "\n",
    "#Saving the updated data frame to MAGs CSV file version 2\n",
    "mag_df2.to_csv('mag_data_v2.csv', index = False)\n",
    "print(\"Updated\") # For confirmation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "63e40689-56e2-4e99-b824-30a9865c6c3b",
   "metadata": {
    "panel-layout": {
     "height": 332.66668701171875,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>SRA_accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Tax_ID</th>\n",
       "      <th>Scientific_name</th>\n",
       "      <th>Sequencing_method</th>\n",
       "      <th>Assembly_Software</th>\n",
       "      <th>Completeness_score</th>\n",
       "      <th>Completeness_software</th>\n",
       "      <th>Contamination_score</th>\n",
       "      <th>Binning_software</th>\n",
       "      <th>Investigation_type</th>\n",
       "      <th>Binning_parameters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>maxbin_SP106_1.016_cleaned.fasta</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned</td>\n",
       "      <td>ERS6626984</td>\n",
       "      <td>SAMEA8944148</td>\n",
       "      <td>157277</td>\n",
       "      <td>uncultured Agrobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>72</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>6</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>maxbin_SP106_1.023_cleaned.fasta</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "      <td>ERS6626983</td>\n",
       "      <td>SAMEA8944147</td>\n",
       "      <td>278208</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>72</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>maxbin_SP106_2.001.fasta</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "      <td>ERS6626844</td>\n",
       "      <td>SAMEA8944008</td>\n",
       "      <td>227322</td>\n",
       "      <td>uncultured Paenibacillus sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>100</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>maxbin_SP106_3.024.fasta</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "      <td>ERS6626982</td>\n",
       "      <td>SAMEA8944146</td>\n",
       "      <td>278208</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>71</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>2</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>maxbin_SP108_3.007_cleaned.fasta</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned</td>\n",
       "      <td>ERS6626962</td>\n",
       "      <td>SAMEA8944126</td>\n",
       "      <td>797541</td>\n",
       "      <td>uncultured Mucilaginibacter sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>63</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>7</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          file_name                          ID  \\\n",
       "0  maxbin_SP106_1.016_cleaned.fasta  maxbin_SP106_1.016_cleaned   \n",
       "1  maxbin_SP106_1.023_cleaned.fasta  maxbin_SP106_1.023_cleaned   \n",
       "2          maxbin_SP106_2.001.fasta          maxbin_SP106_2.001   \n",
       "3          maxbin_SP106_3.024.fasta          maxbin_SP106_3.024   \n",
       "4  maxbin_SP108_3.007_cleaned.fasta  maxbin_SP108_3.007_cleaned   \n",
       "\n",
       "                    Sample_ID SRA_accession     BioSample  Tax_ID  \\\n",
       "0  maxbin_SP106_1.016_cleaned    ERS6626984  SAMEA8944148  157277   \n",
       "1  maxbin_SP106_1.023_cleaned    ERS6626983  SAMEA8944147  278208   \n",
       "2          maxbin_SP106_2.001    ERS6626844  SAMEA8944008  227322   \n",
       "3          maxbin_SP106_3.024    ERS6626982  SAMEA8944146  278208   \n",
       "4  maxbin_SP108_3.007_cleaned    ERS6626962  SAMEA8944126  797541   \n",
       "\n",
       "                   Scientific_name      Sequencing_method Assembly_Software  \\\n",
       "0     uncultured Agrobacterium sp.  Illumina NovaSeq 6000           Megahit   \n",
       "1         uncultured Spirosoma sp.  Illumina NovaSeq 6000           Megahit   \n",
       "2     uncultured Paenibacillus sp.  Illumina NovaSeq 6000           Megahit   \n",
       "3         uncultured Spirosoma sp.  Illumina NovaSeq 6000           Megahit   \n",
       "4  uncultured Mucilaginibacter sp.  Illumina NovaSeq 6000           Megahit   \n",
       "\n",
       "   Completeness_score Completeness_software  Contamination_score  \\\n",
       "0                  72                CheckM                    6   \n",
       "1                  72                CheckM                    1   \n",
       "2                 100                CheckM                    1   \n",
       "3                  71                CheckM                    2   \n",
       "4                  63                CheckM                    7   \n",
       "\n",
       "  Binning_software           Investigation_type Binning_parameters  \n",
       "0           maxbin  metagenome-assembled genome            default  \n",
       "1           maxbin  metagenome-assembled genome            default  \n",
       "2           maxbin  metagenome-assembled genome            default  \n",
       "3           maxbin  metagenome-assembled genome            default  \n",
       "4           maxbin  metagenome-assembled genome            default  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking the updated .csv file\n",
    "mag_df2 = open_file(output_file)\n",
    "mag_df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "c81381fb-a08e-4b3a-8190-2513d3eb1093",
   "metadata": {
    "panel-layout": {
     "height": 0,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All matching files have been renamed\n"
     ]
    }
   ],
   "source": [
    "# Renaming and copying the renamed MAG files mags folder\n",
    "for index, row in mag_df2.iterrows():\n",
    "    new_name = f\"{row['Scientific_name']}_{row['SRA_accession']}.fasta\".replace(' ', '_') # renaming with organism name & accession number and replacing spaces in the file name with '_'\n",
    "    old_name = row['file_name']\n",
    "    old_path = os.path.join(unprocessed_mag, old_name)\n",
    "    new_path = os.path.join(mags, new_name)\n",
    "    \n",
    "    if os.path.exists(old_path):\n",
    "        shutil.copy2(old_path, new_path) # Copying to mags folder\n",
    "    else:\n",
    "        print(f\"File {old_path} does not exist\") # Checking for non available file\n",
    "\n",
    "print(\"All matching files have been renamed\") # Confirmation for all the matched files renaming"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "a2f5bf71-bf6e-40ae-b53d-423a5473d85c",
   "metadata": {
    "panel-layout": {
     "height": 9799.25,
     "visible": true,
     "width": 100
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Bdellovibrionaceae_bacterium_ERS6626579.fasta',\n",
       " 'Bdellovibrionaceae_bacterium_ERS6626826.fasta',\n",
       " 'Burkholderiaceae_bacterium_ERS6626828.fasta',\n",
       " 'Burkholderiaceae_bacterium_ERS6626861.fasta',\n",
       " 'Candidatus_Sericytochromatia_bacterium_ERS6626968.fasta',\n",
       " 'Chitinophagaceae_bacterium_ERS6626975.fasta',\n",
       " 'Mycoplasmatales_bacterium_ERS6626571.fasta',\n",
       " 'Patescibacteria_group_bacterium_ERS6626570.fasta',\n",
       " 'Patescibacteria_group_bacterium_ERS6626572.fasta',\n",
       " 'Patescibacteria_group_bacterium_ERS6626573.fasta',\n",
       " 'Patescibacteria_group_bacterium_ERS6626790.fasta',\n",
       " 'Rhizobiaceae_bacterium_ERS6626796.fasta',\n",
       " 'uncultured_Achromobacter_sp._ERS6626913.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626561.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626565.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626576.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626591.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626592.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626596.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626599.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626607.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626644.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626705.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626715.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626735.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626737.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626738.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626787.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626798.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626819.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626824.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626827.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626841.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626883.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626889.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626918.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626955.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626991.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626997.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6626999.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6627009.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6627016.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6627256.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6627271.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6627284.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6627297.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626564.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626567.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626575.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626593.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626600.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626616.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626617.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626625.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626631.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626637.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626640.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626647.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626656.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626674.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626679.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626684.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626685.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626690.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626701.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626712.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626717.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626718.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626739.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626740.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626745.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626746.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626750.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626751.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626753.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626754.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626761.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626777.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626793.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626800.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626816.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626821.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626825.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626830.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626858.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626912.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626926.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626945.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6627024.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6627027.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6627286.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6627310.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626610.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626648.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626649.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626666.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626670.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626766.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626792.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626805.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626810.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626818.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626838.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626868.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626908.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626910.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626932.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626956.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626958.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626959.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626971.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626977.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626984.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6627249.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6627262.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6627268.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6627285.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6627329.fasta',\n",
       " 'uncultured_Aquabacterium_sp._ERS6627305.fasta',\n",
       " 'uncultured_Aureimonas_sp._ERS6626915.fasta',\n",
       " 'uncultured_Aureimonas_sp._ERS6627306.fasta',\n",
       " 'uncultured_Bacillus_sp._ERS6626947.fasta',\n",
       " 'uncultured_Blastomonas_sp._ERS6626578.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626598.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626654.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626699.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626706.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626728.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626729.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626731.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626732.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626807.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626847.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626880.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626900.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626922.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626970.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626974.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6626980.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6627276.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6627304.fasta',\n",
       " 'uncultured_Brevundimonas_sp._ERS6627341.fasta',\n",
       " 'uncultured_Buchnera_sp._ERS6626646.fasta',\n",
       " 'uncultured_Buchnera_sp._ERS6627326.fasta',\n",
       " 'uncultured_Caulobacter_sp._ERS6626620.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626581.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626590.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626605.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626608.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626609.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626615.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626618.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626662.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626671.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626678.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626682.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626693.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626695.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626697.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626719.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626734.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626736.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626741.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626742.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626743.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626744.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626748.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626749.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626755.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626756.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626758.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626760.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626770.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626776.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626817.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626823.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626849.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626863.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626898.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626903.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626907.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626928.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626933.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626967.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6626996.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6627248.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6627267.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6627296.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6627312.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6627318.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6627325.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6627338.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6627342.fasta',\n",
       " 'uncultured_Chryseobacterium_sp._ERS6627343.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626602.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626630.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626638.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626645.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626650.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626814.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626987.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6627283.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6627300.fasta',\n",
       " 'uncultured_Cupriavidus_sp._ERS6626993.fasta',\n",
       " 'uncultured_Curtobacterium_sp._ERS6626562.fasta',\n",
       " 'uncultured_Curtobacterium_sp._ERS6626839.fasta',\n",
       " 'uncultured_Curtobacterium_sp._ERS6626862.fasta',\n",
       " 'uncultured_Curtobacterium_sp._ERS6626869.fasta',\n",
       " 'uncultured_Curtobacterium_sp._ERS6626875.fasta',\n",
       " 'uncultured_Curtobacterium_sp._ERS6626878.fasta',\n",
       " 'uncultured_Curtobacterium_sp._ERS6626884.fasta',\n",
       " 'uncultured_Curtobacterium_sp._ERS6626897.fasta',\n",
       " 'uncultured_Curtobacterium_sp._ERS6626914.fasta',\n",
       " 'uncultured_Curtobacterium_sp._ERS6627269.fasta',\n",
       " 'uncultured_Curtobacterium_sp._ERS6627303.fasta',\n",
       " 'uncultured_Curtobacterium_sp._ERS6627307.fasta',\n",
       " 'uncultured_Cystobacter_sp._ERS6626771.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6626797.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6626811.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6626820.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6626837.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6627282.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6627293.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6627335.fasta',\n",
       " 'uncultured_Devosia_sp._ERS6626765.fasta',\n",
       " 'uncultured_Devosia_sp._ERS6626961.fasta',\n",
       " 'uncultured_Edaphocola_sp._ERS6626813.fasta',\n",
       " 'uncultured_Empedobacter_sp._ERS6627289.fasta',\n",
       " 'uncultured_Enterobacter_sp._ERS6626583.fasta',\n",
       " 'uncultured_Enterobacter_sp._ERS6626619.fasta',\n",
       " 'uncultured_Enterobacter_sp._ERS6626627.fasta',\n",
       " 'uncultured_Enterobacter_sp._ERS6626629.fasta',\n",
       " 'uncultured_Enterobacter_sp._ERS6626660.fasta',\n",
       " 'uncultured_Enterobacter_sp._ERS6626681.fasta',\n",
       " 'uncultured_Enterobacter_sp._ERS6626683.fasta',\n",
       " 'uncultured_Enterobacter_sp._ERS6626840.fasta',\n",
       " 'uncultured_Enterobacter_sp._ERS6626937.fasta',\n",
       " 'uncultured_Enterobacter_sp._ERS6626992.fasta',\n",
       " 'uncultured_Enterococcus_sp._ERS6626994.fasta',\n",
       " 'uncultured_Enterococcus_sp._ERS6627298.fasta',\n",
       " 'uncultured_Epilithonimonas_sp._ERS6626655.fasta',\n",
       " 'uncultured_Epilithonimonas_sp._ERS6626757.fasta',\n",
       " 'uncultured_Epilithonimonas_sp._ERS6626874.fasta',\n",
       " 'uncultured_Epilithonimonas_sp._ERS6627252.fasta',\n",
       " 'uncultured_Erwinia_sp._ERS6626634.fasta',\n",
       " 'uncultured_Erwinia_sp._ERS6626641.fasta',\n",
       " 'uncultured_Exiguobacterium_sp._ERS6626606.fasta',\n",
       " 'uncultured_Exiguobacterium_sp._ERS6626707.fasta',\n",
       " 'uncultured_Exiguobacterium_sp._ERS6626708.fasta',\n",
       " 'uncultured_Exiguobacterium_sp._ERS6626709.fasta',\n",
       " 'uncultured_Exiguobacterium_sp._ERS6626773.fasta',\n",
       " 'uncultured_Exiguobacterium_sp._ERS6626812.fasta',\n",
       " 'uncultured_Exiguobacterium_sp._ERS6626877.fasta',\n",
       " 'uncultured_Exiguobacterium_sp._ERS6626879.fasta',\n",
       " 'uncultured_Exiguobacterium_sp._ERS6626986.fasta',\n",
       " 'uncultured_Exiguobacterium_sp._ERS6627301.fasta',\n",
       " 'uncultured_Exiguobacterium_sp._ERS6627327.fasta',\n",
       " 'uncultured_Flavobacterium_sp._ERS6626763.fasta',\n",
       " 'uncultured_Flavobacterium_sp._ERS6626951.fasta',\n",
       " 'uncultured_Flavobacterium_sp._ERS6627313.fasta',\n",
       " 'uncultured_Herbaspirillum_sp._ERS6626801.fasta',\n",
       " 'uncultured_Herbaspirillum_sp._ERS6627023.fasta',\n",
       " 'uncultured_Herbaspirillum_sp._ERS6627309.fasta',\n",
       " 'uncultured_Herbaspirillum_sp._ERS6627324.fasta',\n",
       " 'uncultured_Hymenobacter_sp._ERS6626716.fasta',\n",
       " 'uncultured_Hymenobacter_sp._ERS6626733.fasta',\n",
       " 'uncultured_Hymenobacter_sp._ERS6626935.fasta',\n",
       " 'uncultured_Hymenobacter_sp._ERS6627279.fasta',\n",
       " 'uncultured_Klebsiella_sp._ERS6626601.fasta',\n",
       " 'uncultured_Klebsiella_sp._ERS6626611.fasta',\n",
       " 'uncultured_Klebsiella_sp._ERS6626628.fasta',\n",
       " 'uncultured_Leucobacter_sp._ERS6627345.fasta',\n",
       " 'uncultured_Luteibacter_sp._ERS6626809.fasta',\n",
       " 'uncultured_Massilia_sp._ERS6626633.fasta',\n",
       " 'uncultured_Massilia_sp._ERS6626788.fasta',\n",
       " 'uncultured_Massilia_sp._ERS6626950.fasta',\n",
       " 'uncultured_Massilia_sp._ERS6626988.fasta',\n",
       " 'uncultured_Massilia_sp._ERS6627272.fasta',\n",
       " 'uncultured_Methylobacterium_sp._ERS6627288.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626603.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626604.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626665.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626704.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626721.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626781.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626806.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626888.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626906.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626925.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626964.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6627018.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6627019.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6627260.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6627264.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6627311.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6627330.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6627334.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626632.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626762.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626764.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626791.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626799.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626803.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626873.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626894.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626923.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626931.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626962.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626972.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626981.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6627266.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6627275.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6627290.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6627314.fasta',\n",
       " 'uncultured_Myroides_sp._ERS6626694.fasta',\n",
       " 'uncultured_Naasia_sp._ERS6626864.fasta',\n",
       " 'uncultured_Naasia_sp._ERS6627002.fasta',\n",
       " 'uncultured_Naasia_sp._ERS6627012.fasta',\n",
       " 'uncultured_Naasia_sp._ERS6627270.fasta',\n",
       " 'uncultured_Naasia_sp._ERS6627292.fasta',\n",
       " 'uncultured_Naasia_sp._ERS6627328.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626759.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626775.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626778.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626784.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626786.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626794.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626802.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626829.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626852.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626853.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626854.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626865.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626886.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626887.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626893.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626895.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626901.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626917.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626929.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626942.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626965.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626979.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6626989.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6627247.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6627291.fasta',\n",
       " 'uncultured_Novosphingobium_sp._ERS6627336.fasta',\n",
       " 'uncultured_Ochrobactrum_sp._ERS6626652.fasta',\n",
       " 'uncultured_Ochrobactrum_sp._ERS6627020.fasta',\n",
       " 'uncultured_Oxalicibacterium_sp._ERS6626668.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6626585.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6626614.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6626696.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6626698.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6626700.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6626702.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6626703.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6626842.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6626843.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6626844.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6626855.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6626896.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6626905.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6627253.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6627323.fasta',\n",
       " 'uncultured_Paenibacillus_sp._ERS6627333.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6626568.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6626595.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6626623.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6626643.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6626657.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6626661.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6626664.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6626752.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6626902.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6626920.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6626969.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6627004.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6627005.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6627022.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6627332.fasta',\n",
       " 'uncultured_Pedobacter_sp._ERS6626651.fasta',\n",
       " 'uncultured_Pedobacter_sp._ERS6626667.fasta',\n",
       " 'uncultured_Pedobacter_sp._ERS6626919.fasta',\n",
       " 'uncultured_Pedobacter_sp._ERS6626930.fasta',\n",
       " 'uncultured_Pedobacter_sp._ERS6626944.fasta',\n",
       " 'uncultured_Pelomonas_sp._ERS6626569.fasta',\n",
       " 'uncultured_Pelomonas_sp._ERS6626597.fasta',\n",
       " 'uncultured_Pluralibacter_sp._ERS6626635.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626577.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626767.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626850.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626909.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627003.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627320.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627321.fasta',\n",
       " 'uncultured_Pseudocitrobacter_sp._ERS6626990.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626560.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626563.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626566.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626621.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626626.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626639.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626663.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626673.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626675.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626676.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626710.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626713.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626795.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626822.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626882.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626885.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626921.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626934.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626952.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626953.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626976.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626978.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627007.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627014.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627015.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627025.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627026.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627245.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627251.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627265.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627273.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627274.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627287.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627294.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627295.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627315.fasta',\n",
       " 'uncultured_Pseudorhodoferax_sp._ERS6626941.fasta',\n",
       " 'uncultured_Roseateles_sp._ERS6626963.fasta',\n",
       " 'uncultured_Roseateles_sp._ERS6626966.fasta',\n",
       " 'uncultured_Serratia_sp._ERS6626624.fasta',\n",
       " 'uncultured_Serratia_sp._ERS6626636.fasta',\n",
       " 'uncultured_Serratia_sp._ERS6627010.fasta',\n",
       " 'uncultured_Siphonobacter_sp._ERS6626689.fasta',\n",
       " 'uncultured_Siphonobacter_sp._ERS6626714.fasta',\n",
       " 'uncultured_Sphingobacterium_sp._ERS6626677.fasta',\n",
       " 'uncultured_Sphingobacterium_sp._ERS6626711.fasta',\n",
       " 'uncultured_Sphingobacterium_sp._ERS6626722.fasta',\n",
       " 'uncultured_Sphingobacterium_sp._ERS6626789.fasta',\n",
       " 'uncultured_Sphingobacterium_sp._ERS6626939.fasta',\n",
       " 'uncultured_Sphingobium_sp._ERS6626960.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626574.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626580.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626582.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626584.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626586.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626587.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626588.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626589.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626594.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626612.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626622.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626653.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626658.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626669.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626672.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626680.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626686.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626692.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626720.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626768.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626769.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626774.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626780.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626783.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626785.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626831.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626833.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626834.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626835.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626845.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626851.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626856.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626857.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626860.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626866.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626867.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626870.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626871.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626872.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626876.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626881.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626899.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626904.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626911.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626916.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626927.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626936.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626938.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626946.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626948.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626949.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626954.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626957.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626973.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626998.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627000.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627006.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627008.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627013.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627017.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627255.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627257.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627258.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627259.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627261.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627263.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627280.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627281.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627299.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627316.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627317.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6626747.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6626772.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6626782.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6626804.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6626815.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6626890.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6626891.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6626892.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6626982.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6626983.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6626985.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6626995.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6627011.fasta',\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6626659.fasta',\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6626688.fasta',\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6626723.fasta',\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6626727.fasta',\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6627250.fasta',\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6627254.fasta',\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6627308.fasta',\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6627319.fasta',\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6627337.fasta',\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6627344.fasta',\n",
       " 'uncultured_Vagococcus_sp._ERS6626691.fasta',\n",
       " 'uncultured_Variovorax_sp._ERS6626779.fasta',\n",
       " 'uncultured_Variovorax_sp._ERS6627246.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626613.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626642.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626687.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626724.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626725.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626726.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626730.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626808.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626832.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626836.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626846.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626848.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626859.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626924.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6626943.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6627001.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6627021.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6627277.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6627278.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6627322.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6627331.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6627339.fasta',\n",
       " 'uncultured_Xanthomonas_sp._ERS6627340.fasta',\n",
       " 'uncultured_Xylophilus_sp._ERS6626940.fasta',\n",
       " 'uncultured_Xylophilus_sp._ERS6627302.fasta']"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# List the mags folder files to cross check\n",
    "list_file(mags)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "25499ba8-bec5-4f3f-ade6-ab8455ac2ea1",
   "metadata": {
    "panel-layout": {
     "height": 27.135417938232422,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "569"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Cross checking the number of files to verify the 569 MAGs renamed and copied to mags folder\n",
    "len(list_file(mags))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e1ed1dce-8cca-4cc4-b6b8-f39495626902",
   "metadata": {
    "panel-layout": {
     "height": 61.427085876464844,
     "visible": true,
     "width": 100
    }
   },
   "source": [
    "<p style=\"font-family: 'Times New Roman'; font-size: 15px; text-align: justify; width: 100%;\">\n",
    "    The MAGs name are renamed and saved to new folder. The MAG are named after the organism and the SRA accession number for better understanding in the further analysis.\n",
    "</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "868d1bec-28f0-4325-b8ee-ef8965536095",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>SRA_accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Tax_ID</th>\n",
       "      <th>Scientific_name</th>\n",
       "      <th>Sequencing_method</th>\n",
       "      <th>Assembly_Software</th>\n",
       "      <th>Completeness_score</th>\n",
       "      <th>Completeness_software</th>\n",
       "      <th>Contamination_score</th>\n",
       "      <th>Binning_software</th>\n",
       "      <th>Investigation_type</th>\n",
       "      <th>Binning_parameters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>maxbin_SP106_1.016_cleaned.fasta</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned</td>\n",
       "      <td>ERS6626984</td>\n",
       "      <td>SAMEA8944148</td>\n",
       "      <td>157277</td>\n",
       "      <td>uncultured Agrobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>72</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>6</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>maxbin_SP108_3.007_cleaned.fasta</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned</td>\n",
       "      <td>ERS6626962</td>\n",
       "      <td>SAMEA8944126</td>\n",
       "      <td>797541</td>\n",
       "      <td>uncultured Mucilaginibacter sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>63</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>7</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>maxbin_SP135_3.009_sub_cleaned.fasta</td>\n",
       "      <td>maxbin_SP135_3.009_sub_cleaned</td>\n",
       "      <td>maxbin_SP135_3.009_sub_cleaned</td>\n",
       "      <td>ERS6626874</td>\n",
       "      <td>SAMEA8944038</td>\n",
       "      <td>259322</td>\n",
       "      <td>uncultured Epilithonimonas sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>65</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>7</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>maxbin_SP264_3.001.fasta</td>\n",
       "      <td>maxbin_SP264_3.001</td>\n",
       "      <td>maxbin_SP264_3.001</td>\n",
       "      <td>ERS6627327</td>\n",
       "      <td>SAMEA8944550</td>\n",
       "      <td>202669</td>\n",
       "      <td>uncultured Exiguobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>89</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>6</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>maxbin_SP266_1.010.fasta</td>\n",
       "      <td>maxbin_SP266_1.010</td>\n",
       "      <td>maxbin_SP266_1.010</td>\n",
       "      <td>ERS6626972</td>\n",
       "      <td>SAMEA8944136</td>\n",
       "      <td>797541</td>\n",
       "      <td>uncultured Mucilaginibacter sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>66</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>7</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>546</th>\n",
       "      <td>SP77_3_metabat2_genome_mining.18_sub.fasta</td>\n",
       "      <td>SP77_3_metabat2_genome_mining.18_sub</td>\n",
       "      <td>SP77_3_metabat2_genome_mining.18_sub</td>\n",
       "      <td>ERS6626870</td>\n",
       "      <td>SAMEA8944034</td>\n",
       "      <td>158754</td>\n",
       "      <td>uncultured Sphingomonas sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>51</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>10</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>547</th>\n",
       "      <td>SP77_3_metabat2_genome_mining.28.fasta</td>\n",
       "      <td>SP77_3_metabat2_genome_mining.28</td>\n",
       "      <td>SP77_3_metabat2_genome_mining.28</td>\n",
       "      <td>ERS6626938</td>\n",
       "      <td>SAMEA8944102</td>\n",
       "      <td>158754</td>\n",
       "      <td>uncultured Sphingomonas sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>58</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>9</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>553</th>\n",
       "      <td>SP80_2_metabat2_genome_mining.9.fasta</td>\n",
       "      <td>SP80_2_metabat2_genome_mining.9</td>\n",
       "      <td>SP80_2_metabat2_genome_mining.9</td>\n",
       "      <td>ERS6626981</td>\n",
       "      <td>SAMEA8944145</td>\n",
       "      <td>797541</td>\n",
       "      <td>uncultured Mucilaginibacter sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>71</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>8</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>556</th>\n",
       "      <td>SP86_3_metabat2_genome_mining.5.fasta</td>\n",
       "      <td>SP86_3_metabat2_genome_mining.5</td>\n",
       "      <td>SP86_3_metabat2_genome_mining.5</td>\n",
       "      <td>ERS6627282</td>\n",
       "      <td>SAMEA8944505</td>\n",
       "      <td>158789</td>\n",
       "      <td>uncultured Deinococcus sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>80</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>9</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>560</th>\n",
       "      <td>SP92_2_metabat2_genome_mining.23.fasta</td>\n",
       "      <td>SP92_2_metabat2_genome_mining.23</td>\n",
       "      <td>SP92_2_metabat2_genome_mining.23</td>\n",
       "      <td>ERS6626869</td>\n",
       "      <td>SAMEA8944033</td>\n",
       "      <td>331964</td>\n",
       "      <td>uncultured Curtobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>54</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>8</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>71 rows  15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      file_name  \\\n",
       "0              maxbin_SP106_1.016_cleaned.fasta   \n",
       "4              maxbin_SP108_3.007_cleaned.fasta   \n",
       "19         maxbin_SP135_3.009_sub_cleaned.fasta   \n",
       "49                     maxbin_SP264_3.001.fasta   \n",
       "51                     maxbin_SP266_1.010.fasta   \n",
       "..                                          ...   \n",
       "546  SP77_3_metabat2_genome_mining.18_sub.fasta   \n",
       "547      SP77_3_metabat2_genome_mining.28.fasta   \n",
       "553       SP80_2_metabat2_genome_mining.9.fasta   \n",
       "556       SP86_3_metabat2_genome_mining.5.fasta   \n",
       "560      SP92_2_metabat2_genome_mining.23.fasta   \n",
       "\n",
       "                                       ID  \\\n",
       "0              maxbin_SP106_1.016_cleaned   \n",
       "4              maxbin_SP108_3.007_cleaned   \n",
       "19         maxbin_SP135_3.009_sub_cleaned   \n",
       "49                     maxbin_SP264_3.001   \n",
       "51                     maxbin_SP266_1.010   \n",
       "..                                    ...   \n",
       "546  SP77_3_metabat2_genome_mining.18_sub   \n",
       "547      SP77_3_metabat2_genome_mining.28   \n",
       "553       SP80_2_metabat2_genome_mining.9   \n",
       "556       SP86_3_metabat2_genome_mining.5   \n",
       "560      SP92_2_metabat2_genome_mining.23   \n",
       "\n",
       "                                Sample_ID SRA_accession     BioSample  Tax_ID  \\\n",
       "0              maxbin_SP106_1.016_cleaned    ERS6626984  SAMEA8944148  157277   \n",
       "4              maxbin_SP108_3.007_cleaned    ERS6626962  SAMEA8944126  797541   \n",
       "19         maxbin_SP135_3.009_sub_cleaned    ERS6626874  SAMEA8944038  259322   \n",
       "49                     maxbin_SP264_3.001    ERS6627327  SAMEA8944550  202669   \n",
       "51                     maxbin_SP266_1.010    ERS6626972  SAMEA8944136  797541   \n",
       "..                                    ...           ...           ...     ...   \n",
       "546  SP77_3_metabat2_genome_mining.18_sub    ERS6626870  SAMEA8944034  158754   \n",
       "547      SP77_3_metabat2_genome_mining.28    ERS6626938  SAMEA8944102  158754   \n",
       "553       SP80_2_metabat2_genome_mining.9    ERS6626981  SAMEA8944145  797541   \n",
       "556       SP86_3_metabat2_genome_mining.5    ERS6627282  SAMEA8944505  158789   \n",
       "560      SP92_2_metabat2_genome_mining.23    ERS6626869  SAMEA8944033  331964   \n",
       "\n",
       "                     Scientific_name      Sequencing_method Assembly_Software  \\\n",
       "0       uncultured Agrobacterium sp.  Illumina NovaSeq 6000           Megahit   \n",
       "4    uncultured Mucilaginibacter sp.  Illumina NovaSeq 6000           Megahit   \n",
       "19    uncultured Epilithonimonas sp.  Illumina NovaSeq 6000           Megahit   \n",
       "49    uncultured Exiguobacterium sp.  Illumina NovaSeq 6000           Megahit   \n",
       "51   uncultured Mucilaginibacter sp.  Illumina NovaSeq 6000           Megahit   \n",
       "..                               ...                    ...               ...   \n",
       "546      uncultured Sphingomonas sp.  Illumina NovaSeq 6000           Megahit   \n",
       "547      uncultured Sphingomonas sp.  Illumina NovaSeq 6000           Megahit   \n",
       "553  uncultured Mucilaginibacter sp.  Illumina NovaSeq 6000           Megahit   \n",
       "556       uncultured Deinococcus sp.  Illumina NovaSeq 6000           Megahit   \n",
       "560    uncultured Curtobacterium sp.  Illumina NovaSeq 6000           Megahit   \n",
       "\n",
       "     Completeness_score Completeness_software  Contamination_score  \\\n",
       "0                    72                CheckM                    6   \n",
       "4                    63                CheckM                    7   \n",
       "19                   65                CheckM                    7   \n",
       "49                   89                CheckM                    6   \n",
       "51                   66                CheckM                    7   \n",
       "..                  ...                   ...                  ...   \n",
       "546                  51                CheckM                   10   \n",
       "547                  58                CheckM                    9   \n",
       "553                  71                CheckM                    8   \n",
       "556                  80                CheckM                    9   \n",
       "560                  54                CheckM                    8   \n",
       "\n",
       "    Binning_software           Investigation_type Binning_parameters  \n",
       "0             maxbin  metagenome-assembled genome            default  \n",
       "4             maxbin  metagenome-assembled genome            default  \n",
       "19            maxbin  metagenome-assembled genome            default  \n",
       "49            maxbin  metagenome-assembled genome            default  \n",
       "51            maxbin  metagenome-assembled genome            default  \n",
       "..               ...                          ...                ...  \n",
       "546         metabat2  metagenome-assembled genome            default  \n",
       "547         metabat2  metagenome-assembled genome            default  \n",
       "553         metabat2  metagenome-assembled genome            default  \n",
       "556         metabat2  metagenome-assembled genome            default  \n",
       "560         metabat2  metagenome-assembled genome            default  \n",
       "\n",
       "[71 rows x 15 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mag_df2[mag_df2['Contamination_score']>5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "c3591106-4cba-4a3b-9511-4819d88bb78f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>SRA_accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Tax_ID</th>\n",
       "      <th>Scientific_name</th>\n",
       "      <th>Sequencing_method</th>\n",
       "      <th>Assembly_Software</th>\n",
       "      <th>Completeness_score</th>\n",
       "      <th>Completeness_software</th>\n",
       "      <th>Contamination_score</th>\n",
       "      <th>Binning_software</th>\n",
       "      <th>Investigation_type</th>\n",
       "      <th>Binning_parameters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>maxbin_SP106_1.023_cleaned.fasta</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "      <td>ERS6626983</td>\n",
       "      <td>SAMEA8944147</td>\n",
       "      <td>278208</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>72</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>maxbin_SP106_2.001.fasta</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "      <td>ERS6626844</td>\n",
       "      <td>SAMEA8944008</td>\n",
       "      <td>227322</td>\n",
       "      <td>uncultured Paenibacillus sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>100</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>maxbin_SP106_3.024.fasta</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "      <td>ERS6626982</td>\n",
       "      <td>SAMEA8944146</td>\n",
       "      <td>278208</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>71</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>2</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>maxbin_SP112_1.001_cleaned.fasta</td>\n",
       "      <td>maxbin_SP112_1.001_cleaned</td>\n",
       "      <td>maxbin_SP112_1.001_cleaned</td>\n",
       "      <td>ERS6626773</td>\n",
       "      <td>SAMEA8943937</td>\n",
       "      <td>202669</td>\n",
       "      <td>uncultured Exiguobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>94</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>maxbin_SP118_2.002.fasta</td>\n",
       "      <td>maxbin_SP118_2.002</td>\n",
       "      <td>maxbin_SP118_2.002</td>\n",
       "      <td>ERS6626928</td>\n",
       "      <td>SAMEA8944092</td>\n",
       "      <td>259322</td>\n",
       "      <td>uncultured Chryseobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>54</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>2</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>564</th>\n",
       "      <td>SP92_3_metabat2_genome_mining.5_sub.fasta</td>\n",
       "      <td>SP92_3_metabat2_genome_mining.5_sub</td>\n",
       "      <td>SP92_3_metabat2_genome_mining.5_sub</td>\n",
       "      <td>ERS6626796</td>\n",
       "      <td>SAMEA8943960</td>\n",
       "      <td>1913961</td>\n",
       "      <td>Rhizobiaceae bacterium</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>96</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>0</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>565</th>\n",
       "      <td>SP92_3_metabat2_genome_mining.9.fasta</td>\n",
       "      <td>SP92_3_metabat2_genome_mining.9</td>\n",
       "      <td>SP92_3_metabat2_genome_mining.9</td>\n",
       "      <td>ERS6626829</td>\n",
       "      <td>SAMEA8943993</td>\n",
       "      <td>292277</td>\n",
       "      <td>uncultured Novosphingobium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>99</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>0</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>566</th>\n",
       "      <td>SP93_2_metabat2_genome_mining.11.fasta</td>\n",
       "      <td>SP93_2_metabat2_genome_mining.11</td>\n",
       "      <td>SP93_2_metabat2_genome_mining.11</td>\n",
       "      <td>ERS6627002</td>\n",
       "      <td>SAMEA8944166</td>\n",
       "      <td>1805779</td>\n",
       "      <td>uncultured Naasia sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>78</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>0</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567</th>\n",
       "      <td>SP93_2_metabat2_genome_mining.8.fasta</td>\n",
       "      <td>SP93_2_metabat2_genome_mining.8</td>\n",
       "      <td>SP93_2_metabat2_genome_mining.8</td>\n",
       "      <td>ERS6626884</td>\n",
       "      <td>SAMEA8944048</td>\n",
       "      <td>331964</td>\n",
       "      <td>uncultured Curtobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>58</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>568</th>\n",
       "      <td>SP93_3_metabat2_genome_mining.5.fasta</td>\n",
       "      <td>SP93_3_metabat2_genome_mining.5</td>\n",
       "      <td>SP93_3_metabat2_genome_mining.5</td>\n",
       "      <td>ERS6626836</td>\n",
       "      <td>SAMEA8944000</td>\n",
       "      <td>152831</td>\n",
       "      <td>uncultured Xanthomonas sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>99</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>498 rows  15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     file_name  \\\n",
       "1             maxbin_SP106_1.023_cleaned.fasta   \n",
       "2                     maxbin_SP106_2.001.fasta   \n",
       "3                     maxbin_SP106_3.024.fasta   \n",
       "5             maxbin_SP112_1.001_cleaned.fasta   \n",
       "6                     maxbin_SP118_2.002.fasta   \n",
       "..                                         ...   \n",
       "564  SP92_3_metabat2_genome_mining.5_sub.fasta   \n",
       "565      SP92_3_metabat2_genome_mining.9.fasta   \n",
       "566     SP93_2_metabat2_genome_mining.11.fasta   \n",
       "567      SP93_2_metabat2_genome_mining.8.fasta   \n",
       "568      SP93_3_metabat2_genome_mining.5.fasta   \n",
       "\n",
       "                                      ID                            Sample_ID  \\\n",
       "1             maxbin_SP106_1.023_cleaned           maxbin_SP106_1.023_cleaned   \n",
       "2                     maxbin_SP106_2.001                   maxbin_SP106_2.001   \n",
       "3                     maxbin_SP106_3.024                   maxbin_SP106_3.024   \n",
       "5             maxbin_SP112_1.001_cleaned           maxbin_SP112_1.001_cleaned   \n",
       "6                     maxbin_SP118_2.002                   maxbin_SP118_2.002   \n",
       "..                                   ...                                  ...   \n",
       "564  SP92_3_metabat2_genome_mining.5_sub  SP92_3_metabat2_genome_mining.5_sub   \n",
       "565      SP92_3_metabat2_genome_mining.9      SP92_3_metabat2_genome_mining.9   \n",
       "566     SP93_2_metabat2_genome_mining.11     SP93_2_metabat2_genome_mining.11   \n",
       "567      SP93_2_metabat2_genome_mining.8      SP93_2_metabat2_genome_mining.8   \n",
       "568      SP93_3_metabat2_genome_mining.5      SP93_3_metabat2_genome_mining.5   \n",
       "\n",
       "    SRA_accession     BioSample   Tax_ID                  Scientific_name  \\\n",
       "1      ERS6626983  SAMEA8944147   278208         uncultured Spirosoma sp.   \n",
       "2      ERS6626844  SAMEA8944008   227322     uncultured Paenibacillus sp.   \n",
       "3      ERS6626982  SAMEA8944146   278208         uncultured Spirosoma sp.   \n",
       "5      ERS6626773  SAMEA8943937   202669   uncultured Exiguobacterium sp.   \n",
       "6      ERS6626928  SAMEA8944092   259322  uncultured Chryseobacterium sp.   \n",
       "..            ...           ...      ...                              ...   \n",
       "564    ERS6626796  SAMEA8943960  1913961           Rhizobiaceae bacterium   \n",
       "565    ERS6626829  SAMEA8943993   292277   uncultured Novosphingobium sp.   \n",
       "566    ERS6627002  SAMEA8944166  1805779            uncultured Naasia sp.   \n",
       "567    ERS6626884  SAMEA8944048   331964    uncultured Curtobacterium sp.   \n",
       "568    ERS6626836  SAMEA8944000   152831       uncultured Xanthomonas sp.   \n",
       "\n",
       "         Sequencing_method Assembly_Software  Completeness_score  \\\n",
       "1    Illumina NovaSeq 6000           Megahit                  72   \n",
       "2    Illumina NovaSeq 6000           Megahit                 100   \n",
       "3    Illumina NovaSeq 6000           Megahit                  71   \n",
       "5    Illumina NovaSeq 6000           Megahit                  94   \n",
       "6    Illumina NovaSeq 6000           Megahit                  54   \n",
       "..                     ...               ...                 ...   \n",
       "564  Illumina NovaSeq 6000           Megahit                  96   \n",
       "565  Illumina NovaSeq 6000           Megahit                  99   \n",
       "566  Illumina NovaSeq 6000           Megahit                  78   \n",
       "567  Illumina NovaSeq 6000           Megahit                  58   \n",
       "568  Illumina NovaSeq 6000           Megahit                  99   \n",
       "\n",
       "    Completeness_software  Contamination_score Binning_software  \\\n",
       "1                  CheckM                    1           maxbin   \n",
       "2                  CheckM                    1           maxbin   \n",
       "3                  CheckM                    2           maxbin   \n",
       "5                  CheckM                    1           maxbin   \n",
       "6                  CheckM                    2           maxbin   \n",
       "..                    ...                  ...              ...   \n",
       "564                CheckM                    0         metabat2   \n",
       "565                CheckM                    0         metabat2   \n",
       "566                CheckM                    0         metabat2   \n",
       "567                CheckM                    1         metabat2   \n",
       "568                CheckM                    1         metabat2   \n",
       "\n",
       "              Investigation_type Binning_parameters  \n",
       "1    metagenome-assembled genome            default  \n",
       "2    metagenome-assembled genome            default  \n",
       "3    metagenome-assembled genome            default  \n",
       "5    metagenome-assembled genome            default  \n",
       "6    metagenome-assembled genome            default  \n",
       "..                           ...                ...  \n",
       "564  metagenome-assembled genome            default  \n",
       "565  metagenome-assembled genome            default  \n",
       "566  metagenome-assembled genome            default  \n",
       "567  metagenome-assembled genome            default  \n",
       "568  metagenome-assembled genome            default  \n",
       "\n",
       "[498 rows x 15 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_mag = mag_df2[mag_df2['Contamination_score']<=5]\n",
    "filtered_mag"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "aa1c7632-065a-48ec-907e-d7c2ab6f8b73",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\dkb1g24\\AppData\\Local\\Temp\\ipykernel_2644\\1719209096.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  filtered_mag['Genome'] = filtered_mag.apply(lambda row: f\"{row['Scientific_name']}_{row['SRA_accession']}\".replace(' ', '_'), axis=1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>SRA_accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Tax_ID</th>\n",
       "      <th>Scientific_name</th>\n",
       "      <th>Sequencing_method</th>\n",
       "      <th>Assembly_Software</th>\n",
       "      <th>Completeness_score</th>\n",
       "      <th>Completeness_software</th>\n",
       "      <th>Contamination_score</th>\n",
       "      <th>Binning_software</th>\n",
       "      <th>Investigation_type</th>\n",
       "      <th>Binning_parameters</th>\n",
       "      <th>Genome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>maxbin_SP106_1.023_cleaned.fasta</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "      <td>ERS6626983</td>\n",
       "      <td>SAMEA8944147</td>\n",
       "      <td>278208</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>72</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>maxbin_SP106_2.001.fasta</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "      <td>ERS6626844</td>\n",
       "      <td>SAMEA8944008</td>\n",
       "      <td>227322</td>\n",
       "      <td>uncultured Paenibacillus sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>100</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Paenibacillus_sp._ERS6626844</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>maxbin_SP106_3.024.fasta</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "      <td>ERS6626982</td>\n",
       "      <td>SAMEA8944146</td>\n",
       "      <td>278208</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>71</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>2</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>maxbin_SP112_1.001_cleaned.fasta</td>\n",
       "      <td>maxbin_SP112_1.001_cleaned</td>\n",
       "      <td>maxbin_SP112_1.001_cleaned</td>\n",
       "      <td>ERS6626773</td>\n",
       "      <td>SAMEA8943937</td>\n",
       "      <td>202669</td>\n",
       "      <td>uncultured Exiguobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>94</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Exiguobacterium_sp._ERS6626773</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>maxbin_SP118_2.002.fasta</td>\n",
       "      <td>maxbin_SP118_2.002</td>\n",
       "      <td>maxbin_SP118_2.002</td>\n",
       "      <td>ERS6626928</td>\n",
       "      <td>SAMEA8944092</td>\n",
       "      <td>259322</td>\n",
       "      <td>uncultured Chryseobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>54</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>2</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Chryseobacterium_sp._ERS6626928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>564</th>\n",
       "      <td>SP92_3_metabat2_genome_mining.5_sub.fasta</td>\n",
       "      <td>SP92_3_metabat2_genome_mining.5_sub</td>\n",
       "      <td>SP92_3_metabat2_genome_mining.5_sub</td>\n",
       "      <td>ERS6626796</td>\n",
       "      <td>SAMEA8943960</td>\n",
       "      <td>1913961</td>\n",
       "      <td>Rhizobiaceae bacterium</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>96</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>0</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>Rhizobiaceae_bacterium_ERS6626796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>565</th>\n",
       "      <td>SP92_3_metabat2_genome_mining.9.fasta</td>\n",
       "      <td>SP92_3_metabat2_genome_mining.9</td>\n",
       "      <td>SP92_3_metabat2_genome_mining.9</td>\n",
       "      <td>ERS6626829</td>\n",
       "      <td>SAMEA8943993</td>\n",
       "      <td>292277</td>\n",
       "      <td>uncultured Novosphingobium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>99</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>0</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Novosphingobium_sp._ERS6626829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>566</th>\n",
       "      <td>SP93_2_metabat2_genome_mining.11.fasta</td>\n",
       "      <td>SP93_2_metabat2_genome_mining.11</td>\n",
       "      <td>SP93_2_metabat2_genome_mining.11</td>\n",
       "      <td>ERS6627002</td>\n",
       "      <td>SAMEA8944166</td>\n",
       "      <td>1805779</td>\n",
       "      <td>uncultured Naasia sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>78</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>0</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Naasia_sp._ERS6627002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567</th>\n",
       "      <td>SP93_2_metabat2_genome_mining.8.fasta</td>\n",
       "      <td>SP93_2_metabat2_genome_mining.8</td>\n",
       "      <td>SP93_2_metabat2_genome_mining.8</td>\n",
       "      <td>ERS6626884</td>\n",
       "      <td>SAMEA8944048</td>\n",
       "      <td>331964</td>\n",
       "      <td>uncultured Curtobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>58</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Curtobacterium_sp._ERS6626884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>568</th>\n",
       "      <td>SP93_3_metabat2_genome_mining.5.fasta</td>\n",
       "      <td>SP93_3_metabat2_genome_mining.5</td>\n",
       "      <td>SP93_3_metabat2_genome_mining.5</td>\n",
       "      <td>ERS6626836</td>\n",
       "      <td>SAMEA8944000</td>\n",
       "      <td>152831</td>\n",
       "      <td>uncultured Xanthomonas sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>99</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Xanthomonas_sp._ERS6626836</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>498 rows  16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     file_name  \\\n",
       "1             maxbin_SP106_1.023_cleaned.fasta   \n",
       "2                     maxbin_SP106_2.001.fasta   \n",
       "3                     maxbin_SP106_3.024.fasta   \n",
       "5             maxbin_SP112_1.001_cleaned.fasta   \n",
       "6                     maxbin_SP118_2.002.fasta   \n",
       "..                                         ...   \n",
       "564  SP92_3_metabat2_genome_mining.5_sub.fasta   \n",
       "565      SP92_3_metabat2_genome_mining.9.fasta   \n",
       "566     SP93_2_metabat2_genome_mining.11.fasta   \n",
       "567      SP93_2_metabat2_genome_mining.8.fasta   \n",
       "568      SP93_3_metabat2_genome_mining.5.fasta   \n",
       "\n",
       "                                      ID                            Sample_ID  \\\n",
       "1             maxbin_SP106_1.023_cleaned           maxbin_SP106_1.023_cleaned   \n",
       "2                     maxbin_SP106_2.001                   maxbin_SP106_2.001   \n",
       "3                     maxbin_SP106_3.024                   maxbin_SP106_3.024   \n",
       "5             maxbin_SP112_1.001_cleaned           maxbin_SP112_1.001_cleaned   \n",
       "6                     maxbin_SP118_2.002                   maxbin_SP118_2.002   \n",
       "..                                   ...                                  ...   \n",
       "564  SP92_3_metabat2_genome_mining.5_sub  SP92_3_metabat2_genome_mining.5_sub   \n",
       "565      SP92_3_metabat2_genome_mining.9      SP92_3_metabat2_genome_mining.9   \n",
       "566     SP93_2_metabat2_genome_mining.11     SP93_2_metabat2_genome_mining.11   \n",
       "567      SP93_2_metabat2_genome_mining.8      SP93_2_metabat2_genome_mining.8   \n",
       "568      SP93_3_metabat2_genome_mining.5      SP93_3_metabat2_genome_mining.5   \n",
       "\n",
       "    SRA_accession     BioSample   Tax_ID                  Scientific_name  \\\n",
       "1      ERS6626983  SAMEA8944147   278208         uncultured Spirosoma sp.   \n",
       "2      ERS6626844  SAMEA8944008   227322     uncultured Paenibacillus sp.   \n",
       "3      ERS6626982  SAMEA8944146   278208         uncultured Spirosoma sp.   \n",
       "5      ERS6626773  SAMEA8943937   202669   uncultured Exiguobacterium sp.   \n",
       "6      ERS6626928  SAMEA8944092   259322  uncultured Chryseobacterium sp.   \n",
       "..            ...           ...      ...                              ...   \n",
       "564    ERS6626796  SAMEA8943960  1913961           Rhizobiaceae bacterium   \n",
       "565    ERS6626829  SAMEA8943993   292277   uncultured Novosphingobium sp.   \n",
       "566    ERS6627002  SAMEA8944166  1805779            uncultured Naasia sp.   \n",
       "567    ERS6626884  SAMEA8944048   331964    uncultured Curtobacterium sp.   \n",
       "568    ERS6626836  SAMEA8944000   152831       uncultured Xanthomonas sp.   \n",
       "\n",
       "         Sequencing_method Assembly_Software  Completeness_score  \\\n",
       "1    Illumina NovaSeq 6000           Megahit                  72   \n",
       "2    Illumina NovaSeq 6000           Megahit                 100   \n",
       "3    Illumina NovaSeq 6000           Megahit                  71   \n",
       "5    Illumina NovaSeq 6000           Megahit                  94   \n",
       "6    Illumina NovaSeq 6000           Megahit                  54   \n",
       "..                     ...               ...                 ...   \n",
       "564  Illumina NovaSeq 6000           Megahit                  96   \n",
       "565  Illumina NovaSeq 6000           Megahit                  99   \n",
       "566  Illumina NovaSeq 6000           Megahit                  78   \n",
       "567  Illumina NovaSeq 6000           Megahit                  58   \n",
       "568  Illumina NovaSeq 6000           Megahit                  99   \n",
       "\n",
       "    Completeness_software  Contamination_score Binning_software  \\\n",
       "1                  CheckM                    1           maxbin   \n",
       "2                  CheckM                    1           maxbin   \n",
       "3                  CheckM                    2           maxbin   \n",
       "5                  CheckM                    1           maxbin   \n",
       "6                  CheckM                    2           maxbin   \n",
       "..                    ...                  ...              ...   \n",
       "564                CheckM                    0         metabat2   \n",
       "565                CheckM                    0         metabat2   \n",
       "566                CheckM                    0         metabat2   \n",
       "567                CheckM                    1         metabat2   \n",
       "568                CheckM                    1         metabat2   \n",
       "\n",
       "              Investigation_type Binning_parameters  \\\n",
       "1    metagenome-assembled genome            default   \n",
       "2    metagenome-assembled genome            default   \n",
       "3    metagenome-assembled genome            default   \n",
       "5    metagenome-assembled genome            default   \n",
       "6    metagenome-assembled genome            default   \n",
       "..                           ...                ...   \n",
       "564  metagenome-assembled genome            default   \n",
       "565  metagenome-assembled genome            default   \n",
       "566  metagenome-assembled genome            default   \n",
       "567  metagenome-assembled genome            default   \n",
       "568  metagenome-assembled genome            default   \n",
       "\n",
       "                                         Genome  \n",
       "1           uncultured_Spirosoma_sp._ERS6626983  \n",
       "2       uncultured_Paenibacillus_sp._ERS6626844  \n",
       "3           uncultured_Spirosoma_sp._ERS6626982  \n",
       "5     uncultured_Exiguobacterium_sp._ERS6626773  \n",
       "6    uncultured_Chryseobacterium_sp._ERS6626928  \n",
       "..                                          ...  \n",
       "564           Rhizobiaceae_bacterium_ERS6626796  \n",
       "565   uncultured_Novosphingobium_sp._ERS6626829  \n",
       "566            uncultured_Naasia_sp._ERS6627002  \n",
       "567    uncultured_Curtobacterium_sp._ERS6626884  \n",
       "568       uncultured_Xanthomonas_sp._ERS6626836  \n",
       "\n",
       "[498 rows x 16 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_mag['Genome'] = filtered_mag.apply(lambda row: f\"{row['Scientific_name']}_{row['SRA_accession']}\".replace(' ', '_'), axis=1)\n",
    "filtered_mag"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "2510d8d4-f50e-40ea-acc8-7c04b200a755",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\dkb1g24\\AppData\\Local\\Temp\\ipykernel_2644\\3615412678.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  filtered_mag['Genome'] = filtered_mag['Genome'].str.replace('^uncultured_', '', regex=True)\n"
     ]
    }
   ],
   "source": [
    "filtered_mag['Genome'] = filtered_mag['Genome'].str.replace('^uncultured_', '', regex=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "b244a263-8ac3-4174-a0bf-89576ba1e9d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "fil_mag_file = os.path.join(raw_data, 'filtered_mag.csv')\n",
    "filtered_mag.to_csv(fil_mag_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "949d7431-ba08-4af2-ac06-ac9c3db13db9",
   "metadata": {
    "panel-layout": {
     "height": 50.8125,
     "visible": true,
     "width": 100
    }
   },
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\r\n",
    "   Create CoxL Protein Data Bases\r\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "22076f28-e8b5-4ce4-8cd1-e387167eff0a",
   "metadata": {
    "panel-layout": {
     "height": 0,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Form 1 proteins saved to C:/Users/dkb1g24/Documents/mres/Raw_Data\\raw_coxl1.fasta\n",
      " Form 2 proteins saved to C:/Users/dkb1g24/Documents/mres/Raw_Data\\raw_coxl2.fasta\n"
     ]
    }
   ],
   "source": [
    "# Separating form 2 proteins and form 1 proteins\n",
    "## Output file names\n",
    "raw_coxl1_file = os.path.join(raw_data, \"raw_coxl1.fasta\") # form 1 proteins\n",
    "raw_coxl2_file = os.path.join(raw_data, \"raw_coxl2.fasta\") # form 2 proteins\n",
    "\n",
    "## Read all seauences from the input file\n",
    "sequences = list(SeqIO.parse(coxl_fasta_path, \"fasta\"))\n",
    "\n",
    "## Split the sequences\n",
    "coxl1 = sequences[:-2]\n",
    "coxl2 = sequences[-2:] ## we know last two sequences are form 2  CoxL protein\n",
    "\n",
    "## Write last form 2 and form 1 CoxL proteins to separate files\n",
    "SeqIO.write(coxl1, raw_coxl1_file, \"fasta\") # Form 1 CoxL proteins\n",
    "SeqIO.write(coxl2, raw_coxl2_file, \"fasta\") # Form 2 CoxL proteins\n",
    "\n",
    "## For cross checking\n",
    "print(f\" Form 1 proteins saved to {raw_coxl1_file}\")\n",
    "print(f\" Form 2 proteins saved to {raw_coxl2_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "a24bd122-f0a3-4548-a4b2-26dda68b44bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "coxl1_file = os.path.join(raw_data, \"coxl1.fasta\") # form 1 proteins\n",
    "coxl2_file = os.path.join(raw_data, \"coxl2.fasta\")\n",
    "remove_gaps_from_fasta(raw_coxl1_file, coxl1_file)\n",
    "remove_gaps_from_fasta(raw_coxl2_file, coxl2_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "425a83c3-3858-4893-9351-c13216034af6",
   "metadata": {
    "panel-layout": {
     "height": 0,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Database created at C:/Users/dkb1g24/Documents/mres/BLAST/CoxL1\\coxl1_db\n",
      "Database created at C:/Users/dkb1g24/Documents/mres/BLAST/CoxL2\\coxl2_db\n"
     ]
    }
   ],
   "source": [
    "# Create databases\n",
    "create_db(coxl1_file, coxl1_db_folder, \"coxl1_db\")\n",
    "create_db(coxl2_file, coxl2_db_folder, \"coxl2_db\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e91d0938-9db2-47f3-a705-342bc0e9398f",
   "metadata": {
    "panel-layout": {
     "height": 50.8125,
     "visible": true,
     "width": 100
    }
   },
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\r\n",
    "   Prepare List to Upload MAGs to RAST Servers\r\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "babab27b-3371-49a9-bef1-d9e3d5814612",
   "metadata": {
    "panel-layout": {
     "height": 332.66668701171875,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>SRA_accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Tax_ID</th>\n",
       "      <th>Scientific_name</th>\n",
       "      <th>Sequencing_method</th>\n",
       "      <th>Assembly_Software</th>\n",
       "      <th>Completeness_score</th>\n",
       "      <th>Completeness_software</th>\n",
       "      <th>Contamination_score</th>\n",
       "      <th>Binning_software</th>\n",
       "      <th>Investigation_type</th>\n",
       "      <th>Binning_parameters</th>\n",
       "      <th>mag_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>maxbin_SP106_1.016_cleaned.fasta</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned</td>\n",
       "      <td>ERS6626984</td>\n",
       "      <td>SAMEA8944148</td>\n",
       "      <td>157277</td>\n",
       "      <td>uncultured Agrobacterium sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>72</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>6</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Agrobacterium_sp._ERS6626984.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>maxbin_SP106_1.023_cleaned.fasta</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "      <td>ERS6626983</td>\n",
       "      <td>SAMEA8944147</td>\n",
       "      <td>278208</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>72</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626983.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>maxbin_SP106_2.001.fasta</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "      <td>ERS6626844</td>\n",
       "      <td>SAMEA8944008</td>\n",
       "      <td>227322</td>\n",
       "      <td>uncultured Paenibacillus sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>100</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Paenibacillus_sp._ERS6626844.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>maxbin_SP106_3.024.fasta</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "      <td>ERS6626982</td>\n",
       "      <td>SAMEA8944146</td>\n",
       "      <td>278208</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>71</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>2</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626982.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>maxbin_SP108_3.007_cleaned.fasta</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned</td>\n",
       "      <td>ERS6626962</td>\n",
       "      <td>SAMEA8944126</td>\n",
       "      <td>797541</td>\n",
       "      <td>uncultured Mucilaginibacter sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>63</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>7</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626962.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          file_name                          ID  \\\n",
       "0  maxbin_SP106_1.016_cleaned.fasta  maxbin_SP106_1.016_cleaned   \n",
       "1  maxbin_SP106_1.023_cleaned.fasta  maxbin_SP106_1.023_cleaned   \n",
       "2          maxbin_SP106_2.001.fasta          maxbin_SP106_2.001   \n",
       "3          maxbin_SP106_3.024.fasta          maxbin_SP106_3.024   \n",
       "4  maxbin_SP108_3.007_cleaned.fasta  maxbin_SP108_3.007_cleaned   \n",
       "\n",
       "                    Sample_ID SRA_accession     BioSample  Tax_ID  \\\n",
       "0  maxbin_SP106_1.016_cleaned    ERS6626984  SAMEA8944148  157277   \n",
       "1  maxbin_SP106_1.023_cleaned    ERS6626983  SAMEA8944147  278208   \n",
       "2          maxbin_SP106_2.001    ERS6626844  SAMEA8944008  227322   \n",
       "3          maxbin_SP106_3.024    ERS6626982  SAMEA8944146  278208   \n",
       "4  maxbin_SP108_3.007_cleaned    ERS6626962  SAMEA8944126  797541   \n",
       "\n",
       "                   Scientific_name      Sequencing_method Assembly_Software  \\\n",
       "0     uncultured Agrobacterium sp.  Illumina NovaSeq 6000           Megahit   \n",
       "1         uncultured Spirosoma sp.  Illumina NovaSeq 6000           Megahit   \n",
       "2     uncultured Paenibacillus sp.  Illumina NovaSeq 6000           Megahit   \n",
       "3         uncultured Spirosoma sp.  Illumina NovaSeq 6000           Megahit   \n",
       "4  uncultured Mucilaginibacter sp.  Illumina NovaSeq 6000           Megahit   \n",
       "\n",
       "   Completeness_score Completeness_software  Contamination_score  \\\n",
       "0                  72                CheckM                    6   \n",
       "1                  72                CheckM                    1   \n",
       "2                 100                CheckM                    1   \n",
       "3                  71                CheckM                    2   \n",
       "4                  63                CheckM                    7   \n",
       "\n",
       "  Binning_software           Investigation_type Binning_parameters  \\\n",
       "0           maxbin  metagenome-assembled genome            default   \n",
       "1           maxbin  metagenome-assembled genome            default   \n",
       "2           maxbin  metagenome-assembled genome            default   \n",
       "3           maxbin  metagenome-assembled genome            default   \n",
       "4           maxbin  metagenome-assembled genome            default   \n",
       "\n",
       "                                           mag_file  \n",
       "0     uncultured_Agrobacterium_sp._ERS6626984.fasta  \n",
       "1         uncultured_Spirosoma_sp._ERS6626983.fasta  \n",
       "2     uncultured_Paenibacillus_sp._ERS6626844.fasta  \n",
       "3         uncultured_Spirosoma_sp._ERS6626982.fasta  \n",
       "4  uncultured_Mucilaginibacter_sp._ERS6626962.fasta  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Creating a column with mag file names\n",
    "mag_df2['mag_file'] = mag_df2['Scientific_name'].str.replace(' ', '_') + '_' + mag_df2['SRA_accession'].str.replace(' ', '_') + '.fasta'\n",
    "mag_df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "3142c484-0064-40e0-8f6d-da4b3e95552b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>SRA_accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Tax_ID</th>\n",
       "      <th>Scientific_name</th>\n",
       "      <th>Sequencing_method</th>\n",
       "      <th>Assembly_Software</th>\n",
       "      <th>Completeness_score</th>\n",
       "      <th>Completeness_software</th>\n",
       "      <th>Contamination_score</th>\n",
       "      <th>Binning_software</th>\n",
       "      <th>Investigation_type</th>\n",
       "      <th>Binning_parameters</th>\n",
       "      <th>mag_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>537</th>\n",
       "      <td>SP67_2_metabat2_genome_mining.16.fasta</td>\n",
       "      <td>SP67_2_metabat2_genome_mining.16</td>\n",
       "      <td>SP67_2_metabat2_genome_mining.16</td>\n",
       "      <td>ERS6626837</td>\n",
       "      <td>SAMEA8944001</td>\n",
       "      <td>158789</td>\n",
       "      <td>uncultured Deinococcus sp.</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>99</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>1</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626837.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  file_name                                ID  \\\n",
       "537  SP67_2_metabat2_genome_mining.16.fasta  SP67_2_metabat2_genome_mining.16   \n",
       "\n",
       "                            Sample_ID SRA_accession     BioSample  Tax_ID  \\\n",
       "537  SP67_2_metabat2_genome_mining.16    ERS6626837  SAMEA8944001  158789   \n",
       "\n",
       "                Scientific_name      Sequencing_method Assembly_Software  \\\n",
       "537  uncultured Deinococcus sp.  Illumina NovaSeq 6000           Megahit   \n",
       "\n",
       "     Completeness_score Completeness_software  Contamination_score  \\\n",
       "537                  99                CheckM                    1   \n",
       "\n",
       "    Binning_software           Investigation_type Binning_parameters  \\\n",
       "537         metabat2  metagenome-assembled genome            default   \n",
       "\n",
       "                                        mag_file  \n",
       "537  uncultured_Deinococcus_sp._ERS6626837.fasta  "
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mag_df2[mag_df2['SRA_accession'] == 'ERS6626837']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "012a7e2e-7be7-4fd1-9c2c-5a02576471ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>SRA_accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Tax_ID</th>\n",
       "      <th>Scientific_name</th>\n",
       "      <th>Sequencing_method</th>\n",
       "      <th>Assembly_Software</th>\n",
       "      <th>Completeness_score</th>\n",
       "      <th>Completeness_software</th>\n",
       "      <th>Contamination_score</th>\n",
       "      <th>Binning_software</th>\n",
       "      <th>Investigation_type</th>\n",
       "      <th>Binning_parameters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>SP213_2_metabat2_genome_mining.4</td>\n",
       "      <td>ERS6626861</td>\n",
       "      <td>SAMEA8944025</td>\n",
       "      <td>2030806</td>\n",
       "      <td>Burkholderiaceae bacterium</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>72</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>2</td>\n",
       "      <td>metabat2</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Sample_ID SRA_accession     BioSample   Tax_ID  \\\n",
       "301  SP213_2_metabat2_genome_mining.4    ERS6626861  SAMEA8944025  2030806   \n",
       "\n",
       "                Scientific_name      Sequencing_method Assembly_Software  \\\n",
       "301  Burkholderiaceae bacterium  Illumina NovaSeq 6000           Megahit   \n",
       "\n",
       "     Completeness_score Completeness_software  Contamination_score  \\\n",
       "301                  72                CheckM                    2   \n",
       "\n",
       "    Binning_software           Investigation_type Binning_parameters  \n",
       "301         metabat2  metagenome-assembled genome            default  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mag_df[mag_df['SRA_accession'] == 'ERS6626861']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "6d9d53b0-de2c-40f5-be9f-1899305d1e64",
   "metadata": {
    "panel-layout": {
     "height": 332.66668701171875,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mag_file</th>\n",
       "      <th>Scientific_name</th>\n",
       "      <th>SRA_accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Tax_ID</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>Completeness_score</th>\n",
       "      <th>Contamination_score</th>\n",
       "      <th>Sequencing_method</th>\n",
       "      <th>Assembly_Software</th>\n",
       "      <th>Completeness_software</th>\n",
       "      <th>Binning_software</th>\n",
       "      <th>Investigation_type</th>\n",
       "      <th>Binning_parameters</th>\n",
       "      <th>file_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uncultured_Agrobacterium_sp._ERS6626984.fasta</td>\n",
       "      <td>uncultured Agrobacterium sp.</td>\n",
       "      <td>ERS6626984</td>\n",
       "      <td>SAMEA8944148</td>\n",
       "      <td>157277</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned</td>\n",
       "      <td>72</td>\n",
       "      <td>6</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626983.fasta</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>ERS6626983</td>\n",
       "      <td>SAMEA8944147</td>\n",
       "      <td>278208</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "      <td>72</td>\n",
       "      <td>1</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured_Paenibacillus_sp._ERS6626844.fasta</td>\n",
       "      <td>uncultured Paenibacillus sp.</td>\n",
       "      <td>ERS6626844</td>\n",
       "      <td>SAMEA8944008</td>\n",
       "      <td>227322</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "      <td>100</td>\n",
       "      <td>1</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>maxbin_SP106_2.001.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626982.fasta</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>ERS6626982</td>\n",
       "      <td>SAMEA8944146</td>\n",
       "      <td>278208</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "      <td>71</td>\n",
       "      <td>2</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>maxbin_SP106_3.024.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626962.fasta</td>\n",
       "      <td>uncultured Mucilaginibacter sp.</td>\n",
       "      <td>ERS6626962</td>\n",
       "      <td>SAMEA8944126</td>\n",
       "      <td>797541</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned</td>\n",
       "      <td>63</td>\n",
       "      <td>7</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           mag_file  \\\n",
       "0     uncultured_Agrobacterium_sp._ERS6626984.fasta   \n",
       "1         uncultured_Spirosoma_sp._ERS6626983.fasta   \n",
       "2     uncultured_Paenibacillus_sp._ERS6626844.fasta   \n",
       "3         uncultured_Spirosoma_sp._ERS6626982.fasta   \n",
       "4  uncultured_Mucilaginibacter_sp._ERS6626962.fasta   \n",
       "\n",
       "                   Scientific_name SRA_accession     BioSample  Tax_ID  \\\n",
       "0     uncultured Agrobacterium sp.    ERS6626984  SAMEA8944148  157277   \n",
       "1         uncultured Spirosoma sp.    ERS6626983  SAMEA8944147  278208   \n",
       "2     uncultured Paenibacillus sp.    ERS6626844  SAMEA8944008  227322   \n",
       "3         uncultured Spirosoma sp.    ERS6626982  SAMEA8944146  278208   \n",
       "4  uncultured Mucilaginibacter sp.    ERS6626962  SAMEA8944126  797541   \n",
       "\n",
       "                    Sample_ID  Completeness_score  Contamination_score  \\\n",
       "0  maxbin_SP106_1.016_cleaned                  72                    6   \n",
       "1  maxbin_SP106_1.023_cleaned                  72                    1   \n",
       "2          maxbin_SP106_2.001                 100                    1   \n",
       "3          maxbin_SP106_3.024                  71                    2   \n",
       "4  maxbin_SP108_3.007_cleaned                  63                    7   \n",
       "\n",
       "       Sequencing_method Assembly_Software Completeness_software  \\\n",
       "0  Illumina NovaSeq 6000           Megahit                CheckM   \n",
       "1  Illumina NovaSeq 6000           Megahit                CheckM   \n",
       "2  Illumina NovaSeq 6000           Megahit                CheckM   \n",
       "3  Illumina NovaSeq 6000           Megahit                CheckM   \n",
       "4  Illumina NovaSeq 6000           Megahit                CheckM   \n",
       "\n",
       "  Binning_software           Investigation_type Binning_parameters  \\\n",
       "0           maxbin  metagenome-assembled genome            default   \n",
       "1           maxbin  metagenome-assembled genome            default   \n",
       "2           maxbin  metagenome-assembled genome            default   \n",
       "3           maxbin  metagenome-assembled genome            default   \n",
       "4           maxbin  metagenome-assembled genome            default   \n",
       "\n",
       "                          file_name  \n",
       "0  maxbin_SP106_1.016_cleaned.fasta  \n",
       "1  maxbin_SP106_1.023_cleaned.fasta  \n",
       "2          maxbin_SP106_2.001.fasta  \n",
       "3          maxbin_SP106_3.024.fasta  \n",
       "4  maxbin_SP108_3.007_cleaned.fasta  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Reordering the columns\n",
    "mag_df2 = mag_df2[['mag_file', 'Scientific_name', 'SRA_accession', 'BioSample', 'Tax_ID', 'Sample_ID', 'Completeness_score', 'Contamination_score', 'Sequencing_method', 'Assembly_Software', 'Completeness_software', 'Binning_software', 'Investigation_type', 'Binning_parameters', 'file_name']]\n",
    "mag_df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "07b36c68-e732-4f99-97c5-1862c95e5905",
   "metadata": {
    "panel-layout": {
     "height": 332.66668701171875,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mag_file</th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>sra_accession</th>\n",
       "      <th>biosample</th>\n",
       "      <th>tax_id</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>completeness_score</th>\n",
       "      <th>contamination_score</th>\n",
       "      <th>sequencing_method</th>\n",
       "      <th>assembly_software</th>\n",
       "      <th>completeness_software</th>\n",
       "      <th>binning_software</th>\n",
       "      <th>investigation_type</th>\n",
       "      <th>binning_parameters</th>\n",
       "      <th>file_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uncultured_Agrobacterium_sp._ERS6626984.fasta</td>\n",
       "      <td>uncultured Agrobacterium sp.</td>\n",
       "      <td>ERS6626984</td>\n",
       "      <td>SAMEA8944148</td>\n",
       "      <td>157277</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned</td>\n",
       "      <td>72</td>\n",
       "      <td>6</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>maxbin_SP106_1.016_cleaned.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626983.fasta</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>ERS6626983</td>\n",
       "      <td>SAMEA8944147</td>\n",
       "      <td>278208</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned</td>\n",
       "      <td>72</td>\n",
       "      <td>1</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>maxbin_SP106_1.023_cleaned.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured_Paenibacillus_sp._ERS6626844.fasta</td>\n",
       "      <td>uncultured Paenibacillus sp.</td>\n",
       "      <td>ERS6626844</td>\n",
       "      <td>SAMEA8944008</td>\n",
       "      <td>227322</td>\n",
       "      <td>maxbin_SP106_2.001</td>\n",
       "      <td>100</td>\n",
       "      <td>1</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>maxbin_SP106_2.001.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626982.fasta</td>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>ERS6626982</td>\n",
       "      <td>SAMEA8944146</td>\n",
       "      <td>278208</td>\n",
       "      <td>maxbin_SP106_3.024</td>\n",
       "      <td>71</td>\n",
       "      <td>2</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>maxbin_SP106_3.024.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626962.fasta</td>\n",
       "      <td>uncultured Mucilaginibacter sp.</td>\n",
       "      <td>ERS6626962</td>\n",
       "      <td>SAMEA8944126</td>\n",
       "      <td>797541</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned</td>\n",
       "      <td>63</td>\n",
       "      <td>7</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Megahit</td>\n",
       "      <td>CheckM</td>\n",
       "      <td>maxbin</td>\n",
       "      <td>metagenome-assembled genome</td>\n",
       "      <td>default</td>\n",
       "      <td>maxbin_SP108_3.007_cleaned.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           mag_file  \\\n",
       "0     uncultured_Agrobacterium_sp._ERS6626984.fasta   \n",
       "1         uncultured_Spirosoma_sp._ERS6626983.fasta   \n",
       "2     uncultured_Paenibacillus_sp._ERS6626844.fasta   \n",
       "3         uncultured_Spirosoma_sp._ERS6626982.fasta   \n",
       "4  uncultured_Mucilaginibacter_sp._ERS6626962.fasta   \n",
       "\n",
       "                   scientific_name sra_accession     biosample  tax_id  \\\n",
       "0     uncultured Agrobacterium sp.    ERS6626984  SAMEA8944148  157277   \n",
       "1         uncultured Spirosoma sp.    ERS6626983  SAMEA8944147  278208   \n",
       "2     uncultured Paenibacillus sp.    ERS6626844  SAMEA8944008  227322   \n",
       "3         uncultured Spirosoma sp.    ERS6626982  SAMEA8944146  278208   \n",
       "4  uncultured Mucilaginibacter sp.    ERS6626962  SAMEA8944126  797541   \n",
       "\n",
       "                    sample_id  completeness_score  contamination_score  \\\n",
       "0  maxbin_SP106_1.016_cleaned                  72                    6   \n",
       "1  maxbin_SP106_1.023_cleaned                  72                    1   \n",
       "2          maxbin_SP106_2.001                 100                    1   \n",
       "3          maxbin_SP106_3.024                  71                    2   \n",
       "4  maxbin_SP108_3.007_cleaned                  63                    7   \n",
       "\n",
       "       sequencing_method assembly_software completeness_software  \\\n",
       "0  Illumina NovaSeq 6000           Megahit                CheckM   \n",
       "1  Illumina NovaSeq 6000           Megahit                CheckM   \n",
       "2  Illumina NovaSeq 6000           Megahit                CheckM   \n",
       "3  Illumina NovaSeq 6000           Megahit                CheckM   \n",
       "4  Illumina NovaSeq 6000           Megahit                CheckM   \n",
       "\n",
       "  binning_software           investigation_type binning_parameters  \\\n",
       "0           maxbin  metagenome-assembled genome            default   \n",
       "1           maxbin  metagenome-assembled genome            default   \n",
       "2           maxbin  metagenome-assembled genome            default   \n",
       "3           maxbin  metagenome-assembled genome            default   \n",
       "4           maxbin  metagenome-assembled genome            default   \n",
       "\n",
       "                          file_name  \n",
       "0  maxbin_SP106_1.016_cleaned.fasta  \n",
       "1  maxbin_SP106_1.023_cleaned.fasta  \n",
       "2          maxbin_SP106_2.001.fasta  \n",
       "3          maxbin_SP106_3.024.fasta  \n",
       "4  maxbin_SP108_3.007_cleaned.fasta  "
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Convert all column headings to lower case\n",
    "mag_df2.columns = [col.lower() for col in mag_df2.columns]\n",
    "mag_df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "c9d85bb8-a1b6-42aa-bed8-d1f03f222235",
   "metadata": {
    "panel-layout": {
     "height": 170.6666717529297,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>sra_accession</th>\n",
       "      <th>mag_file</th>\n",
       "      <th>tax_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uncultured Agrobacterium sp.</td>\n",
       "      <td>ERS6626984</td>\n",
       "      <td>uncultured_Agrobacterium_sp._ERS6626984.fasta</td>\n",
       "      <td>157277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>ERS6626983</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626983.fasta</td>\n",
       "      <td>278208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured Paenibacillus sp.</td>\n",
       "      <td>ERS6626844</td>\n",
       "      <td>uncultured_Paenibacillus_sp._ERS6626844.fasta</td>\n",
       "      <td>227322</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>ERS6626982</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626982.fasta</td>\n",
       "      <td>278208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured Mucilaginibacter sp.</td>\n",
       "      <td>ERS6626962</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626962.fasta</td>\n",
       "      <td>797541</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   scientific_name sra_accession  \\\n",
       "0     uncultured Agrobacterium sp.    ERS6626984   \n",
       "1         uncultured Spirosoma sp.    ERS6626983   \n",
       "2     uncultured Paenibacillus sp.    ERS6626844   \n",
       "3         uncultured Spirosoma sp.    ERS6626982   \n",
       "4  uncultured Mucilaginibacter sp.    ERS6626962   \n",
       "\n",
       "                                           mag_file  tax_id  \n",
       "0     uncultured_Agrobacterium_sp._ERS6626984.fasta  157277  \n",
       "1         uncultured_Spirosoma_sp._ERS6626983.fasta  278208  \n",
       "2     uncultured_Paenibacillus_sp._ERS6626844.fasta  227322  \n",
       "3         uncultured_Spirosoma_sp._ERS6626982.fasta  278208  \n",
       "4  uncultured_Mucilaginibacter_sp._ERS6626962.fasta  797541  "
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Prepating for rast details file\n",
    "rast_list = mag_df2[['scientific_name', 'sra_accession','mag_file', 'tax_id']]\n",
    "rast_list.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "9996b561-bbe2-4db9-a6bb-12ca423be5ae",
   "metadata": {
    "panel-layout": {
     "height": 170.6666717529297,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>sra_accession</th>\n",
       "      <th>mag_file</th>\n",
       "      <th>tax_id</th>\n",
       "      <th>mag_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uncultured Agrobacterium sp.</td>\n",
       "      <td>ERS6626984</td>\n",
       "      <td>uncultured_Agrobacterium_sp._ERS6626984.fasta</td>\n",
       "      <td>157277</td>\n",
       "      <td>uncultured_Agrobacterium_sp._ERS6626984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>ERS6626983</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626983.fasta</td>\n",
       "      <td>278208</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured Paenibacillus sp.</td>\n",
       "      <td>ERS6626844</td>\n",
       "      <td>uncultured_Paenibacillus_sp._ERS6626844.fasta</td>\n",
       "      <td>227322</td>\n",
       "      <td>uncultured_Paenibacillus_sp._ERS6626844</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured Spirosoma sp.</td>\n",
       "      <td>ERS6626982</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626982.fasta</td>\n",
       "      <td>278208</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured Mucilaginibacter sp.</td>\n",
       "      <td>ERS6626962</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626962.fasta</td>\n",
       "      <td>797541</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626962</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   scientific_name sra_accession  \\\n",
       "0     uncultured Agrobacterium sp.    ERS6626984   \n",
       "1         uncultured Spirosoma sp.    ERS6626983   \n",
       "2     uncultured Paenibacillus sp.    ERS6626844   \n",
       "3         uncultured Spirosoma sp.    ERS6626982   \n",
       "4  uncultured Mucilaginibacter sp.    ERS6626962   \n",
       "\n",
       "                                           mag_file  tax_id  \\\n",
       "0     uncultured_Agrobacterium_sp._ERS6626984.fasta  157277   \n",
       "1         uncultured_Spirosoma_sp._ERS6626983.fasta  278208   \n",
       "2     uncultured_Paenibacillus_sp._ERS6626844.fasta  227322   \n",
       "3         uncultured_Spirosoma_sp._ERS6626982.fasta  278208   \n",
       "4  uncultured_Mucilaginibacter_sp._ERS6626962.fasta  797541   \n",
       "\n",
       "                                     mag_name  \n",
       "0     uncultured_Agrobacterium_sp._ERS6626984  \n",
       "1         uncultured_Spirosoma_sp._ERS6626983  \n",
       "2     uncultured_Paenibacillus_sp._ERS6626844  \n",
       "3         uncultured_Spirosoma_sp._ERS6626982  \n",
       "4  uncultured_Mucilaginibacter_sp._ERS6626962  "
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Copying dataframe to avoid SettingWithCopyWarning\n",
    "rast_list = rast_list.copy()\n",
    "# Create a new column 'mag_name' by removing the last 6 characters from 'mag_file'\n",
    "rast_list['mag_name'] = rast_list['mag_file'].str[:-6]\n",
    "rast_list.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "ad0051b0-8740-48e0-a962-f4f8cf74485a",
   "metadata": {
    "panel-layout": {
     "height": 0,
     "visible": true,
     "width": 100
    }
   },
   "outputs": [],
   "source": [
    "# Save to csv files\n",
    "output = os.path.join(raw_data, 'rast_list.csv')\n",
    "rast_list.to_csv(output, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9f8b139-2651-4eb4-8cf3-0344e8a26c18",
   "metadata": {
    "panel-layout": {
     "height": 61.427085876464844,
     "visible": true,
     "width": 100
    }
   },
   "source": [
    "<p style=\"font-family: 'Times New Roman'; font-size: 15px; text-align: justify; width: 100%;\">\r\n",
    "   The MAG files are uploaded to RAST served and the results are downloaded in fasta amino acids and excel files named after organism and SRA accession..\r\n",
    "</p>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65625b61-89b3-4952-8150-ecdfd39c2cde",
   "metadata": {
    "panel-layout": {
     "height": 50.8125,
     "visible": true,
     "width": 100
    }
   },
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\r\n",
    "   Prepare List for CoverM Analysis\r\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "dd80db61-915f-4717-abcc-837113b90819",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Supplementary Data 1. Information on rice varieties used for metagenome sequencing.</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "      <th>Unnamed: 9</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "      <th>Unnamed: 11</th>\n",
       "      <th>Unnamed: 12</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Bioproject</td>\n",
       "      <td>SRA accession</td>\n",
       "      <td>BioSample</td>\n",
       "      <td>Sample ID</td>\n",
       "      <td>Host species</td>\n",
       "      <td>Sub-population</td>\n",
       "      <td>Group</td>\n",
       "      <td>Country of origin</td>\n",
       "      <td>IRRI seed bank ID</td>\n",
       "      <td>GSOR ID</td>\n",
       "      <td>Sample origin</td>\n",
       "      <td>Sample coordinates</td>\n",
       "      <td>Sampling date</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595519</td>\n",
       "      <td>SAMEA8912636</td>\n",
       "      <td>SP71_1</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>GSOR301997</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595579</td>\n",
       "      <td>SAMEA8912697</td>\n",
       "      <td>SP285_1</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>India</td>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>GSOR301983</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595630</td>\n",
       "      <td>SAMEA8912748</td>\n",
       "      <td>SP71_2</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>GSOR301997</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595690</td>\n",
       "      <td>SAMEA8912808</td>\n",
       "      <td>SP285_2</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>India</td>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>GSOR301983</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595820</td>\n",
       "      <td>SAMEA8912938</td>\n",
       "      <td>SP391_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Cote d'Ivoire</td>\n",
       "      <td>IRGC 121843</td>\n",
       "      <td>GSOR302541</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595822</td>\n",
       "      <td>SAMEA8912940</td>\n",
       "      <td>SP397_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Spain</td>\n",
       "      <td>IRGC 121697</td>\n",
       "      <td>GSOR302400</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595823</td>\n",
       "      <td>SAMEA8912941</td>\n",
       "      <td>SP398_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Brazil</td>\n",
       "      <td>IRGC 121809</td>\n",
       "      <td>GSOR302512</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595824</td>\n",
       "      <td>SAMEA8912942</td>\n",
       "      <td>SP400_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121293</td>\n",
       "      <td>GSOR301999</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595832</td>\n",
       "      <td>SAMEA8912950</td>\n",
       "      <td>SP432_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Cote d'Ivoire</td>\n",
       "      <td>IRGC 121368</td>\n",
       "      <td>GSOR302074</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>331 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Supplementary Data 1. Information on rice varieties used for metagenome sequencing.  \\\n",
       "0                                           Bioproject                                    \n",
       "1                                           PRJEB45634                                    \n",
       "2                                           PRJEB45634                                    \n",
       "3                                           PRJEB45634                                    \n",
       "4                                           PRJEB45634                                    \n",
       "..                                                 ...                                    \n",
       "326                                         PRJEB45634                                    \n",
       "327                                         PRJEB45634                                    \n",
       "328                                         PRJEB45634                                    \n",
       "329                                         PRJEB45634                                    \n",
       "330                                         PRJEB45634                                    \n",
       "\n",
       "        Unnamed: 1    Unnamed: 2 Unnamed: 3    Unnamed: 4         Unnamed: 5  \\\n",
       "0    SRA accession     BioSample  Sample ID  Host species     Sub-population   \n",
       "1       ERS6595519  SAMEA8912636     SP71_1  Oryza sativa     Admixed-Indica   \n",
       "2       ERS6595579  SAMEA8912697    SP285_1  Oryza sativa     Admixed-Indica   \n",
       "3       ERS6595630  SAMEA8912748     SP71_2  Oryza sativa     Admixed-Indica   \n",
       "4       ERS6595690  SAMEA8912808    SP285_2  Oryza sativa     Admixed-Indica   \n",
       "..             ...           ...        ...           ...                ...   \n",
       "326     ERS6595820  SAMEA8912938    SP391_3  Oryza sativa  Tropical japonica   \n",
       "327     ERS6595822  SAMEA8912940    SP397_3  Oryza sativa  Tropical japonica   \n",
       "328     ERS6595823  SAMEA8912941    SP398_3  Oryza sativa  Tropical japonica   \n",
       "329     ERS6595824  SAMEA8912942    SP400_3  Oryza sativa  Tropical japonica   \n",
       "330     ERS6595832  SAMEA8912950    SP432_3  Oryza sativa  Tropical japonica   \n",
       "\n",
       "    Unnamed: 6         Unnamed: 7         Unnamed: 8  Unnamed: 9  \\\n",
       "0        Group  Country of origin  IRRI seed bank ID     GSOR ID   \n",
       "1       indica         Madagascar        IRGC 121291  GSOR301997   \n",
       "2       indica              India        IRGC 121277  GSOR301983   \n",
       "3       indica         Madagascar        IRGC 121291  GSOR301997   \n",
       "4       indica              India        IRGC 121277  GSOR301983   \n",
       "..         ...                ...                ...         ...   \n",
       "326   japonica      Cote d'Ivoire        IRGC 121843  GSOR302541   \n",
       "327   japonica              Spain        IRGC 121697  GSOR302400   \n",
       "328   japonica             Brazil        IRGC 121809  GSOR302512   \n",
       "329   japonica         Madagascar        IRGC 121293  GSOR301999   \n",
       "330   japonica      Cote d'Ivoire        IRGC 121368  GSOR302074   \n",
       "\n",
       "                                           Unnamed: 10          Unnamed: 11  \\\n",
       "0                                        Sample origin  Sample coordinates    \n",
       "1    China, Hunan Province, Hunan Identificaiton Ce...        N28.8, E112.2   \n",
       "2    China, Hunan Province, Hunan Identificaiton Ce...        N28.8, E112.2   \n",
       "3    China, Hunan Province, Hunan Identificaiton Ce...        N28.8, E112.2   \n",
       "4    China, Hunan Province, Hunan Identificaiton Ce...        N28.8, E112.2   \n",
       "..                                                 ...                  ...   \n",
       "326  China, Hunan Province, Hunan Identificaiton Ce...        N28.8, E112.2   \n",
       "327  China, Hunan Province, Hunan Identificaiton Ce...        N28.8, E112.2   \n",
       "328  China, Hunan Province, Hunan Identificaiton Ce...        N28.8, E112.2   \n",
       "329  China, Hunan Province, Hunan Identificaiton Ce...        N28.8, E112.2   \n",
       "330  China, Hunan Province, Hunan Identificaiton Ce...        N28.8, E112.2   \n",
       "\n",
       "           Unnamed: 12  \n",
       "0        Sampling date  \n",
       "1    September 5. 2020  \n",
       "2    September 5. 2020  \n",
       "3    September 5. 2020  \n",
       "4    September 5. 2020  \n",
       "..                 ...  \n",
       "326  September 5. 2020  \n",
       "327  September 5. 2020  \n",
       "328  September 5. 2020  \n",
       "329  September 5. 2020  \n",
       "330  September 5. 2020  \n",
       "\n",
       "[331 rows x 13 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rice_cultivars = open_file(rice_path)\n",
    "rice_cultivars"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "d1431037-2e38-4dec-b22d-55964a930f67",
   "metadata": {},
   "outputs": [],
   "source": [
    "rice_cultivars.columns = rice_cultivars.iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "57cce36e-af90-4f00-af69-a5373c43adce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bioproject</th>\n",
       "      <th>SRA accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Sample ID</th>\n",
       "      <th>Host species</th>\n",
       "      <th>Sub-population</th>\n",
       "      <th>Group</th>\n",
       "      <th>Country of origin</th>\n",
       "      <th>IRRI seed bank ID</th>\n",
       "      <th>GSOR ID</th>\n",
       "      <th>Sample origin</th>\n",
       "      <th>Sample coordinates</th>\n",
       "      <th>Sampling date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Bioproject</td>\n",
       "      <td>SRA accession</td>\n",
       "      <td>BioSample</td>\n",
       "      <td>Sample ID</td>\n",
       "      <td>Host species</td>\n",
       "      <td>Sub-population</td>\n",
       "      <td>Group</td>\n",
       "      <td>Country of origin</td>\n",
       "      <td>IRRI seed bank ID</td>\n",
       "      <td>GSOR ID</td>\n",
       "      <td>Sample origin</td>\n",
       "      <td>Sample coordinates</td>\n",
       "      <td>Sampling date</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595519</td>\n",
       "      <td>SAMEA8912636</td>\n",
       "      <td>SP71_1</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>GSOR301997</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595579</td>\n",
       "      <td>SAMEA8912697</td>\n",
       "      <td>SP285_1</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>India</td>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>GSOR301983</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595630</td>\n",
       "      <td>SAMEA8912748</td>\n",
       "      <td>SP71_2</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>GSOR301997</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595690</td>\n",
       "      <td>SAMEA8912808</td>\n",
       "      <td>SP285_2</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>India</td>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>GSOR301983</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595820</td>\n",
       "      <td>SAMEA8912938</td>\n",
       "      <td>SP391_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Cote d'Ivoire</td>\n",
       "      <td>IRGC 121843</td>\n",
       "      <td>GSOR302541</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595822</td>\n",
       "      <td>SAMEA8912940</td>\n",
       "      <td>SP397_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Spain</td>\n",
       "      <td>IRGC 121697</td>\n",
       "      <td>GSOR302400</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595823</td>\n",
       "      <td>SAMEA8912941</td>\n",
       "      <td>SP398_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Brazil</td>\n",
       "      <td>IRGC 121809</td>\n",
       "      <td>GSOR302512</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595824</td>\n",
       "      <td>SAMEA8912942</td>\n",
       "      <td>SP400_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121293</td>\n",
       "      <td>GSOR301999</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595832</td>\n",
       "      <td>SAMEA8912950</td>\n",
       "      <td>SP432_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Cote d'Ivoire</td>\n",
       "      <td>IRGC 121368</td>\n",
       "      <td>GSOR302074</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>331 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "0    Bioproject  SRA accession     BioSample  Sample ID  Host species  \\\n",
       "0    Bioproject  SRA accession     BioSample  Sample ID  Host species   \n",
       "1    PRJEB45634     ERS6595519  SAMEA8912636     SP71_1  Oryza sativa   \n",
       "2    PRJEB45634     ERS6595579  SAMEA8912697    SP285_1  Oryza sativa   \n",
       "3    PRJEB45634     ERS6595630  SAMEA8912748     SP71_2  Oryza sativa   \n",
       "4    PRJEB45634     ERS6595690  SAMEA8912808    SP285_2  Oryza sativa   \n",
       "..          ...            ...           ...        ...           ...   \n",
       "326  PRJEB45634     ERS6595820  SAMEA8912938    SP391_3  Oryza sativa   \n",
       "327  PRJEB45634     ERS6595822  SAMEA8912940    SP397_3  Oryza sativa   \n",
       "328  PRJEB45634     ERS6595823  SAMEA8912941    SP398_3  Oryza sativa   \n",
       "329  PRJEB45634     ERS6595824  SAMEA8912942    SP400_3  Oryza sativa   \n",
       "330  PRJEB45634     ERS6595832  SAMEA8912950    SP432_3  Oryza sativa   \n",
       "\n",
       "0       Sub-population     Group  Country of origin  IRRI seed bank ID  \\\n",
       "0       Sub-population     Group  Country of origin  IRRI seed bank ID   \n",
       "1       Admixed-Indica    indica         Madagascar        IRGC 121291   \n",
       "2       Admixed-Indica    indica              India        IRGC 121277   \n",
       "3       Admixed-Indica    indica         Madagascar        IRGC 121291   \n",
       "4       Admixed-Indica    indica              India        IRGC 121277   \n",
       "..                 ...       ...                ...                ...   \n",
       "326  Tropical japonica  japonica      Cote d'Ivoire        IRGC 121843   \n",
       "327  Tropical japonica  japonica              Spain        IRGC 121697   \n",
       "328  Tropical japonica  japonica             Brazil        IRGC 121809   \n",
       "329  Tropical japonica  japonica         Madagascar        IRGC 121293   \n",
       "330  Tropical japonica  japonica      Cote d'Ivoire        IRGC 121368   \n",
       "\n",
       "0       GSOR ID                                      Sample origin  \\\n",
       "0       GSOR ID                                      Sample origin   \n",
       "1    GSOR301997  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "2    GSOR301983  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "3    GSOR301997  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "4    GSOR301983  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "..          ...                                                ...   \n",
       "326  GSOR302541  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "327  GSOR302400  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "328  GSOR302512  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "329  GSOR301999  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "330  GSOR302074  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "\n",
       "0    Sample coordinates       Sampling date  \n",
       "0    Sample coordinates       Sampling date  \n",
       "1          N28.8, E112.2  September 5. 2020  \n",
       "2          N28.8, E112.2  September 5. 2020  \n",
       "3          N28.8, E112.2  September 5. 2020  \n",
       "4          N28.8, E112.2  September 5. 2020  \n",
       "..                   ...                ...  \n",
       "326        N28.8, E112.2  September 5. 2020  \n",
       "327        N28.8, E112.2  September 5. 2020  \n",
       "328        N28.8, E112.2  September 5. 2020  \n",
       "329        N28.8, E112.2  September 5. 2020  \n",
       "330        N28.8, E112.2  September 5. 2020  \n",
       "\n",
       "[331 rows x 13 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rice_cultivars"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "71a715e0-93fe-4c83-bf3a-00a89c691ef0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bioproject</th>\n",
       "      <th>SRA accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Sample ID</th>\n",
       "      <th>Host species</th>\n",
       "      <th>Sub-population</th>\n",
       "      <th>Group</th>\n",
       "      <th>Country of origin</th>\n",
       "      <th>IRRI seed bank ID</th>\n",
       "      <th>GSOR ID</th>\n",
       "      <th>Sample origin</th>\n",
       "      <th>Sample coordinates</th>\n",
       "      <th>Sampling date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595519</td>\n",
       "      <td>SAMEA8912636</td>\n",
       "      <td>SP71_1</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>GSOR301997</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595579</td>\n",
       "      <td>SAMEA8912697</td>\n",
       "      <td>SP285_1</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>India</td>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>GSOR301983</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595630</td>\n",
       "      <td>SAMEA8912748</td>\n",
       "      <td>SP71_2</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>GSOR301997</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595690</td>\n",
       "      <td>SAMEA8912808</td>\n",
       "      <td>SP285_2</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>India</td>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>GSOR301983</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595740</td>\n",
       "      <td>SAMEA8912858</td>\n",
       "      <td>SP71_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>GSOR301997</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "0  Bioproject SRA accession     BioSample Sample ID  Host species  \\\n",
       "1  PRJEB45634    ERS6595519  SAMEA8912636    SP71_1  Oryza sativa   \n",
       "2  PRJEB45634    ERS6595579  SAMEA8912697   SP285_1  Oryza sativa   \n",
       "3  PRJEB45634    ERS6595630  SAMEA8912748    SP71_2  Oryza sativa   \n",
       "4  PRJEB45634    ERS6595690  SAMEA8912808   SP285_2  Oryza sativa   \n",
       "5  PRJEB45634    ERS6595740  SAMEA8912858    SP71_3  Oryza sativa   \n",
       "\n",
       "0  Sub-population   Group Country of origin IRRI seed bank ID     GSOR ID  \\\n",
       "1  Admixed-Indica  indica        Madagascar       IRGC 121291  GSOR301997   \n",
       "2  Admixed-Indica  indica             India       IRGC 121277  GSOR301983   \n",
       "3  Admixed-Indica  indica        Madagascar       IRGC 121291  GSOR301997   \n",
       "4  Admixed-Indica  indica             India       IRGC 121277  GSOR301983   \n",
       "5  Admixed-Indica  indica        Madagascar       IRGC 121291  GSOR301997   \n",
       "\n",
       "0                                      Sample origin Sample coordinates   \\\n",
       "1  China, Hunan Province, Hunan Identificaiton Ce...       N28.8, E112.2   \n",
       "2  China, Hunan Province, Hunan Identificaiton Ce...       N28.8, E112.2   \n",
       "3  China, Hunan Province, Hunan Identificaiton Ce...       N28.8, E112.2   \n",
       "4  China, Hunan Province, Hunan Identificaiton Ce...       N28.8, E112.2   \n",
       "5  China, Hunan Province, Hunan Identificaiton Ce...       N28.8, E112.2   \n",
       "\n",
       "0      Sampling date  \n",
       "1  September 5. 2020  \n",
       "2  September 5. 2020  \n",
       "3  September 5. 2020  \n",
       "4  September 5. 2020  \n",
       "5  September 5. 2020  "
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rice_cultivars = rice_cultivars[1:]\n",
    "\n",
    "rice_cultivars.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "6251c1cf-e6c0-4b39-a4de-13ea494d3985",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bioproject</th>\n",
       "      <th>SRA accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Sample ID</th>\n",
       "      <th>Host species</th>\n",
       "      <th>Sub-population</th>\n",
       "      <th>Group</th>\n",
       "      <th>Country of origin</th>\n",
       "      <th>IRRI seed bank ID</th>\n",
       "      <th>GSOR ID</th>\n",
       "      <th>Sample origin</th>\n",
       "      <th>Sample coordinates</th>\n",
       "      <th>Sampling date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595519</td>\n",
       "      <td>SAMEA8912636</td>\n",
       "      <td>SP71_1</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>GSOR301997</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595579</td>\n",
       "      <td>SAMEA8912697</td>\n",
       "      <td>SP285_1</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>India</td>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>GSOR301983</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595630</td>\n",
       "      <td>SAMEA8912748</td>\n",
       "      <td>SP71_2</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>GSOR301997</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595690</td>\n",
       "      <td>SAMEA8912808</td>\n",
       "      <td>SP285_2</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>India</td>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>GSOR301983</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595740</td>\n",
       "      <td>SAMEA8912858</td>\n",
       "      <td>SP71_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>GSOR301997</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "0  Bioproject SRA accession     BioSample Sample ID  Host species  \\\n",
       "0  PRJEB45634    ERS6595519  SAMEA8912636    SP71_1  Oryza sativa   \n",
       "1  PRJEB45634    ERS6595579  SAMEA8912697   SP285_1  Oryza sativa   \n",
       "2  PRJEB45634    ERS6595630  SAMEA8912748    SP71_2  Oryza sativa   \n",
       "3  PRJEB45634    ERS6595690  SAMEA8912808   SP285_2  Oryza sativa   \n",
       "4  PRJEB45634    ERS6595740  SAMEA8912858    SP71_3  Oryza sativa   \n",
       "\n",
       "0  Sub-population   Group Country of origin IRRI seed bank ID     GSOR ID  \\\n",
       "0  Admixed-Indica  indica        Madagascar       IRGC 121291  GSOR301997   \n",
       "1  Admixed-Indica  indica             India       IRGC 121277  GSOR301983   \n",
       "2  Admixed-Indica  indica        Madagascar       IRGC 121291  GSOR301997   \n",
       "3  Admixed-Indica  indica             India       IRGC 121277  GSOR301983   \n",
       "4  Admixed-Indica  indica        Madagascar       IRGC 121291  GSOR301997   \n",
       "\n",
       "0                                      Sample origin Sample coordinates   \\\n",
       "0  China, Hunan Province, Hunan Identificaiton Ce...       N28.8, E112.2   \n",
       "1  China, Hunan Province, Hunan Identificaiton Ce...       N28.8, E112.2   \n",
       "2  China, Hunan Province, Hunan Identificaiton Ce...       N28.8, E112.2   \n",
       "3  China, Hunan Province, Hunan Identificaiton Ce...       N28.8, E112.2   \n",
       "4  China, Hunan Province, Hunan Identificaiton Ce...       N28.8, E112.2   \n",
       "\n",
       "0      Sampling date  \n",
       "0  September 5. 2020  \n",
       "1  September 5. 2020  \n",
       "2  September 5. 2020  \n",
       "3  September 5. 2020  \n",
       "4  September 5. 2020  "
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rice_cultivars.reset_index(drop=True, inplace=True)\n",
    "rice_cultivars.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "8e367e85-d3a7-4b35-b2e6-4954d676a658",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bioproject</th>\n",
       "      <th>SRA accession</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Sample ID</th>\n",
       "      <th>Host species</th>\n",
       "      <th>Sub-population</th>\n",
       "      <th>Group</th>\n",
       "      <th>Country of origin</th>\n",
       "      <th>IRRI seed bank ID</th>\n",
       "      <th>GSOR ID</th>\n",
       "      <th>Sample origin</th>\n",
       "      <th>Sample coordinates</th>\n",
       "      <th>Sampling date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595519</td>\n",
       "      <td>SAMEA8912636</td>\n",
       "      <td>SP71_1</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>GSOR301997</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595579</td>\n",
       "      <td>SAMEA8912697</td>\n",
       "      <td>SP285_1</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>India</td>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>GSOR301983</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595630</td>\n",
       "      <td>SAMEA8912748</td>\n",
       "      <td>SP71_2</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>GSOR301997</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595690</td>\n",
       "      <td>SAMEA8912808</td>\n",
       "      <td>SP285_2</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>India</td>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>GSOR301983</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595740</td>\n",
       "      <td>SAMEA8912858</td>\n",
       "      <td>SP71_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Admixed-Indica</td>\n",
       "      <td>indica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>GSOR301997</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595820</td>\n",
       "      <td>SAMEA8912938</td>\n",
       "      <td>SP391_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Cote d'Ivoire</td>\n",
       "      <td>IRGC 121843</td>\n",
       "      <td>GSOR302541</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595822</td>\n",
       "      <td>SAMEA8912940</td>\n",
       "      <td>SP397_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Spain</td>\n",
       "      <td>IRGC 121697</td>\n",
       "      <td>GSOR302400</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595823</td>\n",
       "      <td>SAMEA8912941</td>\n",
       "      <td>SP398_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Brazil</td>\n",
       "      <td>IRGC 121809</td>\n",
       "      <td>GSOR302512</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595824</td>\n",
       "      <td>SAMEA8912942</td>\n",
       "      <td>SP400_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Madagascar</td>\n",
       "      <td>IRGC 121293</td>\n",
       "      <td>GSOR301999</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>PRJEB45634</td>\n",
       "      <td>ERS6595832</td>\n",
       "      <td>SAMEA8912950</td>\n",
       "      <td>SP432_3</td>\n",
       "      <td>Oryza sativa</td>\n",
       "      <td>Tropical japonica</td>\n",
       "      <td>japonica</td>\n",
       "      <td>Cote d'Ivoire</td>\n",
       "      <td>IRGC 121368</td>\n",
       "      <td>GSOR302074</td>\n",
       "      <td>China, Hunan Province, Hunan Identificaiton Ce...</td>\n",
       "      <td>N28.8, E112.2</td>\n",
       "      <td>September 5. 2020</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>276 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "0    Bioproject SRA accession     BioSample Sample ID  Host species  \\\n",
       "0    PRJEB45634    ERS6595519  SAMEA8912636    SP71_1  Oryza sativa   \n",
       "1    PRJEB45634    ERS6595579  SAMEA8912697   SP285_1  Oryza sativa   \n",
       "2    PRJEB45634    ERS6595630  SAMEA8912748    SP71_2  Oryza sativa   \n",
       "3    PRJEB45634    ERS6595690  SAMEA8912808   SP285_2  Oryza sativa   \n",
       "4    PRJEB45634    ERS6595740  SAMEA8912858    SP71_3  Oryza sativa   \n",
       "..          ...           ...           ...       ...           ...   \n",
       "325  PRJEB45634    ERS6595820  SAMEA8912938   SP391_3  Oryza sativa   \n",
       "326  PRJEB45634    ERS6595822  SAMEA8912940   SP397_3  Oryza sativa   \n",
       "327  PRJEB45634    ERS6595823  SAMEA8912941   SP398_3  Oryza sativa   \n",
       "328  PRJEB45634    ERS6595824  SAMEA8912942   SP400_3  Oryza sativa   \n",
       "329  PRJEB45634    ERS6595832  SAMEA8912950   SP432_3  Oryza sativa   \n",
       "\n",
       "0       Sub-population     Group Country of origin IRRI seed bank ID  \\\n",
       "0       Admixed-Indica    indica        Madagascar       IRGC 121291   \n",
       "1       Admixed-Indica    indica             India       IRGC 121277   \n",
       "2       Admixed-Indica    indica        Madagascar       IRGC 121291   \n",
       "3       Admixed-Indica    indica             India       IRGC 121277   \n",
       "4       Admixed-Indica    indica        Madagascar       IRGC 121291   \n",
       "..                 ...       ...               ...               ...   \n",
       "325  Tropical japonica  japonica     Cote d'Ivoire       IRGC 121843   \n",
       "326  Tropical japonica  japonica             Spain       IRGC 121697   \n",
       "327  Tropical japonica  japonica            Brazil       IRGC 121809   \n",
       "328  Tropical japonica  japonica        Madagascar       IRGC 121293   \n",
       "329  Tropical japonica  japonica     Cote d'Ivoire       IRGC 121368   \n",
       "\n",
       "0       GSOR ID                                      Sample origin  \\\n",
       "0    GSOR301997  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "1    GSOR301983  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "2    GSOR301997  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "3    GSOR301983  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "4    GSOR301997  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "..          ...                                                ...   \n",
       "325  GSOR302541  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "326  GSOR302400  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "327  GSOR302512  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "328  GSOR301999  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "329  GSOR302074  China, Hunan Province, Hunan Identificaiton Ce...   \n",
       "\n",
       "0   Sample coordinates       Sampling date  \n",
       "0         N28.8, E112.2  September 5. 2020  \n",
       "1         N28.8, E112.2  September 5. 2020  \n",
       "2         N28.8, E112.2  September 5. 2020  \n",
       "3         N28.8, E112.2  September 5. 2020  \n",
       "4         N28.8, E112.2  September 5. 2020  \n",
       "..                  ...                ...  \n",
       "325       N28.8, E112.2  September 5. 2020  \n",
       "326       N28.8, E112.2  September 5. 2020  \n",
       "327       N28.8, E112.2  September 5. 2020  \n",
       "328       N28.8, E112.2  September 5. 2020  \n",
       "329       N28.8, E112.2  September 5. 2020  \n",
       "\n",
       "[276 rows x 13 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "refining = rice_cultivars[(rice_cultivars['Group'] == 'indica') | (rice_cultivars['Group'] == 'japonica')]\n",
    "refining"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "f5f84358-0cb4-4198-b064-eb4e43f5462e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IRRI seed bank ID</th>\n",
       "      <th>SRA accession</th>\n",
       "      <th>Group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>ERS6595519</td>\n",
       "      <td>indica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>ERS6595579</td>\n",
       "      <td>indica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>ERS6595630</td>\n",
       "      <td>indica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>ERS6595690</td>\n",
       "      <td>indica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>ERS6595740</td>\n",
       "      <td>indica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>IRGC 121843</td>\n",
       "      <td>ERS6595820</td>\n",
       "      <td>japonica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326</th>\n",
       "      <td>IRGC 121697</td>\n",
       "      <td>ERS6595822</td>\n",
       "      <td>japonica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>IRGC 121809</td>\n",
       "      <td>ERS6595823</td>\n",
       "      <td>japonica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>IRGC 121293</td>\n",
       "      <td>ERS6595824</td>\n",
       "      <td>japonica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>IRGC 121368</td>\n",
       "      <td>ERS6595832</td>\n",
       "      <td>japonica</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>276 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "0   IRRI seed bank ID SRA accession     Group\n",
       "0         IRGC 121291    ERS6595519    indica\n",
       "1         IRGC 121277    ERS6595579    indica\n",
       "2         IRGC 121291    ERS6595630    indica\n",
       "3         IRGC 121277    ERS6595690    indica\n",
       "4         IRGC 121291    ERS6595740    indica\n",
       "..                ...           ...       ...\n",
       "325       IRGC 121843    ERS6595820  japonica\n",
       "326       IRGC 121697    ERS6595822  japonica\n",
       "327       IRGC 121809    ERS6595823  japonica\n",
       "328       IRGC 121293    ERS6595824  japonica\n",
       "329       IRGC 121368    ERS6595832  japonica\n",
       "\n",
       "[276 rows x 3 columns]"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "selection = refining[['IRRI seed bank ID', 'SRA accession', 'Group']]\n",
    "selection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "5461efad-0f82-4a0c-8b44-30c6b194b9e1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SRA accession</th>\n",
       "      <th>New SRA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERS6595506</td>\n",
       "      <td>ERR6050324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERS6595507</td>\n",
       "      <td>ERR6050325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERS6595508</td>\n",
       "      <td>ERR6050326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERS6595509</td>\n",
       "      <td>ERR6050327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERS6595513</td>\n",
       "      <td>ERR6050331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>ERS6595785</td>\n",
       "      <td>ERR6050602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>326</th>\n",
       "      <td>ERS6595797</td>\n",
       "      <td>ERR6050614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>ERS6595808</td>\n",
       "      <td>ERR6050625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>ERS6595813</td>\n",
       "      <td>ERR6050630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>ERS6595816</td>\n",
       "      <td>ERR6050633</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>330 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    SRA accession     New SRA\n",
       "0      ERS6595506  ERR6050324\n",
       "1      ERS6595507  ERR6050325\n",
       "2      ERS6595508  ERR6050326\n",
       "3      ERS6595509  ERR6050327\n",
       "4      ERS6595513  ERR6050331\n",
       "..            ...         ...\n",
       "325    ERS6595785  ERR6050602\n",
       "326    ERS6595797  ERR6050614\n",
       "327    ERS6595808  ERR6050625\n",
       "328    ERS6595813  ERR6050630\n",
       "329    ERS6595816  ERR6050633\n",
       "\n",
       "[330 rows x 2 columns]"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sra = open_file(sra_path)\n",
    "sra"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "970c12d3-dd09-436b-9786-0dbe569e7d5e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IRRI seed bank ID</th>\n",
       "      <th>SRA accession</th>\n",
       "      <th>Group</th>\n",
       "      <th>New SRA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>ERS6595519</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>ERS6595579</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>ERS6595630</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IRGC 121277</td>\n",
       "      <td>ERS6595690</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IRGC 121291</td>\n",
       "      <td>ERS6595740</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050557</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>IRGC 121843</td>\n",
       "      <td>ERS6595820</td>\n",
       "      <td>japonica</td>\n",
       "      <td>ERR6050637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272</th>\n",
       "      <td>IRGC 121697</td>\n",
       "      <td>ERS6595822</td>\n",
       "      <td>japonica</td>\n",
       "      <td>ERR6050639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>IRGC 121809</td>\n",
       "      <td>ERS6595823</td>\n",
       "      <td>japonica</td>\n",
       "      <td>ERR6050640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>IRGC 121293</td>\n",
       "      <td>ERS6595824</td>\n",
       "      <td>japonica</td>\n",
       "      <td>ERR6050641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>IRGC 121368</td>\n",
       "      <td>ERS6595832</td>\n",
       "      <td>japonica</td>\n",
       "      <td>ERR6050649</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>276 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    IRRI seed bank ID SRA accession     Group     New SRA\n",
       "0         IRGC 121291    ERS6595519    indica  ERR6050337\n",
       "1         IRGC 121277    ERS6595579    indica  ERR6050397\n",
       "2         IRGC 121291    ERS6595630    indica  ERR6050447\n",
       "3         IRGC 121277    ERS6595690    indica  ERR6050507\n",
       "4         IRGC 121291    ERS6595740    indica  ERR6050557\n",
       "..                ...           ...       ...         ...\n",
       "271       IRGC 121843    ERS6595820  japonica  ERR6050637\n",
       "272       IRGC 121697    ERS6595822  japonica  ERR6050639\n",
       "273       IRGC 121809    ERS6595823  japonica  ERR6050640\n",
       "274       IRGC 121293    ERS6595824  japonica  ERR6050641\n",
       "275       IRGC 121368    ERS6595832  japonica  ERR6050649\n",
       "\n",
       "[276 rows x 4 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group = selection.merge(sra, on = 'SRA accession').reset_index(drop=True)\n",
    "group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "4aab98a4-dd84-4c74-b731-457c20d708a3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "92"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group['IRRI seed bank ID'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "96b1e981-c052-4423-8c01-6483356fdb50",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IRRI seed bank ID</th>\n",
       "      <th>SRA accession</th>\n",
       "      <th>Group</th>\n",
       "      <th>New SRA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IRGC 117429</td>\n",
       "      <td>ERS6595826</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IRGC 117429</td>\n",
       "      <td>ERS6595606</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IRGC 117429</td>\n",
       "      <td>ERS6595716</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IRGC 117459</td>\n",
       "      <td>ERS6595767</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IRGC 117459</td>\n",
       "      <td>ERS6595657</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>IRGC 121872</td>\n",
       "      <td>ERS6595648</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272</th>\n",
       "      <td>IRGC 121872</td>\n",
       "      <td>ERS6595758</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>IRGC 121884</td>\n",
       "      <td>ERS6595701</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>IRGC 121884</td>\n",
       "      <td>ERS6595591</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>IRGC 121884</td>\n",
       "      <td>ERS6595811</td>\n",
       "      <td>indica</td>\n",
       "      <td>ERR6050628</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>276 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    IRRI seed bank ID SRA accession   Group     New SRA\n",
       "0         IRGC 117429    ERS6595826  indica  ERR6050643\n",
       "1         IRGC 117429    ERS6595606  indica  ERR6050423\n",
       "2         IRGC 117429    ERS6595716  indica  ERR6050533\n",
       "3         IRGC 117459    ERS6595767  indica  ERR6050584\n",
       "4         IRGC 117459    ERS6595657  indica  ERR6050474\n",
       "..                ...           ...     ...         ...\n",
       "271       IRGC 121872    ERS6595648  indica  ERR6050465\n",
       "272       IRGC 121872    ERS6595758  indica  ERR6050575\n",
       "273       IRGC 121884    ERS6595701  indica  ERR6050518\n",
       "274       IRGC 121884    ERS6595591  indica  ERR6050408\n",
       "275       IRGC 121884    ERS6595811  indica  ERR6050628\n",
       "\n",
       "[276 rows x 4 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group.sort_values(by = 'IRRI seed bank ID', inplace = True)\n",
    "group.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "c0d38077-5f47-489b-a200-1abfeb7a24ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_put = os.path.join(raw_data, 'group.csv')\n",
    "group.to_csv(out_put, index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  },
  "panel-cell-order": [
   "006c2bb4-2782-429d-be24-8077f786abee",
   "f6d1abb6-dd3d-4fc6-bb3a-8a1090ba7749",
   "40583b9f-6854-4b59-918b-0ab5d35eaea2",
   "4d6cbe16-ec86-48fc-96bc-78a0db88d699",
   "796e3e5a-34e1-43a1-8f83-7c8086a8cf82",
   "c55ba52a-f246-456d-b0d8-4a1f6ef55eb2",
   "f22fc255-bf9b-407e-9ceb-f17fb96a464d",
   "9c801eb6-9f9f-4759-b17b-bd5d7d56a78f",
   "39fb9526-07ff-43f0-8611-d387674dbb54",
   "16f911cd-d03f-43dd-b43c-3a5e0974e818",
   "b025c965-7e22-4b1d-a419-8f6d802feacb",
   "2289d409-875d-4e7a-839e-e037acd5e4e4",
   "ce696663-deb9-4fbb-8f9d-8de68405494f",
   "af157f04-afe5-4077-9600-f299e0b28665",
   "cebf779f-4d7b-4980-8e27-ac33650a2f52",
   "38441d53-2d13-4730-a55f-2122ffaf193f",
   "12a4c985-738b-4fde-a859-23fc8f5c6390",
   "9275dccd-62cc-4763-9669-ba61cc653517",
   "cc2399ae-20ca-42dc-b4cb-e797a35aa416",
   "c78b635f-549c-4fc2-bc65-e022f0292db7",
   "080b0c95-babb-4cc1-a996-8642d8d5ab31",
   "63e40689-56e2-4e99-b824-30a9865c6c3b",
   "c81381fb-a08e-4b3a-8190-2513d3eb1093",
   "a2f5bf71-bf6e-40ae-b53d-423a5473d85c",
   "25499ba8-bec5-4f3f-ade6-ab8455ac2ea1",
   "e1ed1dce-8cca-4cc4-b6b8-f39495626902",
   "949d7431-ba08-4af2-ac06-ac9c3db13db9",
   "22076f28-e8b5-4ce4-8cd1-e387167eff0a",
   "425a83c3-3858-4893-9351-c13216034af6",
   "e91d0938-9db2-47f3-a705-342bc0e9398f",
   "babab27b-3371-49a9-bef1-d9e3d5814612",
   "6d9d53b0-de2c-40f5-be9f-1899305d1e64",
   "07b36c68-e732-4f99-97c5-1862c95e5905",
   "c9d85bb8-a1b6-42aa-bed8-d1f03f222235",
   "9996b561-bbe2-4db9-a6bb-12ca423be5ae",
   "ad0051b0-8740-48e0-a962-f4f8cf74485a",
   "e9f8b139-2651-4eb4-8cf3-0344e8a26c18",
   "65625b61-89b3-4952-8150-ecdfd39c2cde"
  ]
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
