{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "600332eb-1a44-4825-9071-d4f6e9d71d51",
   "metadata": {},
   "source": [
    "<h1 style=\"font-family: 'Times New Roman'; text-align: center; font-weight: bold;\">\n",
    "    Data Preprocessing\n",
    "</h1></p>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c144f605-6567-4d56-9f4e-cb6b5af5c73c",
   "metadata": {},
   "source": [
    "<div style=\"font-family: 'Times New Roman'; font-size: 15px; text-align: justify; width: 100%;\">\n",
    "  <div>\n",
    "    <span style=\"display: inline-block; width: 100px;\"><b>Date</b></span>: 29<sup>th</sup> January 2024\n",
    "  </div>\n",
    "  <div>\n",
    "    <span style=\"display: inline-block; width: 100px;\"><b>Author</b></span>: Deepan Kanagarajan Babu\n",
    "  </div>\n",
    "  <div>\n",
    "    <span style=\"display: inline-block; width: 100px;\"><b>Description</b></span>: In this document, Checking for the presence of Form 2 CoxL protein in <i>Deinococcus gobiensis</i> I-0 genome. Genome is uploaded to RAST server and the results are downloaded in '.faa' and '.xls' file formats. Trying to find the Aerobic carbon monoxide dehydrogenase activity in the 'function' column of 'Deinococcus_gobiensis_I-0_rast.xls' file. Then the fasta amino acid file will be filtered for the sequences with the CODH function. The filtered fasta amino acid file is used for MOTIF search. Here we use both MOTIF form 1 (AYXCSFR) and form 2 (AYRGAGR). We expect for the presence of only form 2 MOTIF. This genome is from an isolated <i>Deinococcus gobiensis</i> I-0 strain.\n",
    "  </div>\n",
    "</div>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eaa94574-68ed-47d9-ace1-38f2491ad207",
   "metadata": {},
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Required Libraries\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0f66e4c7-ad5a-4cce-aabc-aa81a413f2e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Import required modules\n",
    "import pandas as pd\n",
    "import os\n",
    "from Bio import SeqIO\n",
    "import re"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d25a0f4-92f1-4158-ac80-18ab75a9e99a",
   "metadata": {},
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Import and Process Files\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ac7da021-818a-4050-af9a-505cc0bb179b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>contig_id</th>\n",
       "      <th>feature_id</th>\n",
       "      <th>type</th>\n",
       "      <th>location</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>function</th>\n",
       "      <th>aliases</th>\n",
       "      <th>figfam</th>\n",
       "      <th>evidence_codes</th>\n",
       "      <th>nucleotide_sequence</th>\n",
       "      <th>aa_sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_017771.1</td>\n",
       "      <td>fig|745776.20.peg.1</td>\n",
       "      <td>peg</td>\n",
       "      <td>NC_017771.1_2_523</td>\n",
       "      <td>2</td>\n",
       "      <td>523</td>\n",
       "      <td>+</td>\n",
       "      <td>SOS-response repressor and protease LexA (EC 3...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>isu;DNA_repair,_bacterial_UmuCD_system isu;DNA...</td>\n",
       "      <td>gacctgatcgagtacgacaccgcggagcgccggacagccatcatcc...</td>\n",
       "      <td>DLIEYDTAERRTAIIRLTAKGRREAGVPENGEVPELPFPILGEVAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_017771.1</td>\n",
       "      <td>fig|745776.20.peg.2</td>\n",
       "      <td>peg</td>\n",
       "      <td>NC_017771.1_694_1410</td>\n",
       "      <td>694</td>\n",
       "      <td>1410</td>\n",
       "      <td>+</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>atggccctggaactccgccagcgccgcctgctggccgccctcgcgg...</td>\n",
       "      <td>MALELRQRRLLAALAAHDGEVPNDVLASAVQSPPGVAFGLMVRLCE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_017771.1</td>\n",
       "      <td>fig|745776.20.peg.3</td>\n",
       "      <td>peg</td>\n",
       "      <td>NC_017771.1_2257_1394</td>\n",
       "      <td>2257</td>\n",
       "      <td>1394</td>\n",
       "      <td>-</td>\n",
       "      <td>Transposase</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gtgacgacccgcgatactgcccgactgcatgctgacacgctggctg...</td>\n",
       "      <td>MTTRDTARLHADTLAAHLKTHLPHRRLDALRRLAEVLLALLQAEST...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_017771.1</td>\n",
       "      <td>fig|745776.20.peg.4</td>\n",
       "      <td>peg</td>\n",
       "      <td>NC_017771.1_2317_2469</td>\n",
       "      <td>2317</td>\n",
       "      <td>2469</td>\n",
       "      <td>+</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gtgtcaggtgctgaggcctcgacgtccatcccgctgagtgccaaca...</td>\n",
       "      <td>MSGAEASTSIPLSANTLASQISFLDMPHDTEDQISFPSNFDMSPIS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_017771.1</td>\n",
       "      <td>fig|745776.20.peg.5</td>\n",
       "      <td>peg</td>\n",
       "      <td>NC_017771.1_2522_4462</td>\n",
       "      <td>2522</td>\n",
       "      <td>4462</td>\n",
       "      <td>+</td>\n",
       "      <td>Type I restriction-modification system, DNA-me...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>idu(1);Type_I_Restriction-Modification idu(1);...</td>\n",
       "      <td>atgactcagacgcgcatgggcaacatcatctgggccaccgcagaac...</td>\n",
       "      <td>MTQTRMGNIIWATAELLRGDYKQADYGKVILPMTIARRLDGMAGRH...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4895</th>\n",
       "      <td>NC_017806.1</td>\n",
       "      <td>fig|745776.20.peg.4459</td>\n",
       "      <td>peg</td>\n",
       "      <td>NC_017806.1_52489_52821</td>\n",
       "      <td>52489</td>\n",
       "      <td>52821</td>\n",
       "      <td>+</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>atgacggccacccagaatgtggactacacctatgagctgaccctgc...</td>\n",
       "      <td>MTATQNVDYTYELTLPQILREGMDRHGDRAYVALGAAISAVHGLSP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4896</th>\n",
       "      <td>NC_017806.1</td>\n",
       "      <td>fig|745776.20.peg.4460</td>\n",
       "      <td>peg</td>\n",
       "      <td>NC_017806.1_52903_53088</td>\n",
       "      <td>52903</td>\n",
       "      <td>53088</td>\n",
       "      <td>+</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gtgggcgtggctcggaaaatgctgaccagtgacgacacacctgagc...</td>\n",
       "      <td>MGVARKMLTSDDTPERLEVDARLAVQVLRDLGRVHAAGEHLASLDP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4897</th>\n",
       "      <td>NC_017806.1</td>\n",
       "      <td>fig|745776.20.repeat.78</td>\n",
       "      <td>repeat</td>\n",
       "      <td>NC_017806.1_53368_54602</td>\n",
       "      <td>53368</td>\n",
       "      <td>54602</td>\n",
       "      <td>+</td>\n",
       "      <td>repeat region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cggtaatgctccagagatgacttccagaacctgacccttgtgatgc...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4898</th>\n",
       "      <td>NC_017806.1</td>\n",
       "      <td>fig|745776.20.peg.4461</td>\n",
       "      <td>peg</td>\n",
       "      <td>NC_017806.1_53440_53727</td>\n",
       "      <td>53440</td>\n",
       "      <td>53727</td>\n",
       "      <td>+</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>atgggaaagcagagaaaagtctggagcacggacgtcaaagaagcca...</td>\n",
       "      <td>MGKQRKVWSTDVKEAIVLSVLRGDLGVAEAARQHRVNESLIHTWKT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4899</th>\n",
       "      <td>NC_017806.1</td>\n",
       "      <td>fig|745776.20.peg.4462</td>\n",
       "      <td>peg</td>\n",
       "      <td>NC_017806.1_53802_54581</td>\n",
       "      <td>53802</td>\n",
       "      <td>54581</td>\n",
       "      <td>+</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gtgccgtactggcggctccgggaccatcagcacagcgcgtcgacgc...</td>\n",
       "      <td>MPYWRLRDHQHSASTRCARQEHRDALYEKVRQAALQHPTSGYRLLY...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4900 rows Ã— 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        contig_id               feature_id    type                 location  \\\n",
       "0     NC_017771.1      fig|745776.20.peg.1     peg        NC_017771.1_2_523   \n",
       "1     NC_017771.1      fig|745776.20.peg.2     peg     NC_017771.1_694_1410   \n",
       "2     NC_017771.1      fig|745776.20.peg.3     peg    NC_017771.1_2257_1394   \n",
       "3     NC_017771.1      fig|745776.20.peg.4     peg    NC_017771.1_2317_2469   \n",
       "4     NC_017771.1      fig|745776.20.peg.5     peg    NC_017771.1_2522_4462   \n",
       "...           ...                      ...     ...                      ...   \n",
       "4895  NC_017806.1   fig|745776.20.peg.4459     peg  NC_017806.1_52489_52821   \n",
       "4896  NC_017806.1   fig|745776.20.peg.4460     peg  NC_017806.1_52903_53088   \n",
       "4897  NC_017806.1  fig|745776.20.repeat.78  repeat  NC_017806.1_53368_54602   \n",
       "4898  NC_017806.1   fig|745776.20.peg.4461     peg  NC_017806.1_53440_53727   \n",
       "4899  NC_017806.1   fig|745776.20.peg.4462     peg  NC_017806.1_53802_54581   \n",
       "\n",
       "      start   stop strand                                           function  \\\n",
       "0         2    523      +  SOS-response repressor and protease LexA (EC 3...   \n",
       "1       694   1410      +                               hypothetical protein   \n",
       "2      2257   1394      -                                        Transposase   \n",
       "3      2317   2469      +                               hypothetical protein   \n",
       "4      2522   4462      +  Type I restriction-modification system, DNA-me...   \n",
       "...     ...    ...    ...                                                ...   \n",
       "4895  52489  52821      +                               hypothetical protein   \n",
       "4896  52903  53088      +                               hypothetical protein   \n",
       "4897  53368  54602      +                                      repeat region   \n",
       "4898  53440  53727      +                               hypothetical protein   \n",
       "4899  53802  54581      +                               hypothetical protein   \n",
       "\n",
       "      aliases  figfam                                     evidence_codes  \\\n",
       "0         NaN     NaN  isu;DNA_repair,_bacterial_UmuCD_system isu;DNA...   \n",
       "1         NaN     NaN                                                NaN   \n",
       "2         NaN     NaN                                                NaN   \n",
       "3         NaN     NaN                                                NaN   \n",
       "4         NaN     NaN  idu(1);Type_I_Restriction-Modification idu(1);...   \n",
       "...       ...     ...                                                ...   \n",
       "4895      NaN     NaN                                                NaN   \n",
       "4896      NaN     NaN                                                NaN   \n",
       "4897      NaN     NaN                                                NaN   \n",
       "4898      NaN     NaN                                                NaN   \n",
       "4899      NaN     NaN                                                NaN   \n",
       "\n",
       "                                    nucleotide_sequence  \\\n",
       "0     gacctgatcgagtacgacaccgcggagcgccggacagccatcatcc...   \n",
       "1     atggccctggaactccgccagcgccgcctgctggccgccctcgcgg...   \n",
       "2     gtgacgacccgcgatactgcccgactgcatgctgacacgctggctg...   \n",
       "3     gtgtcaggtgctgaggcctcgacgtccatcccgctgagtgccaaca...   \n",
       "4     atgactcagacgcgcatgggcaacatcatctgggccaccgcagaac...   \n",
       "...                                                 ...   \n",
       "4895  atgacggccacccagaatgtggactacacctatgagctgaccctgc...   \n",
       "4896  gtgggcgtggctcggaaaatgctgaccagtgacgacacacctgagc...   \n",
       "4897  cggtaatgctccagagatgacttccagaacctgacccttgtgatgc...   \n",
       "4898  atgggaaagcagagaaaagtctggagcacggacgtcaaagaagcca...   \n",
       "4899  gtgccgtactggcggctccgggaccatcagcacagcgcgtcgacgc...   \n",
       "\n",
       "                                            aa_sequence  \n",
       "0     DLIEYDTAERRTAIIRLTAKGRREAGVPENGEVPELPFPILGEVAA...  \n",
       "1     MALELRQRRLLAALAAHDGEVPNDVLASAVQSPPGVAFGLMVRLCE...  \n",
       "2     MTTRDTARLHADTLAAHLKTHLPHRRLDALRRLAEVLLALLQAEST...  \n",
       "3     MSGAEASTSIPLSANTLASQISFLDMPHDTEDQISFPSNFDMSPIS...  \n",
       "4     MTQTRMGNIIWATAELLRGDYKQADYGKVILPMTIARRLDGMAGRH...  \n",
       "...                                                 ...  \n",
       "4895  MTATQNVDYTYELTLPQILREGMDRHGDRAYVALGAAISAVHGLSP...  \n",
       "4896  MGVARKMLTSDDTPERLEVDARLAVQVLRDLGRVHAAGEHLASLDP...  \n",
       "4897                                                NaN  \n",
       "4898  MGKQRKVWSTDVKEAIVLSVLRGDLGVAEAARQHRVNESLIHTWKT...  \n",
       "4899  MPYWRLRDHQHSASTRCARQEHRDALYEKVRQAALQHPTSGYRLLY...  \n",
       "\n",
       "[4900 rows x 13 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Import RAST .xls file using pandas\n",
    "deinococcus = pd.read_excel('Deinococcus_gobiensis_I-0_rast.xls')\n",
    "deinococcus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "fb693058-b2a6-46a7-a079-75784d86ff6d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>contig_id</th>\n",
       "      <th>feature_id</th>\n",
       "      <th>type</th>\n",
       "      <th>location</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>function</th>\n",
       "      <th>aliases</th>\n",
       "      <th>figfam</th>\n",
       "      <th>evidence_codes</th>\n",
       "      <th>nucleotide_sequence</th>\n",
       "      <th>aa_sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2983</th>\n",
       "      <td>NC_017790.1</td>\n",
       "      <td>fig|745776.20.peg.2909</td>\n",
       "      <td>peg</td>\n",
       "      <td>NC_017790.1_2693628_2694125</td>\n",
       "      <td>2693628</td>\n",
       "      <td>2694125</td>\n",
       "      <td>+</td>\n",
       "      <td>Aerobic carbon monoxide dehydrogenase (quinone...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>atgaacgtgaccatgaccgtgaacggcaagacgtacacccgtgagg...</td>\n",
       "      <td>MNVTMTVNGKTYTREVEPRRLLVQFLREDLGLTGTHVGCDTSQCGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2984</th>\n",
       "      <td>NC_017790.1</td>\n",
       "      <td>fig|745776.20.peg.2910</td>\n",
       "      <td>peg</td>\n",
       "      <td>NC_017790.1_2694250_2696640</td>\n",
       "      <td>2694250</td>\n",
       "      <td>2696640</td>\n",
       "      <td>+</td>\n",
       "      <td>Aerobic carbon monoxide dehydrogenase (quinone...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>atgaccgaatccagaacggacaagtacgtcgggcaggccctcaagc...</td>\n",
       "      <td>MTESRTDKYVGQALKRKEDPRFITGTGNYTDDIVLHGMLHAAMVRS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2985</th>\n",
       "      <td>NC_017790.1</td>\n",
       "      <td>fig|745776.20.peg.2911</td>\n",
       "      <td>peg</td>\n",
       "      <td>NC_017790.1_2696699_2697493</td>\n",
       "      <td>2696699</td>\n",
       "      <td>2697493</td>\n",
       "      <td>+</td>\n",
       "      <td>Aerobic carbon monoxide dehydrogenase (quinone...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>atgtacccagcccagttcgactaccagaaagcgacgagcgtggatg...</td>\n",
       "      <td>MYPAQFDYQKATSVDDALRALAENPDLKVIAGGHSLLPAMKLRLAQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3198</th>\n",
       "      <td>NC_017790.1</td>\n",
       "      <td>fig|745776.20.peg.3110</td>\n",
       "      <td>peg</td>\n",
       "      <td>NC_017790.1_2882128_2881664</td>\n",
       "      <td>2882128</td>\n",
       "      <td>2881664</td>\n",
       "      <td>-</td>\n",
       "      <td>carbon monoxide dehydrogenase G protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>atgaaactcaactattccggccaggaacacgtcaaggccccgcccg...</td>\n",
       "      <td>MKLNYSGQEHVKAPPAAVWAFVRDPERVARCLPDVQDVQVRDATHM...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        contig_id              feature_id type                     location  \\\n",
       "2983  NC_017790.1  fig|745776.20.peg.2909  peg  NC_017790.1_2693628_2694125   \n",
       "2984  NC_017790.1  fig|745776.20.peg.2910  peg  NC_017790.1_2694250_2696640   \n",
       "2985  NC_017790.1  fig|745776.20.peg.2911  peg  NC_017790.1_2696699_2697493   \n",
       "3198  NC_017790.1  fig|745776.20.peg.3110  peg  NC_017790.1_2882128_2881664   \n",
       "\n",
       "        start     stop strand  \\\n",
       "2983  2693628  2694125      +   \n",
       "2984  2694250  2696640      +   \n",
       "2985  2696699  2697493      +   \n",
       "3198  2882128  2881664      -   \n",
       "\n",
       "                                               function  aliases  figfam  \\\n",
       "2983  Aerobic carbon monoxide dehydrogenase (quinone...      NaN     NaN   \n",
       "2984  Aerobic carbon monoxide dehydrogenase (quinone...      NaN     NaN   \n",
       "2985  Aerobic carbon monoxide dehydrogenase (quinone...      NaN     NaN   \n",
       "3198            carbon monoxide dehydrogenase G protein      NaN     NaN   \n",
       "\n",
       "     evidence_codes                                nucleotide_sequence  \\\n",
       "2983            NaN  atgaacgtgaccatgaccgtgaacggcaagacgtacacccgtgagg...   \n",
       "2984            NaN  atgaccgaatccagaacggacaagtacgtcgggcaggccctcaagc...   \n",
       "2985            NaN  atgtacccagcccagttcgactaccagaaagcgacgagcgtggatg...   \n",
       "3198            NaN  atgaaactcaactattccggccaggaacacgtcaaggccccgcccg...   \n",
       "\n",
       "                                            aa_sequence  \n",
       "2983  MNVTMTVNGKTYTREVEPRRLLVQFLREDLGLTGTHVGCDTSQCGA...  \n",
       "2984  MTESRTDKYVGQALKRKEDPRFITGTGNYTDDIVLHGMLHAAMVRS...  \n",
       "2985  MYPAQFDYQKATSVDDALRALAENPDLKVIAGGHSLLPAMKLRLAQ...  \n",
       "3198  MKLNYSGQEHVKAPPAAVWAFVRDPERVARCLPDVQDVQVRDATHM...  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Searching for string 'carbon monoxide' in 'function' column\n",
    "## Define the string to be searched and the regrex pattern\n",
    "search_strings = [r'carbon monoxide', r'Carbon Monoxide'] # python identify capital and small letters\n",
    "regex_pattern = '|'.join(search_strings)\n",
    "\n",
    "## Search and save the filtered rows to 'coxl' variable\n",
    "coxl = deinococcus[deinococcus['function'].str.contains(regex_pattern, regex=True, na=False)]\n",
    "coxl"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "05ce15a1-8e2d-428e-9af7-80dc3a2db7eb",
   "metadata": {},
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Filtering FASTA Amino Acid file\n",
    "</h2>\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "2114e470-4be0-4124-93b8-725551c54699",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtered sequences have been saved to 'filtered_Deinococcus_gobiensis_I-0_rast.faa'\n"
     ]
    }
   ],
   "source": [
    "# Get the IDs form coxl DataFrame to filter '.faa' file\n",
    "ids_to_keep = set(coxl['feature_id'])\n",
    "\n",
    "# Name the FASTA file\n",
    "filename = \"Deinococcus_gobiensis_I-0_rast.faa\"\n",
    "\n",
    "# Initialize a list to store the results\n",
    "results = []\n",
    "\n",
    "# Process the FASTA file\n",
    "filtered_sequences = []\n",
    "\n",
    "for record in SeqIO.parse(filename, \"fasta\"):\n",
    "    sequence_id = record.id.split(\"Sequence ID: \")[-1]\n",
    "    if sequence_id in ids_to_keep:\n",
    "        filtered_sequences.append(record)\n",
    "        results.append({\n",
    "            \"FASTA_File\": filename,\n",
    "            \"Sequence_ID\": sequence_id\n",
    "        })\n",
    "\n",
    "# Save the filtered sequences to a new FASTA file\n",
    "filtered_fasta_file = f\"filtered_{filename}\"\n",
    "SeqIO.write(filtered_sequences, filtered_fasta_file, \"fasta\")\n",
    "\n",
    "\n",
    "print(f\"Filtered sequences have been saved to '{filtered_fasta_file}'\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "8ee08360-5ea4-4297-8dc6-6bdf1077c0ac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequence ID: fig|745776.20.peg.2909\n",
      "Sequence: MNVTMTVNGKTYTREVEPRRLLVQFLREDLGLTGTHVGCDTSQCGACTVHVGGDAVKSCTLLAVQVGDLPVTTIEGMGTPGDLHPLQTGFWEQHGLQCGFCTPGMIMASAELLRHNPQPSEEEIRFHLEGNYCRCTGYHNIVRAVQQAAGAIREAAPGQSQAADD\n",
      "\n",
      "Sequence ID: fig|745776.20.peg.2910\n",
      "Sequence: MTESRTDKYVGQALKRKEDPRFITGTGNYTDDIVLHGMLHAAMVRSPYPHARITAIDKGSVADFPGVQLVLTGQDVKDAGLGSIPVGWLLPDLKTPAHPAIALEEANHVGDIVAVVVAETRAQAEDAAAALDVSYEALPSVSSAVAALEDGAPVVHDDVPGNVAFRWEIGDEAALNESFRRAHKTVKVKLRNHRLIANAIEPRASLAQFSPASGEYTLHTTSQNPHIHRLILAAFVMNIPEHKLRVISPDVGGGFGSKIFQYQEEVIVLLAAQKLGRPVKWAARRSESFVSDAQGRDHDTETEMAVDENGMMLGLRVSTVANLGAYQTLFSPAVPTYLYGTLCNGVYKLPAVHVKVTGVMTNTVPVDAYRGAGRPEATYAVERTVDVMAHELGEDPAEFRRRNFIQPDEFPYQTPVALVYDSGDYEPALDKALGMMNYPALREEQARMKGGRKILGVGLISYLEACGLAPSALVGQLGAQAGQWESSLVRVHPTGKVELFTGSHSHGQGHETAFAQIAADELQIPIEDIELIHGDTGRMPYGWGTYGSRSAAVGGSALKMALLKVTAKAKKIAAHLLEASEEDVEHEGGVFRIKGAPGQSKSFFDVSLMAHLAHNLPADMEPGLEATAFYDPKNFVYPFGTHIAVVEIDTDTGHVKLRQYGCVDDCGPLINPLIAEGQVHGGIAQGAAQALLEDAAYDEDGSLLAGTFMEYAIPRADDVPSFLIDHTVTPSPHNPLGVKGIGEAGTIASSAAVANAVMDALWHECGIAHLDMPYTSEKVWRAVQAARGAQGQAADD\n",
      "\n",
      "Sequence ID: fig|745776.20.peg.2911\n",
      "Sequence: MYPAQFDYQKATSVDDALRALAENPDLKVIAGGHSLLPAMKLRLAQPPALLDVWGLEELRGIRREGDMWVVGAMTTHAEVLRSDLPLFPEVAGWVGDPMVRNRGTIGGSLAHADPSADYPAAALATGTQFVIRGLDGERTVDADDMFLGMFESAVQPGELLTQIRIPATITASVYEKFRHPASHYAIVGLALARHADGTVRAAYTGAGEKAERLSVLEERLNAGQAAGTGLVGAAGLLGDRFASAEYRAHLVDVLAAKAAERLR\n",
      "\n",
      "Sequence ID: fig|745776.20.peg.3110\n",
      "Sequence: MKLNYSGQEHVKAPPAAVWAFVRDPERVARCLPDVQDVQVRDATHMDATVQVGVGMVRGKFRFGIEVQPEEDAGRVNVKVNGGGLGSVVDLLAGATVVDNGDGTTTLDWTGDATMRGPVATVGGRLLDAQAQKLIQKTFQNMSASVEAASGTLA\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Checking the filtered fasta file\n",
    "fasta_file = \"filtered_Deinococcus_gobiensis_I-0_rast.faa\"\n",
    "\n",
    "# Read the filtered FASTA file\n",
    "for record in SeqIO.parse(fasta_file, \"fasta\"):\n",
    "    print(f\"Sequence ID: {record.id}\")\n",
    "    print(f\"Sequence: {record.seq}\\n\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92cc04be-5458-4e1a-9005-4a0f9a88ba68",
   "metadata": {},
   "source": [
    "## MOTIF Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "bc998b30-61da-4648-9134-31f6604cb953",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No results found for form 1.\n",
      "Deinococcus gobiensis I-0 carry Form2 CoxL protein\n"
     ]
    }
   ],
   "source": [
    "## Checking for last 4 aminoacids match from motif and trying to retrive the 3 amino acids before the match\n",
    "# Defining the motifs as regular expressions\n",
    "form1 = re.compile(r\"(.{3})CSFR\") # last four amino acids of form 1 MOTIF: AYXCSFR\n",
    "form2 = re.compile(r\"(.{3})GAGR\") # Last four amino acids of form 2 MOTIF: AYRGAGR\n",
    "\n",
    "# Function to search for a motif and return results as a list of dictionaries\n",
    "def search_motif(motif, record, motif_name):\n",
    "    matches = motif.finditer(str(record.seq))\n",
    "    results = []\n",
    "    for match in matches:\n",
    "        start = match.start()\n",
    "        end = match.end()\n",
    "        first_three = match.group(1)\n",
    "        results.append({\n",
    "            \"Sequence_ID\": record.id,\n",
    "            \"First_Three_Aminoacids\": first_three,\n",
    "            \"Motif\": motif_name,\n",
    "            \"Subject_Start\": start,\n",
    "            \"Subject_End\": end - 1\n",
    "        })\n",
    "    return results\n",
    "\n",
    "# Initialize lists to store results for each motif\n",
    "all_results1 = []\n",
    "all_results2 = []\n",
    "\n",
    "# Name the FASTA file\n",
    "filename = \"filtered_Deinococcus_gobiensis_I-0_rast.faa\"\n",
    "\n",
    "# Process the FASTA file\n",
    "for record in SeqIO.parse(filename, \"fasta\"):\n",
    "    # Search for motif 1\n",
    "    results1 = search_motif(form1, record, \"CSFR\")\n",
    "    all_results1.extend(results1)\n",
    "    \n",
    "    # Search for motif 2\n",
    "    results2 = search_motif(form2, record, \"GAGR\")\n",
    "    all_results2.extend(results2)\n",
    "\n",
    "# Create DataFrames from the results\n",
    "Form1 = pd.DataFrame(all_results1)\n",
    "Form2 = pd.DataFrame(all_results2)\n",
    "\n",
    "# Comfirmation of Form 1 or Form 2 CoxL protein presence\n",
    "if not Form1.empty:\n",
    "    print(\"Deinococcus gobiensis I-0 carry Form1 CoxL protein\")\n",
    "else:\n",
    "    print(\"No results found for form 1.\")\n",
    "\n",
    "if not Form2.empty:\n",
    "    print(\"Deinococcus gobiensis I-0 carry Form2 CoxL protein\")\n",
    "else:\n",
    "    print(\"No results found for form 2.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "f400160b-4bb5-48ea-8e20-017148ad0e13",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sequence_ID</th>\n",
       "      <th>First_Three_Aminoacids</th>\n",
       "      <th>Motif</th>\n",
       "      <th>Subject_Start</th>\n",
       "      <th>Subject_End</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>fig|745776.20.peg.2910</td>\n",
       "      <td>AYR</td>\n",
       "      <td>GAGR</td>\n",
       "      <td>367</td>\n",
       "      <td>373</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Sequence_ID First_Three_Aminoacids Motif  Subject_Start  \\\n",
       "0  fig|745776.20.peg.2910                    AYR  GAGR            367   \n",
       "\n",
       "   Subject_End  \n",
       "0          373  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking Form 2 matches\n",
    "Form2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2d0493ba-13f4-4825-8ca4-32231f858b44",
   "metadata": {},
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Result\n",
    "</h2>\n",
    "<p style=\"font-family: 'Times New Roman'; font-size: 15px; text-align: justify; width: 100%;\">\n",
    "    As per the analysis <i>Deinococcus gobiensis</i> I-0 carry Form 2 CoxL gene.\n",
    "</p>"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
