{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "1db8d44c-8207-4acd-8606-2b4067106416",
   "metadata": {},
   "source": [
    "<h1 style=\"font-family: 'Times New Roman'; text-align: center; font-weight: bold;\">\n",
    "    CoxL BLAST Output Preprocessing\n",
    "</h1>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7fc1fa06-5c6c-4a86-9c2c-f8e3d33c2cbe",
   "metadata": {},
   "source": [
    "<div style=\"font-family: 'Times New Roman'; font-size: 15px; text-align: justify; width: 100%;\">\n",
    "  <div>\n",
    "    <span style=\"display: inline-block; width: 100px;\"><b>Date</b></span>: 15<sup>th</sup> November 2024\n",
    "  </div>\n",
    "  <div>\n",
    "    <span style=\"display: inline-block; width: 100px;\"><b>Author</b></span>: Deepan Kanagarajan Babu\n",
    "  </div>\n",
    "  <div>\n",
    "    <span style=\"display: inline-block; width: 100px;\"><b>Description</b></span>: In this document, the BLAST-X outputs are analysed. The coxL gene carriers are shortlised and their genes are fetched for further analysis.\n",
    "  </div>\n",
    "</div>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f12ef065-1018-410c-9ddd-f3385b28b84e",
   "metadata": {},
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Required Libraries\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3127ca06-e4ff-4b47-8754-9665b6a4b935",
   "metadata": {},
   "outputs": [],
   "source": [
    "# importing necessary libraries\n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d7c16e81-c26a-4fb6-a631-f8549e430543",
   "metadata": {},
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Required Functions\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d88b38ef-32ae-4ec4-a001-ad49c4d90ed2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Defining single function to open multiple file formats (.csv, .txt, .fasta or .fastq (to display only 1st 100 lines), and .xlsx)\n",
    "def open_file(file_path):\n",
    "    if file_path.endswith('.csv'):\n",
    "        # Read CSV file\n",
    "        data = pd.read_csv(file_path)\n",
    "        return data\n",
    "    elif file_path.endswith('.txt'):\n",
    "        # Read TXT file\n",
    "        try:\n",
    "            with open(file_path, 'r') as file:\n",
    "                data = file.read()\n",
    "            return data\n",
    "        except FileNotFoundError:\n",
    "            raise FileNotFoundError(f\"The file {file_path} does not exist.\")\n",
    "    elif file_path.endswith(('.fasta', '.fastq')):\n",
    "        # Read FASTA or FASTQ file\n",
    "        try:\n",
    "            with open(file_path, 'r') as file:\n",
    "                data = []\n",
    "                for _ in range(100):\n",
    "                    line = file.readline()\n",
    "                    if not line:\n",
    "                        break\n",
    "                    data.append(line.strip())\n",
    "            return data\n",
    "        except FileNotFoundError:\n",
    "            raise FileNotFoundError(f\"The file {file_path} does not exist.\")\n",
    "    elif file_path.endswith('.xlsx'):\n",
    "        # Read XLSX file\n",
    "        try:\n",
    "            data = pd.read_excel(file_path)\n",
    "            return data\n",
    "        except FileNotFoundError:\n",
    "            raise FileNotFoundError(f\"The file {file_path} does not exist.\")\n",
    "    else:\n",
    "        raise ValueError(\"Unsupported file format\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8d92f84f-a64a-46d9-8bd5-4d749053483f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to find and remove empty CSV, FASTA, and words files\n",
    "def empty_files(directory):\n",
    "    empty_files = []\n",
    "\n",
    "    # Iterate over each file in the directory\n",
    "    for filename in os.listdir(directory):\n",
    "        if filename.endswith('.csv') or filename.endswith('.docs') or filename.endswith('.fasta'):\n",
    "            file_path = os.path.join(directory, filename)\n",
    "            # Check if the file is empty\n",
    "            if os.path.getsize(file_path) == 0:\n",
    "                empty_files.append(file_path)\n",
    "                os.remove(file_path)\n",
    "            elif filename.endswith('.csv'):\n",
    "                # Read the CSV file into a dataframe\n",
    "                df = pd.read_csv(file_path)\n",
    "                # Check if the dataframe is empty\n",
    "                if df.empty:\n",
    "                    empty_files.append(file_path)\n",
    "                    os.remove(file_path)\n",
    "    print(f'Done {directory}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "34666814-e624-4df5-a0ed-fdc5f72f178f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to concate CSV files\n",
    "def concatenate_csv(input_directory, output_file_path):\n",
    "    # List to hold dataframes\n",
    "    dataframes = []\n",
    "\n",
    "    # Iterate over each CSV file in the directory\n",
    "    for csv_filename in os.listdir(input_directory):\n",
    "        if csv_filename.endswith('.csv'):\n",
    "            csv_file_path = os.path.join(input_directory, csv_filename)\n",
    "            # Read the CSV file into a dataframe\n",
    "            df = pd.read_csv(csv_file_path)\n",
    "            # Add a new column with the input file name\n",
    "            df['source_file'] = csv_filename\n",
    "            # Append the dataframe to the list\n",
    "            dataframes.append(df)\n",
    "\n",
    "    # Concatenate all dataframes\n",
    "    concatenated_df = pd.concat(dataframes, ignore_index=True)\n",
    "\n",
    "    # Save the concatenated dataframe to a new CSV file\n",
    "    concatenated_df.to_csv(output_file_path, index=False)\n",
    "    # Leave a confirmation\n",
    "    print(f\"{output_file_path} Successful\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7a83af15-6428-4d17-91ac-0351b91f69f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to extract genes from genome when start and end and sequence header are provided\n",
    "def fetch_genes(df, target_folder, output_folder):\n",
    "    for target_file in df['target_file'].unique():\n",
    "        target_df = df[df['target_file'] == target_file]\n",
    "        target_path = os.path.join(target_folder, target_file)\n",
    "        \n",
    "        # Read the genome sequences from the target file\n",
    "        genome_seqs = {record.id: str(record.seq) for record in SeqIO.parse(target_path, \"fasta\")}\n",
    "        \n",
    "        output_records = []\n",
    "        query_index = 1  # Initialize query index for each target file\n",
    "        \n",
    "        for _, row in target_df.iterrows():\n",
    "            qseqid = row['qseqid']\n",
    "            qstart = row['qstart']\n",
    "            qend = row['qend']\n",
    "            sseqid = row['sseqid']\n",
    "            \n",
    "            # Check if the genome sequence was found\n",
    "            genome_seq = genome_seqs.get(qseqid, \"\")\n",
    "            if not genome_seq:\n",
    "                print(f\"Sequence with ID {qseqid} not found in {target_file}\")\n",
    "                continue\n",
    "            \n",
    "            # Extract the gene sequence based on qstart and qend\n",
    "            if qstart > len(genome_seq) or qend > len(genome_seq):\n",
    "                print(f\"Invalid positions for {qseqid} in {target_file}: qstart={qstart}, qend={qend}, length={len(genome_seq)}\")\n",
    "                continue\n",
    "            \n",
    "            if qend < qstart:\n",
    "                gene_seq = genome_seq[qend-1:qstart][::-1]  # Reverse the sequence\n",
    "            else:\n",
    "                gene_seq = genome_seq[qstart-1:qend]\n",
    "            \n",
    "            # Check if the extracted gene sequence is empty\n",
    "            if not gene_seq:\n",
    "                print(f\"Empty gene sequence for {qseqid} in {target_file}: qstart={qstart}, qend={qend}\")\n",
    "                continue\n",
    "            \n",
    "            # Create a new sequence record with modified header and description\n",
    "            seq_record = SeqRecord(\n",
    "                Seq(gene_seq), \n",
    "                id=f\"{qseqid} Query{query_index} qstart:{qstart} qend:{qend}\", \n",
    "                description=sseqid\n",
    "            )\n",
    "            output_records.append(seq_record)\n",
    "            query_index += 1  # Increment query index\n",
    "        \n",
    "        # Save all sequence records to a single .fasta file in the output folder\n",
    "        output_file = os.path.join(output_folder, target_file)\n",
    "        SeqIO.write(output_records, output_file, \"fasta\")\n",
    "        print(f\"Saved {output_file}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ced42dc0-18f5-47ea-971f-d0f6b0ed42dc",
   "metadata": {},
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Required Directories Path\n",
    "</h2>\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "01e811be-ff93-4bde-a073-40d23803de05",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Specify the input folder paths\n",
    "## Path to Metagenomic Assembled Genomes\n",
    "mags = 'MAG'\n",
    "\n",
    "## Form 1 CoxL BLAST results\n",
    "### CSV Output folder\n",
    "coxl1_csv_results = \"CoxL1_Results/csv\"\n",
    "\n",
    "### Words Output folder\n",
    "coxl1_docs_results = \"CoxL1_Results/docs\"\n",
    "\n",
    "### FASTA Output folder\n",
    "coxl1_fasta_results = \"CoxL1_Results/fasta\"\n",
    "\n",
    "## Form 2 CoxL BLAST results\n",
    "### CSV Output folder\n",
    "coxl2_csv_results = \"CoxL2_Results/csv\"\n",
    "\n",
    "### Words Output folder\n",
    "coxl2_docs_results = \"CoxL2_Results/docs\"\n",
    "\n",
    "### FASTA Output folder\n",
    "coxl2_fasta_results = \"CoxL2_Results/fasta\"\n",
    "\n",
    "# Output Directories\n",
    "## Directory to store concatenated BLAST CSV for MOTIF form 1\n",
    "motif_1_csv = \"CoxL1_Results/MOTIF\"\n",
    "## Ensure the output directory exists\n",
    "os.makedirs(motif_1_csv, exist_ok=True)\n",
    "\n",
    "## File Path to store concatenated BLAST CSV for MOTIF form 1\n",
    "motif_1_csv_file_path = \"CoxL1_Results/MOTIF/coxl1_csv.csv\"\n",
    "\n",
    "## Directory to store concatenated BLAST CSV for MOTIF form 2\n",
    "motif_2_csv_file_path = \"CoxL2_Results/MOTIF\"\n",
    "## Ensure the output directory exists\n",
    "os.makedirs(motif_2_csv_file_path, exist_ok=True)\n",
    "\n",
    "## File Path to store concatenated BLAST CSV for MOTIF form 2\n",
    "motif_2_csv_file_path = \"CoxL2_Results/MOTIF/coxl2_csv.csv\"\n",
    "\n",
    "## Directory to store unaligned sequences for MOTIF form 1\n",
    "motif_1_fasta_path = \"CoxL1_Results/MOTIF/fasta\"\n",
    "## Ensure the output directory exists\n",
    "os.makedirs(motif_1_fasta_path, exist_ok=True)\n",
    "\n",
    "## Directory to store unaligned sequences for MOTIF form 2\n",
    "motif_2_fasta_path = \"CoxL2_Results/MOTIF/fasta\"\n",
    "## Ensure the output directory exists\n",
    "os.makedirs(motif_2_fasta_path, exist_ok=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "353d4efb-498b-4dc6-881a-5e2b2691c1c2",
   "metadata": {},
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Preprocessing\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b8af4a66-55db-42a4-84be-98f32439c4f3",
   "metadata": {},
   "source": [
    "<h3 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Remove Empty BLAST Output Files\n",
    "</h3>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d213c79e-9960-49bb-9197-eb5e2ff79c61",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done CoxL1_Results/csv\n",
      "Done CoxL1_Results/fasta\n",
      "Done CoxL1_Results/docs\n"
     ]
    }
   ],
   "source": [
    "# Using the 'empty_files' function defined above, removing empty output files of MOTIF form 1 results\n",
    "## Removing empyt CSV files\n",
    "empty_files(coxl1_csv_results)\n",
    "\n",
    "## Removing empty FASTA files\n",
    "empty_files(coxl1_fasta_results)\n",
    "\n",
    "## Removing empty Words files\n",
    "empty_files(coxl1_docs_results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "56d13201-7dae-4257-8d3a-302829ae07e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done CoxL2_Results/csv\n",
      "Done CoxL2_Results/fasta\n",
      "Done CoxL2_Results/docs\n"
     ]
    }
   ],
   "source": [
    "# Using the 'empty_files' function defined above, removing empty output files of MOTIF form 2 results\n",
    "## Removing empyt CSV files\n",
    "empty_files(coxl2_csv_results)\n",
    "\n",
    "## Removing empty FASTA files\n",
    "empty_files(coxl2_fasta_results)\n",
    "\n",
    "## Removing empty Words files\n",
    "empty_files(coxl2_docs_results)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "465bb054-6840-47e9-bb66-7dc1c80de268",
   "metadata": {},
   "source": [
    "<h3 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Combine BLAST CSV Outputs\n",
    "</h3>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "44221967-d439-421b-bde3-2d4b287123c9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CoxL1_Results/MOTIF/coxl1_csv.csv Successful\n"
     ]
    }
   ],
   "source": [
    "# Concatenate MOTIF 1 CSV results\n",
    "concatenate_csv(coxl1_csv_results, motif_1_csv_file_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "dd9661b9-0e70-4789-b792-a045adb79745",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CoxL2_Results/MOTIF/coxl2_csv.csv Successful\n"
     ]
    }
   ],
   "source": [
    "# Concatenate MOTIF 2 CSV results\n",
    "concatenate_csv(coxl2_csv_results, motif_2_csv_file_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ebf952c9-2174-40b6-8c73-8400b271e20d",
   "metadata": {},
   "source": [
    "<h3 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Fetch genes from MAGs for MOTIF 1 BLAST Results\n",
    "</h3>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "3a35751d-d6b5-49c4-a782-8916e55aec80",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Mycobacterium_marinum_M</td>\n",
       "      <td>24.634</td>\n",
       "      <td>42</td>\n",
       "      <td>5.050000e-43</td>\n",
       "      <td>163.0</td>\n",
       "      <td>4980</td>\n",
       "      <td>2872</td>\n",
       "      <td>53</td>\n",
       "      <td>784</td>\n",
       "      <td>751</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Mycobacterium_tusciae_WP_006247553</td>\n",
       "      <td>24.433</td>\n",
       "      <td>42</td>\n",
       "      <td>2.450000e-41</td>\n",
       "      <td>158.0</td>\n",
       "      <td>4980</td>\n",
       "      <td>2872</td>\n",
       "      <td>53</td>\n",
       "      <td>784</td>\n",
       "      <td>749</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Rhodococcus_opacus_B4</td>\n",
       "      <td>24.074</td>\n",
       "      <td>39</td>\n",
       "      <td>4.120000e-41</td>\n",
       "      <td>157.0</td>\n",
       "      <td>4980</td>\n",
       "      <td>3022</td>\n",
       "      <td>56</td>\n",
       "      <td>735</td>\n",
       "      <td>702</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Micromonospora_aurantiaca_ATCC_27029</td>\n",
       "      <td>24.536</td>\n",
       "      <td>42</td>\n",
       "      <td>4.150000e-41</td>\n",
       "      <td>157.0</td>\n",
       "      <td>4980</td>\n",
       "      <td>2872</td>\n",
       "      <td>53</td>\n",
       "      <td>784</td>\n",
       "      <td>754</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Mycobacterium_ulcerans_Agy99_YP_904347</td>\n",
       "      <td>24.528</td>\n",
       "      <td>42</td>\n",
       "      <td>5.750000e-41</td>\n",
       "      <td>156.0</td>\n",
       "      <td>4953</td>\n",
       "      <td>2872</td>\n",
       "      <td>62</td>\n",
       "      <td>784</td>\n",
       "      <td>742</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    qseqid                                  sseqid  pident  \\\n",
       "0  Genome20_S3Ck127_316360                 Mycobacterium_marinum_M  24.634   \n",
       "1  Genome20_S3Ck127_316360      Mycobacterium_tusciae_WP_006247553  24.433   \n",
       "2  Genome20_S3Ck127_316360                   Rhodococcus_opacus_B4  24.074   \n",
       "3  Genome20_S3Ck127_316360    Micromonospora_aurantiaca_ATCC_27029  24.536   \n",
       "4  Genome20_S3Ck127_316360  Mycobacterium_ulcerans_Agy99_YP_904347  24.528   \n",
       "\n",
       "   qcovs        evalue  bitscore  qstart  qend  sstart  send  length  \\\n",
       "0     42  5.050000e-43     163.0    4980  2872      53   784     751   \n",
       "1     42  2.450000e-41     158.0    4980  2872      53   784     749   \n",
       "2     39  4.120000e-41     157.0    4980  3022      56   735     702   \n",
       "3     42  4.150000e-41     157.0    4980  2872      53   784     754   \n",
       "4     42  5.750000e-41     156.0    4953  2872      62   784     742   \n",
       "\n",
       "                                         source_file  \n",
       "0  Bdellovibrionaceae_bacterium_ERS6626579_scores...  \n",
       "1  Bdellovibrionaceae_bacterium_ERS6626579_scores...  \n",
       "2  Bdellovibrionaceae_bacterium_ERS6626579_scores...  \n",
       "3  Bdellovibrionaceae_bacterium_ERS6626579_scores...  \n",
       "4  Bdellovibrionaceae_bacterium_ERS6626579_scores...  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Open the MOTIF 1 concatenated CSV file\n",
    "coxl1_csv = open_file(motif_1_csv_file_path)\n",
    "coxl1_csv.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "baac3747-2719-49ee-b53e-f9679ed09aee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Bdellovibrionaceae_bacterium_ERS6626579_scores.csv'"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check the format of source_file\n",
    "coxl1_csv['source_file'].unique()[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "bf34ac9b-0261-4ef8-8940-d21096701496",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Mycobacterium_marinum_M</td>\n",
       "      <td>24.634</td>\n",
       "      <td>42</td>\n",
       "      <td>5.050000e-43</td>\n",
       "      <td>163.0</td>\n",
       "      <td>4980</td>\n",
       "      <td>2872</td>\n",
       "      <td>53</td>\n",
       "      <td>784</td>\n",
       "      <td>751</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Mycobacterium_tusciae_WP_006247553</td>\n",
       "      <td>24.433</td>\n",
       "      <td>42</td>\n",
       "      <td>2.450000e-41</td>\n",
       "      <td>158.0</td>\n",
       "      <td>4980</td>\n",
       "      <td>2872</td>\n",
       "      <td>53</td>\n",
       "      <td>784</td>\n",
       "      <td>749</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Rhodococcus_opacus_B4</td>\n",
       "      <td>24.074</td>\n",
       "      <td>39</td>\n",
       "      <td>4.120000e-41</td>\n",
       "      <td>157.0</td>\n",
       "      <td>4980</td>\n",
       "      <td>3022</td>\n",
       "      <td>56</td>\n",
       "      <td>735</td>\n",
       "      <td>702</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Micromonospora_aurantiaca_ATCC_27029</td>\n",
       "      <td>24.536</td>\n",
       "      <td>42</td>\n",
       "      <td>4.150000e-41</td>\n",
       "      <td>157.0</td>\n",
       "      <td>4980</td>\n",
       "      <td>2872</td>\n",
       "      <td>53</td>\n",
       "      <td>784</td>\n",
       "      <td>754</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Mycobacterium_ulcerans_Agy99_YP_904347</td>\n",
       "      <td>24.528</td>\n",
       "      <td>42</td>\n",
       "      <td>5.750000e-41</td>\n",
       "      <td>156.0</td>\n",
       "      <td>4953</td>\n",
       "      <td>2872</td>\n",
       "      <td>62</td>\n",
       "      <td>784</td>\n",
       "      <td>742</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99069</th>\n",
       "      <td>Genome526_k127_368022</td>\n",
       "      <td>Solirubrobacter_sp._URHD008_2</td>\n",
       "      <td>24.074</td>\n",
       "      <td>12</td>\n",
       "      <td>2.330000e-15</td>\n",
       "      <td>75.9</td>\n",
       "      <td>2280</td>\n",
       "      <td>3665</td>\n",
       "      <td>272</td>\n",
       "      <td>738</td>\n",
       "      <td>486</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302_scores.csv</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99070</th>\n",
       "      <td>Genome526_k127_368022</td>\n",
       "      <td>Pseduonocardia_autotrophica_1</td>\n",
       "      <td>27.444</td>\n",
       "      <td>7</td>\n",
       "      <td>1.240000e-14</td>\n",
       "      <td>73.6</td>\n",
       "      <td>2880</td>\n",
       "      <td>3665</td>\n",
       "      <td>486</td>\n",
       "      <td>740</td>\n",
       "      <td>266</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302_scores.csv</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99071</th>\n",
       "      <td>Genome526_k127_368022</td>\n",
       "      <td>Salinarimonas_rosea_DSM_21201_2523196126</td>\n",
       "      <td>21.942</td>\n",
       "      <td>13</td>\n",
       "      <td>7.830000e-14</td>\n",
       "      <td>70.9</td>\n",
       "      <td>2283</td>\n",
       "      <td>3695</td>\n",
       "      <td>291</td>\n",
       "      <td>770</td>\n",
       "      <td>515</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302_scores.csv</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99072</th>\n",
       "      <td>Genome526_k127_368022</td>\n",
       "      <td>Nocardioides_sp._JS614_2</td>\n",
       "      <td>27.106</td>\n",
       "      <td>7</td>\n",
       "      <td>9.760000e-14</td>\n",
       "      <td>70.5</td>\n",
       "      <td>2862</td>\n",
       "      <td>3665</td>\n",
       "      <td>491</td>\n",
       "      <td>749</td>\n",
       "      <td>273</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302_scores.csv</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99073</th>\n",
       "      <td>Genome526_k127_368022</td>\n",
       "      <td>Thermocrispum_municipale_coxL1</td>\n",
       "      <td>24.444</td>\n",
       "      <td>12</td>\n",
       "      <td>7.870000e-13</td>\n",
       "      <td>67.8</td>\n",
       "      <td>2280</td>\n",
       "      <td>3665</td>\n",
       "      <td>275</td>\n",
       "      <td>740</td>\n",
       "      <td>495</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302_scores.csv</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99074 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        qseqid                                    sseqid  \\\n",
       "0      Genome20_S3Ck127_316360                   Mycobacterium_marinum_M   \n",
       "1      Genome20_S3Ck127_316360        Mycobacterium_tusciae_WP_006247553   \n",
       "2      Genome20_S3Ck127_316360                     Rhodococcus_opacus_B4   \n",
       "3      Genome20_S3Ck127_316360      Micromonospora_aurantiaca_ATCC_27029   \n",
       "4      Genome20_S3Ck127_316360    Mycobacterium_ulcerans_Agy99_YP_904347   \n",
       "...                        ...                                       ...   \n",
       "99069    Genome526_k127_368022             Solirubrobacter_sp._URHD008_2   \n",
       "99070    Genome526_k127_368022             Pseduonocardia_autotrophica_1   \n",
       "99071    Genome526_k127_368022  Salinarimonas_rosea_DSM_21201_2523196126   \n",
       "99072    Genome526_k127_368022                  Nocardioides_sp._JS614_2   \n",
       "99073    Genome526_k127_368022            Thermocrispum_municipale_coxL1   \n",
       "\n",
       "       pident  qcovs        evalue  bitscore  qstart  qend  sstart  send  \\\n",
       "0      24.634     42  5.050000e-43     163.0    4980  2872      53   784   \n",
       "1      24.433     42  2.450000e-41     158.0    4980  2872      53   784   \n",
       "2      24.074     39  4.120000e-41     157.0    4980  3022      56   735   \n",
       "3      24.536     42  4.150000e-41     157.0    4980  2872      53   784   \n",
       "4      24.528     42  5.750000e-41     156.0    4953  2872      62   784   \n",
       "...       ...    ...           ...       ...     ...   ...     ...   ...   \n",
       "99069  24.074     12  2.330000e-15      75.9    2280  3665     272   738   \n",
       "99070  27.444      7  1.240000e-14      73.6    2880  3665     486   740   \n",
       "99071  21.942     13  7.830000e-14      70.9    2283  3695     291   770   \n",
       "99072  27.106      7  9.760000e-14      70.5    2862  3665     491   749   \n",
       "99073  24.444     12  7.870000e-13      67.8    2280  3665     275   740   \n",
       "\n",
       "       length                                        source_file  \\\n",
       "0         751  Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "1         749  Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "2         702  Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "3         754  Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "4         742  Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "...       ...                                                ...   \n",
       "99069     486    uncultured_Xylophilus_sp._ERS6627302_scores.csv   \n",
       "99070     266    uncultured_Xylophilus_sp._ERS6627302_scores.csv   \n",
       "99071     515    uncultured_Xylophilus_sp._ERS6627302_scores.csv   \n",
       "99072     273    uncultured_Xylophilus_sp._ERS6627302_scores.csv   \n",
       "99073     495    uncultured_Xylophilus_sp._ERS6627302_scores.csv   \n",
       "\n",
       "                                         target_file  \n",
       "0      Bdellovibrionaceae_bacterium_ERS6626579.fasta  \n",
       "1      Bdellovibrionaceae_bacterium_ERS6626579.fasta  \n",
       "2      Bdellovibrionaceae_bacterium_ERS6626579.fasta  \n",
       "3      Bdellovibrionaceae_bacterium_ERS6626579.fasta  \n",
       "4      Bdellovibrionaceae_bacterium_ERS6626579.fasta  \n",
       "...                                              ...  \n",
       "99069     uncultured_Xylophilus_sp._ERS6627302.fasta  \n",
       "99070     uncultured_Xylophilus_sp._ERS6627302.fasta  \n",
       "99071     uncultured_Xylophilus_sp._ERS6627302.fasta  \n",
       "99072     uncultured_Xylophilus_sp._ERS6627302.fasta  \n",
       "99073     uncultured_Xylophilus_sp._ERS6627302.fasta  \n",
       "\n",
       "[99074 rows x 13 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create a target_file column, which is the file name of genomes, from the source file \n",
    "coxl1_csv['target_file'] = coxl1_csv['source_file'].apply(lambda x: x[:-11] + '.fasta')\n",
    "coxl1_csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "534bfc96-e618-4e13-ac70-d58bd0a1b0a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save the updated CoxL1 CSV file\n",
    "coxl1_csv.to_csv(f'{motif_1_csv_file_path}', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "4941dc0e-b560-475a-95b5-2eded007ccf0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saved CoxL1_Results/MOTIF/fasta\\Bdellovibrionaceae_bacterium_ERS6626579.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\Bdellovibrionaceae_bacterium_ERS6626826.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\Burkholderiaceae_bacterium_ERS6626828.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\Burkholderiaceae_bacterium_ERS6626861.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\Candidatus_Sericytochromatia_bacterium_ERS6626968.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\Rhizobiaceae_bacterium_ERS6626796.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Achromobacter_sp._ERS6626913.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626561.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626565.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626576.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626591.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626592.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626596.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626599.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626607.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626644.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626705.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626715.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626735.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626737.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626738.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626787.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626798.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626819.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626824.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626827.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626841.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626883.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626889.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626918.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626955.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626991.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626997.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626999.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6627009.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6627016.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6627256.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6627271.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6627284.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6627297.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626567.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626575.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626593.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626600.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626616.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626617.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626625.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626631.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626637.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626640.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626647.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626656.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626674.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626684.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626685.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626690.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626701.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626712.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626717.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626718.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626740.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626745.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626746.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626750.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626751.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626753.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626754.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626761.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626800.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626816.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626821.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626825.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626858.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626912.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626926.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6627024.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6627027.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6627286.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6627310.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626610.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626648.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626649.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626666.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626670.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626766.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626805.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626810.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626818.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626838.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626868.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626908.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626910.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626932.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626956.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626958.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626959.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626971.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626977.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626984.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6627249.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6627262.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6627268.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6627285.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6627329.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Aquabacterium_sp._ERS6627305.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Aureimonas_sp._ERS6626915.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Aureimonas_sp._ERS6627306.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Blastomonas_sp._ERS6626578.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Brevundimonas_sp._ERS6626970.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Caulobacter_sp._ERS6626620.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626590.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626605.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626618.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626662.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626671.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626678.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626682.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626693.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626695.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626697.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626719.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626734.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626736.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626741.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626743.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626744.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626748.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626749.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626755.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626760.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626776.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626817.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626823.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626849.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626863.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626898.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626907.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626996.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627248.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627267.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627296.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627312.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627318.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627325.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627338.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627342.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627343.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626602.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626630.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626638.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626645.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626650.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626814.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626987.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6627283.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6627300.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Cupriavidus_sp._ERS6626993.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Curtobacterium_sp._ERS6626562.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Curtobacterium_sp._ERS6626839.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Curtobacterium_sp._ERS6626869.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Curtobacterium_sp._ERS6626884.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Curtobacterium_sp._ERS6626897.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Curtobacterium_sp._ERS6626914.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Curtobacterium_sp._ERS6627307.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Cystobacter_sp._ERS6626771.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6626797.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6626811.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6626820.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6626837.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6627282.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6627293.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6627335.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Devosia_sp._ERS6626765.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Devosia_sp._ERS6626961.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Epilithonimonas_sp._ERS6626655.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Epilithonimonas_sp._ERS6626757.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Epilithonimonas_sp._ERS6626874.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Erwinia_sp._ERS6626634.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Erwinia_sp._ERS6626641.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Herbaspirillum_sp._ERS6626801.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Herbaspirillum_sp._ERS6627023.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Herbaspirillum_sp._ERS6627309.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Herbaspirillum_sp._ERS6627324.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Hymenobacter_sp._ERS6626716.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Hymenobacter_sp._ERS6626733.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Hymenobacter_sp._ERS6626935.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Hymenobacter_sp._ERS6627279.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Klebsiella_sp._ERS6626601.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Luteibacter_sp._ERS6626809.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Massilia_sp._ERS6626633.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Massilia_sp._ERS6626788.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Massilia_sp._ERS6626950.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Massilia_sp._ERS6626988.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Massilia_sp._ERS6627272.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Methylobacterium_sp._ERS6627288.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626603.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626604.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626665.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626704.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626721.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626781.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626806.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626906.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626964.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627018.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627019.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627260.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627264.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627311.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627330.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627334.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626632.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626762.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626764.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626791.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626799.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626803.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626873.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626894.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626923.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626931.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626962.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626972.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626981.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6627266.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6627275.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6627290.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6627314.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Naasia_sp._ERS6626864.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Naasia_sp._ERS6627002.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Naasia_sp._ERS6627012.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Naasia_sp._ERS6627270.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Naasia_sp._ERS6627292.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Naasia_sp._ERS6627328.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626759.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626775.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626778.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626784.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626786.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626794.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626802.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626829.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626852.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626853.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626854.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626865.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626886.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626887.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626893.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626895.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626901.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626917.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626929.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626942.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626965.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626979.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626989.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6627247.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6627291.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6627336.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Ochrobactrum_sp._ERS6626652.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Ochrobactrum_sp._ERS6627020.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Oxalicibacterium_sp._ERS6626668.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626585.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626614.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626696.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626698.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626700.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626702.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626703.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626842.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626843.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626844.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626855.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626905.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6627323.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6627333.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626568.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626595.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626623.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626643.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626657.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626661.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626664.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626752.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626902.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626920.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626969.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6627004.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6627005.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6627022.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6627332.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pedobacter_sp._ERS6626651.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pedobacter_sp._ERS6626667.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pedobacter_sp._ERS6626930.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pelomonas_sp._ERS6626569.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pelomonas_sp._ERS6626597.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pluralibacter_sp._ERS6626635.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6626577.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6626767.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6626850.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6626909.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6627003.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6627320.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6627321.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudocitrobacter_sp._ERS6626990.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626560.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626563.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626566.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626621.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626626.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626639.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626663.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626673.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626675.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626676.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626710.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626713.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626795.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626822.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626882.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626921.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626934.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626952.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626953.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626976.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626978.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627007.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627014.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627015.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627025.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627026.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627245.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627251.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627265.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627273.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627274.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627287.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627294.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627295.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627315.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Pseudorhodoferax_sp._ERS6626941.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Roseateles_sp._ERS6626963.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Roseateles_sp._ERS6626966.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Serratia_sp._ERS6626624.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Serratia_sp._ERS6626636.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Serratia_sp._ERS6627010.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Siphonobacter_sp._ERS6626689.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Siphonobacter_sp._ERS6626714.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingobium_sp._ERS6626960.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626574.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626580.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626582.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626584.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626586.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626587.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626588.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626589.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626594.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626612.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626622.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626653.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626658.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626669.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626672.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626680.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626686.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626692.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626720.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626768.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626769.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626774.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626780.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626783.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626785.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626831.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626833.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626834.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626835.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626845.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626851.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626856.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626857.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626860.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626866.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626867.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626870.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626871.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626872.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626876.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626881.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626899.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626904.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626911.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626916.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626927.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626936.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626938.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626946.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626948.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626949.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626954.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626957.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626973.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626998.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627000.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627006.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627008.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627013.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627017.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627255.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627257.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627258.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627259.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627261.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627263.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627280.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627281.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627299.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627316.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627317.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626747.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626772.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626782.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626804.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626815.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626890.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626891.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626892.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626982.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626983.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626985.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626995.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6627011.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6626688.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6626723.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6626727.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6627250.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6627308.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6627319.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6627344.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Variovorax_sp._ERS6626779.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Variovorax_sp._ERS6627246.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626613.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626642.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626687.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626724.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626725.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626726.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626730.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626832.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626836.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626846.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626848.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626859.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626943.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627001.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627021.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627277.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627278.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627322.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627331.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627339.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627340.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xylophilus_sp._ERS6626940.fasta\n",
      "Saved CoxL1_Results/MOTIF/fasta\\uncultured_Xylophilus_sp._ERS6627302.fasta\n"
     ]
    }
   ],
   "source": [
    "# Fetch the matches from MOTIF 1 BLAST\n",
    "fetch_genes(coxl1_csv, mags, motif_1_fasta_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32537e82-dd36-4484-baae-7f3bd69d2b20",
   "metadata": {},
   "source": [
    "<h3 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Fetch genes from MAGs for MOTIF 2 BLAST Results\n",
    "</h3>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "75007ca7-941e-4b2d-8de9-2cd00f81db6a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>24.931</td>\n",
       "      <td>42</td>\n",
       "      <td>8.800000e-31</td>\n",
       "      <td>119.0</td>\n",
       "      <td>4944</td>\n",
       "      <td>2872</td>\n",
       "      <td>52</td>\n",
       "      <td>754</td>\n",
       "      <td>726</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>24.933</td>\n",
       "      <td>41</td>\n",
       "      <td>1.080000e-26</td>\n",
       "      <td>105.0</td>\n",
       "      <td>4953</td>\n",
       "      <td>2911</td>\n",
       "      <td>48</td>\n",
       "      <td>760</td>\n",
       "      <td>742</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Genome267_k127_201050</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>23.846</td>\n",
       "      <td>21</td>\n",
       "      <td>4.900000e-35</td>\n",
       "      <td>134.0</td>\n",
       "      <td>6807</td>\n",
       "      <td>8963</td>\n",
       "      <td>7</td>\n",
       "      <td>767</td>\n",
       "      <td>780</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826_scores...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Genome267_k127_201050</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>24.791</td>\n",
       "      <td>17</td>\n",
       "      <td>4.110000e-34</td>\n",
       "      <td>130.0</td>\n",
       "      <td>7299</td>\n",
       "      <td>9014</td>\n",
       "      <td>188</td>\n",
       "      <td>766</td>\n",
       "      <td>597</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826_scores...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Genome267_k127_433915</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>24.156</td>\n",
       "      <td>20</td>\n",
       "      <td>1.980000e-34</td>\n",
       "      <td>132.0</td>\n",
       "      <td>3178</td>\n",
       "      <td>995</td>\n",
       "      <td>8</td>\n",
       "      <td>741</td>\n",
       "      <td>770</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826_scores...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    qseqid                                             sseqid  \\\n",
       "0  Genome20_S3Ck127_316360  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...   \n",
       "1  Genome20_S3Ck127_316360        BAB48572_Form2_Mesorhizobium_japonicum_MAFF   \n",
       "2    Genome267_k127_201050        BAB48572_Form2_Mesorhizobium_japonicum_MAFF   \n",
       "3    Genome267_k127_201050  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...   \n",
       "4    Genome267_k127_433915  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...   \n",
       "\n",
       "   pident  qcovs        evalue  bitscore  qstart  qend  sstart  send  length  \\\n",
       "0  24.931     42  8.800000e-31     119.0    4944  2872      52   754     726   \n",
       "1  24.933     41  1.080000e-26     105.0    4953  2911      48   760     742   \n",
       "2  23.846     21  4.900000e-35     134.0    6807  8963       7   767     780   \n",
       "3  24.791     17  4.110000e-34     130.0    7299  9014     188   766     597   \n",
       "4  24.156     20  1.980000e-34     132.0    3178   995       8   741     770   \n",
       "\n",
       "                                         source_file  \n",
       "0  Bdellovibrionaceae_bacterium_ERS6626579_scores...  \n",
       "1  Bdellovibrionaceae_bacterium_ERS6626579_scores...  \n",
       "2  Bdellovibrionaceae_bacterium_ERS6626826_scores...  \n",
       "3  Bdellovibrionaceae_bacterium_ERS6626826_scores...  \n",
       "4  Bdellovibrionaceae_bacterium_ERS6626826_scores...  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Open the MOTIF 2 concatenated results\n",
    "coxl2_csv = open_file(motif_2_csv_file_path)\n",
    "coxl2_csv.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "b0539d66-09f7-4656-9294-11a3a4cb0977",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>24.931</td>\n",
       "      <td>42</td>\n",
       "      <td>8.800000e-31</td>\n",
       "      <td>119.0</td>\n",
       "      <td>4944</td>\n",
       "      <td>2872</td>\n",
       "      <td>52</td>\n",
       "      <td>754</td>\n",
       "      <td>726</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>24.933</td>\n",
       "      <td>41</td>\n",
       "      <td>1.080000e-26</td>\n",
       "      <td>105.0</td>\n",
       "      <td>4953</td>\n",
       "      <td>2911</td>\n",
       "      <td>48</td>\n",
       "      <td>760</td>\n",
       "      <td>742</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Genome267_k127_201050</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>23.846</td>\n",
       "      <td>21</td>\n",
       "      <td>4.900000e-35</td>\n",
       "      <td>134.0</td>\n",
       "      <td>6807</td>\n",
       "      <td>8963</td>\n",
       "      <td>7</td>\n",
       "      <td>767</td>\n",
       "      <td>780</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Genome267_k127_201050</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>24.791</td>\n",
       "      <td>17</td>\n",
       "      <td>4.110000e-34</td>\n",
       "      <td>130.0</td>\n",
       "      <td>7299</td>\n",
       "      <td>9014</td>\n",
       "      <td>188</td>\n",
       "      <td>766</td>\n",
       "      <td>597</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Genome267_k127_433915</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>24.156</td>\n",
       "      <td>20</td>\n",
       "      <td>1.980000e-34</td>\n",
       "      <td>132.0</td>\n",
       "      <td>3178</td>\n",
       "      <td>995</td>\n",
       "      <td>8</td>\n",
       "      <td>741</td>\n",
       "      <td>770</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2194</th>\n",
       "      <td>Genome526_k127_12441</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>31.604</td>\n",
       "      <td>1</td>\n",
       "      <td>4.330000e-07</td>\n",
       "      <td>45.1</td>\n",
       "      <td>1502</td>\n",
       "      <td>903</td>\n",
       "      <td>551</td>\n",
       "      <td>754</td>\n",
       "      <td>212</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302_scores.csv</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2195</th>\n",
       "      <td>Genome526_k127_152992</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>25.142</td>\n",
       "      <td>3</td>\n",
       "      <td>2.390000e-24</td>\n",
       "      <td>102.0</td>\n",
       "      <td>51673</td>\n",
       "      <td>53688</td>\n",
       "      <td>9</td>\n",
       "      <td>688</td>\n",
       "      <td>704</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302_scores.csv</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2196</th>\n",
       "      <td>Genome526_k127_152992</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>23.077</td>\n",
       "      <td>3</td>\n",
       "      <td>2.420000e-17</td>\n",
       "      <td>79.3</td>\n",
       "      <td>51673</td>\n",
       "      <td>53721</td>\n",
       "      <td>7</td>\n",
       "      <td>722</td>\n",
       "      <td>741</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302_scores.csv</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2197</th>\n",
       "      <td>Genome526_k127_368022</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>25.146</td>\n",
       "      <td>13</td>\n",
       "      <td>2.320000e-25</td>\n",
       "      <td>102.0</td>\n",
       "      <td>2199</td>\n",
       "      <td>3680</td>\n",
       "      <td>252</td>\n",
       "      <td>752</td>\n",
       "      <td>513</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302_scores.csv</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2198</th>\n",
       "      <td>Genome526_k127_368022</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>25.631</td>\n",
       "      <td>13</td>\n",
       "      <td>4.800000e-25</td>\n",
       "      <td>102.0</td>\n",
       "      <td>2157</td>\n",
       "      <td>3665</td>\n",
       "      <td>234</td>\n",
       "      <td>718</td>\n",
       "      <td>515</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302_scores.csv</td>\n",
       "      <td>uncultured_Xylophilus_sp._ERS6627302.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2199 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       qseqid  \\\n",
       "0     Genome20_S3Ck127_316360   \n",
       "1     Genome20_S3Ck127_316360   \n",
       "2       Genome267_k127_201050   \n",
       "3       Genome267_k127_201050   \n",
       "4       Genome267_k127_433915   \n",
       "...                       ...   \n",
       "2194     Genome526_k127_12441   \n",
       "2195    Genome526_k127_152992   \n",
       "2196    Genome526_k127_152992   \n",
       "2197    Genome526_k127_368022   \n",
       "2198    Genome526_k127_368022   \n",
       "\n",
       "                                                 sseqid  pident  qcovs  \\\n",
       "0     BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  24.931     42   \n",
       "1           BAB48572_Form2_Mesorhizobium_japonicum_MAFF  24.933     41   \n",
       "2           BAB48572_Form2_Mesorhizobium_japonicum_MAFF  23.846     21   \n",
       "3     BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  24.791     17   \n",
       "4     BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  24.156     20   \n",
       "...                                                 ...     ...    ...   \n",
       "2194  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  31.604      1   \n",
       "2195  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  25.142      3   \n",
       "2196        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  23.077      3   \n",
       "2197        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  25.146     13   \n",
       "2198  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  25.631     13   \n",
       "\n",
       "            evalue  bitscore  qstart   qend  sstart  send  length  \\\n",
       "0     8.800000e-31     119.0    4944   2872      52   754     726   \n",
       "1     1.080000e-26     105.0    4953   2911      48   760     742   \n",
       "2     4.900000e-35     134.0    6807   8963       7   767     780   \n",
       "3     4.110000e-34     130.0    7299   9014     188   766     597   \n",
       "4     1.980000e-34     132.0    3178    995       8   741     770   \n",
       "...            ...       ...     ...    ...     ...   ...     ...   \n",
       "2194  4.330000e-07      45.1    1502    903     551   754     212   \n",
       "2195  2.390000e-24     102.0   51673  53688       9   688     704   \n",
       "2196  2.420000e-17      79.3   51673  53721       7   722     741   \n",
       "2197  2.320000e-25     102.0    2199   3680     252   752     513   \n",
       "2198  4.800000e-25     102.0    2157   3665     234   718     515   \n",
       "\n",
       "                                            source_file  \\\n",
       "0     Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "1     Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "2     Bdellovibrionaceae_bacterium_ERS6626826_scores...   \n",
       "3     Bdellovibrionaceae_bacterium_ERS6626826_scores...   \n",
       "4     Bdellovibrionaceae_bacterium_ERS6626826_scores...   \n",
       "...                                                 ...   \n",
       "2194    uncultured_Xylophilus_sp._ERS6627302_scores.csv   \n",
       "2195    uncultured_Xylophilus_sp._ERS6627302_scores.csv   \n",
       "2196    uncultured_Xylophilus_sp._ERS6627302_scores.csv   \n",
       "2197    uncultured_Xylophilus_sp._ERS6627302_scores.csv   \n",
       "2198    uncultured_Xylophilus_sp._ERS6627302_scores.csv   \n",
       "\n",
       "                                        target_file  \n",
       "0     Bdellovibrionaceae_bacterium_ERS6626579.fasta  \n",
       "1     Bdellovibrionaceae_bacterium_ERS6626579.fasta  \n",
       "2     Bdellovibrionaceae_bacterium_ERS6626826.fasta  \n",
       "3     Bdellovibrionaceae_bacterium_ERS6626826.fasta  \n",
       "4     Bdellovibrionaceae_bacterium_ERS6626826.fasta  \n",
       "...                                             ...  \n",
       "2194     uncultured_Xylophilus_sp._ERS6627302.fasta  \n",
       "2195     uncultured_Xylophilus_sp._ERS6627302.fasta  \n",
       "2196     uncultured_Xylophilus_sp._ERS6627302.fasta  \n",
       "2197     uncultured_Xylophilus_sp._ERS6627302.fasta  \n",
       "2198     uncultured_Xylophilus_sp._ERS6627302.fasta  \n",
       "\n",
       "[2199 rows x 13 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create a target_file column, which the file name of genomes, from the source file \n",
    "coxl2_csv['target_file'] = coxl2_csv['source_file'].apply(lambda x: x[:-11] + '.fasta')\n",
    "coxl2_csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "df26aeb8-c1fe-4391-9d44-4464b29641bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save the updated CoxL2 CSV file\n",
    "coxl2_csv.to_csv(f'{motif_2_csv_file_path}', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "1b9cc069-af61-4bb8-a464-752200fd8643",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saved CoxL2_Results/MOTIF/fasta\\Bdellovibrionaceae_bacterium_ERS6626579.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\Bdellovibrionaceae_bacterium_ERS6626826.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\Burkholderiaceae_bacterium_ERS6626828.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\Burkholderiaceae_bacterium_ERS6626861.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\Candidatus_Sericytochromatia_bacterium_ERS6626968.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\Rhizobiaceae_bacterium_ERS6626796.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Achromobacter_sp._ERS6626913.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626561.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626565.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626576.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626591.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626592.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626596.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626599.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626607.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626644.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626705.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626715.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626735.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626737.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626738.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626787.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626798.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626819.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626824.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626827.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626841.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626883.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626889.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626918.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626955.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626991.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626997.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6626999.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6627009.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6627016.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6627256.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6627271.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6627284.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acidovorax_sp._ERS6627297.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626567.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626575.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626593.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626600.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626616.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626617.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626625.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626631.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626637.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626640.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626647.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626656.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626674.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626684.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626685.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626690.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626701.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626712.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626717.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626718.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626740.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626745.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626746.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626750.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626751.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626753.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626754.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626761.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626800.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626816.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626821.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626825.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626858.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626912.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6626926.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6627024.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6627027.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6627286.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Acinetobacter_sp._ERS6627310.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626610.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626648.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626649.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626666.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626670.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626766.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626805.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626810.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626818.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626838.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626868.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626908.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626910.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626932.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626956.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626958.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626959.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626971.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626977.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6626984.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6627249.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6627262.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6627268.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6627285.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Agrobacterium_sp._ERS6627329.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Aquabacterium_sp._ERS6627305.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Aureimonas_sp._ERS6626915.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Aureimonas_sp._ERS6627306.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Blastomonas_sp._ERS6626578.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626590.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626605.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626618.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626662.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626671.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626678.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626682.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626693.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626695.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626697.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626719.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626734.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626736.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626741.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626743.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626744.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626748.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626749.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626755.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626760.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626776.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626817.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626823.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626849.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626863.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626898.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626907.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6626996.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627248.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627267.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627296.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627312.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627318.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627325.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627338.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627342.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Chryseobacterium_sp._ERS6627343.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626602.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626630.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626638.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626645.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626650.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626814.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6626987.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6627283.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Comamonas_sp._ERS6627300.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Cupriavidus_sp._ERS6626993.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Curtobacterium_sp._ERS6626562.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Curtobacterium_sp._ERS6626839.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Curtobacterium_sp._ERS6626869.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Curtobacterium_sp._ERS6626884.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Curtobacterium_sp._ERS6626914.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Curtobacterium_sp._ERS6627307.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Cystobacter_sp._ERS6626771.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6626797.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6626811.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6626820.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6626837.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6627282.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6627293.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Deinococcus_sp._ERS6627335.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Devosia_sp._ERS6626765.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Devosia_sp._ERS6626961.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Epilithonimonas_sp._ERS6626655.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Epilithonimonas_sp._ERS6626757.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Epilithonimonas_sp._ERS6626874.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Erwinia_sp._ERS6626634.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Erwinia_sp._ERS6626641.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Herbaspirillum_sp._ERS6626801.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Herbaspirillum_sp._ERS6627023.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Herbaspirillum_sp._ERS6627309.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Herbaspirillum_sp._ERS6627324.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Hymenobacter_sp._ERS6626716.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Hymenobacter_sp._ERS6626733.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Hymenobacter_sp._ERS6626935.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Hymenobacter_sp._ERS6627279.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Klebsiella_sp._ERS6626601.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Luteibacter_sp._ERS6626809.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Massilia_sp._ERS6626633.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Massilia_sp._ERS6626788.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Massilia_sp._ERS6626950.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Massilia_sp._ERS6626988.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Massilia_sp._ERS6627272.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Methylobacterium_sp._ERS6627288.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626603.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626604.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626665.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626704.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626721.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626781.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626806.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626906.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6626964.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627018.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627019.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627260.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627264.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627311.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627330.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Microbacterium_sp._ERS6627334.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626632.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626762.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626764.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626791.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626799.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626803.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626873.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626894.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626923.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626931.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626962.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626972.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6626981.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6627266.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6627275.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6627290.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Mucilaginibacter_sp._ERS6627314.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Naasia_sp._ERS6626864.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Naasia_sp._ERS6627002.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Naasia_sp._ERS6627012.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Naasia_sp._ERS6627270.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Naasia_sp._ERS6627292.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Naasia_sp._ERS6627328.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626759.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626775.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626778.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626784.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626786.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626794.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626802.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626829.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626852.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626853.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626854.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626865.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626886.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626887.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626893.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626895.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626901.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626917.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626929.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626942.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626965.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626979.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6626989.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6627247.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6627291.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Novosphingobium_sp._ERS6627336.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Ochrobactrum_sp._ERS6626652.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Ochrobactrum_sp._ERS6627020.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Oxalicibacterium_sp._ERS6626668.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626585.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626614.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626696.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626698.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626700.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626702.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626703.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626842.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626843.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626844.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626855.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6626905.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6627253.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6627323.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Paenibacillus_sp._ERS6627333.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626568.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626595.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626623.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626643.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626657.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626661.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626664.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626752.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626902.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626920.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6626969.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6627004.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6627005.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6627022.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pantoea_sp._ERS6627332.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pedobacter_sp._ERS6626651.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pedobacter_sp._ERS6626667.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pedobacter_sp._ERS6626930.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pelomonas_sp._ERS6626569.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pelomonas_sp._ERS6626597.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pluralibacter_sp._ERS6626635.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6626577.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6626767.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6626850.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6626909.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6627003.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6627320.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudacidovorax_sp._ERS6627321.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudocitrobacter_sp._ERS6626990.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626560.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626563.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626566.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626621.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626626.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626639.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626663.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626673.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626675.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626676.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626710.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626713.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626822.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626882.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626921.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626934.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626952.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626953.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626976.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6626978.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627007.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627014.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627015.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627025.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627026.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627245.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627251.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627265.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627273.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627274.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627287.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627294.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627295.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudomonas_sp._ERS6627315.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Pseudorhodoferax_sp._ERS6626941.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Roseateles_sp._ERS6626963.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Roseateles_sp._ERS6626966.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Serratia_sp._ERS6626624.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Serratia_sp._ERS6626636.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Serratia_sp._ERS6627010.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Siphonobacter_sp._ERS6626689.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Siphonobacter_sp._ERS6626714.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingobium_sp._ERS6626960.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626574.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626580.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626582.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626584.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626586.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626587.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626588.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626589.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626594.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626612.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626622.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626653.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626658.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626669.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626672.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626680.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626686.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626692.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626720.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626768.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626769.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626774.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626780.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626783.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626785.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626831.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626833.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626834.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626835.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626845.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626851.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626856.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626857.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626860.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626866.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626867.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626870.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626871.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626872.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626876.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626881.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626899.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626904.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626911.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626916.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626927.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626936.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626938.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626946.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626948.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626949.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626954.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626957.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626973.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6626998.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627000.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627006.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627008.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627013.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627017.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627255.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627257.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627258.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627259.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627261.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627263.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627280.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627281.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627299.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627316.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Sphingomonas_sp._ERS6627317.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626747.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626772.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626782.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626804.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626815.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626890.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626891.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626892.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626982.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626983.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626985.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6626995.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Spirosoma_sp._ERS6627011.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6626688.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6626723.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6626727.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6627250.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6627308.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6627319.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Stenotrophomonas_sp._ERS6627344.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Variovorax_sp._ERS6626779.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Variovorax_sp._ERS6627246.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626613.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626642.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626687.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626724.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626725.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626726.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626730.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626832.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626836.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626846.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626848.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626859.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6626943.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627001.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627021.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627277.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627278.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627322.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627331.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627339.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xanthomonas_sp._ERS6627340.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xylophilus_sp._ERS6626940.fasta\n",
      "Saved CoxL2_Results/MOTIF/fasta\\uncultured_Xylophilus_sp._ERS6627302.fasta\n"
     ]
    }
   ],
   "source": [
    "# Fetch the matches from MOTIF 2 BLAST\n",
    "fetch_genes(coxl2_csv, mags, motif_2_fasta_path)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
