{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "031ecac3-115a-4659-b262-3f196c884d1c",
   "metadata": {},
   "source": [
    "<h1 style=\"font-family: 'Times New Roman'; text-align: center; font-weight: bold;\">\n",
    "    CoxL MOTIF Search\n",
    "</h1>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "66c468ec-2f77-4a20-9f6f-ee21efa44005",
   "metadata": {},
   "source": [
    "<div style=\"font-family: 'Times New Roman'; font-size: 15px; text-align: justify; width: 100%;\">\n",
    "  <div>\n",
    "    <span style=\"display: inline-block; width: 100px;\"><b>Date</b></span>: 15<sup>th</sup> November 2024\n",
    "  </div>\n",
    "  <div>\n",
    "    <span style=\"display: inline-block; width: 100px;\"><b>Author</b></span>: Deepan Kanagarajan Babu\n",
    "  </div>\n",
    "  <div>\n",
    "    <span style=\"display: inline-block; width: 100px;\"><b>Description</b></span>: In this document, the sequences are further filtered based on the presence of MOTIF regions. The gene sequeces fetched are looked for MOTIF regions. CoxL protein have either form 1 MOTIF ('AYXCSFR') or form 2 MOTIF ('AYRGAGR') not both which differenciates the protein function and their ability. Form 1 is know to involved in CODH. Form 2 has unclear function, but some studies claim, their have divergent function & not performing CODH and some claim it perform CODH but not as effective as form 1 proteins.\n",
    "  </div>\n",
    "</div>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "550aa0eb-192f-4da1-8d30-ae023debd659",
   "metadata": {},
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Required Libraries\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "558ef14e-b626-4288-baca-22ef14eb8608",
   "metadata": {},
   "outputs": [],
   "source": [
    "# importing necessary libraries\n",
    "import os\n",
    "import re\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from Bio import SeqIO\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "94129673-daef-449a-a71b-e75f42249b07",
   "metadata": {},
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Required Functions\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "800a4195-108c-434d-8dcb-468eb5ae7d88",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Defining single function to open multiple file formats (.csv, .txt, .fasta or .fastq (to display only 1st 100 lines), and .xlsx)\n",
    "def open_file(file_path):\n",
    "    if file_path.endswith('.csv'):\n",
    "        # Read CSV file\n",
    "        data = pd.read_csv(file_path)\n",
    "        return data\n",
    "    elif file_path.endswith('.txt'):\n",
    "        # Read TXT file\n",
    "        try:\n",
    "            with open(file_path, 'r') as file:\n",
    "                data = file.read()\n",
    "            return data\n",
    "        except FileNotFoundError:\n",
    "            raise FileNotFoundError(f\"The file {file_path} does not exist.\")\n",
    "    elif file_path.endswith(('.fasta', '.fastq')):\n",
    "        # Read FASTA or FASTQ file\n",
    "        try:\n",
    "            with open(file_path, 'r') as file:\n",
    "                data = []\n",
    "                for _ in range(100):\n",
    "                    line = file.readline()\n",
    "                    if not line:\n",
    "                        break\n",
    "                    data.append(line.strip())\n",
    "            return data\n",
    "        except FileNotFoundError:\n",
    "            raise FileNotFoundError(f\"The file {file_path} does not exist.\")\n",
    "    elif file_path.endswith('.xlsx'):\n",
    "        # Read XLSX file\n",
    "        try:\n",
    "            data = pd.read_excel(file_path)\n",
    "            return data\n",
    "        except FileNotFoundError:\n",
    "            raise FileNotFoundError(f\"The file {file_path} does not exist.\")\n",
    "    else:\n",
    "        raise ValueError(\"Unsupported file format\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "246a1540-a7b6-4ca7-afe9-4d04ffa43e4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to search for 2 motifs with unique labels\n",
    "def search_motifs(directory, pattern1, label1=\"Motif1\", pattern2=None, label2=\"Motif2\", output_file=None):\n",
    "    all_results = []\n",
    "\n",
    "    # Ensure directory exists\n",
    "    if not os.path.isdir(directory):\n",
    "        print(f\"Error: Directory '{directory}' not found.\")\n",
    "        return pd.DataFrame()\n",
    "\n",
    "    # Process each FASTA or FAA file in the directory\n",
    "    for filename in os.listdir(directory):\n",
    "        if filename.endswith((\".fasta\", \".faa\")):\n",
    "            fasta_file = os.path.join(directory, filename)\n",
    "\n",
    "            for record in SeqIO.parse(fasta_file, \"fasta\"):\n",
    "                seq = str(record.seq)\n",
    "\n",
    "                # Search for the first pattern (if provided)\n",
    "                if pattern1:\n",
    "                    for match in pattern1.finditer(seq):\n",
    "                        all_results.append({\n",
    "                            \"File_Name\": filename,\n",
    "                            \"ID\": record.id,\n",
    "                            \"Sequence\": seq,\n",
    "                            \"MOTIF\": match.group(0),\n",
    "                            \"Previous_3_AA\": match.group(1),\n",
    "                            \"Motif_Label\": label1\n",
    "                        })\n",
    "\n",
    "                # Search for the second pattern (if provided)\n",
    "                if pattern2:\n",
    "                    for match in pattern2.finditer(seq):\n",
    "                        all_results.append({\n",
    "                            \"File_Name\": filename,\n",
    "                            \"ID\": record.id,\n",
    "                            \"Sequence\": seq,\n",
    "                            \"MOTIF\": match.group(0),\n",
    "                            \"Previous_3_AA\": match.group(1),\n",
    "                            \"Motif_Label\": label2\n",
    "                        })\n",
    "\n",
    "    # Convert results to DataFrame\n",
    "    df = pd.DataFrame(all_results)\n",
    "\n",
    "    # Set output file name dynamically if not provided\n",
    "    if output_file is None:\n",
    "        output_file = f\"{label1}_{label2}_motif_search.csv\"  # Default to label-based filename\n",
    "\n",
    "    # Save only if results are found\n",
    "    if not df.empty:\n",
    "        df.to_csv(output_file, index=False)\n",
    "        print(f\"Results saved to {output_file}\")\n",
    "    else:\n",
    "        print(\"No results found for motifs.\")\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8fe42ce0-268c-4c87-aa92-408324243cce",
   "metadata": {},
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Required Directories Path\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "09f91fcc-f52e-453e-8890-5a9810c738bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Specify the input folder paths\n",
    "## concatenated MOTIF form 1 BLAST CSV file path\n",
    "motif_1_csv_file_path = \"CoxL1_Results/MOTIF/coxl1_csv.csv\"\n",
    "\n",
    "\n",
    "## concatenated MOTIF form 2 BLAST CSV file path\n",
    "motif_2_csv_file_path = \"CoxL2_Results/MOTIF/coxl2_csv.csv\"\n",
    "\n",
    "## CoxL Form 1 BLAST proteins sequence directory\n",
    "motif_1_proteins = \"CoxL1_Results/MOTIF_proteins\"\n",
    "\n",
    "## CoxL Form 2 BLAST proteins sequence directory\n",
    "motif_2_proteins = \"CoxL2_Results/MOTIF_proteins\"\n",
    "\n",
    "# Specify the output folder paths\n",
    "## Directory to store CoxL Form 1 BLAST MOTIF search results output\n",
    "motif1 = \"CoxL1_Results/MOTIF\"\n",
    "\n",
    "## Directory to store CoxL Form 2 BLAST MOTIF search results output\n",
    "motif2 = \"CoxL2_Results/MOTIF\"\n",
    "\n",
    "## Directory to store unaligned sequences for MOTIF form 1\n",
    "motif_1_fasta_path = \"CoxL1_Results/MOTIF/fasta\"\n",
    "## Ensure the output directory exists\n",
    "os.makedirs(motif_1_fasta_path, exist_ok=True)\n",
    "\n",
    "## Directory to store unaligned sequences for MOTIF form 2\n",
    "motif_2_fasta_path = \"CoxL2_Results/MOTIF/fasta\"\n",
    "## Ensure the output directory exists\n",
    "os.makedirs(motif_2_fasta_path, exist_ok=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "990ec0d3-ebc0-4cee-a0dc-50e35edba290",
   "metadata": {},
   "source": [
    "<h2 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Search MOTIFs\n",
    "</h2>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "adcab3f1-4932-440e-b779-085231f45a74",
   "metadata": {},
   "outputs": [],
   "source": [
    "# MOTIF search pattern\n",
    "## Define the motifs as regular expressions\n",
    "form1 = re.compile(r\"(.{3})CSFR\") # last four amino acids of form 1: AYXCSFR\n",
    "form2 = re.compile(r\"(.{3})GAGR\") # Last four amino acids of form 2: AYRGAGR\n",
    "\n",
    "## Define Label for MOTIFs\n",
    "label1 = \"CoxL_form_1\"\n",
    "label2 = \"CoxL_form_2\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f6b464f6-a54a-403c-b001-37bccdfb33c2",
   "metadata": {},
   "source": [
    "<h3 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Search MOTIFs in CoxL Form 1 BLAST Results\n",
    "</h3>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "c9d0274a-77b7-4db6-9e70-20b822cd477f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Results saved to CoxL1_Results/MOTIF/coxl1_motif_search.csv\n"
     ]
    }
   ],
   "source": [
    "# Search MOTIFs for CoxL Form 1 BLAST results\n",
    "coxl1_motif_results = search_motifs(motif_1_proteins, form1, label1, form2, label2, output_file=f\"{motif1}/coxl1_motif_search.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "3e828a44-2738-456f-92fe-16316dc2610b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5145</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5146</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5147</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5148</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>QRVVALTIEPRSVLAARDPETGRLTVRMSTQMPSGVRDAICAAIGL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5149</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5150 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     File_Name                       ID  \\\n",
       "0     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "1     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "2     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "3     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "4     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "...                                        ...                      ...   \n",
       "5145  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5146  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5147  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5148  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5149  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "\n",
       "                                               Sequence    MOTIF  \\\n",
       "0     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "1     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "2     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "3     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "4     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "...                                                 ...      ...   \n",
       "5145  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  AYRGAGR   \n",
       "5146  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR   \n",
       "5147  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  AYRGAGR   \n",
       "5148  QRVVALTIEPRSVLAARDPETGRLTVRMSTQMPSGVRDAICAAIGL...  AYRGAGR   \n",
       "5149  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR   \n",
       "\n",
       "     Previous_3_AA  Motif_Label  \n",
       "0              ISF  CoxL_form_2  \n",
       "1              ISF  CoxL_form_2  \n",
       "2              ISF  CoxL_form_2  \n",
       "3              ISF  CoxL_form_2  \n",
       "4              ISF  CoxL_form_2  \n",
       "...            ...          ...  \n",
       "5145           AYR  CoxL_form_2  \n",
       "5146           LLS  CoxL_form_2  \n",
       "5147           AYR  CoxL_form_2  \n",
       "5148           AYR  CoxL_form_2  \n",
       "5149           LLS  CoxL_form_2  \n",
       "\n",
       "[5150 rows x 6 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking MOTIF search results\n",
    "coxl1_motif_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "0188d89f-bd82-438f-8bda-849383483f4f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['CoxL_form_2', 'CoxL_form_1'], dtype=object)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking for the presence of Form 1 and 2 MOTIFs in the results\n",
    "coxl1_motif_results['Motif_Label'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "ae673b76-88e9-4ee9-8a64-abcd51da4610",
   "metadata": {},
   "outputs": [],
   "source": [
    "coxl1_motif_output = coxl1_motif_results[coxl1_motif_results['Motif_Label'] == 'CoxL_form_1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "a4564f27-7c47-4984-af21-1021ef578b87",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "      <th>target_file</th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sequence_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626602.faa</td>\n",
       "      <td>Genome43_S3Ck127_66545_1</td>\n",
       "      <td>HKQWNYWARGSSTSSGAGRCARPSFRSALRPRRGRGCRTTCPRRTG...</td>\n",
       "      <td>NGSCSFR</td>\n",
       "      <td>NGS</td>\n",
       "      <td>CoxL_form_1</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626602.fasta</td>\n",
       "      <td>Genome43_S3Ck127_66545</td>\n",
       "      <td>199</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 File_Name                        ID  \\\n",
       "9  uncultured_Comamonas_sp._ERS6626602.faa  Genome43_S3Ck127_66545_1   \n",
       "\n",
       "                                            Sequence    MOTIF Previous_3_AA  \\\n",
       "9  HKQWNYWARGSSTSSGAGRCARPSFRSALRPRRGRGCRTTCPRRTG...  NGSCSFR           NGS   \n",
       "\n",
       "   Motif_Label                                target_file  \\\n",
       "9  CoxL_form_1  uncultured_Comamonas_sp._ERS6626602.fasta   \n",
       "\n",
       "                   qseqid  sequence_length  \n",
       "9  Genome43_S3Ck127_66545              199  "
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coxl1_motif_output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "b543e60a-1137-4e92-9350-cc0618ca7801",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "59"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking for number of unique genomes containing  CoxL Form 2 MOTIF\n",
    "coxl1_motif_results['File_Name'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "9055f1a1-31a4-4620-82ee-a3746934b832",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Genome228_k127_160184_1'"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking the structure of sequence ID\n",
    "coxl1_motif_results['ID'].unique()[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "9ece7c6f-48bf-4183-ae42-0ebfa66930a1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5145</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5146</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5147</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5148</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>QRVVALTIEPRSVLAARDPETGRLTVRMSTQMPSGVRDAICAAIGL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5149</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5150 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     File_Name                       ID  \\\n",
       "0     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "1     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "2     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "3     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "4     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "...                                        ...                      ...   \n",
       "5145  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5146  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5147  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5148  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5149  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "\n",
       "                                               Sequence    MOTIF  \\\n",
       "0     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "1     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "2     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "3     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "4     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "...                                                 ...      ...   \n",
       "5145  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  AYRGAGR   \n",
       "5146  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR   \n",
       "5147  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  AYRGAGR   \n",
       "5148  QRVVALTIEPRSVLAARDPETGRLTVRMSTQMPSGVRDAICAAIGL...  AYRGAGR   \n",
       "5149  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR   \n",
       "\n",
       "     Previous_3_AA  Motif_Label                                 target_file  \n",
       "0              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "1              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "2              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "3              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "4              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "...            ...          ...                                         ...  \n",
       "5145           AYR  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "5146           LLS  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "5147           AYR  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "5148           AYR  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "5149           LLS  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "\n",
       "[5150 rows x 7 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Adding a Column 'target_file' for easy analysis in further steps\n",
    "coxl1_motif_results['target_file'] = coxl1_motif_results['File_Name'].apply(lambda x: x[:-4] + '.fasta')\n",
    "coxl1_motif_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "c764e59d-e339-4a26-b5e6-b1883be6d6df",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "      <th>target_file</th>\n",
       "      <th>qseqid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5145</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5146</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5147</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5148</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>QRVVALTIEPRSVLAARDPETGRLTVRMSTQMPSGVRDAICAAIGL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5149</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5150 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     File_Name                       ID  \\\n",
       "0     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "1     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "2     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "3     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "4     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "...                                        ...                      ...   \n",
       "5145  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5146  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5147  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5148  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5149  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "\n",
       "                                               Sequence    MOTIF  \\\n",
       "0     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "1     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "2     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "3     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "4     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "...                                                 ...      ...   \n",
       "5145  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  AYRGAGR   \n",
       "5146  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR   \n",
       "5147  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  AYRGAGR   \n",
       "5148  QRVVALTIEPRSVLAARDPETGRLTVRMSTQMPSGVRDAICAAIGL...  AYRGAGR   \n",
       "5149  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR   \n",
       "\n",
       "     Previous_3_AA  Motif_Label                                 target_file  \\\n",
       "0              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "1              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "2              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "3              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "4              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "...            ...          ...                                         ...   \n",
       "5145           AYR  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "5146           LLS  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "5147           AYR  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "5148           AYR  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "5149           LLS  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "\n",
       "                     qseqid  \n",
       "0     Genome228_k127_160184  \n",
       "1     Genome228_k127_160184  \n",
       "2     Genome228_k127_160184  \n",
       "3     Genome228_k127_160184  \n",
       "4     Genome228_k127_160184  \n",
       "...                     ...  \n",
       "5145   Genome470_k127_43488  \n",
       "5146   Genome470_k127_43488  \n",
       "5147   Genome470_k127_43488  \n",
       "5148   Genome470_k127_43488  \n",
       "5149   Genome470_k127_43488  \n",
       "\n",
       "[5150 rows x 8 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Adding column 'qseqid' from the updated sequence ID coulmn in the dataframe\n",
    "coxl1_motif_results['qseqid'] = coxl1_motif_results['ID'].apply(lambda x: x[:-2])\n",
    "coxl1_motif_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "afa1e162-d04a-4322-b8be-4185b5295fcc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "      <th>target_file</th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sequence_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5145</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5146</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5147</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5148</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>QRVVALTIEPRSVLAARDPETGRLTVRMSTQMPSGVRDAICAAIGL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5149</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5150 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     File_Name                       ID  \\\n",
       "0     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "1     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "2     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "3     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "4     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "...                                        ...                      ...   \n",
       "5145  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5146  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5147  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5148  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5149  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "\n",
       "                                               Sequence    MOTIF  \\\n",
       "0     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "1     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "2     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "3     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "4     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "...                                                 ...      ...   \n",
       "5145  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  AYRGAGR   \n",
       "5146  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR   \n",
       "5147  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  AYRGAGR   \n",
       "5148  QRVVALTIEPRSVLAARDPETGRLTVRMSTQMPSGVRDAICAAIGL...  AYRGAGR   \n",
       "5149  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR   \n",
       "\n",
       "     Previous_3_AA  Motif_Label                                 target_file  \\\n",
       "0              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "1              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "2              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "3              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "4              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "...            ...          ...                                         ...   \n",
       "5145           AYR  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "5146           LLS  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "5147           AYR  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "5148           AYR  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "5149           LLS  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "\n",
       "                     qseqid  sequence_length  \n",
       "0     Genome228_k127_160184              197  \n",
       "1     Genome228_k127_160184              196  \n",
       "2     Genome228_k127_160184              196  \n",
       "3     Genome228_k127_160184              197  \n",
       "4     Genome228_k127_160184              197  \n",
       "...                     ...              ...  \n",
       "5145   Genome470_k127_43488              761  \n",
       "5146   Genome470_k127_43488              760  \n",
       "5147   Genome470_k127_43488              760  \n",
       "5148   Genome470_k127_43488              571  \n",
       "5149   Genome470_k127_43488               88  \n",
       "\n",
       "[5150 rows x 9 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Finding the length of every sequence and save the value in 'sequence length' column\n",
    "coxl1_motif_results['sequence_length'] = coxl1_motif_results['Sequence'].apply(len)\n",
    "coxl1_motif_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "bb62b170-f57b-4a03-b816-ce0878731b5a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "      <th>target_file</th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sequence_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5078</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>VRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARILSIDTS...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5080</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>VRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARILSIDTS...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5098</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>VRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARILSIDTS...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5108</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>QAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARILSID...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5149</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5150 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     File_Name                       ID  \\\n",
       "0     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "3     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "4     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "5     uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "11    uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "...                                        ...                      ...   \n",
       "5078  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5080  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5098  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5108  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "5149  uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "\n",
       "                                               Sequence    MOTIF  \\\n",
       "0     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "3     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "4     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "5     PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "11    PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR   \n",
       "...                                                 ...      ...   \n",
       "5078  VRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARILSIDTS...  LLSGAGR   \n",
       "5080  VRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARILSIDTS...  LLSGAGR   \n",
       "5098  VRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARILSIDTS...  LLSGAGR   \n",
       "5108  QAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARILSID...  LLSGAGR   \n",
       "5149  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR   \n",
       "\n",
       "     Previous_3_AA  Motif_Label                                 target_file  \\\n",
       "0              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "3              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "4              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "5              ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "11             ISF  CoxL_form_2  uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "...            ...          ...                                         ...   \n",
       "5078           LLS  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "5080           LLS  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "5098           LLS  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "5108           LLS  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "5149           LLS  CoxL_form_2  uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "\n",
       "                     qseqid  sequence_length  \n",
       "0     Genome228_k127_160184              197  \n",
       "3     Genome228_k127_160184              197  \n",
       "4     Genome228_k127_160184              197  \n",
       "5     Genome228_k127_160184              197  \n",
       "11    Genome228_k127_160184              197  \n",
       "...                     ...              ...  \n",
       "5078   Genome470_k127_43488              758  \n",
       "5080   Genome470_k127_43488              758  \n",
       "5098   Genome470_k127_43488              757  \n",
       "5108   Genome470_k127_43488              738  \n",
       "5149   Genome470_k127_43488               88  \n",
       "\n",
       "[5150 rows x 9 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter data frame based on 'File_Name', 'qseqid', 'MOTIF' & 'sequence_length'\n",
    "coxl1_motif_results = coxl1_motif_results.sort_values(by=['File_Name', 'qseqid', 'MOTIF', 'sequence_length'],\n",
    "                                        ascending=[True, True, True, False])\n",
    "coxl1_motif_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "8a17ec18-67de-45f6-be50-48a9dae8fa7f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "      <th>target_file</th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sequence_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.faa</td>\n",
       "      <td>Genome228_k127_160184_1</td>\n",
       "      <td>PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...</td>\n",
       "      <td>ISFGAGR</td>\n",
       "      <td>ISF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.faa</td>\n",
       "      <td>Genome495_k127_317059_1</td>\n",
       "      <td>MGASDFSKLPYIGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFV...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "      <td>Genome495_k127_317059</td>\n",
       "      <td>795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.faa</td>\n",
       "      <td>Genome262_k127_235961_1</td>\n",
       "      <td>DMNQTINLSISKKSAKAGDSIPHESAHLHVTGQATYIDDLPELENT...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.fasta</td>\n",
       "      <td>Genome262_k127_235961</td>\n",
       "      <td>776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626858.faa</td>\n",
       "      <td>Genome299_k127_417201_1</td>\n",
       "      <td>YLNAVELRNLRCKTNTVSNTAYRGFGGPQGMFVIENIIDDIARYLG...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626858.fasta</td>\n",
       "      <td>Genome299_k127_417201</td>\n",
       "      <td>444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Agrobacterium_sp._ERS6626766.faa</td>\n",
       "      <td>Genome207_k127_77525_1</td>\n",
       "      <td>MLPVFVGPAGPLVQAERGEAVVSAAGLRAAAVRDGPSAGVVLSPAA...</td>\n",
       "      <td>GCIGAGR</td>\n",
       "      <td>GCI</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Agrobacterium_sp._ERS6626766.fasta</td>\n",
       "      <td>Genome207_k127_77525</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>uncultured_Stenotrophomonas_sp._ERS6627319.faa</td>\n",
       "      <td>Genome543_k127_120424_1</td>\n",
       "      <td>RSKATYGGRVHGAGRMTASSGVAFGSASRLA*</td>\n",
       "      <td>RVHGAGR</td>\n",
       "      <td>RVH</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Stenotrophomonas_sp._ERS6627319.fasta</td>\n",
       "      <td>Genome543_k127_120424</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.faa</td>\n",
       "      <td>Genome220_k127_244806_1</td>\n",
       "      <td>KRTTLTTEPNPTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.fasta</td>\n",
       "      <td>Genome220_k127_244806</td>\n",
       "      <td>777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.faa</td>\n",
       "      <td>Genome220_k127_244806_1</td>\n",
       "      <td>KRTTLTTEPNPTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAH...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.fasta</td>\n",
       "      <td>Genome220_k127_244806</td>\n",
       "      <td>777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>KRTTLTTEPNPTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>KRTTLTTEPNPTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAH...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>777</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         File_Name                       ID  \\\n",
       "0         uncultured_Acidovorax_sp._ERS6626787.faa  Genome228_k127_160184_1   \n",
       "1         uncultured_Acidovorax_sp._ERS6627271.faa  Genome495_k127_317059_1   \n",
       "2      uncultured_Acinetobacter_sp._ERS6626821.faa  Genome262_k127_235961_1   \n",
       "3      uncultured_Acinetobacter_sp._ERS6626858.faa  Genome299_k127_417201_1   \n",
       "4      uncultured_Agrobacterium_sp._ERS6626766.faa   Genome207_k127_77525_1   \n",
       "..                                             ...                      ...   \n",
       "69  uncultured_Stenotrophomonas_sp._ERS6627319.faa  Genome543_k127_120424_1   \n",
       "70        uncultured_Variovorax_sp._ERS6626779.faa  Genome220_k127_244806_1   \n",
       "71        uncultured_Variovorax_sp._ERS6626779.faa  Genome220_k127_244806_1   \n",
       "72        uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "73        uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "\n",
       "                                             Sequence    MOTIF Previous_3_AA  \\\n",
       "0   PRRWSCWPRGSLISFGAGRYARPSRRRARGLLTGRVCRTTRPHRTG...  ISFGAGR           ISF   \n",
       "1   MGASDFSKLPYIGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFV...  AYRGAGR           AYR   \n",
       "2   DMNQTINLSISKKSAKAGDSIPHESAHLHVTGQATYIDDLPELENT...  EQPGAGR           EQP   \n",
       "3   YLNAVELRNLRCKTNTVSNTAYRGFGGPQGMFVIENIIDDIARYLG...  EQPGAGR           EQP   \n",
       "4   MLPVFVGPAGPLVQAERGEAVVSAAGLRAAAVRDGPSAGVVLSPAA...  GCIGAGR           GCI   \n",
       "..                                                ...      ...           ...   \n",
       "69                   RSKATYGGRVHGAGRMTASSGVAFGSASRLA*  RVHGAGR           RVH   \n",
       "70  KRTTLTTEPNPTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAH...  AYRGAGR           AYR   \n",
       "71  KRTTLTTEPNPTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAH...  LLSGAGR           LLS   \n",
       "72  KRTTLTTEPNPTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAH...  AYRGAGR           AYR   \n",
       "73  KRTTLTTEPNPTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAH...  LLSGAGR           LLS   \n",
       "\n",
       "    Motif_Label                                       target_file  \\\n",
       "0   CoxL_form_2        uncultured_Acidovorax_sp._ERS6626787.fasta   \n",
       "1   CoxL_form_2        uncultured_Acidovorax_sp._ERS6627271.fasta   \n",
       "2   CoxL_form_2     uncultured_Acinetobacter_sp._ERS6626821.fasta   \n",
       "3   CoxL_form_2     uncultured_Acinetobacter_sp._ERS6626858.fasta   \n",
       "4   CoxL_form_2     uncultured_Agrobacterium_sp._ERS6626766.fasta   \n",
       "..          ...                                               ...   \n",
       "69  CoxL_form_2  uncultured_Stenotrophomonas_sp._ERS6627319.fasta   \n",
       "70  CoxL_form_2        uncultured_Variovorax_sp._ERS6626779.fasta   \n",
       "71  CoxL_form_2        uncultured_Variovorax_sp._ERS6626779.fasta   \n",
       "72  CoxL_form_2        uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "73  CoxL_form_2        uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "\n",
       "                   qseqid  sequence_length  \n",
       "0   Genome228_k127_160184              197  \n",
       "1   Genome495_k127_317059              795  \n",
       "2   Genome262_k127_235961              776  \n",
       "3   Genome299_k127_417201              444  \n",
       "4    Genome207_k127_77525              220  \n",
       "..                    ...              ...  \n",
       "69  Genome543_k127_120424               32  \n",
       "70  Genome220_k127_244806              777  \n",
       "71  Genome220_k127_244806              777  \n",
       "72   Genome470_k127_43488              777  \n",
       "73   Genome470_k127_43488              777  \n",
       "\n",
       "[74 rows x 9 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Grouping data frame by 'File_Name', 'qseqid', 'MOTIF' and remove the duplicates, considering the 1st row of every group is the best match\n",
    "coxl1_motif_results = coxl1_motif_results.groupby(['File_Name', 'qseqid', 'MOTIF']).head(1).reset_index(drop=True)\n",
    "coxl1_motif_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "ae2f851b-1538-4be1-b596-873350720ed4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "      <th>target_file</th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sequence_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626602.faa</td>\n",
       "      <td>Genome43_S3Ck127_66545_1</td>\n",
       "      <td>HKQWNYWARGSSTSSGAGRCARPSFRSALRPRRGRGCRTTCPRRTG...</td>\n",
       "      <td>NGSCSFR</td>\n",
       "      <td>NGS</td>\n",
       "      <td>CoxL_form_1</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626602.fasta</td>\n",
       "      <td>Genome43_S3Ck127_66545</td>\n",
       "      <td>199</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 File_Name                        ID  \\\n",
       "9  uncultured_Comamonas_sp._ERS6626602.faa  Genome43_S3Ck127_66545_1   \n",
       "\n",
       "                                            Sequence    MOTIF Previous_3_AA  \\\n",
       "9  HKQWNYWARGSSTSSGAGRCARPSFRSALRPRRGRGCRTTCPRRTG...  NGSCSFR           NGS   \n",
       "\n",
       "   Motif_Label                                target_file  \\\n",
       "9  CoxL_form_1  uncultured_Comamonas_sp._ERS6626602.fasta   \n",
       "\n",
       "                   qseqid  sequence_length  \n",
       "9  Genome43_S3Ck127_66545              199  "
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coxl1_motif_results[coxl1_motif_results['Motif_Label'] == 'CoxL_form_1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "50c761db-51a2-4c56-91b4-907dfb8ea8c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save updated results\n",
    "coxl1_motif_results.to_csv(f\"{motif1}/coxl1_motif_results.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "d1b9b4b2-ebce-4c83-9533-b81edf08199c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['uncultured_Acidovorax_sp._ERS6626787.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6627271.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626821.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626858.fasta',\n",
       " 'uncultured_Agrobacterium_sp._ERS6626766.fasta',\n",
       " 'uncultured_Aureimonas_sp._ERS6626915.fasta',\n",
       " 'uncultured_Aureimonas_sp._ERS6627306.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626602.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626630.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626638.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626814.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6627283.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6626797.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6626820.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6626837.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6627293.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6627335.fasta',\n",
       " 'uncultured_Hymenobacter_sp._ERS6626733.fasta',\n",
       " 'uncultured_Hymenobacter_sp._ERS6627279.fasta',\n",
       " 'uncultured_Massilia_sp._ERS6627272.fasta',\n",
       " 'uncultured_Methylobacterium_sp._ERS6627288.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626906.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626964.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626873.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626923.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626962.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626972.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626981.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6627275.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6627290.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6626643.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6627005.fasta',\n",
       " 'uncultured_Pelomonas_sp._ERS6626569.fasta',\n",
       " 'uncultured_Pluralibacter_sp._ERS6626635.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626577.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626850.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626909.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627003.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627320.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627321.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626673.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627273.fasta',\n",
       " 'uncultured_Pseudorhodoferax_sp._ERS6626941.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626586.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626612.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626669.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626680.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626831.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626845.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626927.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626948.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626957.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627258.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627261.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6626983.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6627011.fasta',\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6627319.fasta',\n",
       " 'uncultured_Variovorax_sp._ERS6626779.fasta',\n",
       " 'uncultured_Variovorax_sp._ERS6627246.fasta']"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Generating a list of Genome file names\n",
    "coxl1_motif_mag_list = list(coxl1_motif_results['target_file'].unique())\n",
    "coxl1_motif_mag_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "8f906937-e785-4c50-b04a-5f212027ae46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "59"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking the length of the genomone file names list\n",
    "len(coxl1_motif_mag_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "5f348991-a963-4ae9-9554-c7bca9f6dc13",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'uncultured_Acidovorax_sp._ERS6626787.fasta': {'Genome228_k127_160184'},\n",
       " 'uncultured_Acidovorax_sp._ERS6627271.fasta': {'Genome495_k127_317059'},\n",
       " 'uncultured_Acinetobacter_sp._ERS6626821.fasta': {'Genome262_k127_235961'},\n",
       " 'uncultured_Acinetobacter_sp._ERS6626858.fasta': {'Genome299_k127_417201'},\n",
       " 'uncultured_Agrobacterium_sp._ERS6626766.fasta': {'Genome207_k127_77525'},\n",
       " 'uncultured_Aureimonas_sp._ERS6626915.fasta': {'Genome356_k127_339079',\n",
       "  'Genome356_k127_498042',\n",
       "  'Genome356_k127_610672'},\n",
       " 'uncultured_Aureimonas_sp._ERS6627306.fasta': {'Genome530_k127_16854'},\n",
       " 'uncultured_Comamonas_sp._ERS6626602.fasta': {'Genome43_S3Ck127_66545'},\n",
       " 'uncultured_Comamonas_sp._ERS6626630.fasta': {'Genome71_S1Ck127_18694'},\n",
       " 'uncultured_Comamonas_sp._ERS6626638.fasta': {'Genome79_S3Ck127_287303'},\n",
       " 'uncultured_Comamonas_sp._ERS6626814.fasta': {'Genome255_k127_46261'},\n",
       " 'uncultured_Comamonas_sp._ERS6627283.fasta': {'Genome507_k127_91973'},\n",
       " 'uncultured_Deinococcus_sp._ERS6626797.fasta': {'Genome238_k127_17753',\n",
       "  'Genome238_k127_54530'},\n",
       " 'uncultured_Deinococcus_sp._ERS6626820.fasta': {'Genome261_k127_146972',\n",
       "  'Genome261_k127_332399'},\n",
       " 'uncultured_Deinococcus_sp._ERS6626837.fasta': {'Genome278_k127_93298'},\n",
       " 'uncultured_Deinococcus_sp._ERS6627293.fasta': {'Genome517_k127_348199'},\n",
       " 'uncultured_Deinococcus_sp._ERS6627335.fasta': {'Genome559_k127_243715'},\n",
       " 'uncultured_Hymenobacter_sp._ERS6626733.fasta': {'Genome174_S1Ck127_276112'},\n",
       " 'uncultured_Hymenobacter_sp._ERS6627279.fasta': {'Genome503_k127_453582'},\n",
       " 'uncultured_Massilia_sp._ERS6627272.fasta': {'Genome496_k127_99185'},\n",
       " 'uncultured_Methylobacterium_sp._ERS6627288.fasta': {'Genome512_k127_160927',\n",
       "  'Genome512_k127_80099'},\n",
       " 'uncultured_Microbacterium_sp._ERS6626906.fasta': {'Genome347_k127_84820'},\n",
       " 'uncultured_Microbacterium_sp._ERS6626964.fasta': {'Genome405_k127_107645'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626873.fasta': {'Genome314_k127_168730'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626923.fasta': {'Genome364_k127_154890'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626962.fasta': {'Genome403_k127_249621'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626972.fasta': {'Genome413_k127_255610'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626981.fasta': {'Genome422_k127_88665'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6627275.fasta': {'Genome499_k127_471593'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6627290.fasta': {'Genome514_k127_4285'},\n",
       " 'uncultured_Pantoea_sp._ERS6626643.fasta': {'Genome84_S2Ck127_371734'},\n",
       " 'uncultured_Pantoea_sp._ERS6627005.fasta': {'Genome446_k127_205538'},\n",
       " 'uncultured_Pelomonas_sp._ERS6626569.fasta': {'Genome10_S2Ck127_387977'},\n",
       " 'uncultured_Pluralibacter_sp._ERS6626635.fasta': {'Genome76_S2Ck127_94054'},\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626577.fasta': {'Genome18_S1Ck127_217968'},\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626850.fasta': {'Genome291_k127_419550'},\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626909.fasta': {'Genome350_k127_213661'},\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627003.fasta': {'Genome444_k127_25498',\n",
       "  'Genome444_k127_2949'},\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627320.fasta': {'Genome544_k127_833201',\n",
       "  'Genome544_k127_857444',\n",
       "  'Genome544_k127_977136'},\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627321.fasta': {'Genome545_k127_208535',\n",
       "  'Genome545_k127_344480'},\n",
       " 'uncultured_Pseudomonas_sp._ERS6626673.fasta': {'Genome114_S3Ck127_13141'},\n",
       " 'uncultured_Pseudomonas_sp._ERS6627273.fasta': {'Genome497_k127_28836'},\n",
       " 'uncultured_Pseudorhodoferax_sp._ERS6626941.fasta': {'Genome382_k127_146993'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6626586.fasta': {'Genome27_S3Ck127_309455'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6626612.fasta': {'Genome53_S1Ck127_314311'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6626669.fasta': {'Genome110_S3Ck127_116759'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6626680.fasta': {'Genome121_S1Ck127_116016'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6626831.fasta': {'Genome272_k127_58089'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6626845.fasta': {'Genome286_k127_26698'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6626927.fasta': {'Genome368_k127_109295'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6626948.fasta': {'Genome389_k127_121851'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6626957.fasta': {'Genome398_k127_96590'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6627258.fasta': {'Genome482_k127_81292'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6627261.fasta': {'Genome485_k127_83547'},\n",
       " 'uncultured_Spirosoma_sp._ERS6626983.fasta': {'Genome424_k127_514686'},\n",
       " 'uncultured_Spirosoma_sp._ERS6627011.fasta': {'Genome452_k127_363663'},\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6627319.fasta': {'Genome543_k127_120424'},\n",
       " 'uncultured_Variovorax_sp._ERS6626779.fasta': {'Genome220_k127_244806'},\n",
       " 'uncultured_Variovorax_sp._ERS6627246.fasta': {'Genome470_k127_43488'}}"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Generating a set containing genome file names with their respective query sequence ID\n",
    "coxl1_motif_id_set = coxl1_motif_results.groupby('target_file')['qseqid'].apply(set).to_dict()\n",
    "coxl1_motif_id_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "cfc62544-acfd-420b-8223-b953792129a5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Target File: uncultured_Acidovorax_sp._ERS6626787.fasta\n",
      "  ID: Genome228_k127_160184, Previous 3 AA: ['ISF']\n",
      "Target File: uncultured_Acidovorax_sp._ERS6627271.fasta\n",
      "  ID: Genome495_k127_317059, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Acinetobacter_sp._ERS6626821.fasta\n",
      "  ID: Genome262_k127_235961, Previous 3 AA: ['EQP']\n",
      "Target File: uncultured_Acinetobacter_sp._ERS6626858.fasta\n",
      "  ID: Genome299_k127_417201, Previous 3 AA: ['EQP']\n",
      "Target File: uncultured_Agrobacterium_sp._ERS6626766.fasta\n",
      "  ID: Genome207_k127_77525, Previous 3 AA: ['GCI']\n",
      "Target File: uncultured_Aureimonas_sp._ERS6626915.fasta\n",
      "  ID: Genome356_k127_339079, Previous 3 AA: ['SGR']\n",
      "  ID: Genome356_k127_498042, Previous 3 AA: ['AYR']\n",
      "  ID: Genome356_k127_610672, Previous 3 AA: ['AGT']\n",
      "Target File: uncultured_Aureimonas_sp._ERS6627306.fasta\n",
      "  ID: Genome530_k127_16854, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Comamonas_sp._ERS6626602.fasta\n",
      "  ID: Genome43_S3Ck127_66545, Previous 3 AA: ['NGS', 'TSS']\n",
      "Target File: uncultured_Comamonas_sp._ERS6626630.fasta\n",
      "  ID: Genome71_S1Ck127_18694, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Comamonas_sp._ERS6626638.fasta\n",
      "  ID: Genome79_S3Ck127_287303, Previous 3 AA: ['PRP', 'RVS']\n",
      "Target File: uncultured_Comamonas_sp._ERS6626814.fasta\n",
      "  ID: Genome255_k127_46261, Previous 3 AA: ['PRP']\n",
      "Target File: uncultured_Comamonas_sp._ERS6627283.fasta\n",
      "  ID: Genome507_k127_91973, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Deinococcus_sp._ERS6626797.fasta\n",
      "  ID: Genome238_k127_17753, Previous 3 AA: ['AYR']\n",
      "  ID: Genome238_k127_54530, Previous 3 AA: ['GFG']\n",
      "Target File: uncultured_Deinococcus_sp._ERS6626820.fasta\n",
      "  ID: Genome261_k127_146972, Previous 3 AA: ['AYR']\n",
      "  ID: Genome261_k127_332399, Previous 3 AA: ['GFG']\n",
      "Target File: uncultured_Deinococcus_sp._ERS6626837.fasta\n",
      "  ID: Genome278_k127_93298, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Deinococcus_sp._ERS6627293.fasta\n",
      "  ID: Genome517_k127_348199, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Deinococcus_sp._ERS6627335.fasta\n",
      "  ID: Genome559_k127_243715, Previous 3 AA: ['GFG']\n",
      "Target File: uncultured_Hymenobacter_sp._ERS6626733.fasta\n",
      "  ID: Genome174_S1Ck127_276112, Previous 3 AA: ['SVI']\n",
      "Target File: uncultured_Hymenobacter_sp._ERS6627279.fasta\n",
      "  ID: Genome503_k127_453582, Previous 3 AA: ['SVI']\n",
      "Target File: uncultured_Massilia_sp._ERS6627272.fasta\n",
      "  ID: Genome496_k127_99185, Previous 3 AA: ['TGR']\n",
      "Target File: uncultured_Methylobacterium_sp._ERS6627288.fasta\n",
      "  ID: Genome512_k127_160927, Previous 3 AA: ['AYR']\n",
      "  ID: Genome512_k127_80099, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Microbacterium_sp._ERS6626906.fasta\n",
      "  ID: Genome347_k127_84820, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Microbacterium_sp._ERS6626964.fasta\n",
      "  ID: Genome405_k127_107645, Previous 3 AA: ['VGS']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6626873.fasta\n",
      "  ID: Genome314_k127_168730, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6626923.fasta\n",
      "  ID: Genome364_k127_154890, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6626962.fasta\n",
      "  ID: Genome403_k127_249621, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6626972.fasta\n",
      "  ID: Genome413_k127_255610, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6626981.fasta\n",
      "  ID: Genome422_k127_88665, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6627275.fasta\n",
      "  ID: Genome499_k127_471593, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6627290.fasta\n",
      "  ID: Genome514_k127_4285, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Pantoea_sp._ERS6626643.fasta\n",
      "  ID: Genome84_S2Ck127_371734, Previous 3 AA: ['GSE']\n",
      "Target File: uncultured_Pantoea_sp._ERS6627005.fasta\n",
      "  ID: Genome446_k127_205538, Previous 3 AA: ['ATV']\n",
      "Target File: uncultured_Pelomonas_sp._ERS6626569.fasta\n",
      "  ID: Genome10_S2Ck127_387977, Previous 3 AA: ['QVR']\n",
      "Target File: uncultured_Pluralibacter_sp._ERS6626635.fasta\n",
      "  ID: Genome76_S2Ck127_94054, Previous 3 AA: ['LAG']\n",
      "Target File: uncultured_Pseudacidovorax_sp._ERS6626577.fasta\n",
      "  ID: Genome18_S1Ck127_217968, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Pseudacidovorax_sp._ERS6626850.fasta\n",
      "  ID: Genome291_k127_419550, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Pseudacidovorax_sp._ERS6626909.fasta\n",
      "  ID: Genome350_k127_213661, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Pseudacidovorax_sp._ERS6627003.fasta\n",
      "  ID: Genome444_k127_25498, Previous 3 AA: ['AYR']\n",
      "  ID: Genome444_k127_2949, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Pseudacidovorax_sp._ERS6627320.fasta\n",
      "  ID: Genome544_k127_833201, Previous 3 AA: ['AYR']\n",
      "  ID: Genome544_k127_857444, Previous 3 AA: ['VAL']\n",
      "  ID: Genome544_k127_977136, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Pseudacidovorax_sp._ERS6627321.fasta\n",
      "  ID: Genome545_k127_208535, Previous 3 AA: ['AYR']\n",
      "  ID: Genome545_k127_344480, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Pseudomonas_sp._ERS6626673.fasta\n",
      "  ID: Genome114_S3Ck127_13141, Previous 3 AA: ['ASP']\n",
      "Target File: uncultured_Pseudomonas_sp._ERS6627273.fasta\n",
      "  ID: Genome497_k127_28836, Previous 3 AA: ['ASP', 'GGA']\n",
      "Target File: uncultured_Pseudorhodoferax_sp._ERS6626941.fasta\n",
      "  ID: Genome382_k127_146993, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6626586.fasta\n",
      "  ID: Genome27_S3Ck127_309455, Previous 3 AA: ['LEG', 'RAG']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6626612.fasta\n",
      "  ID: Genome53_S1Ck127_314311, Previous 3 AA: ['HTE']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6626669.fasta\n",
      "  ID: Genome110_S3Ck127_116759, Previous 3 AA: ['AAL']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6626680.fasta\n",
      "  ID: Genome121_S1Ck127_116016, Previous 3 AA: ['SGH']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6626831.fasta\n",
      "  ID: Genome272_k127_58089, Previous 3 AA: ['VST']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6626845.fasta\n",
      "  ID: Genome286_k127_26698, Previous 3 AA: ['HIV']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6626927.fasta\n",
      "  ID: Genome368_k127_109295, Previous 3 AA: ['GSG']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6626948.fasta\n",
      "  ID: Genome389_k127_121851, Previous 3 AA: ['AAC']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6626957.fasta\n",
      "  ID: Genome398_k127_96590, Previous 3 AA: ['RPR']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6627258.fasta\n",
      "  ID: Genome482_k127_81292, Previous 3 AA: ['VRI']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6627261.fasta\n",
      "  ID: Genome485_k127_83547, Previous 3 AA: ['ATV']\n",
      "Target File: uncultured_Spirosoma_sp._ERS6626983.fasta\n",
      "  ID: Genome424_k127_514686, Previous 3 AA: ['STA']\n",
      "Target File: uncultured_Spirosoma_sp._ERS6627011.fasta\n",
      "  ID: Genome452_k127_363663, Previous 3 AA: ['STA']\n",
      "Target File: uncultured_Stenotrophomonas_sp._ERS6627319.fasta\n",
      "  ID: Genome543_k127_120424, Previous 3 AA: ['RVH']\n",
      "Target File: uncultured_Variovorax_sp._ERS6626779.fasta\n",
      "  ID: Genome220_k127_244806, Previous 3 AA: ['AYR', 'LLS']\n",
      "Target File: uncultured_Variovorax_sp._ERS6627246.fasta\n",
      "  ID: Genome470_k127_43488, Previous 3 AA: ['AYR', 'LLS']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\dkb1g24\\AppData\\Local\\Temp\\ipykernel_13124\\1190382739.py:6: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  coxl1_motif_3aa_dict = coxl1_motif_3aa.groupby('target_file').apply(lambda x: x.set_index('qseqid')['Previous_3_AA'].to_dict()).to_dict()\n"
     ]
    }
   ],
   "source": [
    "# Generate a set within a set containing genome file names with their respective query sequences and the first three amino acid sequences of the MOTIF\n",
    "## Initialize a dictionary to hold the data\n",
    "coxl1_motif_3aa = coxl1_motif_results.groupby(['target_file', 'qseqid'])['Previous_3_AA'].apply(list).reset_index()\n",
    "\n",
    "## Create the dictionary\n",
    "coxl1_motif_3aa_dict = coxl1_motif_3aa.groupby('target_file').apply(lambda x: x.set_index('qseqid')['Previous_3_AA'].to_dict()).to_dict()\n",
    "\n",
    "## Print the resulting dictionary\n",
    "for target_file, ids in coxl1_motif_3aa_dict.items():\n",
    "    print(f\"Target File: {target_file}\")\n",
    "    for id, previous_3_aa in ids.items():\n",
    "        print(f\"  ID: {id}, Previous 3 AA: {previous_3_aa}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e0bf661-c928-485d-89ff-017c7ff15591",
   "metadata": {},
   "source": [
    "<h3 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Filter CoxL Form 1 BLAST Results with MOTIF Search Results\n",
    "</h3>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "df134060-cc8d-4bd2-9e09-4948cbdac50d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Mycobacterium_marinum_M</td>\n",
       "      <td>24.634</td>\n",
       "      <td>42</td>\n",
       "      <td>5.050000e-43</td>\n",
       "      <td>163.0</td>\n",
       "      <td>4980</td>\n",
       "      <td>2872</td>\n",
       "      <td>53</td>\n",
       "      <td>784</td>\n",
       "      <td>751</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Mycobacterium_tusciae_WP_006247553</td>\n",
       "      <td>24.433</td>\n",
       "      <td>42</td>\n",
       "      <td>2.450000e-41</td>\n",
       "      <td>158.0</td>\n",
       "      <td>4980</td>\n",
       "      <td>2872</td>\n",
       "      <td>53</td>\n",
       "      <td>784</td>\n",
       "      <td>749</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Rhodococcus_opacus_B4</td>\n",
       "      <td>24.074</td>\n",
       "      <td>39</td>\n",
       "      <td>4.120000e-41</td>\n",
       "      <td>157.0</td>\n",
       "      <td>4980</td>\n",
       "      <td>3022</td>\n",
       "      <td>56</td>\n",
       "      <td>735</td>\n",
       "      <td>702</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Micromonospora_aurantiaca_ATCC_27029</td>\n",
       "      <td>24.536</td>\n",
       "      <td>42</td>\n",
       "      <td>4.150000e-41</td>\n",
       "      <td>157.0</td>\n",
       "      <td>4980</td>\n",
       "      <td>2872</td>\n",
       "      <td>53</td>\n",
       "      <td>784</td>\n",
       "      <td>754</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>Mycobacterium_ulcerans_Agy99_YP_904347</td>\n",
       "      <td>24.528</td>\n",
       "      <td>42</td>\n",
       "      <td>5.750000e-41</td>\n",
       "      <td>156.0</td>\n",
       "      <td>4953</td>\n",
       "      <td>2872</td>\n",
       "      <td>62</td>\n",
       "      <td>784</td>\n",
       "      <td>742</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    qseqid                                  sseqid  pident  \\\n",
       "0  Genome20_S3Ck127_316360                 Mycobacterium_marinum_M  24.634   \n",
       "1  Genome20_S3Ck127_316360      Mycobacterium_tusciae_WP_006247553  24.433   \n",
       "2  Genome20_S3Ck127_316360                   Rhodococcus_opacus_B4  24.074   \n",
       "3  Genome20_S3Ck127_316360    Micromonospora_aurantiaca_ATCC_27029  24.536   \n",
       "4  Genome20_S3Ck127_316360  Mycobacterium_ulcerans_Agy99_YP_904347  24.528   \n",
       "\n",
       "   qcovs        evalue  bitscore  qstart  qend  sstart  send  length  \\\n",
       "0     42  5.050000e-43     163.0    4980  2872      53   784     751   \n",
       "1     42  2.450000e-41     158.0    4980  2872      53   784     749   \n",
       "2     39  4.120000e-41     157.0    4980  3022      56   735     702   \n",
       "3     42  4.150000e-41     157.0    4980  2872      53   784     754   \n",
       "4     42  5.750000e-41     156.0    4953  2872      62   784     742   \n",
       "\n",
       "                                         source_file  \\\n",
       "0  Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "1  Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "2  Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "3  Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "4  Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "\n",
       "                                     target_file  \n",
       "0  Bdellovibrionaceae_bacterium_ERS6626579.fasta  \n",
       "1  Bdellovibrionaceae_bacterium_ERS6626579.fasta  \n",
       "2  Bdellovibrionaceae_bacterium_ERS6626579.fasta  \n",
       "3  Bdellovibrionaceae_bacterium_ERS6626579.fasta  \n",
       "4  Bdellovibrionaceae_bacterium_ERS6626579.fasta  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Open the MOTIF 1 concatenated BLAST results\n",
    "coxl1_csv = open_file(motif_1_csv_file_path)\n",
    "coxl1_csv.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "e35d1fd3-7ae1-4ee8-9284-d499afb12532",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5188</th>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>Pseudonocardia_dioxanivorans_CB1190_coxL2_(651...</td>\n",
       "      <td>36.242</td>\n",
       "      <td>0</td>\n",
       "      <td>2.990000e-15</td>\n",
       "      <td>79.7</td>\n",
       "      <td>33523</td>\n",
       "      <td>33092</td>\n",
       "      <td>644</td>\n",
       "      <td>792</td>\n",
       "      <td>149</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5189</th>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>Amycolatopsis_thermoflava_N1165_DSM_44574_2512...</td>\n",
       "      <td>34.228</td>\n",
       "      <td>1</td>\n",
       "      <td>8.420000e-13</td>\n",
       "      <td>71.6</td>\n",
       "      <td>33523</td>\n",
       "      <td>33110</td>\n",
       "      <td>632</td>\n",
       "      <td>777</td>\n",
       "      <td>149</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5190</th>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>Amycolatopsis_thermoflava_N1165_DSM_44574_2512...</td>\n",
       "      <td>28.788</td>\n",
       "      <td>1</td>\n",
       "      <td>5.960000e-07</td>\n",
       "      <td>52.4</td>\n",
       "      <td>34717</td>\n",
       "      <td>34043</td>\n",
       "      <td>26</td>\n",
       "      <td>285</td>\n",
       "      <td>264</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5191</th>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>Solirubrobacter_soli_DSM_1</td>\n",
       "      <td>33.566</td>\n",
       "      <td>0</td>\n",
       "      <td>1.040000e-12</td>\n",
       "      <td>71.2</td>\n",
       "      <td>33523</td>\n",
       "      <td>33110</td>\n",
       "      <td>632</td>\n",
       "      <td>774</td>\n",
       "      <td>143</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5192</th>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>Solirubrobacter_sp._URHD008_1</td>\n",
       "      <td>34.416</td>\n",
       "      <td>0</td>\n",
       "      <td>1.080000e-12</td>\n",
       "      <td>71.2</td>\n",
       "      <td>33523</td>\n",
       "      <td>33086</td>\n",
       "      <td>621</td>\n",
       "      <td>774</td>\n",
       "      <td>154</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96485</th>\n",
       "      <td>Genome470_k127_212027</td>\n",
       "      <td>Halobellus_limi_CGMCC_1.10331_:_Ga0070554_102_...</td>\n",
       "      <td>47.500</td>\n",
       "      <td>4</td>\n",
       "      <td>1.390000e-07</td>\n",
       "      <td>48.1</td>\n",
       "      <td>1</td>\n",
       "      <td>120</td>\n",
       "      <td>762</td>\n",
       "      <td>801</td>\n",
       "      <td>40</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96486</th>\n",
       "      <td>Genome470_k127_212027</td>\n",
       "      <td>Pseudonocardia_autotrophica_2</td>\n",
       "      <td>55.000</td>\n",
       "      <td>4</td>\n",
       "      <td>1.920000e-07</td>\n",
       "      <td>47.8</td>\n",
       "      <td>1</td>\n",
       "      <td>120</td>\n",
       "      <td>737</td>\n",
       "      <td>773</td>\n",
       "      <td>40</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96487</th>\n",
       "      <td>Genome470_k127_212027</td>\n",
       "      <td>Pseduonocardia_autotrophica_1</td>\n",
       "      <td>50.000</td>\n",
       "      <td>4</td>\n",
       "      <td>3.370000e-07</td>\n",
       "      <td>47.0</td>\n",
       "      <td>1</td>\n",
       "      <td>120</td>\n",
       "      <td>737</td>\n",
       "      <td>776</td>\n",
       "      <td>40</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96488</th>\n",
       "      <td>Genome470_k127_212027</td>\n",
       "      <td>Alkalilimnicola_ehrlichei_MLHE-1_YP_742401</td>\n",
       "      <td>50.000</td>\n",
       "      <td>4</td>\n",
       "      <td>5.070000e-07</td>\n",
       "      <td>46.6</td>\n",
       "      <td>1</td>\n",
       "      <td>120</td>\n",
       "      <td>750</td>\n",
       "      <td>789</td>\n",
       "      <td>40</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96489</th>\n",
       "      <td>Genome470_k127_212027</td>\n",
       "      <td>Nocardioides_sp._JS614_2</td>\n",
       "      <td>51.282</td>\n",
       "      <td>4</td>\n",
       "      <td>7.040000e-07</td>\n",
       "      <td>45.8</td>\n",
       "      <td>1</td>\n",
       "      <td>117</td>\n",
       "      <td>746</td>\n",
       "      <td>784</td>\n",
       "      <td>39</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21388 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      qseqid  \\\n",
       "5188   Genome228_k127_160184   \n",
       "5189   Genome228_k127_160184   \n",
       "5190   Genome228_k127_160184   \n",
       "5191   Genome228_k127_160184   \n",
       "5192   Genome228_k127_160184   \n",
       "...                      ...   \n",
       "96485  Genome470_k127_212027   \n",
       "96486  Genome470_k127_212027   \n",
       "96487  Genome470_k127_212027   \n",
       "96488  Genome470_k127_212027   \n",
       "96489  Genome470_k127_212027   \n",
       "\n",
       "                                                  sseqid  pident  qcovs  \\\n",
       "5188   Pseudonocardia_dioxanivorans_CB1190_coxL2_(651...  36.242      0   \n",
       "5189   Amycolatopsis_thermoflava_N1165_DSM_44574_2512...  34.228      1   \n",
       "5190   Amycolatopsis_thermoflava_N1165_DSM_44574_2512...  28.788      1   \n",
       "5191                          Solirubrobacter_soli_DSM_1  33.566      0   \n",
       "5192                       Solirubrobacter_sp._URHD008_1  34.416      0   \n",
       "...                                                  ...     ...    ...   \n",
       "96485  Halobellus_limi_CGMCC_1.10331_:_Ga0070554_102_...  47.500      4   \n",
       "96486                      Pseudonocardia_autotrophica_2  55.000      4   \n",
       "96487                      Pseduonocardia_autotrophica_1  50.000      4   \n",
       "96488         Alkalilimnicola_ehrlichei_MLHE-1_YP_742401  50.000      4   \n",
       "96489                           Nocardioides_sp._JS614_2  51.282      4   \n",
       "\n",
       "             evalue  bitscore  qstart   qend  sstart  send  length  \\\n",
       "5188   2.990000e-15      79.7   33523  33092     644   792     149   \n",
       "5189   8.420000e-13      71.6   33523  33110     632   777     149   \n",
       "5190   5.960000e-07      52.4   34717  34043      26   285     264   \n",
       "5191   1.040000e-12      71.2   33523  33110     632   774     143   \n",
       "5192   1.080000e-12      71.2   33523  33086     621   774     154   \n",
       "...             ...       ...     ...    ...     ...   ...     ...   \n",
       "96485  1.390000e-07      48.1       1    120     762   801      40   \n",
       "96486  1.920000e-07      47.8       1    120     737   773      40   \n",
       "96487  3.370000e-07      47.0       1    120     737   776      40   \n",
       "96488  5.070000e-07      46.6       1    120     750   789      40   \n",
       "96489  7.040000e-07      45.8       1    117     746   784      39   \n",
       "\n",
       "                                           source_file  \\\n",
       "5188   uncultured_Acidovorax_sp._ERS6626787_scores.csv   \n",
       "5189   uncultured_Acidovorax_sp._ERS6626787_scores.csv   \n",
       "5190   uncultured_Acidovorax_sp._ERS6626787_scores.csv   \n",
       "5191   uncultured_Acidovorax_sp._ERS6626787_scores.csv   \n",
       "5192   uncultured_Acidovorax_sp._ERS6626787_scores.csv   \n",
       "...                                                ...   \n",
       "96485  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "96486  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "96487  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "96488  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "96489  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "\n",
       "                                      target_file  \n",
       "5188   uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "5189   uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "5190   uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "5191   uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "5192   uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "...                                           ...  \n",
       "96485  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "96486  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "96487  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "96488  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "96489  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "\n",
       "[21388 rows x 13 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter the BLAST results with the genome file names list, to remove the other genomes results\n",
    "coxl1_csv_filtered = coxl1_csv[coxl1_csv['target_file'].isin(coxl1_motif_mag_list)]\n",
    "coxl1_csv_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "e7dbf222-5464-4002-9837-201a54a4f564",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5188</th>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>Pseudonocardia_dioxanivorans_CB1190_coxL2_(651...</td>\n",
       "      <td>36.242</td>\n",
       "      <td>0</td>\n",
       "      <td>2.990000e-15</td>\n",
       "      <td>79.7</td>\n",
       "      <td>33523</td>\n",
       "      <td>33092</td>\n",
       "      <td>644</td>\n",
       "      <td>792</td>\n",
       "      <td>149</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5189</th>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>Amycolatopsis_thermoflava_N1165_DSM_44574_2512...</td>\n",
       "      <td>34.228</td>\n",
       "      <td>1</td>\n",
       "      <td>8.420000e-13</td>\n",
       "      <td>71.6</td>\n",
       "      <td>33523</td>\n",
       "      <td>33110</td>\n",
       "      <td>632</td>\n",
       "      <td>777</td>\n",
       "      <td>149</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5191</th>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>Solirubrobacter_soli_DSM_1</td>\n",
       "      <td>33.566</td>\n",
       "      <td>0</td>\n",
       "      <td>1.040000e-12</td>\n",
       "      <td>71.2</td>\n",
       "      <td>33523</td>\n",
       "      <td>33110</td>\n",
       "      <td>632</td>\n",
       "      <td>774</td>\n",
       "      <td>143</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5192</th>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>Solirubrobacter_sp._URHD008_1</td>\n",
       "      <td>34.416</td>\n",
       "      <td>0</td>\n",
       "      <td>1.080000e-12</td>\n",
       "      <td>71.2</td>\n",
       "      <td>33523</td>\n",
       "      <td>33086</td>\n",
       "      <td>621</td>\n",
       "      <td>774</td>\n",
       "      <td>154</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5193</th>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>Pseudonocardia_autotrophica_2</td>\n",
       "      <td>32.215</td>\n",
       "      <td>0</td>\n",
       "      <td>2.380000e-12</td>\n",
       "      <td>70.1</td>\n",
       "      <td>33523</td>\n",
       "      <td>33092</td>\n",
       "      <td>630</td>\n",
       "      <td>778</td>\n",
       "      <td>149</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96140</th>\n",
       "      <td>Genome470_k127_46816</td>\n",
       "      <td>Alicyclobacillus_pomorum_DSM_14955_2513792691</td>\n",
       "      <td>22.444</td>\n",
       "      <td>19</td>\n",
       "      <td>2.300000e-07</td>\n",
       "      <td>48.5</td>\n",
       "      <td>4199</td>\n",
       "      <td>5248</td>\n",
       "      <td>23</td>\n",
       "      <td>399</td>\n",
       "      <td>401</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96141</th>\n",
       "      <td>Genome470_k127_46816</td>\n",
       "      <td>Conexibacter_woesei_DSM_14684_(YP_003394821)</td>\n",
       "      <td>27.511</td>\n",
       "      <td>12</td>\n",
       "      <td>3.230000e-07</td>\n",
       "      <td>48.1</td>\n",
       "      <td>4598</td>\n",
       "      <td>5266</td>\n",
       "      <td>196</td>\n",
       "      <td>405</td>\n",
       "      <td>229</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96142</th>\n",
       "      <td>Genome470_k127_46816</td>\n",
       "      <td>Micromonospora_aurantiaca_ATCC_27029</td>\n",
       "      <td>23.614</td>\n",
       "      <td>20</td>\n",
       "      <td>3.980000e-07</td>\n",
       "      <td>47.8</td>\n",
       "      <td>4172</td>\n",
       "      <td>5266</td>\n",
       "      <td>23</td>\n",
       "      <td>415</td>\n",
       "      <td>415</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96143</th>\n",
       "      <td>Genome470_k127_46816</td>\n",
       "      <td>Thermocrispum_municipale_coxL2</td>\n",
       "      <td>23.833</td>\n",
       "      <td>20</td>\n",
       "      <td>4.750000e-07</td>\n",
       "      <td>47.8</td>\n",
       "      <td>4199</td>\n",
       "      <td>5266</td>\n",
       "      <td>27</td>\n",
       "      <td>410</td>\n",
       "      <td>407</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96144</th>\n",
       "      <td>Genome470_k127_46816</td>\n",
       "      <td>Halobellus_limi_CGMCC_1.10331_:_Ga0070554_102_...</td>\n",
       "      <td>28.221</td>\n",
       "      <td>9</td>\n",
       "      <td>9.390000e-07</td>\n",
       "      <td>46.6</td>\n",
       "      <td>4556</td>\n",
       "      <td>5026</td>\n",
       "      <td>206</td>\n",
       "      <td>356</td>\n",
       "      <td>163</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21388 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      qseqid  \\\n",
       "5188   Genome228_k127_160184   \n",
       "5189   Genome228_k127_160184   \n",
       "5191   Genome228_k127_160184   \n",
       "5192   Genome228_k127_160184   \n",
       "5193   Genome228_k127_160184   \n",
       "...                      ...   \n",
       "96140   Genome470_k127_46816   \n",
       "96141   Genome470_k127_46816   \n",
       "96142   Genome470_k127_46816   \n",
       "96143   Genome470_k127_46816   \n",
       "96144   Genome470_k127_46816   \n",
       "\n",
       "                                                  sseqid  pident  qcovs  \\\n",
       "5188   Pseudonocardia_dioxanivorans_CB1190_coxL2_(651...  36.242      0   \n",
       "5189   Amycolatopsis_thermoflava_N1165_DSM_44574_2512...  34.228      1   \n",
       "5191                          Solirubrobacter_soli_DSM_1  33.566      0   \n",
       "5192                       Solirubrobacter_sp._URHD008_1  34.416      0   \n",
       "5193                       Pseudonocardia_autotrophica_2  32.215      0   \n",
       "...                                                  ...     ...    ...   \n",
       "96140      Alicyclobacillus_pomorum_DSM_14955_2513792691  22.444     19   \n",
       "96141       Conexibacter_woesei_DSM_14684_(YP_003394821)  27.511     12   \n",
       "96142               Micromonospora_aurantiaca_ATCC_27029  23.614     20   \n",
       "96143                     Thermocrispum_municipale_coxL2  23.833     20   \n",
       "96144  Halobellus_limi_CGMCC_1.10331_:_Ga0070554_102_...  28.221      9   \n",
       "\n",
       "             evalue  bitscore  qstart   qend  sstart  send  length  \\\n",
       "5188   2.990000e-15      79.7   33523  33092     644   792     149   \n",
       "5189   8.420000e-13      71.6   33523  33110     632   777     149   \n",
       "5191   1.040000e-12      71.2   33523  33110     632   774     143   \n",
       "5192   1.080000e-12      71.2   33523  33086     621   774     154   \n",
       "5193   2.380000e-12      70.1   33523  33092     630   778     149   \n",
       "...             ...       ...     ...    ...     ...   ...     ...   \n",
       "96140  2.300000e-07      48.5    4199   5248      23   399     401   \n",
       "96141  3.230000e-07      48.1    4598   5266     196   405     229   \n",
       "96142  3.980000e-07      47.8    4172   5266      23   415     415   \n",
       "96143  4.750000e-07      47.8    4199   5266      27   410     407   \n",
       "96144  9.390000e-07      46.6    4556   5026     206   356     163   \n",
       "\n",
       "                                           source_file  \\\n",
       "5188   uncultured_Acidovorax_sp._ERS6626787_scores.csv   \n",
       "5189   uncultured_Acidovorax_sp._ERS6626787_scores.csv   \n",
       "5191   uncultured_Acidovorax_sp._ERS6626787_scores.csv   \n",
       "5192   uncultured_Acidovorax_sp._ERS6626787_scores.csv   \n",
       "5193   uncultured_Acidovorax_sp._ERS6626787_scores.csv   \n",
       "...                                                ...   \n",
       "96140  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "96141  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "96142  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "96143  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "96144  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "\n",
       "                                      target_file  \n",
       "5188   uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "5189   uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "5191   uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "5192   uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "5193   uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "...                                           ...  \n",
       "96140  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "96141  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "96142  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "96143  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "96144  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "\n",
       "[21388 rows x 13 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Sort the blast results based on genome file name ('target_file'), query sequence id ('qseqid'), E-value, Bitscore, & Percentage Identity ('pident') \n",
    "coxl1_csv_sorted = coxl1_csv_filtered.sort_values(by=['target_file', 'qseqid', 'evalue', 'bitscore', 'pident'],\n",
    "                                        ascending=[True, True, True, False, False])\n",
    "\n",
    "coxl1_csv_sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "5e7cad62-2a14-4529-8ddb-a0e77e19c009",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>Pseudonocardia_dioxanivorans_CB1190_coxL2_(651...</td>\n",
       "      <td>36.242</td>\n",
       "      <td>0</td>\n",
       "      <td>2.990000e-15</td>\n",
       "      <td>79.7</td>\n",
       "      <td>33523</td>\n",
       "      <td>33092</td>\n",
       "      <td>644</td>\n",
       "      <td>792</td>\n",
       "      <td>149</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Genome228_k127_35392</td>\n",
       "      <td>Meiothermus_rufus_DSM_22234_2523182758</td>\n",
       "      <td>26.582</td>\n",
       "      <td>3</td>\n",
       "      <td>7.500000e-36</td>\n",
       "      <td>145.0</td>\n",
       "      <td>48735</td>\n",
       "      <td>46567</td>\n",
       "      <td>16</td>\n",
       "      <td>787</td>\n",
       "      <td>790</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Genome495_k127_217386</td>\n",
       "      <td>Sulfolobus_solfataricus_98/2</td>\n",
       "      <td>23.488</td>\n",
       "      <td>10</td>\n",
       "      <td>7.110000e-13</td>\n",
       "      <td>67.8</td>\n",
       "      <td>30</td>\n",
       "      <td>1124</td>\n",
       "      <td>28</td>\n",
       "      <td>423</td>\n",
       "      <td>430</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Genome495_k127_317059</td>\n",
       "      <td>Mycobacterium_smegmatis_MC2_155</td>\n",
       "      <td>43.511</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>591.0</td>\n",
       "      <td>29003</td>\n",
       "      <td>31342</td>\n",
       "      <td>27</td>\n",
       "      <td>796</td>\n",
       "      <td>786</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Genome495_k127_374228</td>\n",
       "      <td>BKZW01_10191_Dictyobacter_vulcani_W12</td>\n",
       "      <td>25.616</td>\n",
       "      <td>15</td>\n",
       "      <td>1.010000e-44</td>\n",
       "      <td>170.0</td>\n",
       "      <td>9123</td>\n",
       "      <td>7084</td>\n",
       "      <td>20</td>\n",
       "      <td>720</td>\n",
       "      <td>730</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>Genome470_k127_290350</td>\n",
       "      <td>Geoarchaeota_archaeon_OSPB-1_2504814364</td>\n",
       "      <td>25.548</td>\n",
       "      <td>10</td>\n",
       "      <td>6.960000e-41</td>\n",
       "      <td>159.0</td>\n",
       "      <td>1764</td>\n",
       "      <td>3992</td>\n",
       "      <td>23</td>\n",
       "      <td>764</td>\n",
       "      <td>775</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>Genome470_k127_338375</td>\n",
       "      <td>BKZW01_10191_Dictyobacter_vulcani_W12</td>\n",
       "      <td>41.550</td>\n",
       "      <td>53</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>595.0</td>\n",
       "      <td>2342</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>777</td>\n",
       "      <td>787</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>Genome470_k127_380329</td>\n",
       "      <td>BKZW01_10191_Dictyobacter_vulcani_W12</td>\n",
       "      <td>27.982</td>\n",
       "      <td>4</td>\n",
       "      <td>2.830000e-35</td>\n",
       "      <td>142.0</td>\n",
       "      <td>27723</td>\n",
       "      <td>25837</td>\n",
       "      <td>100</td>\n",
       "      <td>771</td>\n",
       "      <td>679</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>BKZW01_10191_Dictyobacter_vulcani_W12</td>\n",
       "      <td>36.086</td>\n",
       "      <td>32</td>\n",
       "      <td>5.640000e-125</td>\n",
       "      <td>408.0</td>\n",
       "      <td>1355</td>\n",
       "      <td>3637</td>\n",
       "      <td>8</td>\n",
       "      <td>776</td>\n",
       "      <td>787</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>Genome470_k127_46816</td>\n",
       "      <td>Sulfolobus_islandicus_YN1551</td>\n",
       "      <td>23.039</td>\n",
       "      <td>20</td>\n",
       "      <td>1.100000e-10</td>\n",
       "      <td>59.3</td>\n",
       "      <td>4169</td>\n",
       "      <td>5251</td>\n",
       "      <td>17</td>\n",
       "      <td>402</td>\n",
       "      <td>408</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>264 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    qseqid                                             sseqid  \\\n",
       "0    Genome228_k127_160184  Pseudonocardia_dioxanivorans_CB1190_coxL2_(651...   \n",
       "1     Genome228_k127_35392             Meiothermus_rufus_DSM_22234_2523182758   \n",
       "2    Genome495_k127_217386                       Sulfolobus_solfataricus_98/2   \n",
       "3    Genome495_k127_317059                    Mycobacterium_smegmatis_MC2_155   \n",
       "4    Genome495_k127_374228              BKZW01_10191_Dictyobacter_vulcani_W12   \n",
       "..                     ...                                                ...   \n",
       "259  Genome470_k127_290350            Geoarchaeota_archaeon_OSPB-1_2504814364   \n",
       "260  Genome470_k127_338375              BKZW01_10191_Dictyobacter_vulcani_W12   \n",
       "261  Genome470_k127_380329              BKZW01_10191_Dictyobacter_vulcani_W12   \n",
       "262   Genome470_k127_43488              BKZW01_10191_Dictyobacter_vulcani_W12   \n",
       "263   Genome470_k127_46816                       Sulfolobus_islandicus_YN1551   \n",
       "\n",
       "     pident  qcovs         evalue  bitscore  qstart   qend  sstart  send  \\\n",
       "0    36.242      0   2.990000e-15      79.7   33523  33092     644   792   \n",
       "1    26.582      3   7.500000e-36     145.0   48735  46567      16   787   \n",
       "2    23.488     10   7.110000e-13      67.8      30   1124      28   423   \n",
       "3    43.511      7   0.000000e+00     591.0   29003  31342      27   796   \n",
       "4    25.616     15   1.010000e-44     170.0    9123   7084      20   720   \n",
       "..      ...    ...            ...       ...     ...    ...     ...   ...   \n",
       "259  25.548     10   6.960000e-41     159.0    1764   3992      23   764   \n",
       "260  41.550     53   0.000000e+00     595.0    2342      6       7   777   \n",
       "261  27.982      4   2.830000e-35     142.0   27723  25837     100   771   \n",
       "262  36.086     32  5.640000e-125     408.0    1355   3637       8   776   \n",
       "263  23.039     20   1.100000e-10      59.3    4169   5251      17   402   \n",
       "\n",
       "     length                                      source_file  \\\n",
       "0       149  uncultured_Acidovorax_sp._ERS6626787_scores.csv   \n",
       "1       790  uncultured_Acidovorax_sp._ERS6626787_scores.csv   \n",
       "2       430  uncultured_Acidovorax_sp._ERS6627271_scores.csv   \n",
       "3       786  uncultured_Acidovorax_sp._ERS6627271_scores.csv   \n",
       "4       730  uncultured_Acidovorax_sp._ERS6627271_scores.csv   \n",
       "..      ...                                              ...   \n",
       "259     775  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "260     787  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "261     679  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "262     787  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "263     408  uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "\n",
       "                                    target_file  \n",
       "0    uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "1    uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "2    uncultured_Acidovorax_sp._ERS6627271.fasta  \n",
       "3    uncultured_Acidovorax_sp._ERS6627271.fasta  \n",
       "4    uncultured_Acidovorax_sp._ERS6627271.fasta  \n",
       "..                                          ...  \n",
       "259  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "260  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "261  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "262  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "263  uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "\n",
       "[264 rows x 13 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Group the data frame by 'target_file' & 'qseqid' and considering the 1st row of every group is the best results, removing other duplicates\n",
    "coxl1_csv_sorted = coxl1_csv_sorted.groupby(['target_file', 'qseqid']).head(1).reset_index(drop=True)\n",
    "coxl1_csv_sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "1d01eab8-2656-4361-a152-07c081d5ae58",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome228_k127_160184</td>\n",
       "      <td>Pseudonocardia_dioxanivorans_CB1190_coxL2_(651...</td>\n",
       "      <td>36.242</td>\n",
       "      <td>0</td>\n",
       "      <td>2.990000e-15</td>\n",
       "      <td>79.7</td>\n",
       "      <td>33523</td>\n",
       "      <td>33092</td>\n",
       "      <td>644</td>\n",
       "      <td>792</td>\n",
       "      <td>149</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6626787.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Genome495_k127_317059</td>\n",
       "      <td>Mycobacterium_smegmatis_MC2_155</td>\n",
       "      <td>43.511</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>591.0</td>\n",
       "      <td>29003</td>\n",
       "      <td>31342</td>\n",
       "      <td>27</td>\n",
       "      <td>796</td>\n",
       "      <td>786</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Genome262_k127_235961</td>\n",
       "      <td>BKZW01_10191_Dictyobacter_vulcani_W12</td>\n",
       "      <td>27.422</td>\n",
       "      <td>7</td>\n",
       "      <td>9.230000e-56</td>\n",
       "      <td>205.0</td>\n",
       "      <td>6207</td>\n",
       "      <td>8249</td>\n",
       "      <td>20</td>\n",
       "      <td>717</td>\n",
       "      <td>733</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821_scores...</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Genome299_k127_417201</td>\n",
       "      <td>BKZW01_10191_Dictyobacter_vulcani_W12</td>\n",
       "      <td>26.842</td>\n",
       "      <td>40</td>\n",
       "      <td>9.620000e-28</td>\n",
       "      <td>113.0</td>\n",
       "      <td>121</td>\n",
       "      <td>1212</td>\n",
       "      <td>362</td>\n",
       "      <td>717</td>\n",
       "      <td>380</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626858_scores...</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626858.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Genome207_k127_77525</td>\n",
       "      <td>Rhodococcus_opacus_B4</td>\n",
       "      <td>31.536</td>\n",
       "      <td>96</td>\n",
       "      <td>2.350000e-19</td>\n",
       "      <td>82.8</td>\n",
       "      <td>1063</td>\n",
       "      <td>23</td>\n",
       "      <td>431</td>\n",
       "      <td>791</td>\n",
       "      <td>371</td>\n",
       "      <td>uncultured_Agrobacterium_sp._ERS6626766_scores...</td>\n",
       "      <td>uncultured_Agrobacterium_sp._ERS6626766.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>Genome424_k127_514686</td>\n",
       "      <td>Edaphobacter_aggregans_DSM_19364_2571099646</td>\n",
       "      <td>26.106</td>\n",
       "      <td>76</td>\n",
       "      <td>2.250000e-39</td>\n",
       "      <td>149.0</td>\n",
       "      <td>1902</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>655</td>\n",
       "      <td>678</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626983_scores.csv</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6626983.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>Genome452_k127_363663</td>\n",
       "      <td>Methyloferula_stellata_WP_020174914</td>\n",
       "      <td>26.638</td>\n",
       "      <td>22</td>\n",
       "      <td>6.140000e-13</td>\n",
       "      <td>65.9</td>\n",
       "      <td>641</td>\n",
       "      <td>9</td>\n",
       "      <td>20</td>\n",
       "      <td>240</td>\n",
       "      <td>229</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6627011_scores.csv</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6627011.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>Genome543_k127_120424</td>\n",
       "      <td>Sphaerobacter_thermophilus_DSM_20745_ZP_04495982</td>\n",
       "      <td>27.835</td>\n",
       "      <td>1</td>\n",
       "      <td>7.470000e-42</td>\n",
       "      <td>164.0</td>\n",
       "      <td>20040</td>\n",
       "      <td>17932</td>\n",
       "      <td>12</td>\n",
       "      <td>769</td>\n",
       "      <td>776</td>\n",
       "      <td>uncultured_Stenotrophomonas_sp._ERS6627319_sco...</td>\n",
       "      <td>uncultured_Stenotrophomonas_sp._ERS6627319.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>Genome220_k127_244806</td>\n",
       "      <td>BKZW01_10191_Dictyobacter_vulcani_W12</td>\n",
       "      <td>36.086</td>\n",
       "      <td>12</td>\n",
       "      <td>4.310000e-125</td>\n",
       "      <td>410.0</td>\n",
       "      <td>11575</td>\n",
       "      <td>13857</td>\n",
       "      <td>8</td>\n",
       "      <td>776</td>\n",
       "      <td>787</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>BKZW01_10191_Dictyobacter_vulcani_W12</td>\n",
       "      <td>36.086</td>\n",
       "      <td>32</td>\n",
       "      <td>5.640000e-125</td>\n",
       "      <td>408.0</td>\n",
       "      <td>1355</td>\n",
       "      <td>3637</td>\n",
       "      <td>8</td>\n",
       "      <td>776</td>\n",
       "      <td>787</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   qseqid                                             sseqid  \\\n",
       "0   Genome228_k127_160184  Pseudonocardia_dioxanivorans_CB1190_coxL2_(651...   \n",
       "1   Genome495_k127_317059                    Mycobacterium_smegmatis_MC2_155   \n",
       "2   Genome262_k127_235961              BKZW01_10191_Dictyobacter_vulcani_W12   \n",
       "3   Genome299_k127_417201              BKZW01_10191_Dictyobacter_vulcani_W12   \n",
       "4    Genome207_k127_77525                              Rhodococcus_opacus_B4   \n",
       "..                    ...                                                ...   \n",
       "63  Genome424_k127_514686        Edaphobacter_aggregans_DSM_19364_2571099646   \n",
       "64  Genome452_k127_363663                Methyloferula_stellata_WP_020174914   \n",
       "65  Genome543_k127_120424   Sphaerobacter_thermophilus_DSM_20745_ZP_04495982   \n",
       "66  Genome220_k127_244806              BKZW01_10191_Dictyobacter_vulcani_W12   \n",
       "67   Genome470_k127_43488              BKZW01_10191_Dictyobacter_vulcani_W12   \n",
       "\n",
       "    pident  qcovs         evalue  bitscore  qstart   qend  sstart  send  \\\n",
       "0   36.242      0   2.990000e-15      79.7   33523  33092     644   792   \n",
       "1   43.511      7   0.000000e+00     591.0   29003  31342      27   796   \n",
       "2   27.422      7   9.230000e-56     205.0    6207   8249      20   717   \n",
       "3   26.842     40   9.620000e-28     113.0     121   1212     362   717   \n",
       "4   31.536     96   2.350000e-19      82.8    1063     23     431   791   \n",
       "..     ...    ...            ...       ...     ...    ...     ...   ...   \n",
       "63  26.106     76   2.250000e-39     149.0    1902     13       5   655   \n",
       "64  26.638     22   6.140000e-13      65.9     641      9      20   240   \n",
       "65  27.835      1   7.470000e-42     164.0   20040  17932      12   769   \n",
       "66  36.086     12  4.310000e-125     410.0   11575  13857       8   776   \n",
       "67  36.086     32  5.640000e-125     408.0    1355   3637       8   776   \n",
       "\n",
       "    length                                        source_file  \\\n",
       "0      149    uncultured_Acidovorax_sp._ERS6626787_scores.csv   \n",
       "1      786    uncultured_Acidovorax_sp._ERS6627271_scores.csv   \n",
       "2      733  uncultured_Acinetobacter_sp._ERS6626821_scores...   \n",
       "3      380  uncultured_Acinetobacter_sp._ERS6626858_scores...   \n",
       "4      371  uncultured_Agrobacterium_sp._ERS6626766_scores...   \n",
       "..     ...                                                ...   \n",
       "63     678     uncultured_Spirosoma_sp._ERS6626983_scores.csv   \n",
       "64     229     uncultured_Spirosoma_sp._ERS6627011_scores.csv   \n",
       "65     776  uncultured_Stenotrophomonas_sp._ERS6627319_sco...   \n",
       "66     787    uncultured_Variovorax_sp._ERS6626779_scores.csv   \n",
       "67     787    uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "\n",
       "                                         target_file  \n",
       "0         uncultured_Acidovorax_sp._ERS6626787.fasta  \n",
       "1         uncultured_Acidovorax_sp._ERS6627271.fasta  \n",
       "2      uncultured_Acinetobacter_sp._ERS6626821.fasta  \n",
       "3      uncultured_Acinetobacter_sp._ERS6626858.fasta  \n",
       "4      uncultured_Agrobacterium_sp._ERS6626766.fasta  \n",
       "..                                               ...  \n",
       "63         uncultured_Spirosoma_sp._ERS6626983.fasta  \n",
       "64         uncultured_Spirosoma_sp._ERS6627011.fasta  \n",
       "65  uncultured_Stenotrophomonas_sp._ERS6627319.fasta  \n",
       "66        uncultured_Variovorax_sp._ERS6626779.fasta  \n",
       "67        uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "\n",
       "[68 rows x 13 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filtering the data frame with a genome file names and query sequence id set, to remove the other sequences which don't contain MOTIF\n",
    "## Create a empty set to store fileted rows\n",
    "filtered_rows = []\n",
    "## Filter the data frame\n",
    "for key, values in coxl1_motif_id_set.items():\n",
    "    filtered_rows.append(coxl1_csv_sorted[(coxl1_csv_sorted['target_file'] == key) & (coxl1_csv_sorted['qseqid'].isin(values))])\n",
    "    ## Combine the filtered results\n",
    "    filtered_coxl1_csv = pd.concat(filtered_rows).reset_index(drop=True)\n",
    "## Check the output\n",
    "filtered_coxl1_csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "f7a1503e-a4d2-41fb-8f60-44248096b184",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save the filtered data frame for further analysis\n",
    "filtered_coxl1_csv.to_csv(f\"{motif1}/filtered_coxl1_csv.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e11f57c7-ea03-4448-842b-cd1f009c54f3",
   "metadata": {},
   "source": [
    "<h3 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Search MOTIFs in CoxL Form 2 BLAST Results\n",
    "</h3>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "91a40bcf-cd3e-4ec4-9940-7117565c6838",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Results saved to CoxL2_Results/MOTIF/coxl2_motif_search.csv\n"
     ]
    }
   ],
   "source": [
    "# Search MOTIFs for CoxL Form 2 BLAST results \n",
    "coxl2_motif_results = search_motifs(motif_2_proteins, form1, label1, form2, label2, output_file=f\"{motif2}/coxl2_motif_search.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "5d5f3477-bc2b-45ed-91c7-8da28c32a8ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.faa</td>\n",
       "      <td>Genome302_k127_117597_1</td>\n",
       "      <td>MAGARRHQAAEDVVGAGRDLKRLVRPGREDDEADAAAAVAPPPPVR...</td>\n",
       "      <td>DVVGAGR</td>\n",
       "      <td>DVV</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.faa</td>\n",
       "      <td>Genome495_k127_317059_1</td>\n",
       "      <td>IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.faa</td>\n",
       "      <td>Genome495_k127_317059_1</td>\n",
       "      <td>IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.faa</td>\n",
       "      <td>Genome262_k127_235961_1</td>\n",
       "      <td>HQVSVESRRMGGGFGGKESQSAQWACIASLAAQKTGRPCKLRLDRD...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.faa</td>\n",
       "      <td>Genome262_k127_235961_1</td>\n",
       "      <td>NTMHLAVGFSSCAKGKISKFDLDAVRQADGVHAVFSAKDIDVENNW...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.faa</td>\n",
       "      <td>Genome220_k127_244806_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       File_Name                       ID  \\\n",
       "0      Burkholderiaceae_bacterium_ERS6626861.faa  Genome302_k127_117597_1   \n",
       "1       uncultured_Acidovorax_sp._ERS6627271.faa  Genome495_k127_317059_1   \n",
       "2       uncultured_Acidovorax_sp._ERS6627271.faa  Genome495_k127_317059_1   \n",
       "3    uncultured_Acinetobacter_sp._ERS6626821.faa  Genome262_k127_235961_1   \n",
       "4    uncultured_Acinetobacter_sp._ERS6626821.faa  Genome262_k127_235961_1   \n",
       "..                                           ...                      ...   \n",
       "104     uncultured_Variovorax_sp._ERS6626779.faa  Genome220_k127_244806_1   \n",
       "105     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "106     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "107     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "108     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "\n",
       "                                              Sequence    MOTIF Previous_3_AA  \\\n",
       "0    MAGARRHQAAEDVVGAGRDLKRLVRPGREDDEADAAAAVAPPPPVR...  DVVGAGR           DVV   \n",
       "1    IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...  AYRGAGR           AYR   \n",
       "2    IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...  AYRGAGR           AYR   \n",
       "3    HQVSVESRRMGGGFGGKESQSAQWACIASLAAQKTGRPCKLRLDRD...  EQPGAGR           EQP   \n",
       "4    NTMHLAVGFSSCAKGKISKFDLDAVRQADGVHAVFSAKDIDVENNW...  EQPGAGR           EQP   \n",
       "..                                                 ...      ...           ...   \n",
       "104  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  AYRGAGR           AYR   \n",
       "105  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR           LLS   \n",
       "106  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  AYRGAGR           AYR   \n",
       "107  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  LLSGAGR           LLS   \n",
       "108  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  AYRGAGR           AYR   \n",
       "\n",
       "     Motif_Label  \n",
       "0    CoxL_form_2  \n",
       "1    CoxL_form_2  \n",
       "2    CoxL_form_2  \n",
       "3    CoxL_form_2  \n",
       "4    CoxL_form_2  \n",
       "..           ...  \n",
       "104  CoxL_form_2  \n",
       "105  CoxL_form_2  \n",
       "106  CoxL_form_2  \n",
       "107  CoxL_form_2  \n",
       "108  CoxL_form_2  \n",
       "\n",
       "[109 rows x 6 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking the MOTIF search results output\n",
    "coxl2_motif_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "5e70fb0a-22b2-4fa3-a53b-c6dbf349fb20",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['CoxL_form_2'], dtype=object)"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking for the presence of MOTIF form 1 and 2\n",
    "coxl2_motif_results['Motif_Label'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "7428ac4e-bc06-49cb-be3f-4fad2b3e17e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "48"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking for number of genomes containing MOTIF form 2\n",
    "coxl2_motif_results['File_Name'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "6d3793a1-1f33-46e8-97b3-0bc9930dffd5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.faa</td>\n",
       "      <td>Genome302_k127_117597_1</td>\n",
       "      <td>MAGARRHQAAEDVVGAGRDLKRLVRPGREDDEADAAAAVAPPPPVR...</td>\n",
       "      <td>DVVGAGR</td>\n",
       "      <td>DVV</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.faa</td>\n",
       "      <td>Genome495_k127_317059_1</td>\n",
       "      <td>IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.faa</td>\n",
       "      <td>Genome495_k127_317059_1</td>\n",
       "      <td>IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.faa</td>\n",
       "      <td>Genome262_k127_235961_1</td>\n",
       "      <td>HQVSVESRRMGGGFGGKESQSAQWACIASLAAQKTGRPCKLRLDRD...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.faa</td>\n",
       "      <td>Genome262_k127_235961_1</td>\n",
       "      <td>NTMHLAVGFSSCAKGKISKFDLDAVRQADGVHAVFSAKDIDVENNW...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.faa</td>\n",
       "      <td>Genome220_k127_244806_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       File_Name                       ID  \\\n",
       "0      Burkholderiaceae_bacterium_ERS6626861.faa  Genome302_k127_117597_1   \n",
       "1       uncultured_Acidovorax_sp._ERS6627271.faa  Genome495_k127_317059_1   \n",
       "2       uncultured_Acidovorax_sp._ERS6627271.faa  Genome495_k127_317059_1   \n",
       "3    uncultured_Acinetobacter_sp._ERS6626821.faa  Genome262_k127_235961_1   \n",
       "4    uncultured_Acinetobacter_sp._ERS6626821.faa  Genome262_k127_235961_1   \n",
       "..                                           ...                      ...   \n",
       "104     uncultured_Variovorax_sp._ERS6626779.faa  Genome220_k127_244806_1   \n",
       "105     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "106     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "107     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "108     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "\n",
       "                                              Sequence    MOTIF Previous_3_AA  \\\n",
       "0    MAGARRHQAAEDVVGAGRDLKRLVRPGREDDEADAAAAVAPPPPVR...  DVVGAGR           DVV   \n",
       "1    IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...  AYRGAGR           AYR   \n",
       "2    IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...  AYRGAGR           AYR   \n",
       "3    HQVSVESRRMGGGFGGKESQSAQWACIASLAAQKTGRPCKLRLDRD...  EQPGAGR           EQP   \n",
       "4    NTMHLAVGFSSCAKGKISKFDLDAVRQADGVHAVFSAKDIDVENNW...  EQPGAGR           EQP   \n",
       "..                                                 ...      ...           ...   \n",
       "104  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  AYRGAGR           AYR   \n",
       "105  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR           LLS   \n",
       "106  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  AYRGAGR           AYR   \n",
       "107  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  LLSGAGR           LLS   \n",
       "108  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  AYRGAGR           AYR   \n",
       "\n",
       "     Motif_Label                                    target_file  \n",
       "0    CoxL_form_2    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "1    CoxL_form_2     uncultured_Acidovorax_sp._ERS6627271.fasta  \n",
       "2    CoxL_form_2     uncultured_Acidovorax_sp._ERS6627271.fasta  \n",
       "3    CoxL_form_2  uncultured_Acinetobacter_sp._ERS6626821.fasta  \n",
       "4    CoxL_form_2  uncultured_Acinetobacter_sp._ERS6626821.fasta  \n",
       "..           ...                                            ...  \n",
       "104  CoxL_form_2     uncultured_Variovorax_sp._ERS6626779.fasta  \n",
       "105  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "106  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "107  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "108  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "\n",
       "[109 rows x 7 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Generating genome file name columns ('target_file')\n",
    "coxl2_motif_results['target_file'] = coxl2_motif_results['File_Name'].apply(lambda x: x[:-4] + '.fasta')\n",
    "coxl2_motif_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "71a251c8-bf73-4a79-b140-abd25fef2945",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "      <th>target_file</th>\n",
       "      <th>qseqid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.faa</td>\n",
       "      <td>Genome302_k127_117597_1</td>\n",
       "      <td>MAGARRHQAAEDVVGAGRDLKRLVRPGREDDEADAAAAVAPPPPVR...</td>\n",
       "      <td>DVVGAGR</td>\n",
       "      <td>DVV</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "      <td>Genome302_k127_117597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.faa</td>\n",
       "      <td>Genome495_k127_317059_1</td>\n",
       "      <td>IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "      <td>Genome495_k127_317059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.faa</td>\n",
       "      <td>Genome495_k127_317059_1</td>\n",
       "      <td>IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "      <td>Genome495_k127_317059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.faa</td>\n",
       "      <td>Genome262_k127_235961_1</td>\n",
       "      <td>HQVSVESRRMGGGFGGKESQSAQWACIASLAAQKTGRPCKLRLDRD...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.fasta</td>\n",
       "      <td>Genome262_k127_235961</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.faa</td>\n",
       "      <td>Genome262_k127_235961_1</td>\n",
       "      <td>NTMHLAVGFSSCAKGKISKFDLDAVRQADGVHAVFSAKDIDVENNW...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.fasta</td>\n",
       "      <td>Genome262_k127_235961</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.faa</td>\n",
       "      <td>Genome220_k127_244806_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.fasta</td>\n",
       "      <td>Genome220_k127_244806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       File_Name                       ID  \\\n",
       "0      Burkholderiaceae_bacterium_ERS6626861.faa  Genome302_k127_117597_1   \n",
       "1       uncultured_Acidovorax_sp._ERS6627271.faa  Genome495_k127_317059_1   \n",
       "2       uncultured_Acidovorax_sp._ERS6627271.faa  Genome495_k127_317059_1   \n",
       "3    uncultured_Acinetobacter_sp._ERS6626821.faa  Genome262_k127_235961_1   \n",
       "4    uncultured_Acinetobacter_sp._ERS6626821.faa  Genome262_k127_235961_1   \n",
       "..                                           ...                      ...   \n",
       "104     uncultured_Variovorax_sp._ERS6626779.faa  Genome220_k127_244806_1   \n",
       "105     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "106     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "107     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "108     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "\n",
       "                                              Sequence    MOTIF Previous_3_AA  \\\n",
       "0    MAGARRHQAAEDVVGAGRDLKRLVRPGREDDEADAAAAVAPPPPVR...  DVVGAGR           DVV   \n",
       "1    IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...  AYRGAGR           AYR   \n",
       "2    IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...  AYRGAGR           AYR   \n",
       "3    HQVSVESRRMGGGFGGKESQSAQWACIASLAAQKTGRPCKLRLDRD...  EQPGAGR           EQP   \n",
       "4    NTMHLAVGFSSCAKGKISKFDLDAVRQADGVHAVFSAKDIDVENNW...  EQPGAGR           EQP   \n",
       "..                                                 ...      ...           ...   \n",
       "104  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  AYRGAGR           AYR   \n",
       "105  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR           LLS   \n",
       "106  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  AYRGAGR           AYR   \n",
       "107  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  LLSGAGR           LLS   \n",
       "108  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  AYRGAGR           AYR   \n",
       "\n",
       "     Motif_Label                                    target_file  \\\n",
       "0    CoxL_form_2    Burkholderiaceae_bacterium_ERS6626861.fasta   \n",
       "1    CoxL_form_2     uncultured_Acidovorax_sp._ERS6627271.fasta   \n",
       "2    CoxL_form_2     uncultured_Acidovorax_sp._ERS6627271.fasta   \n",
       "3    CoxL_form_2  uncultured_Acinetobacter_sp._ERS6626821.fasta   \n",
       "4    CoxL_form_2  uncultured_Acinetobacter_sp._ERS6626821.fasta   \n",
       "..           ...                                            ...   \n",
       "104  CoxL_form_2     uncultured_Variovorax_sp._ERS6626779.fasta   \n",
       "105  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "106  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "107  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "108  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "\n",
       "                    qseqid  \n",
       "0    Genome302_k127_117597  \n",
       "1    Genome495_k127_317059  \n",
       "2    Genome495_k127_317059  \n",
       "3    Genome262_k127_235961  \n",
       "4    Genome262_k127_235961  \n",
       "..                     ...  \n",
       "104  Genome220_k127_244806  \n",
       "105   Genome470_k127_43488  \n",
       "106   Genome470_k127_43488  \n",
       "107   Genome470_k127_43488  \n",
       "108   Genome470_k127_43488  \n",
       "\n",
       "[109 rows x 8 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Generating input query sequence ID coulmn from updated sequence ID column ('ID')\n",
    "coxl2_motif_results['qseqid'] = coxl2_motif_results['ID'].apply(lambda x: x[:-2])\n",
    "coxl2_motif_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "b20a7436-f653-4794-9d77-54590cf68ca6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "      <th>target_file</th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sequence_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.faa</td>\n",
       "      <td>Genome302_k127_117597_1</td>\n",
       "      <td>MAGARRHQAAEDVVGAGRDLKRLVRPGREDDEADAAAAVAPPPPVR...</td>\n",
       "      <td>DVVGAGR</td>\n",
       "      <td>DVV</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "      <td>Genome302_k127_117597</td>\n",
       "      <td>183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.faa</td>\n",
       "      <td>Genome495_k127_317059_1</td>\n",
       "      <td>IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "      <td>Genome495_k127_317059</td>\n",
       "      <td>783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.faa</td>\n",
       "      <td>Genome495_k127_317059_1</td>\n",
       "      <td>IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "      <td>Genome495_k127_317059</td>\n",
       "      <td>781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.faa</td>\n",
       "      <td>Genome262_k127_235961_1</td>\n",
       "      <td>HQVSVESRRMGGGFGGKESQSAQWACIASLAAQKTGRPCKLRLDRD...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.fasta</td>\n",
       "      <td>Genome262_k127_235961</td>\n",
       "      <td>537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.faa</td>\n",
       "      <td>Genome262_k127_235961_1</td>\n",
       "      <td>NTMHLAVGFSSCAKGKISKFDLDAVRQADGVHAVFSAKDIDVENNW...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.fasta</td>\n",
       "      <td>Genome262_k127_235961</td>\n",
       "      <td>663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.faa</td>\n",
       "      <td>Genome220_k127_244806_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.fasta</td>\n",
       "      <td>Genome220_k127_244806</td>\n",
       "      <td>767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>767</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       File_Name                       ID  \\\n",
       "0      Burkholderiaceae_bacterium_ERS6626861.faa  Genome302_k127_117597_1   \n",
       "1       uncultured_Acidovorax_sp._ERS6627271.faa  Genome495_k127_317059_1   \n",
       "2       uncultured_Acidovorax_sp._ERS6627271.faa  Genome495_k127_317059_1   \n",
       "3    uncultured_Acinetobacter_sp._ERS6626821.faa  Genome262_k127_235961_1   \n",
       "4    uncultured_Acinetobacter_sp._ERS6626821.faa  Genome262_k127_235961_1   \n",
       "..                                           ...                      ...   \n",
       "104     uncultured_Variovorax_sp._ERS6626779.faa  Genome220_k127_244806_1   \n",
       "105     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "106     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "107     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "108     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "\n",
       "                                              Sequence    MOTIF Previous_3_AA  \\\n",
       "0    MAGARRHQAAEDVVGAGRDLKRLVRPGREDDEADAAAAVAPPPPVR...  DVVGAGR           DVV   \n",
       "1    IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...  AYRGAGR           AYR   \n",
       "2    IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...  AYRGAGR           AYR   \n",
       "3    HQVSVESRRMGGGFGGKESQSAQWACIASLAAQKTGRPCKLRLDRD...  EQPGAGR           EQP   \n",
       "4    NTMHLAVGFSSCAKGKISKFDLDAVRQADGVHAVFSAKDIDVENNW...  EQPGAGR           EQP   \n",
       "..                                                 ...      ...           ...   \n",
       "104  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  AYRGAGR           AYR   \n",
       "105  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR           LLS   \n",
       "106  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  AYRGAGR           AYR   \n",
       "107  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  LLSGAGR           LLS   \n",
       "108  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  AYRGAGR           AYR   \n",
       "\n",
       "     Motif_Label                                    target_file  \\\n",
       "0    CoxL_form_2    Burkholderiaceae_bacterium_ERS6626861.fasta   \n",
       "1    CoxL_form_2     uncultured_Acidovorax_sp._ERS6627271.fasta   \n",
       "2    CoxL_form_2     uncultured_Acidovorax_sp._ERS6627271.fasta   \n",
       "3    CoxL_form_2  uncultured_Acinetobacter_sp._ERS6626821.fasta   \n",
       "4    CoxL_form_2  uncultured_Acinetobacter_sp._ERS6626821.fasta   \n",
       "..           ...                                            ...   \n",
       "104  CoxL_form_2     uncultured_Variovorax_sp._ERS6626779.fasta   \n",
       "105  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "106  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "107  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "108  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "\n",
       "                    qseqid  sequence_length  \n",
       "0    Genome302_k127_117597              183  \n",
       "1    Genome495_k127_317059              783  \n",
       "2    Genome495_k127_317059              781  \n",
       "3    Genome262_k127_235961              537  \n",
       "4    Genome262_k127_235961              663  \n",
       "..                     ...              ...  \n",
       "104  Genome220_k127_244806              767  \n",
       "105   Genome470_k127_43488              763  \n",
       "106   Genome470_k127_43488              763  \n",
       "107   Genome470_k127_43488              767  \n",
       "108   Genome470_k127_43488              767  \n",
       "\n",
       "[109 rows x 9 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Finding the length of every sequence and save the value in 'sequence length' column\n",
    "coxl2_motif_results['sequence_length'] = coxl2_motif_results['Sequence'].apply(len)\n",
    "coxl2_motif_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "7545c996-ffe5-46f9-bf85-f2928592df69",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "      <th>target_file</th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sequence_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.faa</td>\n",
       "      <td>Genome302_k127_117597_1</td>\n",
       "      <td>MAGARRHQAAEDVVGAGRDLKRLVRPGREDDEADAAAAVAPPPPVR...</td>\n",
       "      <td>DVVGAGR</td>\n",
       "      <td>DVV</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "      <td>Genome302_k127_117597</td>\n",
       "      <td>183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.faa</td>\n",
       "      <td>Genome495_k127_317059_1</td>\n",
       "      <td>IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "      <td>Genome495_k127_317059</td>\n",
       "      <td>783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.faa</td>\n",
       "      <td>Genome495_k127_317059_1</td>\n",
       "      <td>IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "      <td>Genome495_k127_317059</td>\n",
       "      <td>781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.faa</td>\n",
       "      <td>Genome262_k127_235961_1</td>\n",
       "      <td>NTMHLAVGFSSCAKGKISKFDLDAVRQADGVHAVFSAKDIDVENNW...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.fasta</td>\n",
       "      <td>Genome262_k127_235961</td>\n",
       "      <td>663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.faa</td>\n",
       "      <td>Genome262_k127_235961_1</td>\n",
       "      <td>HQVSVESRRMGGGFGGKESQSAQWACIASLAAQKTGRPCKLRLDRD...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.fasta</td>\n",
       "      <td>Genome262_k127_235961</td>\n",
       "      <td>537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.faa</td>\n",
       "      <td>Genome220_k127_244806_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.fasta</td>\n",
       "      <td>Genome220_k127_244806</td>\n",
       "      <td>763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>763</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       File_Name                       ID  \\\n",
       "0      Burkholderiaceae_bacterium_ERS6626861.faa  Genome302_k127_117597_1   \n",
       "1       uncultured_Acidovorax_sp._ERS6627271.faa  Genome495_k127_317059_1   \n",
       "2       uncultured_Acidovorax_sp._ERS6627271.faa  Genome495_k127_317059_1   \n",
       "4    uncultured_Acinetobacter_sp._ERS6626821.faa  Genome262_k127_235961_1   \n",
       "3    uncultured_Acinetobacter_sp._ERS6626821.faa  Genome262_k127_235961_1   \n",
       "..                                           ...                      ...   \n",
       "101     uncultured_Variovorax_sp._ERS6626779.faa  Genome220_k127_244806_1   \n",
       "108     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "106     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "107     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "105     uncultured_Variovorax_sp._ERS6627246.faa   Genome470_k127_43488_1   \n",
       "\n",
       "                                              Sequence    MOTIF Previous_3_AA  \\\n",
       "0    MAGARRHQAAEDVVGAGRDLKRLVRPGREDDEADAAAAVAPPPPVR...  DVVGAGR           DVV   \n",
       "1    IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...  AYRGAGR           AYR   \n",
       "2    IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...  AYRGAGR           AYR   \n",
       "4    NTMHLAVGFSSCAKGKISKFDLDAVRQADGVHAVFSAKDIDVENNW...  EQPGAGR           EQP   \n",
       "3    HQVSVESRRMGGGFGGKESQSAQWACIASLAAQKTGRPCKLRLDRD...  EQPGAGR           EQP   \n",
       "..                                                 ...      ...           ...   \n",
       "101  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR           LLS   \n",
       "108  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  AYRGAGR           AYR   \n",
       "106  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  AYRGAGR           AYR   \n",
       "107  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  LLSGAGR           LLS   \n",
       "105  GSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPHARIL...  LLSGAGR           LLS   \n",
       "\n",
       "     Motif_Label                                    target_file  \\\n",
       "0    CoxL_form_2    Burkholderiaceae_bacterium_ERS6626861.fasta   \n",
       "1    CoxL_form_2     uncultured_Acidovorax_sp._ERS6627271.fasta   \n",
       "2    CoxL_form_2     uncultured_Acidovorax_sp._ERS6627271.fasta   \n",
       "4    CoxL_form_2  uncultured_Acinetobacter_sp._ERS6626821.fasta   \n",
       "3    CoxL_form_2  uncultured_Acinetobacter_sp._ERS6626821.fasta   \n",
       "..           ...                                            ...   \n",
       "101  CoxL_form_2     uncultured_Variovorax_sp._ERS6626779.fasta   \n",
       "108  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "106  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "107  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "105  CoxL_form_2     uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "\n",
       "                    qseqid  sequence_length  \n",
       "0    Genome302_k127_117597              183  \n",
       "1    Genome495_k127_317059              783  \n",
       "2    Genome495_k127_317059              781  \n",
       "4    Genome262_k127_235961              663  \n",
       "3    Genome262_k127_235961              537  \n",
       "..                     ...              ...  \n",
       "101  Genome220_k127_244806              763  \n",
       "108   Genome470_k127_43488              767  \n",
       "106   Genome470_k127_43488              763  \n",
       "107   Genome470_k127_43488              767  \n",
       "105   Genome470_k127_43488              763  \n",
       "\n",
       "[109 rows x 9 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter data frame based on 'File_Name', 'qseqid', 'MOTIF' & 'sequence_length'\n",
    "coxl2_motif_results = coxl2_motif_results.sort_values(by=['File_Name', 'qseqid', 'MOTIF', 'sequence_length'],\n",
    "                                        ascending=[True, True, True, False])\n",
    "coxl2_motif_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "535a904e-bb36-4af9-a803-c7259b14f6ef",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File_Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>MOTIF</th>\n",
       "      <th>Previous_3_AA</th>\n",
       "      <th>Motif_Label</th>\n",
       "      <th>target_file</th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sequence_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.faa</td>\n",
       "      <td>Genome302_k127_117597_1</td>\n",
       "      <td>MAGARRHQAAEDVVGAGRDLKRLVRPGREDDEADAAAAVAPPPPVR...</td>\n",
       "      <td>DVVGAGR</td>\n",
       "      <td>DVV</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "      <td>Genome302_k127_117597</td>\n",
       "      <td>183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.faa</td>\n",
       "      <td>Genome495_k127_317059_1</td>\n",
       "      <td>IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "      <td>Genome495_k127_317059</td>\n",
       "      <td>783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.faa</td>\n",
       "      <td>Genome262_k127_235961_1</td>\n",
       "      <td>NTMHLAVGFSSCAKGKISKFDLDAVRQADGVHAVFSAKDIDVENNW...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.fasta</td>\n",
       "      <td>Genome262_k127_235961</td>\n",
       "      <td>663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626858.faa</td>\n",
       "      <td>Genome299_k127_417201_1</td>\n",
       "      <td>ELRNLRCKTNTVSNTAYRGFGGPQGMFVIENIIDDIARYLGCDPIE...</td>\n",
       "      <td>EQPGAGR</td>\n",
       "      <td>EQP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626858.fasta</td>\n",
       "      <td>Genome299_k127_417201</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uncultured_Aureimonas_sp._ERS6626915.faa</td>\n",
       "      <td>Genome356_k127_498042_1</td>\n",
       "      <td>LAMIAAEKLKRPVRWVADRNEHFLADTHGRANLATATMALDAKGRF...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Aureimonas_sp._ERS6626915.fasta</td>\n",
       "      <td>Genome356_k127_498042</td>\n",
       "      <td>498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>uncultured_Aureimonas_sp._ERS6626915.faa</td>\n",
       "      <td>Genome356_k127_610672_1</td>\n",
       "      <td>IGKGIDRFEGPLKVTGRARYAYENMTGPEVAYGFVVTAGTGAGRIK...</td>\n",
       "      <td>AGTGAGR</td>\n",
       "      <td>AGT</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Aureimonas_sp._ERS6626915.fasta</td>\n",
       "      <td>Genome356_k127_610672</td>\n",
       "      <td>370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>uncultured_Aureimonas_sp._ERS6627306.faa</td>\n",
       "      <td>Genome530_k127_16854_1</td>\n",
       "      <td>MGIEGIGARVLRKEDRRFITGRGRYTDDMVVPGMKYAAFVRSPHAH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Aureimonas_sp._ERS6627306.fasta</td>\n",
       "      <td>Genome530_k127_16854</td>\n",
       "      <td>780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626630.faa</td>\n",
       "      <td>Genome71_S1Ck127_18694_1</td>\n",
       "      <td>IGEPVKRKEDYRFLTGAGQYTDDIALAAQAHAVFVRSPHAHARVRS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626630.fasta</td>\n",
       "      <td>Genome71_S1Ck127_18694</td>\n",
       "      <td>782</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626638.faa</td>\n",
       "      <td>Genome79_S3Ck127_287303_1</td>\n",
       "      <td>PATPADVAAIRTRPAVGGNARPGQRAPGTGKRQRPGPARRAGRATP...</td>\n",
       "      <td>PRPGAGR</td>\n",
       "      <td>PRP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626638.fasta</td>\n",
       "      <td>Genome79_S3Ck127_287303</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626638.faa</td>\n",
       "      <td>Genome79_S3Ck127_287303_3</td>\n",
       "      <td>GVAHSVCRGSCRRRRRRAGGGTTPPTRARPTAAACRRASRARPSAT...</td>\n",
       "      <td>RVSGAGR</td>\n",
       "      <td>RVS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626638.fasta</td>\n",
       "      <td>Genome79_S3Ck127_287303</td>\n",
       "      <td>294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626814.faa</td>\n",
       "      <td>Genome255_k127_46261_1</td>\n",
       "      <td>PATPADIAAIRTRPTVGRNARPGQRVTGTGGREVPTPARRAGRATP...</td>\n",
       "      <td>PRPGAGR</td>\n",
       "      <td>PRP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626814.fasta</td>\n",
       "      <td>Genome255_k127_46261</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>uncultured_Comamonas_sp._ERS6627283.faa</td>\n",
       "      <td>Genome507_k127_91973_1</td>\n",
       "      <td>IGEPVKRKEDYRFLTGAGQYTDDIALAAQAHAVFVRSPHAHARVRS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6627283.fasta</td>\n",
       "      <td>Genome507_k127_91973</td>\n",
       "      <td>782</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626797.faa</td>\n",
       "      <td>Genome238_k127_17753_1</td>\n",
       "      <td>VGQALKRKEDPRFITGTGNYTDDIVLHGMLHAAMVRSPYPHARITA...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626797.fasta</td>\n",
       "      <td>Genome238_k127_17753</td>\n",
       "      <td>771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626797.faa</td>\n",
       "      <td>Genome238_k127_54530_1</td>\n",
       "      <td>PRMTSVGRPVKRIEGRAKVLGKARFAAEYPIRGLVYAAPVLSTIPK...</td>\n",
       "      <td>GFGGAGR</td>\n",
       "      <td>GFG</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626797.fasta</td>\n",
       "      <td>Genome238_k127_54530</td>\n",
       "      <td>713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626820.faa</td>\n",
       "      <td>Genome261_k127_146972_1</td>\n",
       "      <td>VGQALKRKEDPRFITGTGNYTDDIVLHGMLHAAMVRSPYPHARITA...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626820.fasta</td>\n",
       "      <td>Genome261_k127_146972</td>\n",
       "      <td>771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626820.faa</td>\n",
       "      <td>Genome261_k127_332399_1</td>\n",
       "      <td>MTSVGRPVKRIEGRAKVLGKARFAAEYPIRGLVYAAPVLSTIPKGR...</td>\n",
       "      <td>GFGGAGR</td>\n",
       "      <td>GFG</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626820.fasta</td>\n",
       "      <td>Genome261_k127_332399</td>\n",
       "      <td>317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626837.faa</td>\n",
       "      <td>Genome278_k127_93298_1</td>\n",
       "      <td>VGQALKRKEDPRFITGTGNYTDDIVLHGMLHAAMVRSPYPHARITA...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626837.fasta</td>\n",
       "      <td>Genome278_k127_93298</td>\n",
       "      <td>775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6627293.faa</td>\n",
       "      <td>Genome517_k127_348199_1</td>\n",
       "      <td>VGQALKRKEDPRFITGTGNYTDDIVLHGMLHAAMVRSPYPHARITA...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6627293.fasta</td>\n",
       "      <td>Genome517_k127_348199</td>\n",
       "      <td>771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6627335.faa</td>\n",
       "      <td>Genome559_k127_243715_1</td>\n",
       "      <td>PRMTSVGRPVKRIEGRAKVLGKARFAAEYPIRGLVYAAPVLSTIPK...</td>\n",
       "      <td>GFGGAGR</td>\n",
       "      <td>GFG</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6627335.fasta</td>\n",
       "      <td>Genome559_k127_243715</td>\n",
       "      <td>713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>uncultured_Hymenobacter_sp._ERS6626733.faa</td>\n",
       "      <td>Genome174_S1Ck127_276112_1</td>\n",
       "      <td>VGKPLVRVEGREKVTGKAKYAAEFALPGLTYGVFTTSPIAKGKITS...</td>\n",
       "      <td>SVIGAGR</td>\n",
       "      <td>SVI</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Hymenobacter_sp._ERS6626733.fasta</td>\n",
       "      <td>Genome174_S1Ck127_276112</td>\n",
       "      <td>722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>uncultured_Hymenobacter_sp._ERS6627279.faa</td>\n",
       "      <td>Genome503_k127_453582_1</td>\n",
       "      <td>VGKPLVRVEGREKVTGKAKYAAEFALPGLTYGVFTTSPIAKGKITR...</td>\n",
       "      <td>SVIGAGR</td>\n",
       "      <td>SVI</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Hymenobacter_sp._ERS6627279.fasta</td>\n",
       "      <td>Genome503_k127_453582</td>\n",
       "      <td>722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>uncultured_Methylobacterium_sp._ERS6627288.faa</td>\n",
       "      <td>Genome512_k127_160927_1</td>\n",
       "      <td>MTVTKFGLGQPLRRVEDRRLITGEGNYGDDHAPEGCLHAALLRSPH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Methylobacterium_sp._ERS6627288.fasta</td>\n",
       "      <td>Genome512_k127_160927</td>\n",
       "      <td>759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>uncultured_Methylobacterium_sp._ERS6627288.faa</td>\n",
       "      <td>Genome512_k127_80099_1</td>\n",
       "      <td>PQKFGIGQPVRRAEDPVLVRGEGRYTDDLALDGQAYGVFVRSPVAH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Methylobacterium_sp._ERS6627288.fasta</td>\n",
       "      <td>Genome512_k127_80099</td>\n",
       "      <td>768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>uncultured_Microbacterium_sp._ERS6626906.faa</td>\n",
       "      <td>Genome347_k127_84820_1</td>\n",
       "      <td>DHAPEGCLHAALLRSPHAHATFTITDVEAARAMPGVHLVLTAEDVA...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Microbacterium_sp._ERS6626906.fasta</td>\n",
       "      <td>Genome347_k127_84820</td>\n",
       "      <td>347</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>uncultured_Microbacterium_sp._ERS6626964.faa</td>\n",
       "      <td>Genome405_k127_107645_1</td>\n",
       "      <td>VGASVIPEAARRLVQGREPFTLDEPVPGGPPLVLRVVTSPHAHARV...</td>\n",
       "      <td>VGSGAGR</td>\n",
       "      <td>VGS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Microbacterium_sp._ERS6626964.fasta</td>\n",
       "      <td>Genome405_k127_107645</td>\n",
       "      <td>768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626873.faa</td>\n",
       "      <td>Genome314_k127_168730_1</td>\n",
       "      <td>IAPPPFTDGIDRVDGRLKVTGAAKYSAEYDMPNMSYAVFALSTITK...</td>\n",
       "      <td>GVFGAGR</td>\n",
       "      <td>GVF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626873.fasta</td>\n",
       "      <td>Genome314_k127_168730</td>\n",
       "      <td>713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626923.faa</td>\n",
       "      <td>Genome364_k127_154890_1</td>\n",
       "      <td>IDRVDGRLKVTGAAKYAAEFEVPGMTYAVFANSTITKGTIKNIDTK...</td>\n",
       "      <td>GVFGAGR</td>\n",
       "      <td>GVF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626923.fasta</td>\n",
       "      <td>Genome364_k127_154890</td>\n",
       "      <td>488</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626962.faa</td>\n",
       "      <td>Genome403_k127_249621_1</td>\n",
       "      <td>IDRVDGRLKVTGAAKYAAEFEVPGMTYAVFANSTITKGTIKTMDIK...</td>\n",
       "      <td>GVFGAGR</td>\n",
       "      <td>GVF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626962.fasta</td>\n",
       "      <td>Genome403_k127_249621</td>\n",
       "      <td>589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626972.faa</td>\n",
       "      <td>Genome413_k127_255610_1</td>\n",
       "      <td>IDRVDGRLKVTGAAKYAAEFEVPGMTYAVFANSTITKGTIKTIDTK...</td>\n",
       "      <td>GVFGAGR</td>\n",
       "      <td>GVF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626972.fasta</td>\n",
       "      <td>Genome413_k127_255610</td>\n",
       "      <td>701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626981.faa</td>\n",
       "      <td>Genome422_k127_88665_1</td>\n",
       "      <td>IDRVDGRLKVTGAAKYAAEFEVPGMTYAVFANSTITKGTIKTIDTK...</td>\n",
       "      <td>GVFGAGR</td>\n",
       "      <td>GVF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626981.fasta</td>\n",
       "      <td>Genome422_k127_88665</td>\n",
       "      <td>701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6627275.faa</td>\n",
       "      <td>Genome499_k127_471593_1</td>\n",
       "      <td>IDRVDGRLKVTGAAKYAAEFEVPGMTYAVFANSTITKGTIKTIDTK...</td>\n",
       "      <td>GVFGAGR</td>\n",
       "      <td>GVF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6627275.fasta</td>\n",
       "      <td>Genome499_k127_471593</td>\n",
       "      <td>701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6627290.faa</td>\n",
       "      <td>Genome514_k127_4285_1</td>\n",
       "      <td>IDRVDGRLKVTGAAKYAAEFEVPGMTYAVFANSTITKGTIKTIDIK...</td>\n",
       "      <td>GVFGAGR</td>\n",
       "      <td>GVF</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6627290.fasta</td>\n",
       "      <td>Genome514_k127_4285</td>\n",
       "      <td>701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>uncultured_Pantoea_sp._ERS6626643.faa</td>\n",
       "      <td>Genome84_S2Ck127_371734_2</td>\n",
       "      <td>MRRLRAVTVCLCGAGGGPRETAVSAAGTRRIVTTETSITTGSEGAG...</td>\n",
       "      <td>GSEGAGR</td>\n",
       "      <td>GSE</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pantoea_sp._ERS6626643.fasta</td>\n",
       "      <td>Genome84_S2Ck127_371734</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>uncultured_Pantoea_sp._ERS6627005.faa</td>\n",
       "      <td>Genome446_k127_205538_3</td>\n",
       "      <td>LRGSYGRDRTARAGRRLPLLSIAATVGAGRCVWCAATGCICHQEAA...</td>\n",
       "      <td>ATVGAGR</td>\n",
       "      <td>ATV</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pantoea_sp._ERS6627005.fasta</td>\n",
       "      <td>Genome446_k127_205538</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>uncultured_Pelomonas_sp._ERS6626569.faa</td>\n",
       "      <td>Genome10_S2Ck127_387977_3</td>\n",
       "      <td>SPSATASPRTSTTRAPTTRPATGPRSGRGGSSRSGTGPTSGEAGRA...</td>\n",
       "      <td>QVRGAGR</td>\n",
       "      <td>QVR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pelomonas_sp._ERS6626569.fasta</td>\n",
       "      <td>Genome10_S2Ck127_387977</td>\n",
       "      <td>253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6626577.faa</td>\n",
       "      <td>Genome18_S1Ck127_217968_1</td>\n",
       "      <td>IGEAIRRKEDVRFLTGAGNYTGDVVLANQAHAVFLRSPHAHAAITS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6626577.fasta</td>\n",
       "      <td>Genome18_S1Ck127_217968</td>\n",
       "      <td>777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6626850.faa</td>\n",
       "      <td>Genome291_k127_419550_1</td>\n",
       "      <td>IGEAIRRKEDVRFLTGAGNYTGDVVLAHQAHAVFLRSPHAHAAITS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6626850.fasta</td>\n",
       "      <td>Genome291_k127_419550</td>\n",
       "      <td>777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6626909.faa</td>\n",
       "      <td>Genome350_k127_213661_1</td>\n",
       "      <td>RFGSGQAVRRLEDEALLAGRGRYTDDIRPRDCAALAFQRSPHPHAR...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6626909.fasta</td>\n",
       "      <td>Genome350_k127_213661</td>\n",
       "      <td>765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627003.faa</td>\n",
       "      <td>Genome444_k127_25498_1</td>\n",
       "      <td>VKWTGERSEEFLTTTHGRDVRARAAMALDADGRVLALRVHSDANVG...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627003.fasta</td>\n",
       "      <td>Genome444_k127_25498</td>\n",
       "      <td>490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627003.faa</td>\n",
       "      <td>Genome444_k127_2949_1</td>\n",
       "      <td>IGEAIRRKEDVRFLTGAGNYTGDVVLAHQAHAVFLRSPHAHAAITS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627003.fasta</td>\n",
       "      <td>Genome444_k127_2949</td>\n",
       "      <td>710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627320.faa</td>\n",
       "      <td>Genome544_k127_833201_1</td>\n",
       "      <td>IGEAIRRKEDLRFLTGAGNYTGDVVLANQAHAVFLRSPHAHAAIQS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627320.fasta</td>\n",
       "      <td>Genome544_k127_833201</td>\n",
       "      <td>777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627320.faa</td>\n",
       "      <td>Genome544_k127_857444_1</td>\n",
       "      <td>GKETPRVDGIAKVTGQAKYAAEFQLPNLAYGFLVLSTVAKGRITRV...</td>\n",
       "      <td>VALGAGR</td>\n",
       "      <td>VAL</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627320.fasta</td>\n",
       "      <td>Genome544_k127_857444</td>\n",
       "      <td>721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627320.faa</td>\n",
       "      <td>Genome544_k127_977136_1</td>\n",
       "      <td>RFGSGQAVRRLEDEALLAGRGRYTDDIRLQDCAALAFQRSPHPHAR...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627320.fasta</td>\n",
       "      <td>Genome544_k127_977136</td>\n",
       "      <td>724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627321.faa</td>\n",
       "      <td>Genome545_k127_208535_1</td>\n",
       "      <td>RFGSGQAVRRLEDEALLAGRGRYTDDIRLQGCATLAFQRSPHPHAR...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627321.fasta</td>\n",
       "      <td>Genome545_k127_208535</td>\n",
       "      <td>765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627321.faa</td>\n",
       "      <td>Genome545_k127_344480_1</td>\n",
       "      <td>IGEAIRRKEDLRFLTGAGNYTGDVVLANQAHAVFLRSPHAHAAITS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627321.fasta</td>\n",
       "      <td>Genome545_k127_344480</td>\n",
       "      <td>777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>uncultured_Pseudomonas_sp._ERS6626673.faa</td>\n",
       "      <td>Genome114_S3Ck127_13141_2</td>\n",
       "      <td>MAPTAVRPDKEVGLRVAPRLRPPAPRPAELLVPMAAVLCVPAAEAA...</td>\n",
       "      <td>ASPGAGR</td>\n",
       "      <td>ASP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudomonas_sp._ERS6626673.fasta</td>\n",
       "      <td>Genome114_S3Ck127_13141</td>\n",
       "      <td>478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>uncultured_Pseudomonas_sp._ERS6627273.faa</td>\n",
       "      <td>Genome497_k127_28836_2</td>\n",
       "      <td>MACGPLLRGRRDASVRVLAVRGREGTAHTAAEAAAAGRRPVGEAEP...</td>\n",
       "      <td>ASPGAGR</td>\n",
       "      <td>ASP</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudomonas_sp._ERS6627273.fasta</td>\n",
       "      <td>Genome497_k127_28836</td>\n",
       "      <td>414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>uncultured_Pseudomonas_sp._ERS6627273.faa</td>\n",
       "      <td>Genome497_k127_28836_2</td>\n",
       "      <td>MACGPLLRGRRDASVRVLAVRGREGTAHTAAEAAAAGRRPVGEAEP...</td>\n",
       "      <td>GGAGAGR</td>\n",
       "      <td>GGA</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudomonas_sp._ERS6627273.fasta</td>\n",
       "      <td>Genome497_k127_28836</td>\n",
       "      <td>414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>uncultured_Pseudorhodoferax_sp._ERS6626941.faa</td>\n",
       "      <td>Genome382_k127_146993_1</td>\n",
       "      <td>SPTRFGSGQAVRRLEDDALLAGRGKFTDDVVPQPPGAQLRAVFVRS...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Pseudorhodoferax_sp._ERS6626941.fasta</td>\n",
       "      <td>Genome382_k127_146993</td>\n",
       "      <td>767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6626680.faa</td>\n",
       "      <td>Genome121_S1Ck127_116016_1</td>\n",
       "      <td>MHAAPGARRASRWQAAGAPCVGAPFHATPGRPAFRRGGVVVTRPPR...</td>\n",
       "      <td>SGHGAGR</td>\n",
       "      <td>SGH</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6626680.fasta</td>\n",
       "      <td>Genome121_S1Ck127_116016</td>\n",
       "      <td>287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6626831.faa</td>\n",
       "      <td>Genome272_k127_58089_2</td>\n",
       "      <td>MAAFAFLAARLGVKRAVVVRHARYNPRPAPAVREGNAAAFGTLVST...</td>\n",
       "      <td>VSTGAGR</td>\n",
       "      <td>VST</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6626831.fasta</td>\n",
       "      <td>Genome272_k127_58089</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6626957.faa</td>\n",
       "      <td>Genome398_k127_96590_2</td>\n",
       "      <td>MPRQQPLARWQEVTVEELALWVAEPVKERLEERALYLAPWAARPRG...</td>\n",
       "      <td>RPRGAGR</td>\n",
       "      <td>RPR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6626957.fasta</td>\n",
       "      <td>Genome398_k127_96590</td>\n",
       "      <td>132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6627258.faa</td>\n",
       "      <td>Genome482_k127_81292_1</td>\n",
       "      <td>IGKATDRIDGPRKVSGQAPYAYERHDVVANQAYGVIVGAAIAKGRI...</td>\n",
       "      <td>VRIGAGR</td>\n",
       "      <td>VRI</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6627258.fasta</td>\n",
       "      <td>Genome482_k127_81292</td>\n",
       "      <td>680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6627011.faa</td>\n",
       "      <td>Genome452_k127_363663_1</td>\n",
       "      <td>SNEDGRYDVSPGRASTAGAGRAIRLAT*</td>\n",
       "      <td>STAGAGR</td>\n",
       "      <td>STA</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6627011.fasta</td>\n",
       "      <td>Genome452_k127_363663</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>uncultured_Stenotrophomonas_sp._ERS6626688.faa</td>\n",
       "      <td>Genome129_S3Ck127_15574_1</td>\n",
       "      <td>KATYGGRVHGAGRMTASSGVAFGSASRLA*</td>\n",
       "      <td>RVHGAGR</td>\n",
       "      <td>RVH</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Stenotrophomonas_sp._ERS6626688.fasta</td>\n",
       "      <td>Genome129_S3Ck127_15574</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>uncultured_Stenotrophomonas_sp._ERS6627319.faa</td>\n",
       "      <td>Genome543_k127_120424_1</td>\n",
       "      <td>KATYGGRVHGAGRMTASSGVAFGSASRLA*</td>\n",
       "      <td>RVHGAGR</td>\n",
       "      <td>RVH</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Stenotrophomonas_sp._ERS6627319.fasta</td>\n",
       "      <td>Genome543_k127_120424</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.faa</td>\n",
       "      <td>Genome220_k127_244806_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.fasta</td>\n",
       "      <td>Genome220_k127_244806</td>\n",
       "      <td>767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.faa</td>\n",
       "      <td>Genome220_k127_244806_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.fasta</td>\n",
       "      <td>Genome220_k127_244806</td>\n",
       "      <td>767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>AYRGAGR</td>\n",
       "      <td>AYR</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.faa</td>\n",
       "      <td>Genome470_k127_43488_1</td>\n",
       "      <td>PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...</td>\n",
       "      <td>LLSGAGR</td>\n",
       "      <td>LLS</td>\n",
       "      <td>CoxL_form_2</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>767</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         File_Name  \\\n",
       "0        Burkholderiaceae_bacterium_ERS6626861.faa   \n",
       "1         uncultured_Acidovorax_sp._ERS6627271.faa   \n",
       "2      uncultured_Acinetobacter_sp._ERS6626821.faa   \n",
       "3      uncultured_Acinetobacter_sp._ERS6626858.faa   \n",
       "4         uncultured_Aureimonas_sp._ERS6626915.faa   \n",
       "5         uncultured_Aureimonas_sp._ERS6626915.faa   \n",
       "6         uncultured_Aureimonas_sp._ERS6627306.faa   \n",
       "7          uncultured_Comamonas_sp._ERS6626630.faa   \n",
       "8          uncultured_Comamonas_sp._ERS6626638.faa   \n",
       "9          uncultured_Comamonas_sp._ERS6626638.faa   \n",
       "10         uncultured_Comamonas_sp._ERS6626814.faa   \n",
       "11         uncultured_Comamonas_sp._ERS6627283.faa   \n",
       "12       uncultured_Deinococcus_sp._ERS6626797.faa   \n",
       "13       uncultured_Deinococcus_sp._ERS6626797.faa   \n",
       "14       uncultured_Deinococcus_sp._ERS6626820.faa   \n",
       "15       uncultured_Deinococcus_sp._ERS6626820.faa   \n",
       "16       uncultured_Deinococcus_sp._ERS6626837.faa   \n",
       "17       uncultured_Deinococcus_sp._ERS6627293.faa   \n",
       "18       uncultured_Deinococcus_sp._ERS6627335.faa   \n",
       "19      uncultured_Hymenobacter_sp._ERS6626733.faa   \n",
       "20      uncultured_Hymenobacter_sp._ERS6627279.faa   \n",
       "21  uncultured_Methylobacterium_sp._ERS6627288.faa   \n",
       "22  uncultured_Methylobacterium_sp._ERS6627288.faa   \n",
       "23    uncultured_Microbacterium_sp._ERS6626906.faa   \n",
       "24    uncultured_Microbacterium_sp._ERS6626964.faa   \n",
       "25  uncultured_Mucilaginibacter_sp._ERS6626873.faa   \n",
       "26  uncultured_Mucilaginibacter_sp._ERS6626923.faa   \n",
       "27  uncultured_Mucilaginibacter_sp._ERS6626962.faa   \n",
       "28  uncultured_Mucilaginibacter_sp._ERS6626972.faa   \n",
       "29  uncultured_Mucilaginibacter_sp._ERS6626981.faa   \n",
       "30  uncultured_Mucilaginibacter_sp._ERS6627275.faa   \n",
       "31  uncultured_Mucilaginibacter_sp._ERS6627290.faa   \n",
       "32           uncultured_Pantoea_sp._ERS6626643.faa   \n",
       "33           uncultured_Pantoea_sp._ERS6627005.faa   \n",
       "34         uncultured_Pelomonas_sp._ERS6626569.faa   \n",
       "35   uncultured_Pseudacidovorax_sp._ERS6626577.faa   \n",
       "36   uncultured_Pseudacidovorax_sp._ERS6626850.faa   \n",
       "37   uncultured_Pseudacidovorax_sp._ERS6626909.faa   \n",
       "38   uncultured_Pseudacidovorax_sp._ERS6627003.faa   \n",
       "39   uncultured_Pseudacidovorax_sp._ERS6627003.faa   \n",
       "40   uncultured_Pseudacidovorax_sp._ERS6627320.faa   \n",
       "41   uncultured_Pseudacidovorax_sp._ERS6627320.faa   \n",
       "42   uncultured_Pseudacidovorax_sp._ERS6627320.faa   \n",
       "43   uncultured_Pseudacidovorax_sp._ERS6627321.faa   \n",
       "44   uncultured_Pseudacidovorax_sp._ERS6627321.faa   \n",
       "45       uncultured_Pseudomonas_sp._ERS6626673.faa   \n",
       "46       uncultured_Pseudomonas_sp._ERS6627273.faa   \n",
       "47       uncultured_Pseudomonas_sp._ERS6627273.faa   \n",
       "48  uncultured_Pseudorhodoferax_sp._ERS6626941.faa   \n",
       "49      uncultured_Sphingomonas_sp._ERS6626680.faa   \n",
       "50      uncultured_Sphingomonas_sp._ERS6626831.faa   \n",
       "51      uncultured_Sphingomonas_sp._ERS6626957.faa   \n",
       "52      uncultured_Sphingomonas_sp._ERS6627258.faa   \n",
       "53         uncultured_Spirosoma_sp._ERS6627011.faa   \n",
       "54  uncultured_Stenotrophomonas_sp._ERS6626688.faa   \n",
       "55  uncultured_Stenotrophomonas_sp._ERS6627319.faa   \n",
       "56        uncultured_Variovorax_sp._ERS6626779.faa   \n",
       "57        uncultured_Variovorax_sp._ERS6626779.faa   \n",
       "58        uncultured_Variovorax_sp._ERS6627246.faa   \n",
       "59        uncultured_Variovorax_sp._ERS6627246.faa   \n",
       "\n",
       "                            ID  \\\n",
       "0      Genome302_k127_117597_1   \n",
       "1      Genome495_k127_317059_1   \n",
       "2      Genome262_k127_235961_1   \n",
       "3      Genome299_k127_417201_1   \n",
       "4      Genome356_k127_498042_1   \n",
       "5      Genome356_k127_610672_1   \n",
       "6       Genome530_k127_16854_1   \n",
       "7     Genome71_S1Ck127_18694_1   \n",
       "8    Genome79_S3Ck127_287303_1   \n",
       "9    Genome79_S3Ck127_287303_3   \n",
       "10      Genome255_k127_46261_1   \n",
       "11      Genome507_k127_91973_1   \n",
       "12      Genome238_k127_17753_1   \n",
       "13      Genome238_k127_54530_1   \n",
       "14     Genome261_k127_146972_1   \n",
       "15     Genome261_k127_332399_1   \n",
       "16      Genome278_k127_93298_1   \n",
       "17     Genome517_k127_348199_1   \n",
       "18     Genome559_k127_243715_1   \n",
       "19  Genome174_S1Ck127_276112_1   \n",
       "20     Genome503_k127_453582_1   \n",
       "21     Genome512_k127_160927_1   \n",
       "22      Genome512_k127_80099_1   \n",
       "23      Genome347_k127_84820_1   \n",
       "24     Genome405_k127_107645_1   \n",
       "25     Genome314_k127_168730_1   \n",
       "26     Genome364_k127_154890_1   \n",
       "27     Genome403_k127_249621_1   \n",
       "28     Genome413_k127_255610_1   \n",
       "29      Genome422_k127_88665_1   \n",
       "30     Genome499_k127_471593_1   \n",
       "31       Genome514_k127_4285_1   \n",
       "32   Genome84_S2Ck127_371734_2   \n",
       "33     Genome446_k127_205538_3   \n",
       "34   Genome10_S2Ck127_387977_3   \n",
       "35   Genome18_S1Ck127_217968_1   \n",
       "36     Genome291_k127_419550_1   \n",
       "37     Genome350_k127_213661_1   \n",
       "38      Genome444_k127_25498_1   \n",
       "39       Genome444_k127_2949_1   \n",
       "40     Genome544_k127_833201_1   \n",
       "41     Genome544_k127_857444_1   \n",
       "42     Genome544_k127_977136_1   \n",
       "43     Genome545_k127_208535_1   \n",
       "44     Genome545_k127_344480_1   \n",
       "45   Genome114_S3Ck127_13141_2   \n",
       "46      Genome497_k127_28836_2   \n",
       "47      Genome497_k127_28836_2   \n",
       "48     Genome382_k127_146993_1   \n",
       "49  Genome121_S1Ck127_116016_1   \n",
       "50      Genome272_k127_58089_2   \n",
       "51      Genome398_k127_96590_2   \n",
       "52      Genome482_k127_81292_1   \n",
       "53     Genome452_k127_363663_1   \n",
       "54   Genome129_S3Ck127_15574_1   \n",
       "55     Genome543_k127_120424_1   \n",
       "56     Genome220_k127_244806_1   \n",
       "57     Genome220_k127_244806_1   \n",
       "58      Genome470_k127_43488_1   \n",
       "59      Genome470_k127_43488_1   \n",
       "\n",
       "                                             Sequence    MOTIF Previous_3_AA  \\\n",
       "0   MAGARRHQAAEDVVGAGRDLKRLVRPGREDDEADAAAAVAPPPPVR...  DVVGAGR           DVV   \n",
       "1   IGEALKRKEDYRFLTGAGQYTDDVVLAAQCHAVFVRSPHAHAKINS...  AYRGAGR           AYR   \n",
       "2   NTMHLAVGFSSCAKGKISKFDLDAVRQADGVHAVFSAKDIDVENNW...  EQPGAGR           EQP   \n",
       "3   ELRNLRCKTNTVSNTAYRGFGGPQGMFVIENIIDDIARYLGCDPIE...  EQPGAGR           EQP   \n",
       "4   LAMIAAEKLKRPVRWVADRNEHFLADTHGRANLATATMALDAKGRF...  AYRGAGR           AYR   \n",
       "5   IGKGIDRFEGPLKVTGRARYAYENMTGPEVAYGFVVTAGTGAGRIK...  AGTGAGR           AGT   \n",
       "6   MGIEGIGARVLRKEDRRFITGRGRYTDDMVVPGMKYAAFVRSPHAH...  AYRGAGR           AYR   \n",
       "7   IGEPVKRKEDYRFLTGAGQYTDDIALAAQAHAVFVRSPHAHARVRS...  AYRGAGR           AYR   \n",
       "8   PATPADVAAIRTRPAVGGNARPGQRAPGTGKRQRPGPARRAGRATP...  PRPGAGR           PRP   \n",
       "9   GVAHSVCRGSCRRRRRRAGGGTTPPTRARPTAAACRRASRARPSAT...  RVSGAGR           RVS   \n",
       "10  PATPADIAAIRTRPTVGRNARPGQRVTGTGGREVPTPARRAGRATP...  PRPGAGR           PRP   \n",
       "11  IGEPVKRKEDYRFLTGAGQYTDDIALAAQAHAVFVRSPHAHARVRS...  AYRGAGR           AYR   \n",
       "12  VGQALKRKEDPRFITGTGNYTDDIVLHGMLHAAMVRSPYPHARITA...  AYRGAGR           AYR   \n",
       "13  PRMTSVGRPVKRIEGRAKVLGKARFAAEYPIRGLVYAAPVLSTIPK...  GFGGAGR           GFG   \n",
       "14  VGQALKRKEDPRFITGTGNYTDDIVLHGMLHAAMVRSPYPHARITA...  AYRGAGR           AYR   \n",
       "15  MTSVGRPVKRIEGRAKVLGKARFAAEYPIRGLVYAAPVLSTIPKGR...  GFGGAGR           GFG   \n",
       "16  VGQALKRKEDPRFITGTGNYTDDIVLHGMLHAAMVRSPYPHARITA...  AYRGAGR           AYR   \n",
       "17  VGQALKRKEDPRFITGTGNYTDDIVLHGMLHAAMVRSPYPHARITA...  AYRGAGR           AYR   \n",
       "18  PRMTSVGRPVKRIEGRAKVLGKARFAAEYPIRGLVYAAPVLSTIPK...  GFGGAGR           GFG   \n",
       "19  VGKPLVRVEGREKVTGKAKYAAEFALPGLTYGVFTTSPIAKGKITS...  SVIGAGR           SVI   \n",
       "20  VGKPLVRVEGREKVTGKAKYAAEFALPGLTYGVFTTSPIAKGKITR...  SVIGAGR           SVI   \n",
       "21  MTVTKFGLGQPLRRVEDRRLITGEGNYGDDHAPEGCLHAALLRSPH...  AYRGAGR           AYR   \n",
       "22  PQKFGIGQPVRRAEDPVLVRGEGRYTDDLALDGQAYGVFVRSPVAH...  AYRGAGR           AYR   \n",
       "23  DHAPEGCLHAALLRSPHAHATFTITDVEAARAMPGVHLVLTAEDVA...  AYRGAGR           AYR   \n",
       "24  VGASVIPEAARRLVQGREPFTLDEPVPGGPPLVLRVVTSPHAHARV...  VGSGAGR           VGS   \n",
       "25  IAPPPFTDGIDRVDGRLKVTGAAKYSAEYDMPNMSYAVFALSTITK...  GVFGAGR           GVF   \n",
       "26  IDRVDGRLKVTGAAKYAAEFEVPGMTYAVFANSTITKGTIKNIDTK...  GVFGAGR           GVF   \n",
       "27  IDRVDGRLKVTGAAKYAAEFEVPGMTYAVFANSTITKGTIKTMDIK...  GVFGAGR           GVF   \n",
       "28  IDRVDGRLKVTGAAKYAAEFEVPGMTYAVFANSTITKGTIKTIDTK...  GVFGAGR           GVF   \n",
       "29  IDRVDGRLKVTGAAKYAAEFEVPGMTYAVFANSTITKGTIKTIDTK...  GVFGAGR           GVF   \n",
       "30  IDRVDGRLKVTGAAKYAAEFEVPGMTYAVFANSTITKGTIKTIDTK...  GVFGAGR           GVF   \n",
       "31  IDRVDGRLKVTGAAKYAAEFEVPGMTYAVFANSTITKGTIKTIDIK...  GVFGAGR           GVF   \n",
       "32  MRRLRAVTVCLCGAGGGPRETAVSAAGTRRIVTTETSITTGSEGAG...  GSEGAGR           GSE   \n",
       "33  LRGSYGRDRTARAGRRLPLLSIAATVGAGRCVWCAATGCICHQEAA...  ATVGAGR           ATV   \n",
       "34  SPSATASPRTSTTRAPTTRPATGPRSGRGGSSRSGTGPTSGEAGRA...  QVRGAGR           QVR   \n",
       "35  IGEAIRRKEDVRFLTGAGNYTGDVVLANQAHAVFLRSPHAHAAITS...  AYRGAGR           AYR   \n",
       "36  IGEAIRRKEDVRFLTGAGNYTGDVVLAHQAHAVFLRSPHAHAAITS...  AYRGAGR           AYR   \n",
       "37  RFGSGQAVRRLEDEALLAGRGRYTDDIRPRDCAALAFQRSPHPHAR...  AYRGAGR           AYR   \n",
       "38  VKWTGERSEEFLTTTHGRDVRARAAMALDADGRVLALRVHSDANVG...  AYRGAGR           AYR   \n",
       "39  IGEAIRRKEDVRFLTGAGNYTGDVVLAHQAHAVFLRSPHAHAAITS...  AYRGAGR           AYR   \n",
       "40  IGEAIRRKEDLRFLTGAGNYTGDVVLANQAHAVFLRSPHAHAAIQS...  AYRGAGR           AYR   \n",
       "41  GKETPRVDGIAKVTGQAKYAAEFQLPNLAYGFLVLSTVAKGRITRV...  VALGAGR           VAL   \n",
       "42  RFGSGQAVRRLEDEALLAGRGRYTDDIRLQDCAALAFQRSPHPHAR...  AYRGAGR           AYR   \n",
       "43  RFGSGQAVRRLEDEALLAGRGRYTDDIRLQGCATLAFQRSPHPHAR...  AYRGAGR           AYR   \n",
       "44  IGEAIRRKEDLRFLTGAGNYTGDVVLANQAHAVFLRSPHAHAAITS...  AYRGAGR           AYR   \n",
       "45  MAPTAVRPDKEVGLRVAPRLRPPAPRPAELLVPMAAVLCVPAAEAA...  ASPGAGR           ASP   \n",
       "46  MACGPLLRGRRDASVRVLAVRGREGTAHTAAEAAAAGRRPVGEAEP...  ASPGAGR           ASP   \n",
       "47  MACGPLLRGRRDASVRVLAVRGREGTAHTAAEAAAAGRRPVGEAEP...  GGAGAGR           GGA   \n",
       "48  SPTRFGSGQAVRRLEDDALLAGRGKFTDDVVPQPPGAQLRAVFVRS...  AYRGAGR           AYR   \n",
       "49  MHAAPGARRASRWQAAGAPCVGAPFHATPGRPAFRRGGVVVTRPPR...  SGHGAGR           SGH   \n",
       "50  MAAFAFLAARLGVKRAVVVRHARYNPRPAPAVREGNAAAFGTLVST...  VSTGAGR           VST   \n",
       "51  MPRQQPLARWQEVTVEELALWVAEPVKERLEERALYLAPWAARPRG...  RPRGAGR           RPR   \n",
       "52  IGKATDRIDGPRKVSGQAPYAYERHDVVANQAYGVIVGAAIAKGRI...  VRIGAGR           VRI   \n",
       "53                       SNEDGRYDVSPGRASTAGAGRAIRLAT*  STAGAGR           STA   \n",
       "54                     KATYGGRVHGAGRMTASSGVAFGSASRLA*  RVHGAGR           RVH   \n",
       "55                     KATYGGRVHGAGRMTASSGVAFGSASRLA*  RVHGAGR           RVH   \n",
       "56  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  AYRGAGR           AYR   \n",
       "57  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  LLSGAGR           LLS   \n",
       "58  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  AYRGAGR           AYR   \n",
       "59  PTRFGSGQAVRRLEDESLLSGAGRYTDDVTLPEQAHLVFLRSSYPH...  LLSGAGR           LLS   \n",
       "\n",
       "    Motif_Label                                       target_file  \\\n",
       "0   CoxL_form_2       Burkholderiaceae_bacterium_ERS6626861.fasta   \n",
       "1   CoxL_form_2        uncultured_Acidovorax_sp._ERS6627271.fasta   \n",
       "2   CoxL_form_2     uncultured_Acinetobacter_sp._ERS6626821.fasta   \n",
       "3   CoxL_form_2     uncultured_Acinetobacter_sp._ERS6626858.fasta   \n",
       "4   CoxL_form_2        uncultured_Aureimonas_sp._ERS6626915.fasta   \n",
       "5   CoxL_form_2        uncultured_Aureimonas_sp._ERS6626915.fasta   \n",
       "6   CoxL_form_2        uncultured_Aureimonas_sp._ERS6627306.fasta   \n",
       "7   CoxL_form_2         uncultured_Comamonas_sp._ERS6626630.fasta   \n",
       "8   CoxL_form_2         uncultured_Comamonas_sp._ERS6626638.fasta   \n",
       "9   CoxL_form_2         uncultured_Comamonas_sp._ERS6626638.fasta   \n",
       "10  CoxL_form_2         uncultured_Comamonas_sp._ERS6626814.fasta   \n",
       "11  CoxL_form_2         uncultured_Comamonas_sp._ERS6627283.fasta   \n",
       "12  CoxL_form_2       uncultured_Deinococcus_sp._ERS6626797.fasta   \n",
       "13  CoxL_form_2       uncultured_Deinococcus_sp._ERS6626797.fasta   \n",
       "14  CoxL_form_2       uncultured_Deinococcus_sp._ERS6626820.fasta   \n",
       "15  CoxL_form_2       uncultured_Deinococcus_sp._ERS6626820.fasta   \n",
       "16  CoxL_form_2       uncultured_Deinococcus_sp._ERS6626837.fasta   \n",
       "17  CoxL_form_2       uncultured_Deinococcus_sp._ERS6627293.fasta   \n",
       "18  CoxL_form_2       uncultured_Deinococcus_sp._ERS6627335.fasta   \n",
       "19  CoxL_form_2      uncultured_Hymenobacter_sp._ERS6626733.fasta   \n",
       "20  CoxL_form_2      uncultured_Hymenobacter_sp._ERS6627279.fasta   \n",
       "21  CoxL_form_2  uncultured_Methylobacterium_sp._ERS6627288.fasta   \n",
       "22  CoxL_form_2  uncultured_Methylobacterium_sp._ERS6627288.fasta   \n",
       "23  CoxL_form_2    uncultured_Microbacterium_sp._ERS6626906.fasta   \n",
       "24  CoxL_form_2    uncultured_Microbacterium_sp._ERS6626964.fasta   \n",
       "25  CoxL_form_2  uncultured_Mucilaginibacter_sp._ERS6626873.fasta   \n",
       "26  CoxL_form_2  uncultured_Mucilaginibacter_sp._ERS6626923.fasta   \n",
       "27  CoxL_form_2  uncultured_Mucilaginibacter_sp._ERS6626962.fasta   \n",
       "28  CoxL_form_2  uncultured_Mucilaginibacter_sp._ERS6626972.fasta   \n",
       "29  CoxL_form_2  uncultured_Mucilaginibacter_sp._ERS6626981.fasta   \n",
       "30  CoxL_form_2  uncultured_Mucilaginibacter_sp._ERS6627275.fasta   \n",
       "31  CoxL_form_2  uncultured_Mucilaginibacter_sp._ERS6627290.fasta   \n",
       "32  CoxL_form_2           uncultured_Pantoea_sp._ERS6626643.fasta   \n",
       "33  CoxL_form_2           uncultured_Pantoea_sp._ERS6627005.fasta   \n",
       "34  CoxL_form_2         uncultured_Pelomonas_sp._ERS6626569.fasta   \n",
       "35  CoxL_form_2   uncultured_Pseudacidovorax_sp._ERS6626577.fasta   \n",
       "36  CoxL_form_2   uncultured_Pseudacidovorax_sp._ERS6626850.fasta   \n",
       "37  CoxL_form_2   uncultured_Pseudacidovorax_sp._ERS6626909.fasta   \n",
       "38  CoxL_form_2   uncultured_Pseudacidovorax_sp._ERS6627003.fasta   \n",
       "39  CoxL_form_2   uncultured_Pseudacidovorax_sp._ERS6627003.fasta   \n",
       "40  CoxL_form_2   uncultured_Pseudacidovorax_sp._ERS6627320.fasta   \n",
       "41  CoxL_form_2   uncultured_Pseudacidovorax_sp._ERS6627320.fasta   \n",
       "42  CoxL_form_2   uncultured_Pseudacidovorax_sp._ERS6627320.fasta   \n",
       "43  CoxL_form_2   uncultured_Pseudacidovorax_sp._ERS6627321.fasta   \n",
       "44  CoxL_form_2   uncultured_Pseudacidovorax_sp._ERS6627321.fasta   \n",
       "45  CoxL_form_2       uncultured_Pseudomonas_sp._ERS6626673.fasta   \n",
       "46  CoxL_form_2       uncultured_Pseudomonas_sp._ERS6627273.fasta   \n",
       "47  CoxL_form_2       uncultured_Pseudomonas_sp._ERS6627273.fasta   \n",
       "48  CoxL_form_2  uncultured_Pseudorhodoferax_sp._ERS6626941.fasta   \n",
       "49  CoxL_form_2      uncultured_Sphingomonas_sp._ERS6626680.fasta   \n",
       "50  CoxL_form_2      uncultured_Sphingomonas_sp._ERS6626831.fasta   \n",
       "51  CoxL_form_2      uncultured_Sphingomonas_sp._ERS6626957.fasta   \n",
       "52  CoxL_form_2      uncultured_Sphingomonas_sp._ERS6627258.fasta   \n",
       "53  CoxL_form_2         uncultured_Spirosoma_sp._ERS6627011.fasta   \n",
       "54  CoxL_form_2  uncultured_Stenotrophomonas_sp._ERS6626688.fasta   \n",
       "55  CoxL_form_2  uncultured_Stenotrophomonas_sp._ERS6627319.fasta   \n",
       "56  CoxL_form_2        uncultured_Variovorax_sp._ERS6626779.fasta   \n",
       "57  CoxL_form_2        uncultured_Variovorax_sp._ERS6626779.fasta   \n",
       "58  CoxL_form_2        uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "59  CoxL_form_2        uncultured_Variovorax_sp._ERS6627246.fasta   \n",
       "\n",
       "                      qseqid  sequence_length  \n",
       "0      Genome302_k127_117597              183  \n",
       "1      Genome495_k127_317059              783  \n",
       "2      Genome262_k127_235961              663  \n",
       "3      Genome299_k127_417201              438  \n",
       "4      Genome356_k127_498042              498  \n",
       "5      Genome356_k127_610672              370  \n",
       "6       Genome530_k127_16854              780  \n",
       "7     Genome71_S1Ck127_18694              782  \n",
       "8    Genome79_S3Ck127_287303              200  \n",
       "9    Genome79_S3Ck127_287303              294  \n",
       "10      Genome255_k127_46261              223  \n",
       "11      Genome507_k127_91973              782  \n",
       "12      Genome238_k127_17753              771  \n",
       "13      Genome238_k127_54530              713  \n",
       "14     Genome261_k127_146972              771  \n",
       "15     Genome261_k127_332399              317  \n",
       "16      Genome278_k127_93298              775  \n",
       "17     Genome517_k127_348199              771  \n",
       "18     Genome559_k127_243715              713  \n",
       "19  Genome174_S1Ck127_276112              722  \n",
       "20     Genome503_k127_453582              722  \n",
       "21     Genome512_k127_160927              759  \n",
       "22      Genome512_k127_80099              768  \n",
       "23      Genome347_k127_84820              347  \n",
       "24     Genome405_k127_107645              768  \n",
       "25     Genome314_k127_168730              713  \n",
       "26     Genome364_k127_154890              488  \n",
       "27     Genome403_k127_249621              589  \n",
       "28     Genome413_k127_255610              701  \n",
       "29      Genome422_k127_88665              701  \n",
       "30     Genome499_k127_471593              701  \n",
       "31       Genome514_k127_4285              701  \n",
       "32   Genome84_S2Ck127_371734               87  \n",
       "33     Genome446_k127_205538               87  \n",
       "34   Genome10_S2Ck127_387977              253  \n",
       "35   Genome18_S1Ck127_217968              777  \n",
       "36     Genome291_k127_419550              777  \n",
       "37     Genome350_k127_213661              765  \n",
       "38      Genome444_k127_25498              490  \n",
       "39       Genome444_k127_2949              710  \n",
       "40     Genome544_k127_833201              777  \n",
       "41     Genome544_k127_857444              721  \n",
       "42     Genome544_k127_977136              724  \n",
       "43     Genome545_k127_208535              765  \n",
       "44     Genome545_k127_344480              777  \n",
       "45   Genome114_S3Ck127_13141              478  \n",
       "46      Genome497_k127_28836              414  \n",
       "47      Genome497_k127_28836              414  \n",
       "48     Genome382_k127_146993              767  \n",
       "49  Genome121_S1Ck127_116016              287  \n",
       "50      Genome272_k127_58089              216  \n",
       "51      Genome398_k127_96590              132  \n",
       "52      Genome482_k127_81292              680  \n",
       "53     Genome452_k127_363663               28  \n",
       "54   Genome129_S3Ck127_15574               30  \n",
       "55     Genome543_k127_120424               30  \n",
       "56     Genome220_k127_244806              767  \n",
       "57     Genome220_k127_244806              767  \n",
       "58      Genome470_k127_43488              767  \n",
       "59      Genome470_k127_43488              767  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Group the data frame by 'File_Name', 'qseqid' & 'MOTIF', considering 1st row of the group as best match removing other duplicates\n",
    "coxl2_motif_results = coxl2_motif_results.groupby(['File_Name', 'qseqid', 'MOTIF']).head(1).reset_index(drop=True)\n",
    "\n",
    "coxl2_motif_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "1d753d67-e356-4051-95bc-5627af6abca6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save filtered data frame\n",
    "coxl2_motif_results.to_csv(f\"{motif2}/coxl2_motif_results.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "c332eb3e-3fd6-4cf1-b2b4-8bd06f68d667",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "48\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['Burkholderiaceae_bacterium_ERS6626861.fasta',\n",
       " 'uncultured_Acidovorax_sp._ERS6627271.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626821.fasta',\n",
       " 'uncultured_Acinetobacter_sp._ERS6626858.fasta',\n",
       " 'uncultured_Aureimonas_sp._ERS6626915.fasta',\n",
       " 'uncultured_Aureimonas_sp._ERS6627306.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626630.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626638.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6626814.fasta',\n",
       " 'uncultured_Comamonas_sp._ERS6627283.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6626797.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6626820.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6626837.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6627293.fasta',\n",
       " 'uncultured_Deinococcus_sp._ERS6627335.fasta',\n",
       " 'uncultured_Hymenobacter_sp._ERS6626733.fasta',\n",
       " 'uncultured_Hymenobacter_sp._ERS6627279.fasta',\n",
       " 'uncultured_Methylobacterium_sp._ERS6627288.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626906.fasta',\n",
       " 'uncultured_Microbacterium_sp._ERS6626964.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626873.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626923.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626962.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626972.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626981.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6627275.fasta',\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6627290.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6626643.fasta',\n",
       " 'uncultured_Pantoea_sp._ERS6627005.fasta',\n",
       " 'uncultured_Pelomonas_sp._ERS6626569.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626577.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626850.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626909.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627003.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627320.fasta',\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627321.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6626673.fasta',\n",
       " 'uncultured_Pseudomonas_sp._ERS6627273.fasta',\n",
       " 'uncultured_Pseudorhodoferax_sp._ERS6626941.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626680.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626831.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6626957.fasta',\n",
       " 'uncultured_Sphingomonas_sp._ERS6627258.fasta',\n",
       " 'uncultured_Spirosoma_sp._ERS6627011.fasta',\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6626688.fasta',\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6627319.fasta',\n",
       " 'uncultured_Variovorax_sp._ERS6626779.fasta',\n",
       " 'uncultured_Variovorax_sp._ERS6627246.fasta']"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Genereating genome files name list\n",
    "coxl2_motif_mag_list = list(coxl2_motif_results['target_file'].unique())\n",
    "print(len(coxl2_motif_mag_list))\n",
    "coxl2_motif_mag_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "d1cd4181-7591-4dd4-8933-3a71c3e4a1b2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Burkholderiaceae_bacterium_ERS6626861.fasta': {'Genome302_k127_117597'},\n",
       " 'uncultured_Acidovorax_sp._ERS6627271.fasta': {'Genome495_k127_317059'},\n",
       " 'uncultured_Acinetobacter_sp._ERS6626821.fasta': {'Genome262_k127_235961'},\n",
       " 'uncultured_Acinetobacter_sp._ERS6626858.fasta': {'Genome299_k127_417201'},\n",
       " 'uncultured_Aureimonas_sp._ERS6626915.fasta': {'Genome356_k127_498042',\n",
       "  'Genome356_k127_610672'},\n",
       " 'uncultured_Aureimonas_sp._ERS6627306.fasta': {'Genome530_k127_16854'},\n",
       " 'uncultured_Comamonas_sp._ERS6626630.fasta': {'Genome71_S1Ck127_18694'},\n",
       " 'uncultured_Comamonas_sp._ERS6626638.fasta': {'Genome79_S3Ck127_287303'},\n",
       " 'uncultured_Comamonas_sp._ERS6626814.fasta': {'Genome255_k127_46261'},\n",
       " 'uncultured_Comamonas_sp._ERS6627283.fasta': {'Genome507_k127_91973'},\n",
       " 'uncultured_Deinococcus_sp._ERS6626797.fasta': {'Genome238_k127_17753',\n",
       "  'Genome238_k127_54530'},\n",
       " 'uncultured_Deinococcus_sp._ERS6626820.fasta': {'Genome261_k127_146972',\n",
       "  'Genome261_k127_332399'},\n",
       " 'uncultured_Deinococcus_sp._ERS6626837.fasta': {'Genome278_k127_93298'},\n",
       " 'uncultured_Deinococcus_sp._ERS6627293.fasta': {'Genome517_k127_348199'},\n",
       " 'uncultured_Deinococcus_sp._ERS6627335.fasta': {'Genome559_k127_243715'},\n",
       " 'uncultured_Hymenobacter_sp._ERS6626733.fasta': {'Genome174_S1Ck127_276112'},\n",
       " 'uncultured_Hymenobacter_sp._ERS6627279.fasta': {'Genome503_k127_453582'},\n",
       " 'uncultured_Methylobacterium_sp._ERS6627288.fasta': {'Genome512_k127_160927',\n",
       "  'Genome512_k127_80099'},\n",
       " 'uncultured_Microbacterium_sp._ERS6626906.fasta': {'Genome347_k127_84820'},\n",
       " 'uncultured_Microbacterium_sp._ERS6626964.fasta': {'Genome405_k127_107645'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626873.fasta': {'Genome314_k127_168730'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626923.fasta': {'Genome364_k127_154890'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626962.fasta': {'Genome403_k127_249621'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626972.fasta': {'Genome413_k127_255610'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6626981.fasta': {'Genome422_k127_88665'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6627275.fasta': {'Genome499_k127_471593'},\n",
       " 'uncultured_Mucilaginibacter_sp._ERS6627290.fasta': {'Genome514_k127_4285'},\n",
       " 'uncultured_Pantoea_sp._ERS6626643.fasta': {'Genome84_S2Ck127_371734'},\n",
       " 'uncultured_Pantoea_sp._ERS6627005.fasta': {'Genome446_k127_205538'},\n",
       " 'uncultured_Pelomonas_sp._ERS6626569.fasta': {'Genome10_S2Ck127_387977'},\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626577.fasta': {'Genome18_S1Ck127_217968'},\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626850.fasta': {'Genome291_k127_419550'},\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6626909.fasta': {'Genome350_k127_213661'},\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627003.fasta': {'Genome444_k127_25498',\n",
       "  'Genome444_k127_2949'},\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627320.fasta': {'Genome544_k127_833201',\n",
       "  'Genome544_k127_857444',\n",
       "  'Genome544_k127_977136'},\n",
       " 'uncultured_Pseudacidovorax_sp._ERS6627321.fasta': {'Genome545_k127_208535',\n",
       "  'Genome545_k127_344480'},\n",
       " 'uncultured_Pseudomonas_sp._ERS6626673.fasta': {'Genome114_S3Ck127_13141'},\n",
       " 'uncultured_Pseudomonas_sp._ERS6627273.fasta': {'Genome497_k127_28836'},\n",
       " 'uncultured_Pseudorhodoferax_sp._ERS6626941.fasta': {'Genome382_k127_146993'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6626680.fasta': {'Genome121_S1Ck127_116016'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6626831.fasta': {'Genome272_k127_58089'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6626957.fasta': {'Genome398_k127_96590'},\n",
       " 'uncultured_Sphingomonas_sp._ERS6627258.fasta': {'Genome482_k127_81292'},\n",
       " 'uncultured_Spirosoma_sp._ERS6627011.fasta': {'Genome452_k127_363663'},\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6626688.fasta': {'Genome129_S3Ck127_15574'},\n",
       " 'uncultured_Stenotrophomonas_sp._ERS6627319.fasta': {'Genome543_k127_120424'},\n",
       " 'uncultured_Variovorax_sp._ERS6626779.fasta': {'Genome220_k127_244806'},\n",
       " 'uncultured_Variovorax_sp._ERS6627246.fasta': {'Genome470_k127_43488'}}"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Generating a set with genome file names and its query sequence ID\n",
    "coxl2_motif_id_set = coxl2_motif_results.groupby('target_file')['qseqid'].apply(set).to_dict()\n",
    "coxl2_motif_id_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "1b541c3f-4094-4c38-8592-4dabb1cdd93e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Target File: Burkholderiaceae_bacterium_ERS6626861.fasta\n",
      "  ID: Genome302_k127_117597, Previous 3 AA: ['DVV']\n",
      "Target File: uncultured_Acidovorax_sp._ERS6627271.fasta\n",
      "  ID: Genome495_k127_317059, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Acinetobacter_sp._ERS6626821.fasta\n",
      "  ID: Genome262_k127_235961, Previous 3 AA: ['EQP']\n",
      "Target File: uncultured_Acinetobacter_sp._ERS6626858.fasta\n",
      "  ID: Genome299_k127_417201, Previous 3 AA: ['EQP']\n",
      "Target File: uncultured_Aureimonas_sp._ERS6626915.fasta\n",
      "  ID: Genome356_k127_498042, Previous 3 AA: ['AYR']\n",
      "  ID: Genome356_k127_610672, Previous 3 AA: ['AGT']\n",
      "Target File: uncultured_Aureimonas_sp._ERS6627306.fasta\n",
      "  ID: Genome530_k127_16854, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Comamonas_sp._ERS6626630.fasta\n",
      "  ID: Genome71_S1Ck127_18694, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Comamonas_sp._ERS6626638.fasta\n",
      "  ID: Genome79_S3Ck127_287303, Previous 3 AA: ['PRP', 'RVS']\n",
      "Target File: uncultured_Comamonas_sp._ERS6626814.fasta\n",
      "  ID: Genome255_k127_46261, Previous 3 AA: ['PRP']\n",
      "Target File: uncultured_Comamonas_sp._ERS6627283.fasta\n",
      "  ID: Genome507_k127_91973, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Deinococcus_sp._ERS6626797.fasta\n",
      "  ID: Genome238_k127_17753, Previous 3 AA: ['AYR']\n",
      "  ID: Genome238_k127_54530, Previous 3 AA: ['GFG']\n",
      "Target File: uncultured_Deinococcus_sp._ERS6626820.fasta\n",
      "  ID: Genome261_k127_146972, Previous 3 AA: ['AYR']\n",
      "  ID: Genome261_k127_332399, Previous 3 AA: ['GFG']\n",
      "Target File: uncultured_Deinococcus_sp._ERS6626837.fasta\n",
      "  ID: Genome278_k127_93298, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Deinococcus_sp._ERS6627293.fasta\n",
      "  ID: Genome517_k127_348199, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Deinococcus_sp._ERS6627335.fasta\n",
      "  ID: Genome559_k127_243715, Previous 3 AA: ['GFG']\n",
      "Target File: uncultured_Hymenobacter_sp._ERS6626733.fasta\n",
      "  ID: Genome174_S1Ck127_276112, Previous 3 AA: ['SVI']\n",
      "Target File: uncultured_Hymenobacter_sp._ERS6627279.fasta\n",
      "  ID: Genome503_k127_453582, Previous 3 AA: ['SVI']\n",
      "Target File: uncultured_Methylobacterium_sp._ERS6627288.fasta\n",
      "  ID: Genome512_k127_160927, Previous 3 AA: ['AYR']\n",
      "  ID: Genome512_k127_80099, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Microbacterium_sp._ERS6626906.fasta\n",
      "  ID: Genome347_k127_84820, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Microbacterium_sp._ERS6626964.fasta\n",
      "  ID: Genome405_k127_107645, Previous 3 AA: ['VGS']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6626873.fasta\n",
      "  ID: Genome314_k127_168730, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6626923.fasta\n",
      "  ID: Genome364_k127_154890, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6626962.fasta\n",
      "  ID: Genome403_k127_249621, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6626972.fasta\n",
      "  ID: Genome413_k127_255610, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6626981.fasta\n",
      "  ID: Genome422_k127_88665, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6627275.fasta\n",
      "  ID: Genome499_k127_471593, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Mucilaginibacter_sp._ERS6627290.fasta\n",
      "  ID: Genome514_k127_4285, Previous 3 AA: ['GVF']\n",
      "Target File: uncultured_Pantoea_sp._ERS6626643.fasta\n",
      "  ID: Genome84_S2Ck127_371734, Previous 3 AA: ['GSE']\n",
      "Target File: uncultured_Pantoea_sp._ERS6627005.fasta\n",
      "  ID: Genome446_k127_205538, Previous 3 AA: ['ATV']\n",
      "Target File: uncultured_Pelomonas_sp._ERS6626569.fasta\n",
      "  ID: Genome10_S2Ck127_387977, Previous 3 AA: ['QVR']\n",
      "Target File: uncultured_Pseudacidovorax_sp._ERS6626577.fasta\n",
      "  ID: Genome18_S1Ck127_217968, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Pseudacidovorax_sp._ERS6626850.fasta\n",
      "  ID: Genome291_k127_419550, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Pseudacidovorax_sp._ERS6626909.fasta\n",
      "  ID: Genome350_k127_213661, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Pseudacidovorax_sp._ERS6627003.fasta\n",
      "  ID: Genome444_k127_25498, Previous 3 AA: ['AYR']\n",
      "  ID: Genome444_k127_2949, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Pseudacidovorax_sp._ERS6627320.fasta\n",
      "  ID: Genome544_k127_833201, Previous 3 AA: ['AYR']\n",
      "  ID: Genome544_k127_857444, Previous 3 AA: ['VAL']\n",
      "  ID: Genome544_k127_977136, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Pseudacidovorax_sp._ERS6627321.fasta\n",
      "  ID: Genome545_k127_208535, Previous 3 AA: ['AYR']\n",
      "  ID: Genome545_k127_344480, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Pseudomonas_sp._ERS6626673.fasta\n",
      "  ID: Genome114_S3Ck127_13141, Previous 3 AA: ['ASP']\n",
      "Target File: uncultured_Pseudomonas_sp._ERS6627273.fasta\n",
      "  ID: Genome497_k127_28836, Previous 3 AA: ['ASP', 'GGA']\n",
      "Target File: uncultured_Pseudorhodoferax_sp._ERS6626941.fasta\n",
      "  ID: Genome382_k127_146993, Previous 3 AA: ['AYR']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6626680.fasta\n",
      "  ID: Genome121_S1Ck127_116016, Previous 3 AA: ['SGH']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6626831.fasta\n",
      "  ID: Genome272_k127_58089, Previous 3 AA: ['VST']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6626957.fasta\n",
      "  ID: Genome398_k127_96590, Previous 3 AA: ['RPR']\n",
      "Target File: uncultured_Sphingomonas_sp._ERS6627258.fasta\n",
      "  ID: Genome482_k127_81292, Previous 3 AA: ['VRI']\n",
      "Target File: uncultured_Spirosoma_sp._ERS6627011.fasta\n",
      "  ID: Genome452_k127_363663, Previous 3 AA: ['STA']\n",
      "Target File: uncultured_Stenotrophomonas_sp._ERS6626688.fasta\n",
      "  ID: Genome129_S3Ck127_15574, Previous 3 AA: ['RVH']\n",
      "Target File: uncultured_Stenotrophomonas_sp._ERS6627319.fasta\n",
      "  ID: Genome543_k127_120424, Previous 3 AA: ['RVH']\n",
      "Target File: uncultured_Variovorax_sp._ERS6626779.fasta\n",
      "  ID: Genome220_k127_244806, Previous 3 AA: ['AYR', 'LLS']\n",
      "Target File: uncultured_Variovorax_sp._ERS6627246.fasta\n",
      "  ID: Genome470_k127_43488, Previous 3 AA: ['AYR', 'LLS']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\dkb1g24\\AppData\\Local\\Temp\\ipykernel_13124\\983491136.py:6: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  coxl2_motif_3aa_dict = coxl2_motif_3aa.groupby('target_file').apply(lambda x: x.set_index('qseqid')['Previous_3_AA'].to_dict()).to_dict()\n"
     ]
    }
   ],
   "source": [
    "# Generate a set within a set containing genome file names with their respective query sequences and the first three amino acid sequences of the MOTIF\n",
    "## Initialize a dictionary to hold the data\n",
    "coxl2_motif_3aa = coxl2_motif_results.groupby(['target_file', 'qseqid'])['Previous_3_AA'].apply(list).reset_index()\n",
    "\n",
    "## Create the dictionary\n",
    "coxl2_motif_3aa_dict = coxl2_motif_3aa.groupby('target_file').apply(lambda x: x.set_index('qseqid')['Previous_3_AA'].to_dict()).to_dict()\n",
    "\n",
    "## Print the resulting dictionary\n",
    "for target_file, ids in coxl2_motif_3aa_dict.items():\n",
    "    print(f\"Target File: {target_file}\")\n",
    "    for id, previous_3_aa in ids.items():\n",
    "        print(f\"  ID: {id}, Previous 3 AA: {previous_3_aa}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fce5e818-cec7-4daf-8993-4c84216d0139",
   "metadata": {},
   "source": [
    "<h3 style=\"font-family: 'Times New Roman'; font-weight: bold;\">\n",
    "    Filter CoxL Form 2 BLAST Results with MOTIF Search Results\n",
    "</h3>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "2600204b-c6ef-462f-8825-8b1aa7a4c6ba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>24.931</td>\n",
       "      <td>42</td>\n",
       "      <td>8.800000e-31</td>\n",
       "      <td>119.0</td>\n",
       "      <td>4944</td>\n",
       "      <td>2872</td>\n",
       "      <td>52</td>\n",
       "      <td>754</td>\n",
       "      <td>726</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Genome20_S3Ck127_316360</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>24.933</td>\n",
       "      <td>41</td>\n",
       "      <td>1.080000e-26</td>\n",
       "      <td>105.0</td>\n",
       "      <td>4953</td>\n",
       "      <td>2911</td>\n",
       "      <td>48</td>\n",
       "      <td>760</td>\n",
       "      <td>742</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626579.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Genome267_k127_201050</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>23.846</td>\n",
       "      <td>21</td>\n",
       "      <td>4.900000e-35</td>\n",
       "      <td>134.0</td>\n",
       "      <td>6807</td>\n",
       "      <td>8963</td>\n",
       "      <td>7</td>\n",
       "      <td>767</td>\n",
       "      <td>780</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Genome267_k127_201050</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>24.791</td>\n",
       "      <td>17</td>\n",
       "      <td>4.110000e-34</td>\n",
       "      <td>130.0</td>\n",
       "      <td>7299</td>\n",
       "      <td>9014</td>\n",
       "      <td>188</td>\n",
       "      <td>766</td>\n",
       "      <td>597</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Genome267_k127_433915</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>24.156</td>\n",
       "      <td>20</td>\n",
       "      <td>1.980000e-34</td>\n",
       "      <td>132.0</td>\n",
       "      <td>3178</td>\n",
       "      <td>995</td>\n",
       "      <td>8</td>\n",
       "      <td>741</td>\n",
       "      <td>770</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826_scores...</td>\n",
       "      <td>Bdellovibrionaceae_bacterium_ERS6626826.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    qseqid                                             sseqid  \\\n",
       "0  Genome20_S3Ck127_316360  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...   \n",
       "1  Genome20_S3Ck127_316360        BAB48572_Form2_Mesorhizobium_japonicum_MAFF   \n",
       "2    Genome267_k127_201050        BAB48572_Form2_Mesorhizobium_japonicum_MAFF   \n",
       "3    Genome267_k127_201050  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...   \n",
       "4    Genome267_k127_433915  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...   \n",
       "\n",
       "   pident  qcovs        evalue  bitscore  qstart  qend  sstart  send  length  \\\n",
       "0  24.931     42  8.800000e-31     119.0    4944  2872      52   754     726   \n",
       "1  24.933     41  1.080000e-26     105.0    4953  2911      48   760     742   \n",
       "2  23.846     21  4.900000e-35     134.0    6807  8963       7   767     780   \n",
       "3  24.791     17  4.110000e-34     130.0    7299  9014     188   766     597   \n",
       "4  24.156     20  1.980000e-34     132.0    3178   995       8   741     770   \n",
       "\n",
       "                                         source_file  \\\n",
       "0  Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "1  Bdellovibrionaceae_bacterium_ERS6626579_scores...   \n",
       "2  Bdellovibrionaceae_bacterium_ERS6626826_scores...   \n",
       "3  Bdellovibrionaceae_bacterium_ERS6626826_scores...   \n",
       "4  Bdellovibrionaceae_bacterium_ERS6626826_scores...   \n",
       "\n",
       "                                     target_file  \n",
       "0  Bdellovibrionaceae_bacterium_ERS6626579.fasta  \n",
       "1  Bdellovibrionaceae_bacterium_ERS6626579.fasta  \n",
       "2  Bdellovibrionaceae_bacterium_ERS6626826.fasta  \n",
       "3  Bdellovibrionaceae_bacterium_ERS6626826.fasta  \n",
       "4  Bdellovibrionaceae_bacterium_ERS6626826.fasta  "
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Open the MOTIF 2 concatenated BLAST results\n",
    "coxl2_csv = open_file(motif_2_csv_file_path)\n",
    "coxl2_csv.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "7a0e80bc-480e-448a-8aff-2ae30d5c063e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Genome302_k127_440979</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>50.296</td>\n",
       "      <td>15</td>\n",
       "      <td>4.210000e-43</td>\n",
       "      <td>157.0</td>\n",
       "      <td>509</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>172</td>\n",
       "      <td>169</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Genome302_k127_440979</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>43.195</td>\n",
       "      <td>16</td>\n",
       "      <td>1.120000e-32</td>\n",
       "      <td>124.0</td>\n",
       "      <td>509</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>170</td>\n",
       "      <td>169</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Genome302_k127_117597</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>27.599</td>\n",
       "      <td>28</td>\n",
       "      <td>2.150000e-31</td>\n",
       "      <td>121.0</td>\n",
       "      <td>5027</td>\n",
       "      <td>3483</td>\n",
       "      <td>264</td>\n",
       "      <td>758</td>\n",
       "      <td>529</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Genome302_k127_117597</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>25.149</td>\n",
       "      <td>26</td>\n",
       "      <td>6.280000e-21</td>\n",
       "      <td>87.4</td>\n",
       "      <td>5015</td>\n",
       "      <td>3558</td>\n",
       "      <td>272</td>\n",
       "      <td>759</td>\n",
       "      <td>505</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Genome302_k127_836765</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>64.516</td>\n",
       "      <td>12</td>\n",
       "      <td>3.680000e-71</td>\n",
       "      <td>244.0</td>\n",
       "      <td>5633</td>\n",
       "      <td>4983</td>\n",
       "      <td>567</td>\n",
       "      <td>778</td>\n",
       "      <td>217</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2138</th>\n",
       "      <td>Genome470_k127_338375</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>43.406</td>\n",
       "      <td>53</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>578.0</td>\n",
       "      <td>2342</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>759</td>\n",
       "      <td>781</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>Genome470_k127_380329</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>26.543</td>\n",
       "      <td>3</td>\n",
       "      <td>2.760000e-19</td>\n",
       "      <td>85.5</td>\n",
       "      <td>27726</td>\n",
       "      <td>25918</td>\n",
       "      <td>97</td>\n",
       "      <td>723</td>\n",
       "      <td>648</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>Genome470_k127_380329</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>23.981</td>\n",
       "      <td>3</td>\n",
       "      <td>6.110000e-16</td>\n",
       "      <td>74.7</td>\n",
       "      <td>27669</td>\n",
       "      <td>25918</td>\n",
       "      <td>120</td>\n",
       "      <td>746</td>\n",
       "      <td>638</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2141</th>\n",
       "      <td>Genome470_k127_212027</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>70.000</td>\n",
       "      <td>4</td>\n",
       "      <td>1.200000e-12</td>\n",
       "      <td>59.3</td>\n",
       "      <td>1</td>\n",
       "      <td>120</td>\n",
       "      <td>738</td>\n",
       "      <td>774</td>\n",
       "      <td>40</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>Genome470_k127_212027</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>56.098</td>\n",
       "      <td>4</td>\n",
       "      <td>4.480000e-09</td>\n",
       "      <td>47.8</td>\n",
       "      <td>1</td>\n",
       "      <td>120</td>\n",
       "      <td>715</td>\n",
       "      <td>755</td>\n",
       "      <td>41</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>380 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     qseqid  \\\n",
       "12    Genome302_k127_440979   \n",
       "13    Genome302_k127_440979   \n",
       "14    Genome302_k127_117597   \n",
       "15    Genome302_k127_117597   \n",
       "16    Genome302_k127_836765   \n",
       "...                     ...   \n",
       "2138  Genome470_k127_338375   \n",
       "2139  Genome470_k127_380329   \n",
       "2140  Genome470_k127_380329   \n",
       "2141  Genome470_k127_212027   \n",
       "2142  Genome470_k127_212027   \n",
       "\n",
       "                                                 sseqid  pident  qcovs  \\\n",
       "12          BAB48572_Form2_Mesorhizobium_japonicum_MAFF  50.296     15   \n",
       "13    BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  43.195     16   \n",
       "14    BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  27.599     28   \n",
       "15          BAB48572_Form2_Mesorhizobium_japonicum_MAFF  25.149     26   \n",
       "16          BAB48572_Form2_Mesorhizobium_japonicum_MAFF  64.516     12   \n",
       "...                                                 ...     ...    ...   \n",
       "2138  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  43.406     53   \n",
       "2139  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  26.543      3   \n",
       "2140        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  23.981      3   \n",
       "2141        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  70.000      4   \n",
       "2142  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  56.098      4   \n",
       "\n",
       "            evalue  bitscore  qstart   qend  sstart  send  length  \\\n",
       "12    4.210000e-43     157.0     509      9       6   172     169   \n",
       "13    1.120000e-32     124.0     509      3       8   170     169   \n",
       "14    2.150000e-31     121.0    5027   3483     264   758     529   \n",
       "15    6.280000e-21      87.4    5015   3558     272   759     505   \n",
       "16    3.680000e-71     244.0    5633   4983     567   778     217   \n",
       "...            ...       ...     ...    ...     ...   ...     ...   \n",
       "2138  0.000000e+00     578.0    2342      9       8   759     781   \n",
       "2139  2.760000e-19      85.5   27726  25918      97   723     648   \n",
       "2140  6.110000e-16      74.7   27669  25918     120   746     638   \n",
       "2141  1.200000e-12      59.3       1    120     738   774      40   \n",
       "2142  4.480000e-09      47.8       1    120     715   755      41   \n",
       "\n",
       "                                           source_file  \\\n",
       "12    Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "13    Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "14    Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "15    Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "16    Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "...                                                ...   \n",
       "2138   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "2139   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "2140   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "2141   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "2142   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "\n",
       "                                      target_file  \n",
       "12    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "13    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "14    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "15    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "16    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "...                                           ...  \n",
       "2138   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "2139   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "2140   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "2141   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "2142   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "\n",
       "[380 rows x 13 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter MOTIF form 2 BLAST results with genome files that contain Form 2 MOTIFs\n",
    "coxl2_csv_filtered = coxl2_csv[coxl2_csv['target_file'].isin(coxl2_motif_mag_list)]\n",
    "coxl2_csv_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "d8edda47-4671-4b36-a16d-6f7cf688d232",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Genome302_k127_117597</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>27.599</td>\n",
       "      <td>28</td>\n",
       "      <td>2.150000e-31</td>\n",
       "      <td>121.0</td>\n",
       "      <td>5027</td>\n",
       "      <td>3483</td>\n",
       "      <td>264</td>\n",
       "      <td>758</td>\n",
       "      <td>529</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Genome302_k127_117597</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>25.149</td>\n",
       "      <td>26</td>\n",
       "      <td>6.280000e-21</td>\n",
       "      <td>87.4</td>\n",
       "      <td>5015</td>\n",
       "      <td>3558</td>\n",
       "      <td>272</td>\n",
       "      <td>759</td>\n",
       "      <td>505</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Genome302_k127_440979</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>50.296</td>\n",
       "      <td>15</td>\n",
       "      <td>4.210000e-43</td>\n",
       "      <td>157.0</td>\n",
       "      <td>509</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>172</td>\n",
       "      <td>169</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Genome302_k127_440979</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>43.195</td>\n",
       "      <td>16</td>\n",
       "      <td>1.120000e-32</td>\n",
       "      <td>124.0</td>\n",
       "      <td>509</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>170</td>\n",
       "      <td>169</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Genome302_k127_499603</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>39.371</td>\n",
       "      <td>30</td>\n",
       "      <td>4.730000e-139</td>\n",
       "      <td>444.0</td>\n",
       "      <td>7271</td>\n",
       "      <td>4965</td>\n",
       "      <td>5</td>\n",
       "      <td>780</td>\n",
       "      <td>795</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2138</th>\n",
       "      <td>Genome470_k127_338375</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>43.406</td>\n",
       "      <td>53</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>578.0</td>\n",
       "      <td>2342</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>759</td>\n",
       "      <td>781</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>Genome470_k127_380329</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>26.543</td>\n",
       "      <td>3</td>\n",
       "      <td>2.760000e-19</td>\n",
       "      <td>85.5</td>\n",
       "      <td>27726</td>\n",
       "      <td>25918</td>\n",
       "      <td>97</td>\n",
       "      <td>723</td>\n",
       "      <td>648</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2140</th>\n",
       "      <td>Genome470_k127_380329</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>23.981</td>\n",
       "      <td>3</td>\n",
       "      <td>6.110000e-16</td>\n",
       "      <td>74.7</td>\n",
       "      <td>27669</td>\n",
       "      <td>25918</td>\n",
       "      <td>120</td>\n",
       "      <td>746</td>\n",
       "      <td>638</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2135</th>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>40.759</td>\n",
       "      <td>32</td>\n",
       "      <td>1.710000e-153</td>\n",
       "      <td>485.0</td>\n",
       "      <td>1349</td>\n",
       "      <td>3637</td>\n",
       "      <td>5</td>\n",
       "      <td>778</td>\n",
       "      <td>790</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2136</th>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>40.233</td>\n",
       "      <td>32</td>\n",
       "      <td>1.530000e-138</td>\n",
       "      <td>442.0</td>\n",
       "      <td>1337</td>\n",
       "      <td>3637</td>\n",
       "      <td>3</td>\n",
       "      <td>759</td>\n",
       "      <td>773</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>380 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     qseqid  \\\n",
       "14    Genome302_k127_117597   \n",
       "15    Genome302_k127_117597   \n",
       "12    Genome302_k127_440979   \n",
       "13    Genome302_k127_440979   \n",
       "18    Genome302_k127_499603   \n",
       "...                     ...   \n",
       "2138  Genome470_k127_338375   \n",
       "2139  Genome470_k127_380329   \n",
       "2140  Genome470_k127_380329   \n",
       "2135   Genome470_k127_43488   \n",
       "2136   Genome470_k127_43488   \n",
       "\n",
       "                                                 sseqid  pident  qcovs  \\\n",
       "14    BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  27.599     28   \n",
       "15          BAB48572_Form2_Mesorhizobium_japonicum_MAFF  25.149     26   \n",
       "12          BAB48572_Form2_Mesorhizobium_japonicum_MAFF  50.296     15   \n",
       "13    BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  43.195     16   \n",
       "18          BAB48572_Form2_Mesorhizobium_japonicum_MAFF  39.371     30   \n",
       "...                                                 ...     ...    ...   \n",
       "2138  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  43.406     53   \n",
       "2139  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  26.543      3   \n",
       "2140        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  23.981      3   \n",
       "2135        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  40.759     32   \n",
       "2136  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  40.233     32   \n",
       "\n",
       "             evalue  bitscore  qstart   qend  sstart  send  length  \\\n",
       "14     2.150000e-31     121.0    5027   3483     264   758     529   \n",
       "15     6.280000e-21      87.4    5015   3558     272   759     505   \n",
       "12     4.210000e-43     157.0     509      9       6   172     169   \n",
       "13     1.120000e-32     124.0     509      3       8   170     169   \n",
       "18    4.730000e-139     444.0    7271   4965       5   780     795   \n",
       "...             ...       ...     ...    ...     ...   ...     ...   \n",
       "2138   0.000000e+00     578.0    2342      9       8   759     781   \n",
       "2139   2.760000e-19      85.5   27726  25918      97   723     648   \n",
       "2140   6.110000e-16      74.7   27669  25918     120   746     638   \n",
       "2135  1.710000e-153     485.0    1349   3637       5   778     790   \n",
       "2136  1.530000e-138     442.0    1337   3637       3   759     773   \n",
       "\n",
       "                                           source_file  \\\n",
       "14    Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "15    Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "12    Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "13    Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "18    Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "...                                                ...   \n",
       "2138   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "2139   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "2140   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "2135   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "2136   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "\n",
       "                                      target_file  \n",
       "14    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "15    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "12    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "13    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "18    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "...                                           ...  \n",
       "2138   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "2139   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "2140   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "2135   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "2136   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "\n",
       "[380 rows x 13 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Sort the BLAST results based on genome file name, query sequence id, E-value, Bitscore, & percentage Identity\n",
    "coxl2_csv_sorted = coxl2_csv_filtered.sort_values(by=['target_file', 'qseqid', 'evalue', 'bitscore', 'pident'],\n",
    "                                        ascending=[True, True, True, False, False])\n",
    "\n",
    "coxl2_csv_sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "0c0d5635-28a0-48d3-ae84-a155a43de665",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome302_k127_117597</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>27.599</td>\n",
       "      <td>28</td>\n",
       "      <td>2.150000e-31</td>\n",
       "      <td>121.0</td>\n",
       "      <td>5027</td>\n",
       "      <td>3483</td>\n",
       "      <td>264</td>\n",
       "      <td>758</td>\n",
       "      <td>529</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Genome302_k127_440979</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>50.296</td>\n",
       "      <td>15</td>\n",
       "      <td>4.210000e-43</td>\n",
       "      <td>157.0</td>\n",
       "      <td>509</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>172</td>\n",
       "      <td>169</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Genome302_k127_499603</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>39.371</td>\n",
       "      <td>30</td>\n",
       "      <td>4.730000e-139</td>\n",
       "      <td>444.0</td>\n",
       "      <td>7271</td>\n",
       "      <td>4965</td>\n",
       "      <td>5</td>\n",
       "      <td>780</td>\n",
       "      <td>795</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Genome302_k127_836765</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>64.516</td>\n",
       "      <td>12</td>\n",
       "      <td>3.680000e-71</td>\n",
       "      <td>244.0</td>\n",
       "      <td>5633</td>\n",
       "      <td>4983</td>\n",
       "      <td>567</td>\n",
       "      <td>778</td>\n",
       "      <td>217</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Genome495_k127_217386</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>24.510</td>\n",
       "      <td>6</td>\n",
       "      <td>3.890000e-07</td>\n",
       "      <td>43.5</td>\n",
       "      <td>450</td>\n",
       "      <td>1058</td>\n",
       "      <td>204</td>\n",
       "      <td>398</td>\n",
       "      <td>204</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>Genome470_k127_221644</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>28.889</td>\n",
       "      <td>1</td>\n",
       "      <td>4.050000e-07</td>\n",
       "      <td>45.1</td>\n",
       "      <td>432</td>\n",
       "      <td>40</td>\n",
       "      <td>663</td>\n",
       "      <td>777</td>\n",
       "      <td>135</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>Genome470_k127_290350</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>28.371</td>\n",
       "      <td>8</td>\n",
       "      <td>1.420000e-33</td>\n",
       "      <td>130.0</td>\n",
       "      <td>2379</td>\n",
       "      <td>4031</td>\n",
       "      <td>229</td>\n",
       "      <td>758</td>\n",
       "      <td>571</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>Genome470_k127_338375</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>58.537</td>\n",
       "      <td>53</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>903.0</td>\n",
       "      <td>2342</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>778</td>\n",
       "      <td>779</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>Genome470_k127_380329</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>26.543</td>\n",
       "      <td>3</td>\n",
       "      <td>2.760000e-19</td>\n",
       "      <td>85.5</td>\n",
       "      <td>27726</td>\n",
       "      <td>25918</td>\n",
       "      <td>97</td>\n",
       "      <td>723</td>\n",
       "      <td>648</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>40.759</td>\n",
       "      <td>32</td>\n",
       "      <td>1.710000e-153</td>\n",
       "      <td>485.0</td>\n",
       "      <td>1349</td>\n",
       "      <td>3637</td>\n",
       "      <td>5</td>\n",
       "      <td>778</td>\n",
       "      <td>790</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>189 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    qseqid                                             sseqid  \\\n",
       "0    Genome302_k127_117597  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...   \n",
       "1    Genome302_k127_440979        BAB48572_Form2_Mesorhizobium_japonicum_MAFF   \n",
       "2    Genome302_k127_499603        BAB48572_Form2_Mesorhizobium_japonicum_MAFF   \n",
       "3    Genome302_k127_836765        BAB48572_Form2_Mesorhizobium_japonicum_MAFF   \n",
       "4    Genome495_k127_217386  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...   \n",
       "..                     ...                                                ...   \n",
       "184  Genome470_k127_221644        BAB48572_Form2_Mesorhizobium_japonicum_MAFF   \n",
       "185  Genome470_k127_290350  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...   \n",
       "186  Genome470_k127_338375        BAB48572_Form2_Mesorhizobium_japonicum_MAFF   \n",
       "187  Genome470_k127_380329  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...   \n",
       "188   Genome470_k127_43488        BAB48572_Form2_Mesorhizobium_japonicum_MAFF   \n",
       "\n",
       "     pident  qcovs         evalue  bitscore  qstart   qend  sstart  send  \\\n",
       "0    27.599     28   2.150000e-31     121.0    5027   3483     264   758   \n",
       "1    50.296     15   4.210000e-43     157.0     509      9       6   172   \n",
       "2    39.371     30  4.730000e-139     444.0    7271   4965       5   780   \n",
       "3    64.516     12   3.680000e-71     244.0    5633   4983     567   778   \n",
       "4    24.510      6   3.890000e-07      43.5     450   1058     204   398   \n",
       "..      ...    ...            ...       ...     ...    ...     ...   ...   \n",
       "184  28.889      1   4.050000e-07      45.1     432     40     663   777   \n",
       "185  28.371      8   1.420000e-33     130.0    2379   4031     229   758   \n",
       "186  58.537     53   0.000000e+00     903.0    2342      9       6   778   \n",
       "187  26.543      3   2.760000e-19      85.5   27726  25918      97   723   \n",
       "188  40.759     32  1.710000e-153     485.0    1349   3637       5   778   \n",
       "\n",
       "     length                                       source_file  \\\n",
       "0       529  Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "1       169  Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "2       795  Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "3       217  Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "4       204   uncultured_Acidovorax_sp._ERS6627271_scores.csv   \n",
       "..      ...                                               ...   \n",
       "184     135   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "185     571   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "186     779   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "187     648   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "188     790   uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "\n",
       "                                     target_file  \n",
       "0    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "1    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "2    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "3    Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "4     uncultured_Acidovorax_sp._ERS6627271.fasta  \n",
       "..                                           ...  \n",
       "184   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "185   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "186   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "187   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "188   uncultured_Variovorax_sp._ERS6627246.fasta  \n",
       "\n",
       "[189 rows x 13 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Group the BLAST results by genome file names and query sequence ID, considering 1st row of every group is the best match removing duplicates\n",
    "coxl2_csv_sorted = coxl2_csv_sorted.groupby(['target_file', 'qseqid']).head(1).reset_index(drop=True)\n",
    "coxl2_csv_sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "ed2d08b5-7e5f-4f19-828b-d01b11153c67",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>qcovs</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>length</th>\n",
       "      <th>source_file</th>\n",
       "      <th>target_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome302_k127_117597</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>27.599</td>\n",
       "      <td>28</td>\n",
       "      <td>2.150000e-31</td>\n",
       "      <td>121.0</td>\n",
       "      <td>5027</td>\n",
       "      <td>3483</td>\n",
       "      <td>264</td>\n",
       "      <td>758</td>\n",
       "      <td>529</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861_scores.csv</td>\n",
       "      <td>Burkholderiaceae_bacterium_ERS6626861.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Genome495_k127_317059</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>61.175</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>905.0</td>\n",
       "      <td>28991</td>\n",
       "      <td>31339</td>\n",
       "      <td>6</td>\n",
       "      <td>781</td>\n",
       "      <td>783</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271_scores.csv</td>\n",
       "      <td>uncultured_Acidovorax_sp._ERS6627271.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Genome262_k127_235961</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>27.289</td>\n",
       "      <td>5</td>\n",
       "      <td>1.710000e-33</td>\n",
       "      <td>130.0</td>\n",
       "      <td>6804</td>\n",
       "      <td>8414</td>\n",
       "      <td>243</td>\n",
       "      <td>776</td>\n",
       "      <td>568</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821_scores...</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626821.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Genome299_k127_417201</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>27.152</td>\n",
       "      <td>48</td>\n",
       "      <td>1.100000e-29</td>\n",
       "      <td>114.0</td>\n",
       "      <td>79</td>\n",
       "      <td>1377</td>\n",
       "      <td>348</td>\n",
       "      <td>776</td>\n",
       "      <td>453</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626858_scores...</td>\n",
       "      <td>uncultured_Acinetobacter_sp._ERS6626858.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Genome356_k127_498042</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>56.338</td>\n",
       "      <td>41</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>552.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1491</td>\n",
       "      <td>264</td>\n",
       "      <td>760</td>\n",
       "      <td>497</td>\n",
       "      <td>uncultured_Aureimonas_sp._ERS6626915_scores.csv</td>\n",
       "      <td>uncultured_Aureimonas_sp._ERS6626915.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Genome356_k127_610672</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>24.757</td>\n",
       "      <td>30</td>\n",
       "      <td>6.790000e-14</td>\n",
       "      <td>63.9</td>\n",
       "      <td>2298</td>\n",
       "      <td>3407</td>\n",
       "      <td>6</td>\n",
       "      <td>404</td>\n",
       "      <td>412</td>\n",
       "      <td>uncultured_Aureimonas_sp._ERS6626915_scores.csv</td>\n",
       "      <td>uncultured_Aureimonas_sp._ERS6626915.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Genome530_k127_16854</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>78.590</td>\n",
       "      <td>24</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1218.0</td>\n",
       "      <td>7296</td>\n",
       "      <td>9635</td>\n",
       "      <td>1</td>\n",
       "      <td>780</td>\n",
       "      <td>780</td>\n",
       "      <td>uncultured_Aureimonas_sp._ERS6627306_scores.csv</td>\n",
       "      <td>uncultured_Aureimonas_sp._ERS6627306.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Genome71_S1Ck127_18694</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>59.591</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>882.0</td>\n",
       "      <td>29663</td>\n",
       "      <td>32008</td>\n",
       "      <td>6</td>\n",
       "      <td>780</td>\n",
       "      <td>782</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626630_scores.csv</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626630.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Genome79_S3Ck127_287303</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>24.521</td>\n",
       "      <td>2</td>\n",
       "      <td>4.890000e-26</td>\n",
       "      <td>108.0</td>\n",
       "      <td>67567</td>\n",
       "      <td>65405</td>\n",
       "      <td>19</td>\n",
       "      <td>759</td>\n",
       "      <td>783</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626638_scores.csv</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626638.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Genome255_k127_46261</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>24.036</td>\n",
       "      <td>12</td>\n",
       "      <td>4.940000e-28</td>\n",
       "      <td>112.0</td>\n",
       "      <td>13099</td>\n",
       "      <td>10937</td>\n",
       "      <td>19</td>\n",
       "      <td>759</td>\n",
       "      <td>778</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626814_scores.csv</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6626814.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Genome507_k127_91973</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>60.102</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>882.0</td>\n",
       "      <td>28856</td>\n",
       "      <td>31201</td>\n",
       "      <td>6</td>\n",
       "      <td>780</td>\n",
       "      <td>782</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6627283_scores.csv</td>\n",
       "      <td>uncultured_Comamonas_sp._ERS6627283.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Genome238_k127_17753</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>52.577</td>\n",
       "      <td>32</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>741.0</td>\n",
       "      <td>3607</td>\n",
       "      <td>5919</td>\n",
       "      <td>6</td>\n",
       "      <td>774</td>\n",
       "      <td>776</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626797_scores.csv</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626797.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Genome238_k127_54530</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>25.984</td>\n",
       "      <td>68</td>\n",
       "      <td>3.110000e-30</td>\n",
       "      <td>116.0</td>\n",
       "      <td>124</td>\n",
       "      <td>2256</td>\n",
       "      <td>3</td>\n",
       "      <td>741</td>\n",
       "      <td>762</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626797_scores.csv</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626797.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Genome261_k127_146972</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>52.577</td>\n",
       "      <td>36</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>742.0</td>\n",
       "      <td>2595</td>\n",
       "      <td>4907</td>\n",
       "      <td>6</td>\n",
       "      <td>774</td>\n",
       "      <td>776</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626820_scores.csv</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626820.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Genome261_k127_332399</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>27.515</td>\n",
       "      <td>53</td>\n",
       "      <td>4.100000e-07</td>\n",
       "      <td>40.4</td>\n",
       "      <td>800</td>\n",
       "      <td>1750</td>\n",
       "      <td>3</td>\n",
       "      <td>327</td>\n",
       "      <td>338</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626820_scores.csv</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626820.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Genome278_k127_93298</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>52.308</td>\n",
       "      <td>25</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>743.0</td>\n",
       "      <td>5439</td>\n",
       "      <td>7763</td>\n",
       "      <td>6</td>\n",
       "      <td>778</td>\n",
       "      <td>780</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626837_scores.csv</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6626837.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Genome517_k127_348199</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>52.448</td>\n",
       "      <td>14</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>740.0</td>\n",
       "      <td>9389</td>\n",
       "      <td>11701</td>\n",
       "      <td>6</td>\n",
       "      <td>774</td>\n",
       "      <td>776</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6627293_scores.csv</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6627293.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Genome559_k127_243715</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>25.624</td>\n",
       "      <td>11</td>\n",
       "      <td>6.900000e-30</td>\n",
       "      <td>118.0</td>\n",
       "      <td>1353</td>\n",
       "      <td>3485</td>\n",
       "      <td>3</td>\n",
       "      <td>741</td>\n",
       "      <td>761</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6627335_scores.csv</td>\n",
       "      <td>uncultured_Deinococcus_sp._ERS6627335.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Genome174_S1Ck127_276112</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>25.581</td>\n",
       "      <td>7</td>\n",
       "      <td>1.190000e-33</td>\n",
       "      <td>131.0</td>\n",
       "      <td>13377</td>\n",
       "      <td>15542</td>\n",
       "      <td>6</td>\n",
       "      <td>741</td>\n",
       "      <td>774</td>\n",
       "      <td>uncultured_Hymenobacter_sp._ERS6626733_scores.csv</td>\n",
       "      <td>uncultured_Hymenobacter_sp._ERS6626733.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Genome503_k127_453582</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>25.032</td>\n",
       "      <td>4</td>\n",
       "      <td>5.040000e-33</td>\n",
       "      <td>130.0</td>\n",
       "      <td>35476</td>\n",
       "      <td>37641</td>\n",
       "      <td>6</td>\n",
       "      <td>741</td>\n",
       "      <td>775</td>\n",
       "      <td>uncultured_Hymenobacter_sp._ERS6627279_scores.csv</td>\n",
       "      <td>uncultured_Hymenobacter_sp._ERS6627279.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Genome512_k127_160927</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>56.373</td>\n",
       "      <td>9</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>686.0</td>\n",
       "      <td>12460</td>\n",
       "      <td>14736</td>\n",
       "      <td>1</td>\n",
       "      <td>755</td>\n",
       "      <td>761</td>\n",
       "      <td>uncultured_Methylobacterium_sp._ERS6627288_sco...</td>\n",
       "      <td>uncultured_Methylobacterium_sp._ERS6627288.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Genome512_k127_80099</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>42.692</td>\n",
       "      <td>23</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>566.0</td>\n",
       "      <td>319</td>\n",
       "      <td>2607</td>\n",
       "      <td>5</td>\n",
       "      <td>778</td>\n",
       "      <td>780</td>\n",
       "      <td>uncultured_Methylobacterium_sp._ERS6627288_sco...</td>\n",
       "      <td>uncultured_Methylobacterium_sp._ERS6627288.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Genome347_k127_84820</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>57.184</td>\n",
       "      <td>99</td>\n",
       "      <td>1.120000e-95</td>\n",
       "      <td>290.0</td>\n",
       "      <td>3</td>\n",
       "      <td>1043</td>\n",
       "      <td>30</td>\n",
       "      <td>373</td>\n",
       "      <td>348</td>\n",
       "      <td>uncultured_Microbacterium_sp._ERS6626906_score...</td>\n",
       "      <td>uncultured_Microbacterium_sp._ERS6626906.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Genome405_k127_107645</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>24.741</td>\n",
       "      <td>20</td>\n",
       "      <td>1.070000e-29</td>\n",
       "      <td>117.0</td>\n",
       "      <td>1951</td>\n",
       "      <td>4140</td>\n",
       "      <td>42</td>\n",
       "      <td>774</td>\n",
       "      <td>772</td>\n",
       "      <td>uncultured_Microbacterium_sp._ERS6626964_score...</td>\n",
       "      <td>uncultured_Microbacterium_sp._ERS6626964.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Genome314_k127_168730</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>24.936</td>\n",
       "      <td>49</td>\n",
       "      <td>1.160000e-35</td>\n",
       "      <td>134.0</td>\n",
       "      <td>90</td>\n",
       "      <td>2192</td>\n",
       "      <td>10</td>\n",
       "      <td>770</td>\n",
       "      <td>778</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626873_sco...</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626873.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Genome364_k127_154890</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>23.941</td>\n",
       "      <td>50</td>\n",
       "      <td>5.330000e-26</td>\n",
       "      <td>102.0</td>\n",
       "      <td>1461</td>\n",
       "      <td>2924</td>\n",
       "      <td>10</td>\n",
       "      <td>535</td>\n",
       "      <td>543</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626923_sco...</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626923.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Genome403_k127_249621</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>22.639</td>\n",
       "      <td>35</td>\n",
       "      <td>6.040000e-24</td>\n",
       "      <td>97.1</td>\n",
       "      <td>3318</td>\n",
       "      <td>5084</td>\n",
       "      <td>10</td>\n",
       "      <td>660</td>\n",
       "      <td>667</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626962_sco...</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626962.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Genome413_k127_255610</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>24.649</td>\n",
       "      <td>31</td>\n",
       "      <td>2.200000e-35</td>\n",
       "      <td>134.0</td>\n",
       "      <td>766</td>\n",
       "      <td>2868</td>\n",
       "      <td>10</td>\n",
       "      <td>770</td>\n",
       "      <td>783</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626972_sco...</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626972.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Genome422_k127_88665</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>24.904</td>\n",
       "      <td>37</td>\n",
       "      <td>1.720000e-34</td>\n",
       "      <td>131.0</td>\n",
       "      <td>2567</td>\n",
       "      <td>4669</td>\n",
       "      <td>10</td>\n",
       "      <td>770</td>\n",
       "      <td>783</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626981_sco...</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6626981.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Genome499_k127_471593</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>25.032</td>\n",
       "      <td>12</td>\n",
       "      <td>8.510000e-34</td>\n",
       "      <td>130.0</td>\n",
       "      <td>9895</td>\n",
       "      <td>11997</td>\n",
       "      <td>10</td>\n",
       "      <td>770</td>\n",
       "      <td>783</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6627275_sco...</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6627275.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Genome514_k127_4285</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>24.519</td>\n",
       "      <td>8</td>\n",
       "      <td>4.750000e-33</td>\n",
       "      <td>129.0</td>\n",
       "      <td>12070</td>\n",
       "      <td>14172</td>\n",
       "      <td>10</td>\n",
       "      <td>770</td>\n",
       "      <td>779</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6627290_sco...</td>\n",
       "      <td>uncultured_Mucilaginibacter_sp._ERS6627290.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Genome84_S2Ck127_371734</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>25.182</td>\n",
       "      <td>1</td>\n",
       "      <td>1.250000e-40</td>\n",
       "      <td>157.0</td>\n",
       "      <td>67684</td>\n",
       "      <td>65420</td>\n",
       "      <td>5</td>\n",
       "      <td>778</td>\n",
       "      <td>822</td>\n",
       "      <td>uncultured_Pantoea_sp._ERS6626643_scores.csv</td>\n",
       "      <td>uncultured_Pantoea_sp._ERS6626643.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Genome446_k127_205538</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>26.224</td>\n",
       "      <td>39</td>\n",
       "      <td>2.420000e-32</td>\n",
       "      <td>123.0</td>\n",
       "      <td>4241</td>\n",
       "      <td>2586</td>\n",
       "      <td>229</td>\n",
       "      <td>759</td>\n",
       "      <td>572</td>\n",
       "      <td>uncultured_Pantoea_sp._ERS6627005_scores.csv</td>\n",
       "      <td>uncultured_Pantoea_sp._ERS6627005.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Genome10_S2Ck127_387977</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>26.786</td>\n",
       "      <td>18</td>\n",
       "      <td>7.870000e-34</td>\n",
       "      <td>130.0</td>\n",
       "      <td>8881</td>\n",
       "      <td>6842</td>\n",
       "      <td>21</td>\n",
       "      <td>701</td>\n",
       "      <td>728</td>\n",
       "      <td>uncultured_Pelomonas_sp._ERS6626569_scores.csv</td>\n",
       "      <td>uncultured_Pelomonas_sp._ERS6626569.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Genome18_S1Ck127_217968</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>58.151</td>\n",
       "      <td>12</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>842.0</td>\n",
       "      <td>9592</td>\n",
       "      <td>11922</td>\n",
       "      <td>6</td>\n",
       "      <td>778</td>\n",
       "      <td>779</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6626577_scor...</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6626577.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Genome291_k127_419550</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>58.280</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>841.0</td>\n",
       "      <td>11300</td>\n",
       "      <td>13630</td>\n",
       "      <td>6</td>\n",
       "      <td>778</td>\n",
       "      <td>779</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6626850_scor...</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6626850.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Genome350_k127_213661</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>39.570</td>\n",
       "      <td>42</td>\n",
       "      <td>2.300000e-149</td>\n",
       "      <td>469.0</td>\n",
       "      <td>2088</td>\n",
       "      <td>4379</td>\n",
       "      <td>5</td>\n",
       "      <td>780</td>\n",
       "      <td>791</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6626909_scor...</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6626909.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Genome444_k127_25498</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>39.922</td>\n",
       "      <td>41</td>\n",
       "      <td>1.110000e-106</td>\n",
       "      <td>343.0</td>\n",
       "      <td>3</td>\n",
       "      <td>1472</td>\n",
       "      <td>280</td>\n",
       "      <td>780</td>\n",
       "      <td>516</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627003_scor...</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627003.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Genome444_k127_2949</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>58.006</td>\n",
       "      <td>16</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>758.0</td>\n",
       "      <td>11306</td>\n",
       "      <td>13435</td>\n",
       "      <td>6</td>\n",
       "      <td>714</td>\n",
       "      <td>712</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627003_scor...</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627003.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Genome544_k127_833201</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>58.280</td>\n",
       "      <td>9</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>839.0</td>\n",
       "      <td>18213</td>\n",
       "      <td>20543</td>\n",
       "      <td>6</td>\n",
       "      <td>778</td>\n",
       "      <td>779</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627320_scor...</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627320.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Genome544_k127_857444</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>26.302</td>\n",
       "      <td>1</td>\n",
       "      <td>1.970000e-40</td>\n",
       "      <td>155.0</td>\n",
       "      <td>66280</td>\n",
       "      <td>68442</td>\n",
       "      <td>9</td>\n",
       "      <td>753</td>\n",
       "      <td>768</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627320_scor...</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627320.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Genome544_k127_977136</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>40.161</td>\n",
       "      <td>3</td>\n",
       "      <td>4.140000e-137</td>\n",
       "      <td>442.0</td>\n",
       "      <td>17983</td>\n",
       "      <td>20148</td>\n",
       "      <td>5</td>\n",
       "      <td>745</td>\n",
       "      <td>747</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627320_scor...</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627320.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Genome545_k127_208535</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>39.849</td>\n",
       "      <td>4</td>\n",
       "      <td>2.320000e-156</td>\n",
       "      <td>497.0</td>\n",
       "      <td>19488</td>\n",
       "      <td>21779</td>\n",
       "      <td>5</td>\n",
       "      <td>780</td>\n",
       "      <td>793</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627321_scor...</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627321.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Genome545_k127_344480</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>57.841</td>\n",
       "      <td>13</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>835.0</td>\n",
       "      <td>7171</td>\n",
       "      <td>9501</td>\n",
       "      <td>6</td>\n",
       "      <td>778</td>\n",
       "      <td>778</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627321_scor...</td>\n",
       "      <td>uncultured_Pseudacidovorax_sp._ERS6627321.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Genome114_S3Ck127_13141</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>26.904</td>\n",
       "      <td>0</td>\n",
       "      <td>8.510000e-38</td>\n",
       "      <td>148.0</td>\n",
       "      <td>380855</td>\n",
       "      <td>378579</td>\n",
       "      <td>7</td>\n",
       "      <td>763</td>\n",
       "      <td>788</td>\n",
       "      <td>uncultured_Pseudomonas_sp._ERS6626673_scores.csv</td>\n",
       "      <td>uncultured_Pseudomonas_sp._ERS6626673.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Genome497_k127_28836</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>26.777</td>\n",
       "      <td>4</td>\n",
       "      <td>1.830000e-38</td>\n",
       "      <td>147.0</td>\n",
       "      <td>29714</td>\n",
       "      <td>27438</td>\n",
       "      <td>7</td>\n",
       "      <td>763</td>\n",
       "      <td>788</td>\n",
       "      <td>uncultured_Pseudomonas_sp._ERS6627273_scores.csv</td>\n",
       "      <td>uncultured_Pseudomonas_sp._ERS6627273.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Genome382_k127_146993</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>41.138</td>\n",
       "      <td>28</td>\n",
       "      <td>9.520000e-141</td>\n",
       "      <td>449.0</td>\n",
       "      <td>3454</td>\n",
       "      <td>5754</td>\n",
       "      <td>2</td>\n",
       "      <td>755</td>\n",
       "      <td>773</td>\n",
       "      <td>uncultured_Pseudorhodoferax_sp._ERS6626941_sco...</td>\n",
       "      <td>uncultured_Pseudorhodoferax_sp._ERS6626941.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Genome121_S1Ck127_116016</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>23.934</td>\n",
       "      <td>1</td>\n",
       "      <td>5.580000e-13</td>\n",
       "      <td>67.0</td>\n",
       "      <td>212177</td>\n",
       "      <td>210096</td>\n",
       "      <td>8</td>\n",
       "      <td>712</td>\n",
       "      <td>727</td>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6626680_scores.csv</td>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6626680.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Genome272_k127_58089</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>27.503</td>\n",
       "      <td>5</td>\n",
       "      <td>1.420000e-30</td>\n",
       "      <td>122.0</td>\n",
       "      <td>8274</td>\n",
       "      <td>6172</td>\n",
       "      <td>8</td>\n",
       "      <td>750</td>\n",
       "      <td>749</td>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6626831_scores.csv</td>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6626831.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>Genome398_k127_96590</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>25.033</td>\n",
       "      <td>19</td>\n",
       "      <td>2.690000e-32</td>\n",
       "      <td>125.0</td>\n",
       "      <td>2042</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>746</td>\n",
       "      <td>755</td>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6626957_scores.csv</td>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6626957.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Genome482_k127_81292</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>26.861</td>\n",
       "      <td>23</td>\n",
       "      <td>3.460000e-25</td>\n",
       "      <td>102.0</td>\n",
       "      <td>6101</td>\n",
       "      <td>7999</td>\n",
       "      <td>52</td>\n",
       "      <td>718</td>\n",
       "      <td>685</td>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6627258_scores.csv</td>\n",
       "      <td>uncultured_Sphingomonas_sp._ERS6627258.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>Genome452_k127_363663</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>28.030</td>\n",
       "      <td>13</td>\n",
       "      <td>2.260000e-09</td>\n",
       "      <td>48.5</td>\n",
       "      <td>638</td>\n",
       "      <td>264</td>\n",
       "      <td>6</td>\n",
       "      <td>136</td>\n",
       "      <td>132</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6627011_scores.csv</td>\n",
       "      <td>uncultured_Spirosoma_sp._ERS6627011.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Genome129_S3Ck127_15574</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>27.131</td>\n",
       "      <td>1</td>\n",
       "      <td>1.320000e-26</td>\n",
       "      <td>112.0</td>\n",
       "      <td>91848</td>\n",
       "      <td>89911</td>\n",
       "      <td>51</td>\n",
       "      <td>735</td>\n",
       "      <td>704</td>\n",
       "      <td>uncultured_Stenotrophomonas_sp._ERS6626688_sco...</td>\n",
       "      <td>uncultured_Stenotrophomonas_sp._ERS6626688.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Genome543_k127_120424</td>\n",
       "      <td>BAB48572_Form2_Rhodopseudomonas_palustris_BisA...</td>\n",
       "      <td>27.557</td>\n",
       "      <td>1</td>\n",
       "      <td>3.370000e-28</td>\n",
       "      <td>115.0</td>\n",
       "      <td>19902</td>\n",
       "      <td>17965</td>\n",
       "      <td>51</td>\n",
       "      <td>735</td>\n",
       "      <td>704</td>\n",
       "      <td>uncultured_Stenotrophomonas_sp._ERS6627319_sco...</td>\n",
       "      <td>uncultured_Stenotrophomonas_sp._ERS6627319.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>Genome220_k127_244806</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>40.759</td>\n",
       "      <td>12</td>\n",
       "      <td>1.100000e-153</td>\n",
       "      <td>488.0</td>\n",
       "      <td>11569</td>\n",
       "      <td>13857</td>\n",
       "      <td>5</td>\n",
       "      <td>778</td>\n",
       "      <td>790</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6626779.fasta</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>Genome470_k127_43488</td>\n",
       "      <td>BAB48572_Form2_Mesorhizobium_japonicum_MAFF</td>\n",
       "      <td>40.759</td>\n",
       "      <td>32</td>\n",
       "      <td>1.710000e-153</td>\n",
       "      <td>485.0</td>\n",
       "      <td>1349</td>\n",
       "      <td>3637</td>\n",
       "      <td>5</td>\n",
       "      <td>778</td>\n",
       "      <td>790</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246_scores.csv</td>\n",
       "      <td>uncultured_Variovorax_sp._ERS6627246.fasta</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      qseqid  \\\n",
       "0      Genome302_k127_117597   \n",
       "1      Genome495_k127_317059   \n",
       "2      Genome262_k127_235961   \n",
       "3      Genome299_k127_417201   \n",
       "4      Genome356_k127_498042   \n",
       "5      Genome356_k127_610672   \n",
       "6       Genome530_k127_16854   \n",
       "7     Genome71_S1Ck127_18694   \n",
       "8    Genome79_S3Ck127_287303   \n",
       "9       Genome255_k127_46261   \n",
       "10      Genome507_k127_91973   \n",
       "11      Genome238_k127_17753   \n",
       "12      Genome238_k127_54530   \n",
       "13     Genome261_k127_146972   \n",
       "14     Genome261_k127_332399   \n",
       "15      Genome278_k127_93298   \n",
       "16     Genome517_k127_348199   \n",
       "17     Genome559_k127_243715   \n",
       "18  Genome174_S1Ck127_276112   \n",
       "19     Genome503_k127_453582   \n",
       "20     Genome512_k127_160927   \n",
       "21      Genome512_k127_80099   \n",
       "22      Genome347_k127_84820   \n",
       "23     Genome405_k127_107645   \n",
       "24     Genome314_k127_168730   \n",
       "25     Genome364_k127_154890   \n",
       "26     Genome403_k127_249621   \n",
       "27     Genome413_k127_255610   \n",
       "28      Genome422_k127_88665   \n",
       "29     Genome499_k127_471593   \n",
       "30       Genome514_k127_4285   \n",
       "31   Genome84_S2Ck127_371734   \n",
       "32     Genome446_k127_205538   \n",
       "33   Genome10_S2Ck127_387977   \n",
       "34   Genome18_S1Ck127_217968   \n",
       "35     Genome291_k127_419550   \n",
       "36     Genome350_k127_213661   \n",
       "37      Genome444_k127_25498   \n",
       "38       Genome444_k127_2949   \n",
       "39     Genome544_k127_833201   \n",
       "40     Genome544_k127_857444   \n",
       "41     Genome544_k127_977136   \n",
       "42     Genome545_k127_208535   \n",
       "43     Genome545_k127_344480   \n",
       "44   Genome114_S3Ck127_13141   \n",
       "45      Genome497_k127_28836   \n",
       "46     Genome382_k127_146993   \n",
       "47  Genome121_S1Ck127_116016   \n",
       "48      Genome272_k127_58089   \n",
       "49      Genome398_k127_96590   \n",
       "50      Genome482_k127_81292   \n",
       "51     Genome452_k127_363663   \n",
       "52   Genome129_S3Ck127_15574   \n",
       "53     Genome543_k127_120424   \n",
       "54     Genome220_k127_244806   \n",
       "55      Genome470_k127_43488   \n",
       "\n",
       "                                               sseqid  pident  qcovs  \\\n",
       "0   BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  27.599     28   \n",
       "1         BAB48572_Form2_Mesorhizobium_japonicum_MAFF  61.175      7   \n",
       "2         BAB48572_Form2_Mesorhizobium_japonicum_MAFF  27.289      5   \n",
       "3         BAB48572_Form2_Mesorhizobium_japonicum_MAFF  27.152     48   \n",
       "4   BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  56.338     41   \n",
       "5         BAB48572_Form2_Mesorhizobium_japonicum_MAFF  24.757     30   \n",
       "6         BAB48572_Form2_Mesorhizobium_japonicum_MAFF  78.590     24   \n",
       "7         BAB48572_Form2_Mesorhizobium_japonicum_MAFF  59.591      5   \n",
       "8         BAB48572_Form2_Mesorhizobium_japonicum_MAFF  24.521      2   \n",
       "9         BAB48572_Form2_Mesorhizobium_japonicum_MAFF  24.036     12   \n",
       "10        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  60.102      5   \n",
       "11        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  52.577     32   \n",
       "12        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  25.984     68   \n",
       "13        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  52.577     36   \n",
       "14        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  27.515     53   \n",
       "15        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  52.308     25   \n",
       "16        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  52.448     14   \n",
       "17        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  25.624     11   \n",
       "18        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  25.581      7   \n",
       "19        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  25.032      4   \n",
       "20  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  56.373      9   \n",
       "21        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  42.692     23   \n",
       "22  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  57.184     99   \n",
       "23        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  24.741     20   \n",
       "24        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  24.936     49   \n",
       "25        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  23.941     50   \n",
       "26        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  22.639     35   \n",
       "27        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  24.649     31   \n",
       "28        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  24.904     37   \n",
       "29        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  25.032     12   \n",
       "30        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  24.519      8   \n",
       "31        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  25.182      1   \n",
       "32  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  26.224     39   \n",
       "33  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  26.786     18   \n",
       "34        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  58.151     12   \n",
       "35        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  58.280      2   \n",
       "36        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  39.570     42   \n",
       "37        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  39.922     41   \n",
       "38        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  58.006     16   \n",
       "39        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  58.280      9   \n",
       "40  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  26.302      1   \n",
       "41        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  40.161      3   \n",
       "42        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  39.849      4   \n",
       "43        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  57.841     13   \n",
       "44  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  26.904      0   \n",
       "45  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  26.777      4   \n",
       "46  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  41.138     28   \n",
       "47  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  23.934      1   \n",
       "48  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  27.503      5   \n",
       "49        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  25.033     19   \n",
       "50  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  26.861     23   \n",
       "51        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  28.030     13   \n",
       "52  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  27.131      1   \n",
       "53  BAB48572_Form2_Rhodopseudomonas_palustris_BisA...  27.557      1   \n",
       "54        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  40.759     12   \n",
       "55        BAB48572_Form2_Mesorhizobium_japonicum_MAFF  40.759     32   \n",
       "\n",
       "           evalue  bitscore  qstart    qend  sstart  send  length  \\\n",
       "0    2.150000e-31     121.0    5027    3483     264   758     529   \n",
       "1    0.000000e+00     905.0   28991   31339       6   781     783   \n",
       "2    1.710000e-33     130.0    6804    8414     243   776     568   \n",
       "3    1.100000e-29     114.0      79    1377     348   776     453   \n",
       "4    0.000000e+00     552.0       1    1491     264   760     497   \n",
       "5    6.790000e-14      63.9    2298    3407       6   404     412   \n",
       "6    0.000000e+00    1218.0    7296    9635       1   780     780   \n",
       "7    0.000000e+00     882.0   29663   32008       6   780     782   \n",
       "8    4.890000e-26     108.0   67567   65405      19   759     783   \n",
       "9    4.940000e-28     112.0   13099   10937      19   759     778   \n",
       "10   0.000000e+00     882.0   28856   31201       6   780     782   \n",
       "11   0.000000e+00     741.0    3607    5919       6   774     776   \n",
       "12   3.110000e-30     116.0     124    2256       3   741     762   \n",
       "13   0.000000e+00     742.0    2595    4907       6   774     776   \n",
       "14   4.100000e-07      40.4     800    1750       3   327     338   \n",
       "15   0.000000e+00     743.0    5439    7763       6   778     780   \n",
       "16   0.000000e+00     740.0    9389   11701       6   774     776   \n",
       "17   6.900000e-30     118.0    1353    3485       3   741     761   \n",
       "18   1.190000e-33     131.0   13377   15542       6   741     774   \n",
       "19   5.040000e-33     130.0   35476   37641       6   741     775   \n",
       "20   0.000000e+00     686.0   12460   14736       1   755     761   \n",
       "21   0.000000e+00     566.0     319    2607       5   778     780   \n",
       "22   1.120000e-95     290.0       3    1043      30   373     348   \n",
       "23   1.070000e-29     117.0    1951    4140      42   774     772   \n",
       "24   1.160000e-35     134.0      90    2192      10   770     778   \n",
       "25   5.330000e-26     102.0    1461    2924      10   535     543   \n",
       "26   6.040000e-24      97.1    3318    5084      10   660     667   \n",
       "27   2.200000e-35     134.0     766    2868      10   770     783   \n",
       "28   1.720000e-34     131.0    2567    4669      10   770     783   \n",
       "29   8.510000e-34     130.0    9895   11997      10   770     783   \n",
       "30   4.750000e-33     129.0   12070   14172      10   770     779   \n",
       "31   1.250000e-40     157.0   67684   65420       5   778     822   \n",
       "32   2.420000e-32     123.0    4241    2586     229   759     572   \n",
       "33   7.870000e-34     130.0    8881    6842      21   701     728   \n",
       "34   0.000000e+00     842.0    9592   11922       6   778     779   \n",
       "35   0.000000e+00     841.0   11300   13630       6   778     779   \n",
       "36  2.300000e-149     469.0    2088    4379       5   780     791   \n",
       "37  1.110000e-106     343.0       3    1472     280   780     516   \n",
       "38   0.000000e+00     758.0   11306   13435       6   714     712   \n",
       "39   0.000000e+00     839.0   18213   20543       6   778     779   \n",
       "40   1.970000e-40     155.0   66280   68442       9   753     768   \n",
       "41  4.140000e-137     442.0   17983   20148       5   745     747   \n",
       "42  2.320000e-156     497.0   19488   21779       5   780     793   \n",
       "43   0.000000e+00     835.0    7171    9501       6   778     778   \n",
       "44   8.510000e-38     148.0  380855  378579       7   763     788   \n",
       "45   1.830000e-38     147.0   29714   27438       7   763     788   \n",
       "46  9.520000e-141     449.0    3454    5754       2   755     773   \n",
       "47   5.580000e-13      67.0  212177  210096       8   712     727   \n",
       "48   1.420000e-30     122.0    8274    6172       8   750     749   \n",
       "49   2.690000e-32     125.0    2042       3       6   746     755   \n",
       "50   3.460000e-25     102.0    6101    7999      52   718     685   \n",
       "51   2.260000e-09      48.5     638     264       6   136     132   \n",
       "52   1.320000e-26     112.0   91848   89911      51   735     704   \n",
       "53   3.370000e-28     115.0   19902   17965      51   735     704   \n",
       "54  1.100000e-153     488.0   11569   13857       5   778     790   \n",
       "55  1.710000e-153     485.0    1349    3637       5   778     790   \n",
       "\n",
       "                                          source_file  \\\n",
       "0    Burkholderiaceae_bacterium_ERS6626861_scores.csv   \n",
       "1     uncultured_Acidovorax_sp._ERS6627271_scores.csv   \n",
       "2   uncultured_Acinetobacter_sp._ERS6626821_scores...   \n",
       "3   uncultured_Acinetobacter_sp._ERS6626858_scores...   \n",
       "4     uncultured_Aureimonas_sp._ERS6626915_scores.csv   \n",
       "5     uncultured_Aureimonas_sp._ERS6626915_scores.csv   \n",
       "6     uncultured_Aureimonas_sp._ERS6627306_scores.csv   \n",
       "7      uncultured_Comamonas_sp._ERS6626630_scores.csv   \n",
       "8      uncultured_Comamonas_sp._ERS6626638_scores.csv   \n",
       "9      uncultured_Comamonas_sp._ERS6626814_scores.csv   \n",
       "10     uncultured_Comamonas_sp._ERS6627283_scores.csv   \n",
       "11   uncultured_Deinococcus_sp._ERS6626797_scores.csv   \n",
       "12   uncultured_Deinococcus_sp._ERS6626797_scores.csv   \n",
       "13   uncultured_Deinococcus_sp._ERS6626820_scores.csv   \n",
       "14   uncultured_Deinococcus_sp._ERS6626820_scores.csv   \n",
       "15   uncultured_Deinococcus_sp._ERS6626837_scores.csv   \n",
       "16   uncultured_Deinococcus_sp._ERS6627293_scores.csv   \n",
       "17   uncultured_Deinococcus_sp._ERS6627335_scores.csv   \n",
       "18  uncultured_Hymenobacter_sp._ERS6626733_scores.csv   \n",
       "19  uncultured_Hymenobacter_sp._ERS6627279_scores.csv   \n",
       "20  uncultured_Methylobacterium_sp._ERS6627288_sco...   \n",
       "21  uncultured_Methylobacterium_sp._ERS6627288_sco...   \n",
       "22  uncultured_Microbacterium_sp._ERS6626906_score...   \n",
       "23  uncultured_Microbacterium_sp._ERS6626964_score...   \n",
       "24  uncultured_Mucilaginibacter_sp._ERS6626873_sco...   \n",
       "25  uncultured_Mucilaginibacter_sp._ERS6626923_sco...   \n",
       "26  uncultured_Mucilaginibacter_sp._ERS6626962_sco...   \n",
       "27  uncultured_Mucilaginibacter_sp._ERS6626972_sco...   \n",
       "28  uncultured_Mucilaginibacter_sp._ERS6626981_sco...   \n",
       "29  uncultured_Mucilaginibacter_sp._ERS6627275_sco...   \n",
       "30  uncultured_Mucilaginibacter_sp._ERS6627290_sco...   \n",
       "31       uncultured_Pantoea_sp._ERS6626643_scores.csv   \n",
       "32       uncultured_Pantoea_sp._ERS6627005_scores.csv   \n",
       "33     uncultured_Pelomonas_sp._ERS6626569_scores.csv   \n",
       "34  uncultured_Pseudacidovorax_sp._ERS6626577_scor...   \n",
       "35  uncultured_Pseudacidovorax_sp._ERS6626850_scor...   \n",
       "36  uncultured_Pseudacidovorax_sp._ERS6626909_scor...   \n",
       "37  uncultured_Pseudacidovorax_sp._ERS6627003_scor...   \n",
       "38  uncultured_Pseudacidovorax_sp._ERS6627003_scor...   \n",
       "39  uncultured_Pseudacidovorax_sp._ERS6627320_scor...   \n",
       "40  uncultured_Pseudacidovorax_sp._ERS6627320_scor...   \n",
       "41  uncultured_Pseudacidovorax_sp._ERS6627320_scor...   \n",
       "42  uncultured_Pseudacidovorax_sp._ERS6627321_scor...   \n",
       "43  uncultured_Pseudacidovorax_sp._ERS6627321_scor...   \n",
       "44   uncultured_Pseudomonas_sp._ERS6626673_scores.csv   \n",
       "45   uncultured_Pseudomonas_sp._ERS6627273_scores.csv   \n",
       "46  uncultured_Pseudorhodoferax_sp._ERS6626941_sco...   \n",
       "47  uncultured_Sphingomonas_sp._ERS6626680_scores.csv   \n",
       "48  uncultured_Sphingomonas_sp._ERS6626831_scores.csv   \n",
       "49  uncultured_Sphingomonas_sp._ERS6626957_scores.csv   \n",
       "50  uncultured_Sphingomonas_sp._ERS6627258_scores.csv   \n",
       "51     uncultured_Spirosoma_sp._ERS6627011_scores.csv   \n",
       "52  uncultured_Stenotrophomonas_sp._ERS6626688_sco...   \n",
       "53  uncultured_Stenotrophomonas_sp._ERS6627319_sco...   \n",
       "54    uncultured_Variovorax_sp._ERS6626779_scores.csv   \n",
       "55    uncultured_Variovorax_sp._ERS6627246_scores.csv   \n",
       "\n",
       "                                         target_file  \n",
       "0        Burkholderiaceae_bacterium_ERS6626861.fasta  \n",
       "1         uncultured_Acidovorax_sp._ERS6627271.fasta  \n",
       "2      uncultured_Acinetobacter_sp._ERS6626821.fasta  \n",
       "3      uncultured_Acinetobacter_sp._ERS6626858.fasta  \n",
       "4         uncultured_Aureimonas_sp._ERS6626915.fasta  \n",
       "5         uncultured_Aureimonas_sp._ERS6626915.fasta  \n",
       "6         uncultured_Aureimonas_sp._ERS6627306.fasta  \n",
       "7          uncultured_Comamonas_sp._ERS6626630.fasta  \n",
       "8          uncultured_Comamonas_sp._ERS6626638.fasta  \n",
       "9          uncultured_Comamonas_sp._ERS6626814.fasta  \n",
       "10         uncultured_Comamonas_sp._ERS6627283.fasta  \n",
       "11       uncultured_Deinococcus_sp._ERS6626797.fasta  \n",
       "12       uncultured_Deinococcus_sp._ERS6626797.fasta  \n",
       "13       uncultured_Deinococcus_sp._ERS6626820.fasta  \n",
       "14       uncultured_Deinococcus_sp._ERS6626820.fasta  \n",
       "15       uncultured_Deinococcus_sp._ERS6626837.fasta  \n",
       "16       uncultured_Deinococcus_sp._ERS6627293.fasta  \n",
       "17       uncultured_Deinococcus_sp._ERS6627335.fasta  \n",
       "18      uncultured_Hymenobacter_sp._ERS6626733.fasta  \n",
       "19      uncultured_Hymenobacter_sp._ERS6627279.fasta  \n",
       "20  uncultured_Methylobacterium_sp._ERS6627288.fasta  \n",
       "21  uncultured_Methylobacterium_sp._ERS6627288.fasta  \n",
       "22    uncultured_Microbacterium_sp._ERS6626906.fasta  \n",
       "23    uncultured_Microbacterium_sp._ERS6626964.fasta  \n",
       "24  uncultured_Mucilaginibacter_sp._ERS6626873.fasta  \n",
       "25  uncultured_Mucilaginibacter_sp._ERS6626923.fasta  \n",
       "26  uncultured_Mucilaginibacter_sp._ERS6626962.fasta  \n",
       "27  uncultured_Mucilaginibacter_sp._ERS6626972.fasta  \n",
       "28  uncultured_Mucilaginibacter_sp._ERS6626981.fasta  \n",
       "29  uncultured_Mucilaginibacter_sp._ERS6627275.fasta  \n",
       "30  uncultured_Mucilaginibacter_sp._ERS6627290.fasta  \n",
       "31           uncultured_Pantoea_sp._ERS6626643.fasta  \n",
       "32           uncultured_Pantoea_sp._ERS6627005.fasta  \n",
       "33         uncultured_Pelomonas_sp._ERS6626569.fasta  \n",
       "34   uncultured_Pseudacidovorax_sp._ERS6626577.fasta  \n",
       "35   uncultured_Pseudacidovorax_sp._ERS6626850.fasta  \n",
       "36   uncultured_Pseudacidovorax_sp._ERS6626909.fasta  \n",
       "37   uncultured_Pseudacidovorax_sp._ERS6627003.fasta  \n",
       "38   uncultured_Pseudacidovorax_sp._ERS6627003.fasta  \n",
       "39   uncultured_Pseudacidovorax_sp._ERS6627320.fasta  \n",
       "40   uncultured_Pseudacidovorax_sp._ERS6627320.fasta  \n",
       "41   uncultured_Pseudacidovorax_sp._ERS6627320.fasta  \n",
       "42   uncultured_Pseudacidovorax_sp._ERS6627321.fasta  \n",
       "43   uncultured_Pseudacidovorax_sp._ERS6627321.fasta  \n",
       "44       uncultured_Pseudomonas_sp._ERS6626673.fasta  \n",
       "45       uncultured_Pseudomonas_sp._ERS6627273.fasta  \n",
       "46  uncultured_Pseudorhodoferax_sp._ERS6626941.fasta  \n",
       "47      uncultured_Sphingomonas_sp._ERS6626680.fasta  \n",
       "48      uncultured_Sphingomonas_sp._ERS6626831.fasta  \n",
       "49      uncultured_Sphingomonas_sp._ERS6626957.fasta  \n",
       "50      uncultured_Sphingomonas_sp._ERS6627258.fasta  \n",
       "51         uncultured_Spirosoma_sp._ERS6627011.fasta  \n",
       "52  uncultured_Stenotrophomonas_sp._ERS6626688.fasta  \n",
       "53  uncultured_Stenotrophomonas_sp._ERS6627319.fasta  \n",
       "54        uncultured_Variovorax_sp._ERS6626779.fasta  \n",
       "55        uncultured_Variovorax_sp._ERS6627246.fasta  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filtering the data frame with a genome file names and query sequence id set, to remove the other sequences which don't contain MOTIF\n",
    "## Create a empty set to store fileted rows\n",
    "filtered_rows = []\n",
    "## Filter the data frame\n",
    "for key, values in coxl2_motif_id_set.items():\n",
    "    filtered_rows.append(coxl2_csv_sorted[(coxl2_csv_sorted['target_file'] == key) & (coxl2_csv_sorted['qseqid'].isin(values))])\n",
    "    ## Combine the filtered results\n",
    "    filtered_coxl2_csv = pd.concat(filtered_rows).reset_index(drop=True)\n",
    "## Check the output\n",
    "filtered_coxl2_csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "57bc1dee-b99e-4cfc-a2f4-c4c4d1d9264a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save the filtered data frame\n",
    "filtered_coxl2_csv.to_csv(f\"{motif2}/filtered_coxl2_csv.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
